{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.read_csv('Methylation.csv').set_index('attrib_name').T.to_csv('Methylationoutput.csv',index='none')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import data handling libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# set up for plotting figures\n",
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE5</th>\n",
       "      <th>BAGE</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>MIR7-1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4548</td>\n",
       "      <td>0.2310</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>-0.4737</td>\n",
       "      <td>-0.4817</td>\n",
       "      <td>0.4142</td>\n",
       "      <td>-0.4366</td>\n",
       "      <td>-0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>0.3420</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>-0.4313</td>\n",
       "      <td>-0.4692</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.3972</td>\n",
       "      <td>-0.2806</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>0.4373</td>\n",
       "      <td>-0.4527</td>\n",
       "      <td>-0.4643</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>0.3857</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.3C.AALJ</td>\n",
       "      <td>-0.4285</td>\n",
       "      <td>-0.4697</td>\n",
       "      <td>0.2664</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>-0.0637</td>\n",
       "      <td>-0.4783</td>\n",
       "      <td>0.3738</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4416</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>0.3319</td>\n",
       "      <td>0.1929</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.3C.AALK</td>\n",
       "      <td>-0.4043</td>\n",
       "      <td>-0.4714</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>-0.0440</td>\n",
       "      <td>-0.4827</td>\n",
       "      <td>0.3704</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4558</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.3457</td>\n",
       "      <td>0.2445</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.4H.AAAK</td>\n",
       "      <td>-0.4083</td>\n",
       "      <td>-0.4651</td>\n",
       "      <td>0.1717</td>\n",
       "      <td>0.3406</td>\n",
       "      <td>-0.0880</td>\n",
       "      <td>-0.4788</td>\n",
       "      <td>0.4143</td>\n",
       "      <td>-0.4442</td>\n",
       "      <td>-0.4675</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.3542</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>TCGA.5L.AAT0</td>\n",
       "      <td>-0.4151</td>\n",
       "      <td>-0.4685</td>\n",
       "      <td>0.2419</td>\n",
       "      <td>0.3673</td>\n",
       "      <td>-0.3054</td>\n",
       "      <td>-0.4827</td>\n",
       "      <td>0.4148</td>\n",
       "      <td>-0.4482</td>\n",
       "      <td>-0.4606</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>0.2846</td>\n",
       "      <td>-0.0858</td>\n",
       "      <td>-0.0858</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>TCGA.5L.AAT1</td>\n",
       "      <td>-0.4100</td>\n",
       "      <td>-0.4643</td>\n",
       "      <td>0.2020</td>\n",
       "      <td>0.3746</td>\n",
       "      <td>-0.0647</td>\n",
       "      <td>-0.4829</td>\n",
       "      <td>0.4315</td>\n",
       "      <td>-0.4191</td>\n",
       "      <td>-0.4593</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0867</td>\n",
       "      <td>0.0867</td>\n",
       "      <td>0.0867</td>\n",
       "      <td>0.0867</td>\n",
       "      <td>0.0867</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>0.2412</td>\n",
       "      <td>-0.0495</td>\n",
       "      <td>-0.0495</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>TCGA.5T.A9QA</td>\n",
       "      <td>-0.4451</td>\n",
       "      <td>-0.4495</td>\n",
       "      <td>0.3196</td>\n",
       "      <td>0.3975</td>\n",
       "      <td>0.1776</td>\n",
       "      <td>-0.4827</td>\n",
       "      <td>0.4280</td>\n",
       "      <td>-0.4399</td>\n",
       "      <td>-0.4629</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2175</td>\n",
       "      <td>0.2175</td>\n",
       "      <td>0.2175</td>\n",
       "      <td>0.2175</td>\n",
       "      <td>0.2175</td>\n",
       "      <td>0.3522</td>\n",
       "      <td>0.1939</td>\n",
       "      <td>0.0582</td>\n",
       "      <td>0.0582</td>\n",
       "      <td>-0.3693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>TCGA.A1.A0SB</td>\n",
       "      <td>-0.4654</td>\n",
       "      <td>-0.4484</td>\n",
       "      <td>0.1363</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>-0.4838</td>\n",
       "      <td>-0.4861</td>\n",
       "      <td>0.3223</td>\n",
       "      <td>-0.4716</td>\n",
       "      <td>-0.4790</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3220</td>\n",
       "      <td>0.3220</td>\n",
       "      <td>0.3220</td>\n",
       "      <td>0.3220</td>\n",
       "      <td>0.3220</td>\n",
       "      <td>0.3816</td>\n",
       "      <td>0.4396</td>\n",
       "      <td>0.1478</td>\n",
       "      <td>0.1478</td>\n",
       "      <td>-0.4334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>TCGA.A1.A0SE</td>\n",
       "      <td>-0.4414</td>\n",
       "      <td>-0.4620</td>\n",
       "      <td>0.2364</td>\n",
       "      <td>0.3922</td>\n",
       "      <td>-0.0145</td>\n",
       "      <td>-0.4841</td>\n",
       "      <td>0.4294</td>\n",
       "      <td>-0.4627</td>\n",
       "      <td>-0.4733</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>0.3947</td>\n",
       "      <td>0.4486</td>\n",
       "      <td>-0.1328</td>\n",
       "      <td>-0.1328</td>\n",
       "      <td>-0.3822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>TCGA.A1.A0SF</td>\n",
       "      <td>-0.4510</td>\n",
       "      <td>-0.4640</td>\n",
       "      <td>0.2014</td>\n",
       "      <td>0.3796</td>\n",
       "      <td>-0.2818</td>\n",
       "      <td>-0.4848</td>\n",
       "      <td>0.4393</td>\n",
       "      <td>-0.4717</td>\n",
       "      <td>-0.4743</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0929</td>\n",
       "      <td>0.0929</td>\n",
       "      <td>0.0929</td>\n",
       "      <td>0.0929</td>\n",
       "      <td>0.0929</td>\n",
       "      <td>0.3885</td>\n",
       "      <td>0.4100</td>\n",
       "      <td>-0.0653</td>\n",
       "      <td>-0.0653</td>\n",
       "      <td>-0.4346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>TCGA.A1.A0SG</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4811</td>\n",
       "      <td>0.2578</td>\n",
       "      <td>0.3802</td>\n",
       "      <td>-0.1725</td>\n",
       "      <td>-0.4875</td>\n",
       "      <td>0.4135</td>\n",
       "      <td>-0.4769</td>\n",
       "      <td>-0.4785</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>0.3908</td>\n",
       "      <td>0.4529</td>\n",
       "      <td>-0.1917</td>\n",
       "      <td>-0.1917</td>\n",
       "      <td>-0.4281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>TCGA.A1.A0SH</td>\n",
       "      <td>-0.4553</td>\n",
       "      <td>-0.4609</td>\n",
       "      <td>0.0838</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>-0.1679</td>\n",
       "      <td>-0.4841</td>\n",
       "      <td>0.4383</td>\n",
       "      <td>-0.4652</td>\n",
       "      <td>-0.4721</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0805</td>\n",
       "      <td>0.0805</td>\n",
       "      <td>0.0805</td>\n",
       "      <td>0.0805</td>\n",
       "      <td>0.0805</td>\n",
       "      <td>0.4014</td>\n",
       "      <td>0.4516</td>\n",
       "      <td>-0.0954</td>\n",
       "      <td>-0.0954</td>\n",
       "      <td>-0.3637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>TCGA.A1.A0SI</td>\n",
       "      <td>-0.4487</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>0.2003</td>\n",
       "      <td>0.3522</td>\n",
       "      <td>-0.0607</td>\n",
       "      <td>-0.4835</td>\n",
       "      <td>0.4514</td>\n",
       "      <td>-0.4674</td>\n",
       "      <td>-0.4744</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.3959</td>\n",
       "      <td>0.4546</td>\n",
       "      <td>-0.1176</td>\n",
       "      <td>-0.1176</td>\n",
       "      <td>-0.4302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>TCGA.A1.A0SJ</td>\n",
       "      <td>-0.4487</td>\n",
       "      <td>-0.4723</td>\n",
       "      <td>0.2851</td>\n",
       "      <td>0.4104</td>\n",
       "      <td>0.1005</td>\n",
       "      <td>-0.4823</td>\n",
       "      <td>0.4518</td>\n",
       "      <td>-0.4645</td>\n",
       "      <td>-0.4613</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1237</td>\n",
       "      <td>0.1237</td>\n",
       "      <td>0.1237</td>\n",
       "      <td>0.1237</td>\n",
       "      <td>0.1237</td>\n",
       "      <td>0.4427</td>\n",
       "      <td>0.4311</td>\n",
       "      <td>-0.0756</td>\n",
       "      <td>-0.0756</td>\n",
       "      <td>-0.2048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>TCGA.A1.A0SK</td>\n",
       "      <td>-0.4189</td>\n",
       "      <td>-0.4774</td>\n",
       "      <td>0.4379</td>\n",
       "      <td>-0.2939</td>\n",
       "      <td>-0.4816</td>\n",
       "      <td>-0.4842</td>\n",
       "      <td>0.3460</td>\n",
       "      <td>-0.4655</td>\n",
       "      <td>-0.4698</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>0.4266</td>\n",
       "      <td>0.4579</td>\n",
       "      <td>0.0205</td>\n",
       "      <td>0.0205</td>\n",
       "      <td>-0.2232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>TCGA.A1.A0SM</td>\n",
       "      <td>-0.4018</td>\n",
       "      <td>-0.4725</td>\n",
       "      <td>0.1609</td>\n",
       "      <td>0.3909</td>\n",
       "      <td>0.0325</td>\n",
       "      <td>-0.4828</td>\n",
       "      <td>0.4525</td>\n",
       "      <td>-0.4518</td>\n",
       "      <td>-0.4653</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2307</td>\n",
       "      <td>0.2307</td>\n",
       "      <td>0.2307</td>\n",
       "      <td>0.2307</td>\n",
       "      <td>0.2307</td>\n",
       "      <td>0.4035</td>\n",
       "      <td>0.4453</td>\n",
       "      <td>-0.0037</td>\n",
       "      <td>-0.0037</td>\n",
       "      <td>-0.2431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>TCGA.A1.A0SN</td>\n",
       "      <td>-0.4522</td>\n",
       "      <td>-0.4729</td>\n",
       "      <td>0.2267</td>\n",
       "      <td>0.4081</td>\n",
       "      <td>-0.0196</td>\n",
       "      <td>-0.4876</td>\n",
       "      <td>0.4171</td>\n",
       "      <td>-0.4648</td>\n",
       "      <td>-0.4760</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.3983</td>\n",
       "      <td>0.4621</td>\n",
       "      <td>0.0284</td>\n",
       "      <td>0.0284</td>\n",
       "      <td>-0.4373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>TCGA.A1.A0SO</td>\n",
       "      <td>-0.1983</td>\n",
       "      <td>-0.4751</td>\n",
       "      <td>0.4300</td>\n",
       "      <td>0.4168</td>\n",
       "      <td>-0.4856</td>\n",
       "      <td>-0.4864</td>\n",
       "      <td>0.4608</td>\n",
       "      <td>-0.4788</td>\n",
       "      <td>-0.4771</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1650</td>\n",
       "      <td>0.1650</td>\n",
       "      <td>0.1650</td>\n",
       "      <td>0.1650</td>\n",
       "      <td>0.1650</td>\n",
       "      <td>0.4241</td>\n",
       "      <td>0.4348</td>\n",
       "      <td>0.0783</td>\n",
       "      <td>0.0783</td>\n",
       "      <td>-0.2705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>TCGA.A1.A0SP</td>\n",
       "      <td>-0.4070</td>\n",
       "      <td>-0.4613</td>\n",
       "      <td>0.2350</td>\n",
       "      <td>0.4068</td>\n",
       "      <td>-0.4497</td>\n",
       "      <td>-0.4846</td>\n",
       "      <td>0.4462</td>\n",
       "      <td>-0.4534</td>\n",
       "      <td>-0.4658</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1428</td>\n",
       "      <td>0.1428</td>\n",
       "      <td>0.1428</td>\n",
       "      <td>0.1428</td>\n",
       "      <td>0.1428</td>\n",
       "      <td>0.3926</td>\n",
       "      <td>0.4398</td>\n",
       "      <td>-0.0790</td>\n",
       "      <td>-0.0790</td>\n",
       "      <td>-0.3134</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows × 20107 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     attrib_name    RBL2   VDAC3   ACTN1  ATP2A1   SFRP1   NIPA2  MAN1B1  \\\n",
       "0   TCGA.3C.AAAU -0.4444 -0.4548  0.2310  0.3568 -0.4737 -0.4817  0.4142   \n",
       "1   TCGA.3C.AALI -0.4313 -0.4692  0.2859  0.3972 -0.2806 -0.4740  0.4373   \n",
       "2   TCGA.3C.AALJ -0.4285 -0.4697  0.2664  0.3482 -0.0637 -0.4783  0.3738   \n",
       "3   TCGA.3C.AALK -0.4043 -0.4714  0.2174  0.3533 -0.0440 -0.4827  0.3704   \n",
       "4   TCGA.4H.AAAK -0.4083 -0.4651  0.1717  0.3406 -0.0880 -0.4788  0.4143   \n",
       "5   TCGA.5L.AAT0 -0.4151 -0.4685  0.2419  0.3673 -0.3054 -0.4827  0.4148   \n",
       "6   TCGA.5L.AAT1 -0.4100 -0.4643  0.2020  0.3746 -0.0647 -0.4829  0.4315   \n",
       "7   TCGA.5T.A9QA -0.4451 -0.4495  0.3196  0.3975  0.1776 -0.4827  0.4280   \n",
       "8   TCGA.A1.A0SB -0.4654 -0.4484  0.1363  0.3337 -0.4838 -0.4861  0.3223   \n",
       "9   TCGA.A1.A0SE -0.4414 -0.4620  0.2364  0.3922 -0.0145 -0.4841  0.4294   \n",
       "10  TCGA.A1.A0SF -0.4510 -0.4640  0.2014  0.3796 -0.2818 -0.4848  0.4393   \n",
       "11  TCGA.A1.A0SG -0.4444 -0.4811  0.2578  0.3802 -0.1725 -0.4875  0.4135   \n",
       "12  TCGA.A1.A0SH -0.4553 -0.4609  0.0838  0.3630 -0.1679 -0.4841  0.4383   \n",
       "13  TCGA.A1.A0SI -0.4487 -0.4688  0.2003  0.3522 -0.0607 -0.4835  0.4514   \n",
       "14  TCGA.A1.A0SJ -0.4487 -0.4723  0.2851  0.4104  0.1005 -0.4823  0.4518   \n",
       "15  TCGA.A1.A0SK -0.4189 -0.4774  0.4379 -0.2939 -0.4816 -0.4842  0.3460   \n",
       "16  TCGA.A1.A0SM -0.4018 -0.4725  0.1609  0.3909  0.0325 -0.4828  0.4525   \n",
       "17  TCGA.A1.A0SN -0.4522 -0.4729  0.2267  0.4081 -0.0196 -0.4876  0.4171   \n",
       "18  TCGA.A1.A0SO -0.1983 -0.4751  0.4300  0.4168 -0.4856 -0.4864  0.4608   \n",
       "19  TCGA.A1.A0SP -0.4070 -0.4613  0.2350  0.4068 -0.4497 -0.4846  0.4462   \n",
       "\n",
       "    TSEN34  LRRC16A  ...   BAGE5    BAGE   BAGE2   BAGE4   BAGE3  MIR637  \\\n",
       "0  -0.4366  -0.4610  ... -0.0420 -0.0420 -0.0420 -0.0420 -0.0420  0.3420   \n",
       "1  -0.4527  -0.4643  ... -0.0640 -0.0640 -0.0640 -0.0640 -0.0640  0.3857   \n",
       "2  -0.4430  -0.4416  ... -0.0873 -0.0873 -0.0873 -0.0873 -0.0873  0.3319   \n",
       "3  -0.4430  -0.4558  ...  0.1426  0.1426  0.1426  0.1426  0.1426  0.3457   \n",
       "4  -0.4442  -0.4675  ...  0.0218  0.0218  0.0218  0.0218  0.0218  0.3542   \n",
       "5  -0.4482  -0.4606  ...  0.0954  0.0954  0.0954  0.0954  0.0954  0.3482   \n",
       "6  -0.4191  -0.4593  ...  0.0867  0.0867  0.0867  0.0867  0.0867  0.3250   \n",
       "7  -0.4399  -0.4629  ...  0.2175  0.2175  0.2175  0.2175  0.2175  0.3522   \n",
       "8  -0.4716  -0.4790  ...  0.3220  0.3220  0.3220  0.3220  0.3220  0.3816   \n",
       "9  -0.4627  -0.4733  ...  0.0132  0.0132  0.0132  0.0132  0.0132  0.3947   \n",
       "10 -0.4717  -0.4743  ...  0.0929  0.0929  0.0929  0.0929  0.0929  0.3885   \n",
       "11 -0.4769  -0.4785  ... -0.0925 -0.0925 -0.0925 -0.0925 -0.0925  0.3908   \n",
       "12 -0.4652  -0.4721  ...  0.0805  0.0805  0.0805  0.0805  0.0805  0.4014   \n",
       "13 -0.4674  -0.4744  ...  0.0678  0.0678  0.0678  0.0678  0.0678  0.3959   \n",
       "14 -0.4645  -0.4613  ...  0.1237  0.1237  0.1237  0.1237  0.1237  0.4427   \n",
       "15 -0.4655  -0.4698  ... -0.0425 -0.0425 -0.0425 -0.0425 -0.0425  0.4266   \n",
       "16 -0.4518  -0.4653  ...  0.2307  0.2307  0.2307  0.2307  0.2307  0.4035   \n",
       "17 -0.4648  -0.4760  ...  0.1269  0.1269  0.1269  0.1269  0.1269  0.3983   \n",
       "18 -0.4788  -0.4771  ...  0.1650  0.1650  0.1650  0.1650  0.1650  0.4241   \n",
       "19 -0.4534  -0.4658  ...  0.1428  0.1428  0.1428  0.1428  0.1428  0.3926   \n",
       "\n",
       "    LOC100130932  GOLGA8F  GOLGA8G  MIR7-1  \n",
       "0         0.2383  -0.1168  -0.1168     NaN  \n",
       "1         0.3783  -0.1271  -0.1271     NaN  \n",
       "2         0.1929   0.0066   0.0066     NaN  \n",
       "3         0.2445  -0.0071  -0.0071     NaN  \n",
       "4         0.3128  -0.0929  -0.0929     NaN  \n",
       "5         0.2846  -0.0858  -0.0858     NaN  \n",
       "6         0.2412  -0.0495  -0.0495     NaN  \n",
       "7         0.1939   0.0582   0.0582 -0.3693  \n",
       "8         0.4396   0.1478   0.1478 -0.4334  \n",
       "9         0.4486  -0.1328  -0.1328 -0.3822  \n",
       "10        0.4100  -0.0653  -0.0653 -0.4346  \n",
       "11        0.4529  -0.1917  -0.1917 -0.4281  \n",
       "12        0.4516  -0.0954  -0.0954 -0.3637  \n",
       "13        0.4546  -0.1176  -0.1176 -0.4302  \n",
       "14        0.4311  -0.0756  -0.0756 -0.2048  \n",
       "15        0.4579   0.0205   0.0205 -0.2232  \n",
       "16        0.4453  -0.0037  -0.0037 -0.2431  \n",
       "17        0.4621   0.0284   0.0284 -0.4373  \n",
       "18        0.4348   0.0783   0.0783 -0.2705  \n",
       "19        0.4398  -0.0790  -0.0790 -0.3134  \n",
       "\n",
       "[20 rows x 20107 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read Methylation data\n",
    "methylationData = pd.read_csv('Methylationoutput.csv')\n",
    "\n",
    "methylationData = methylationData.rename(columns = {\"Unnamed: 0\": \"attrib_name\"}) \n",
    "methylationData[0:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      TCGA.3C.AAAU\n",
       "1      TCGA.3C.AALI\n",
       "2      TCGA.3C.AALJ\n",
       "3      TCGA.3C.AALK\n",
       "4      TCGA.4H.AAAK\n",
       "           ...     \n",
       "778    TCGA.WT.AB44\n",
       "779    TCGA.XX.A899\n",
       "780    TCGA.XX.A89A\n",
       "781    TCGA.Z7.A8R5\n",
       "782    TCGA.Z7.A8R6\n",
       "Name: attrib_name, Length: 783, dtype: object"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylationData['attrib_name']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reading Clinical data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read the file; this assumes it's in the same directory as this notebook\n",
    "clinicalData = pd.read_csv('clinical_transpose.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>years_to_birth</th>\n",
       "      <th>Tumor_purity</th>\n",
       "      <th>pathologic_stage</th>\n",
       "      <th>pathology_T_stage</th>\n",
       "      <th>pathology_N_stage</th>\n",
       "      <th>pathology_M_stage</th>\n",
       "      <th>histological_type</th>\n",
       "      <th>number_of_lymph_nodes</th>\n",
       "      <th>PAM50</th>\n",
       "      <th>...</th>\n",
       "      <th>PR.Status</th>\n",
       "      <th>HER2.Status</th>\n",
       "      <th>gender</th>\n",
       "      <th>radiation_therapy</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Median_overall_survival</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "      <th>overallsurvival</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.5L.AAT0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>0.6501</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>hispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1477.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1477,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.5L.AAT1</td>\n",
       "      <td>63.0</td>\n",
       "      <td>0.5553</td>\n",
       "      <td>stageiv</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m1</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>hispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1471.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1471,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.A1.A0SP</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.6913</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>584.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>584,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.A2.A04V</td>\n",
       "      <td>39.0</td>\n",
       "      <td>0.8341</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1920.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1920,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.A2.A04Y</td>\n",
       "      <td>53.0</td>\n",
       "      <td>0.7318</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LumB</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1099.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1099,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>TCGA.A2.A0CQ</td>\n",
       "      <td>62.0</td>\n",
       "      <td>0.7701</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t1</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>blackorafricanamerican</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2695.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2695,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>TCGA.A2.A1G4</td>\n",
       "      <td>71.0</td>\n",
       "      <td>0.8879</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>2.0</td>\n",
       "      <td>LumB</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>595.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>595,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>TCGA.A2.A25A</td>\n",
       "      <td>44.0</td>\n",
       "      <td>0.5950</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3276.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3276,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>TCGA.A7.A0CD</td>\n",
       "      <td>66.0</td>\n",
       "      <td>0.8525</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t1</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1165.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1165,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>TCGA.A7.A13G</td>\n",
       "      <td>79.0</td>\n",
       "      <td>0.9357</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>718.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>718,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>TCGA.A7.A26E</td>\n",
       "      <td>71.0</td>\n",
       "      <td>0.7913</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>954.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>954,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>TCGA.A7.A26F</td>\n",
       "      <td>55.0</td>\n",
       "      <td>0.6312</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t1</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>other,specify</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>738.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>738,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>TCGA.A7.A26H</td>\n",
       "      <td>72.0</td>\n",
       "      <td>0.7811</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t1</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>724.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>724,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>TCGA.A7.A26I</td>\n",
       "      <td>65.0</td>\n",
       "      <td>0.7381</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>661.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>661,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>TCGA.A7.A2KD</td>\n",
       "      <td>53.0</td>\n",
       "      <td>0.6932</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>8.0</td>\n",
       "      <td>LumB</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>blackorafricanamerican</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>679.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>679,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>TCGA.A7.A3J1</td>\n",
       "      <td>63.0</td>\n",
       "      <td>0.6485</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t1</td>\n",
       "      <td>n0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>343.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>343,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>TCGA.A7.A426</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.7037</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>13.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>364,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>TCGA.A7.A5ZX</td>\n",
       "      <td>48.0</td>\n",
       "      <td>0.8360</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n3</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>21.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>0.0</td>\n",
       "      <td>336.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>336,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>TCGA.A8.A06T</td>\n",
       "      <td>74.0</td>\n",
       "      <td>0.7562</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1614.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1614,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>TCGA.A8.A06U</td>\n",
       "      <td>80.0</td>\n",
       "      <td>0.7982</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LumA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>883.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>883,1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     attrib_name  years_to_birth  Tumor_purity pathologic_stage  \\\n",
       "0   TCGA.5L.AAT0            42.0        0.6501          stageii   \n",
       "1   TCGA.5L.AAT1            63.0        0.5553          stageiv   \n",
       "2   TCGA.A1.A0SP            40.0        0.6913          stageii   \n",
       "3   TCGA.A2.A04V            39.0        0.8341          stageii   \n",
       "4   TCGA.A2.A04Y            53.0        0.7318          stageii   \n",
       "5   TCGA.A2.A0CQ            62.0        0.7701           stagei   \n",
       "6   TCGA.A2.A1G4            71.0        0.8879         stageiii   \n",
       "7   TCGA.A2.A25A            44.0        0.5950          stageii   \n",
       "8   TCGA.A7.A0CD            66.0        0.8525           stagei   \n",
       "9   TCGA.A7.A13G            79.0        0.9357          stageii   \n",
       "10  TCGA.A7.A26E            71.0        0.7913         stageiii   \n",
       "11  TCGA.A7.A26F            55.0        0.6312           stagei   \n",
       "12  TCGA.A7.A26H            72.0        0.7811          stageii   \n",
       "13  TCGA.A7.A26I            65.0        0.7381          stageii   \n",
       "14  TCGA.A7.A2KD            53.0        0.6932         stageiii   \n",
       "15  TCGA.A7.A3J1            63.0        0.6485           stagei   \n",
       "16  TCGA.A7.A426            50.0        0.7037         stageiii   \n",
       "17  TCGA.A7.A5ZX            48.0        0.8360         stageiii   \n",
       "18  TCGA.A8.A06T            74.0        0.7562         stageiii   \n",
       "19  TCGA.A8.A06U            80.0        0.7982          stageii   \n",
       "\n",
       "   pathology_T_stage pathology_N_stage pathology_M_stage  \\\n",
       "0                 t2                n0                m0   \n",
       "1                 t2                n0                m1   \n",
       "2                 t2                n0                m0   \n",
       "3                 t2                n0                m0   \n",
       "4                 t2                n1                m0   \n",
       "5                 t1                n0                m0   \n",
       "6                 t3                n1                m0   \n",
       "7                 t2                n0                m0   \n",
       "8                 t1                n0                m0   \n",
       "9                 t2                n0               NaN   \n",
       "10                t3                n1               NaN   \n",
       "11                t1                n0                m0   \n",
       "12                t1                n1                m0   \n",
       "13                t2                n0                m0   \n",
       "14                t2                n2               NaN   \n",
       "15                t1                n0               NaN   \n",
       "16                t3                n3               NaN   \n",
       "17                t2                n3                m0   \n",
       "18                t3                n1                m0   \n",
       "19                t2                n1                m0   \n",
       "\n",
       "               histological_type  number_of_lymph_nodes  PAM50  ... PR.Status  \\\n",
       "0   infiltratinglobularcarcinoma                    0.0    NaN  ...       NaN   \n",
       "1   infiltratinglobularcarcinoma                    0.0    NaN  ...       NaN   \n",
       "2    infiltratingductalcarcinoma                    0.0  Basal  ...       NaN   \n",
       "3    infiltratingductalcarcinoma                    0.0   LumA  ...       NaN   \n",
       "4    infiltratingductalcarcinoma                    1.0   LumB  ...       NaN   \n",
       "5    infiltratingductalcarcinoma                    0.0   LumA  ...       NaN   \n",
       "6    infiltratingductalcarcinoma                    2.0   LumB  ...       NaN   \n",
       "7   infiltratinglobularcarcinoma                    0.0    NaN  ...       NaN   \n",
       "8    infiltratingductalcarcinoma                    0.0   LumA  ...  Positive   \n",
       "9    infiltratingductalcarcinoma                    0.0   LumA  ...       NaN   \n",
       "10  infiltratinglobularcarcinoma                    0.0   LumA  ...       NaN   \n",
       "11                 other,specify                    0.0  Basal  ...       NaN   \n",
       "12   infiltratingductalcarcinoma                    0.0   LumA  ...       NaN   \n",
       "13   infiltratingductalcarcinoma                    0.0  Basal  ...       NaN   \n",
       "14   infiltratingductalcarcinoma                    8.0   LumB  ...       NaN   \n",
       "15  infiltratinglobularcarcinoma                    0.0   LumA  ...       NaN   \n",
       "16  infiltratinglobularcarcinoma                   13.0   LumA  ...       NaN   \n",
       "17  infiltratinglobularcarcinoma                   21.0   LumA  ...       NaN   \n",
       "18   infiltratingductalcarcinoma                    NaN   LumA  ...       NaN   \n",
       "19   infiltratingductalcarcinoma                    NaN   LumA  ...       NaN   \n",
       "\n",
       "   HER2.Status  gender radiation_therapy                    race  \\\n",
       "0          NaN  female               yes                   white   \n",
       "1          NaN  female                no                   white   \n",
       "2          NaN  female               NaN                     NaN   \n",
       "3          NaN  female                no                   white   \n",
       "4          NaN  female               yes                   white   \n",
       "5          NaN  female               yes  blackorafricanamerican   \n",
       "6          NaN  female                no                   white   \n",
       "7          NaN  female               yes                   white   \n",
       "8     Negative  female               yes                   white   \n",
       "9          NaN  female                no                   white   \n",
       "10         NaN  female               yes                   white   \n",
       "11         NaN  female               yes                   white   \n",
       "12         NaN  female                no                   white   \n",
       "13         NaN  female               yes                   white   \n",
       "14         NaN  female               yes  blackorafricanamerican   \n",
       "15         NaN  female               yes                   white   \n",
       "16         NaN  female               yes                   white   \n",
       "17         NaN  female               yes                   white   \n",
       "18         NaN  female                no                     NaN   \n",
       "19         NaN  female                no                     NaN   \n",
       "\n",
       "              ethnicity Median_overall_survival  overall_survival  status  \\\n",
       "0      hispanicorlatino                     0.0            1477.0     0.0   \n",
       "1      hispanicorlatino                     0.0            1471.0     0.0   \n",
       "2   nothispanicorlatino                     0.0             584.0     0.0   \n",
       "3   nothispanicorlatino                     0.0            1920.0     1.0   \n",
       "4                   NaN                     0.0            1099.0     0.0   \n",
       "5   nothispanicorlatino                     0.0            2695.0     0.0   \n",
       "6   nothispanicorlatino                     0.0             595.0     0.0   \n",
       "7   nothispanicorlatino                     0.0            3276.0     0.0   \n",
       "8   nothispanicorlatino                     0.0            1165.0     0.0   \n",
       "9   nothispanicorlatino                     0.0             718.0     0.0   \n",
       "10  nothispanicorlatino                     0.0             954.0     0.0   \n",
       "11  nothispanicorlatino                     0.0             738.0     0.0   \n",
       "12  nothispanicorlatino                     0.0             724.0     0.0   \n",
       "13  nothispanicorlatino                     0.0             661.0     0.0   \n",
       "14  nothispanicorlatino                     0.0             679.0     0.0   \n",
       "15  nothispanicorlatino                     0.0             343.0     0.0   \n",
       "16  nothispanicorlatino                     0.0             364.0     0.0   \n",
       "17  nothispanicorlatino                     0.0             336.0     0.0   \n",
       "18                  NaN                     0.0            1614.0     0.0   \n",
       "19                  NaN                     0.0             883.0     1.0   \n",
       "\n",
       "    overallsurvival  \n",
       "0            1477,0  \n",
       "1            1471,0  \n",
       "2             584,0  \n",
       "3            1920,1  \n",
       "4            1099,0  \n",
       "5            2695,0  \n",
       "6             595,0  \n",
       "7            3276,0  \n",
       "8            1165,0  \n",
       "9             718,0  \n",
       "10            954,0  \n",
       "11            738,0  \n",
       "12            724,0  \n",
       "13            661,0  \n",
       "14            679,0  \n",
       "15            343,0  \n",
       "16            364,0  \n",
       "17            336,0  \n",
       "18           1614,0  \n",
       "19            883,1  \n",
       "\n",
       "[20 rows x 21 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinicalData[0:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1097 entries, 0 to 1096\n",
      "Data columns (total 21 columns):\n",
      "attrib_name                1097 non-null object\n",
      "years_to_birth             1082 non-null float64\n",
      "Tumor_purity               1086 non-null float64\n",
      "pathologic_stage           1075 non-null object\n",
      "pathology_T_stage          1094 non-null object\n",
      "pathology_N_stage          1077 non-null object\n",
      "pathology_M_stage          934 non-null object\n",
      "histological_type          1095 non-null object\n",
      "number_of_lymph_nodes      929 non-null float64\n",
      "PAM50                      826 non-null object\n",
      "ER.Status                  104 non-null object\n",
      "PR.Status                  105 non-null object\n",
      "HER2.Status                104 non-null object\n",
      "gender                     1097 non-null object\n",
      "radiation_therapy          1002 non-null object\n",
      "race                       1001 non-null object\n",
      "ethnicity                  923 non-null object\n",
      "Median_overall_survival    1055 non-null float64\n",
      "overall_survival           1055 non-null float64\n",
      "status                     1055 non-null float64\n",
      "overallsurvival            1097 non-null object\n",
      "dtypes: float64(6), object(15)\n",
      "memory usage: 180.1+ KB\n"
     ]
    }
   ],
   "source": [
    "# this will print some basic stats on the data\n",
    "clinicalData.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>years_to_birth</th>\n",
       "      <th>Tumor_purity</th>\n",
       "      <th>number_of_lymph_nodes</th>\n",
       "      <th>Median_overall_survival</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>count</td>\n",
       "      <td>1082.000000</td>\n",
       "      <td>1086.000000</td>\n",
       "      <td>929.000000</td>\n",
       "      <td>1055.000000</td>\n",
       "      <td>1055.000000</td>\n",
       "      <td>1055.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mean</td>\n",
       "      <td>58.596118</td>\n",
       "      <td>0.728815</td>\n",
       "      <td>2.362756</td>\n",
       "      <td>0.032227</td>\n",
       "      <td>1294.534597</td>\n",
       "      <td>0.143128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>std</td>\n",
       "      <td>13.193799</td>\n",
       "      <td>0.135894</td>\n",
       "      <td>4.633874</td>\n",
       "      <td>0.176687</td>\n",
       "      <td>1189.770845</td>\n",
       "      <td>0.350369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>min</td>\n",
       "      <td>26.000000</td>\n",
       "      <td>0.134400</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>30.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25%</td>\n",
       "      <td>49.000000</td>\n",
       "      <td>0.656150</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>499.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50%</td>\n",
       "      <td>59.000000</td>\n",
       "      <td>0.746600</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>904.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75%</td>\n",
       "      <td>68.000000</td>\n",
       "      <td>0.825250</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1728.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>max</td>\n",
       "      <td>90.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>35.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>8605.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       years_to_birth  Tumor_purity  number_of_lymph_nodes  \\\n",
       "count     1082.000000   1086.000000             929.000000   \n",
       "mean        58.596118      0.728815               2.362756   \n",
       "std         13.193799      0.135894               4.633874   \n",
       "min         26.000000      0.134400               0.000000   \n",
       "25%         49.000000      0.656150               0.000000   \n",
       "50%         59.000000      0.746600               1.000000   \n",
       "75%         68.000000      0.825250               2.000000   \n",
       "max         90.000000      1.000000              35.000000   \n",
       "\n",
       "       Median_overall_survival  overall_survival       status  \n",
       "count              1055.000000       1055.000000  1055.000000  \n",
       "mean                  0.032227       1294.534597     0.143128  \n",
       "std                   0.176687       1189.770845     0.350369  \n",
       "min                   0.000000         30.000000     0.000000  \n",
       "25%                   0.000000        499.000000     0.000000  \n",
       "50%                   0.000000        904.000000     0.000000  \n",
       "75%                   0.000000       1728.000000     0.000000  \n",
       "max                   1.000000       8605.000000     1.000000  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# this will print basic statistical information about each numeric column\n",
    "clinicalData.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 0.,  1., nan])"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# let's make sure that the \"status\" column is really structured like we think by looking at the 'unique' values\n",
    "clinicalData['status'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number who did not survive: 151.0\n",
      "liklihood of non-survival: 0.14312796208530806\n"
     ]
    }
   ],
   "source": [
    "# let's see how many people survived; we can use \"sum\" since the value is 1 or 0\n",
    "print('number who did not survive:', clinicalData['status'].sum())\n",
    "pnSurvived = clinicalData['status'].sum() / clinicalData['status'].count()\n",
    "print('liklihood of non-survival:', pnSurvived)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Removing unhelpful duplicates.\n",
    "1.overallsurvival is repeatation column.\n",
    "2.attrib_name is use case number, unique, not relevant here.\n",
    "3.Median_overall_survival not relevant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "cleanClinicalData = clinicalData.drop(['overallsurvival', 'Median_overall_survival'], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>years_to_birth</th>\n",
       "      <th>Tumor_purity</th>\n",
       "      <th>pathologic_stage</th>\n",
       "      <th>pathology_T_stage</th>\n",
       "      <th>pathology_N_stage</th>\n",
       "      <th>pathology_M_stage</th>\n",
       "      <th>histological_type</th>\n",
       "      <th>number_of_lymph_nodes</th>\n",
       "      <th>PAM50</th>\n",
       "      <th>ER.Status</th>\n",
       "      <th>PR.Status</th>\n",
       "      <th>HER2.Status</th>\n",
       "      <th>gender</th>\n",
       "      <th>radiation_therapy</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.5L.AAT0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>0.6501</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>hispanicorlatino</td>\n",
       "      <td>1477.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.5L.AAT1</td>\n",
       "      <td>63.0</td>\n",
       "      <td>0.5553</td>\n",
       "      <td>stageiv</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m1</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>hispanicorlatino</td>\n",
       "      <td>1471.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.A1.A0SP</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.6913</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>584.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.A2.A04V</td>\n",
       "      <td>39.0</td>\n",
       "      <td>0.8341</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>1920.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.A2.A04Y</td>\n",
       "      <td>53.0</td>\n",
       "      <td>0.7318</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LumB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1099.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>TCGA.A2.A0CQ</td>\n",
       "      <td>62.0</td>\n",
       "      <td>0.7701</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t1</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>blackorafricanamerican</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>2695.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>TCGA.A2.A1G4</td>\n",
       "      <td>71.0</td>\n",
       "      <td>0.8879</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>2.0</td>\n",
       "      <td>LumB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>595.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>TCGA.A2.A25A</td>\n",
       "      <td>44.0</td>\n",
       "      <td>0.5950</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>3276.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>TCGA.A7.A0CD</td>\n",
       "      <td>66.0</td>\n",
       "      <td>0.8525</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t1</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>1165.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>TCGA.A7.A13G</td>\n",
       "      <td>79.0</td>\n",
       "      <td>0.9357</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>718.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>TCGA.A7.A26E</td>\n",
       "      <td>71.0</td>\n",
       "      <td>0.7913</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>954.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>TCGA.A7.A26F</td>\n",
       "      <td>55.0</td>\n",
       "      <td>0.6312</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t1</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>other,specify</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>738.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>TCGA.A7.A26H</td>\n",
       "      <td>72.0</td>\n",
       "      <td>0.7811</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t1</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>724.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>TCGA.A7.A26I</td>\n",
       "      <td>65.0</td>\n",
       "      <td>0.7381</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>661.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>TCGA.A7.A2KD</td>\n",
       "      <td>53.0</td>\n",
       "      <td>0.6932</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>8.0</td>\n",
       "      <td>LumB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>blackorafricanamerican</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>679.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>TCGA.A7.A3J1</td>\n",
       "      <td>63.0</td>\n",
       "      <td>0.6485</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t1</td>\n",
       "      <td>n0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>343.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>TCGA.A7.A426</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.7037</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>13.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>364.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>TCGA.A7.A5ZX</td>\n",
       "      <td>48.0</td>\n",
       "      <td>0.8360</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n3</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratinglobularcarcinoma</td>\n",
       "      <td>21.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>yes</td>\n",
       "      <td>white</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>336.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>TCGA.A8.A06T</td>\n",
       "      <td>74.0</td>\n",
       "      <td>0.7562</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1614.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>TCGA.A8.A06U</td>\n",
       "      <td>80.0</td>\n",
       "      <td>0.7982</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t2</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>infiltratingductalcarcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LumA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>883.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     attrib_name  years_to_birth  Tumor_purity pathologic_stage  \\\n",
       "0   TCGA.5L.AAT0            42.0        0.6501          stageii   \n",
       "1   TCGA.5L.AAT1            63.0        0.5553          stageiv   \n",
       "2   TCGA.A1.A0SP            40.0        0.6913          stageii   \n",
       "3   TCGA.A2.A04V            39.0        0.8341          stageii   \n",
       "4   TCGA.A2.A04Y            53.0        0.7318          stageii   \n",
       "5   TCGA.A2.A0CQ            62.0        0.7701           stagei   \n",
       "6   TCGA.A2.A1G4            71.0        0.8879         stageiii   \n",
       "7   TCGA.A2.A25A            44.0        0.5950          stageii   \n",
       "8   TCGA.A7.A0CD            66.0        0.8525           stagei   \n",
       "9   TCGA.A7.A13G            79.0        0.9357          stageii   \n",
       "10  TCGA.A7.A26E            71.0        0.7913         stageiii   \n",
       "11  TCGA.A7.A26F            55.0        0.6312           stagei   \n",
       "12  TCGA.A7.A26H            72.0        0.7811          stageii   \n",
       "13  TCGA.A7.A26I            65.0        0.7381          stageii   \n",
       "14  TCGA.A7.A2KD            53.0        0.6932         stageiii   \n",
       "15  TCGA.A7.A3J1            63.0        0.6485           stagei   \n",
       "16  TCGA.A7.A426            50.0        0.7037         stageiii   \n",
       "17  TCGA.A7.A5ZX            48.0        0.8360         stageiii   \n",
       "18  TCGA.A8.A06T            74.0        0.7562         stageiii   \n",
       "19  TCGA.A8.A06U            80.0        0.7982          stageii   \n",
       "\n",
       "   pathology_T_stage pathology_N_stage pathology_M_stage  \\\n",
       "0                 t2                n0                m0   \n",
       "1                 t2                n0                m1   \n",
       "2                 t2                n0                m0   \n",
       "3                 t2                n0                m0   \n",
       "4                 t2                n1                m0   \n",
       "5                 t1                n0                m0   \n",
       "6                 t3                n1                m0   \n",
       "7                 t2                n0                m0   \n",
       "8                 t1                n0                m0   \n",
       "9                 t2                n0               NaN   \n",
       "10                t3                n1               NaN   \n",
       "11                t1                n0                m0   \n",
       "12                t1                n1                m0   \n",
       "13                t2                n0                m0   \n",
       "14                t2                n2               NaN   \n",
       "15                t1                n0               NaN   \n",
       "16                t3                n3               NaN   \n",
       "17                t2                n3                m0   \n",
       "18                t3                n1                m0   \n",
       "19                t2                n1                m0   \n",
       "\n",
       "               histological_type  number_of_lymph_nodes  PAM50 ER.Status  \\\n",
       "0   infiltratinglobularcarcinoma                    0.0    NaN       NaN   \n",
       "1   infiltratinglobularcarcinoma                    0.0    NaN       NaN   \n",
       "2    infiltratingductalcarcinoma                    0.0  Basal       NaN   \n",
       "3    infiltratingductalcarcinoma                    0.0   LumA       NaN   \n",
       "4    infiltratingductalcarcinoma                    1.0   LumB       NaN   \n",
       "5    infiltratingductalcarcinoma                    0.0   LumA       NaN   \n",
       "6    infiltratingductalcarcinoma                    2.0   LumB       NaN   \n",
       "7   infiltratinglobularcarcinoma                    0.0    NaN       NaN   \n",
       "8    infiltratingductalcarcinoma                    0.0   LumA  Positive   \n",
       "9    infiltratingductalcarcinoma                    0.0   LumA       NaN   \n",
       "10  infiltratinglobularcarcinoma                    0.0   LumA       NaN   \n",
       "11                 other,specify                    0.0  Basal       NaN   \n",
       "12   infiltratingductalcarcinoma                    0.0   LumA       NaN   \n",
       "13   infiltratingductalcarcinoma                    0.0  Basal       NaN   \n",
       "14   infiltratingductalcarcinoma                    8.0   LumB       NaN   \n",
       "15  infiltratinglobularcarcinoma                    0.0   LumA       NaN   \n",
       "16  infiltratinglobularcarcinoma                   13.0   LumA       NaN   \n",
       "17  infiltratinglobularcarcinoma                   21.0   LumA       NaN   \n",
       "18   infiltratingductalcarcinoma                    NaN   LumA       NaN   \n",
       "19   infiltratingductalcarcinoma                    NaN   LumA       NaN   \n",
       "\n",
       "   PR.Status HER2.Status  gender radiation_therapy                    race  \\\n",
       "0        NaN         NaN  female               yes                   white   \n",
       "1        NaN         NaN  female                no                   white   \n",
       "2        NaN         NaN  female               NaN                     NaN   \n",
       "3        NaN         NaN  female                no                   white   \n",
       "4        NaN         NaN  female               yes                   white   \n",
       "5        NaN         NaN  female               yes  blackorafricanamerican   \n",
       "6        NaN         NaN  female                no                   white   \n",
       "7        NaN         NaN  female               yes                   white   \n",
       "8   Positive    Negative  female               yes                   white   \n",
       "9        NaN         NaN  female                no                   white   \n",
       "10       NaN         NaN  female               yes                   white   \n",
       "11       NaN         NaN  female               yes                   white   \n",
       "12       NaN         NaN  female                no                   white   \n",
       "13       NaN         NaN  female               yes                   white   \n",
       "14       NaN         NaN  female               yes  blackorafricanamerican   \n",
       "15       NaN         NaN  female               yes                   white   \n",
       "16       NaN         NaN  female               yes                   white   \n",
       "17       NaN         NaN  female               yes                   white   \n",
       "18       NaN         NaN  female                no                     NaN   \n",
       "19       NaN         NaN  female                no                     NaN   \n",
       "\n",
       "              ethnicity  overall_survival  status  \n",
       "0      hispanicorlatino            1477.0     0.0  \n",
       "1      hispanicorlatino            1471.0     0.0  \n",
       "2   nothispanicorlatino             584.0     0.0  \n",
       "3   nothispanicorlatino            1920.0     1.0  \n",
       "4                   NaN            1099.0     0.0  \n",
       "5   nothispanicorlatino            2695.0     0.0  \n",
       "6   nothispanicorlatino             595.0     0.0  \n",
       "7   nothispanicorlatino            3276.0     0.0  \n",
       "8   nothispanicorlatino            1165.0     0.0  \n",
       "9   nothispanicorlatino             718.0     0.0  \n",
       "10  nothispanicorlatino             954.0     0.0  \n",
       "11  nothispanicorlatino             738.0     0.0  \n",
       "12  nothispanicorlatino             724.0     0.0  \n",
       "13  nothispanicorlatino             661.0     0.0  \n",
       "14  nothispanicorlatino             679.0     0.0  \n",
       "15  nothispanicorlatino             343.0     0.0  \n",
       "16  nothispanicorlatino             364.0     0.0  \n",
       "17  nothispanicorlatino             336.0     0.0  \n",
       "18                  NaN            1614.0     0.0  \n",
       "19                  NaN             883.0     1.0  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleanClinicalData[0:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#removing where survival status is not present\n",
    "cleanClinicalData=cleanClinicalData[cleanClinicalData['status'].notnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>years_to_birth</th>\n",
       "      <th>Tumor_purity</th>\n",
       "      <th>pathologic_stage</th>\n",
       "      <th>pathology_T_stage</th>\n",
       "      <th>pathology_N_stage</th>\n",
       "      <th>pathology_M_stage</th>\n",
       "      <th>histological_type</th>\n",
       "      <th>number_of_lymph_nodes</th>\n",
       "      <th>PAM50</th>\n",
       "      <th>ER.Status</th>\n",
       "      <th>PR.Status</th>\n",
       "      <th>HER2.Status</th>\n",
       "      <th>gender</th>\n",
       "      <th>radiation_therapy</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [attrib_name, years_to_birth, Tumor_purity, pathologic_stage, pathology_T_stage, pathology_N_stage, pathology_M_stage, histological_type, number_of_lymph_nodes, PAM50, ER.Status, PR.Status, HER2.Status, gender, radiation_therapy, race, ethnicity, overall_survival, status]\n",
       "Index: []"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleanClinicalData[cleanClinicalData['status'].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 1055 entries, 0 to 1096\n",
      "Data columns (total 19 columns):\n",
      "attrib_name              1055 non-null object\n",
      "years_to_birth           1041 non-null float64\n",
      "Tumor_purity             1044 non-null float64\n",
      "pathologic_stage         1033 non-null object\n",
      "pathology_T_stage        1052 non-null object\n",
      "pathology_N_stage        1035 non-null object\n",
      "pathology_M_stage        893 non-null object\n",
      "histological_type        1053 non-null object\n",
      "number_of_lymph_nodes    902 non-null float64\n",
      "PAM50                    801 non-null object\n",
      "ER.Status                99 non-null object\n",
      "PR.Status                100 non-null object\n",
      "HER2.Status              99 non-null object\n",
      "gender                   1055 non-null object\n",
      "radiation_therapy        976 non-null object\n",
      "race                     968 non-null object\n",
      "ethnicity                890 non-null object\n",
      "overall_survival         1055 non-null float64\n",
      "status                   1055 non-null float64\n",
      "dtypes: float64(5), object(14)\n",
      "memory usage: 164.8+ KB\n"
     ]
    }
   ],
   "source": [
    "cleanClinicalData.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[<matplotlib.axes._subplots.AxesSubplot object at 0x7ff46b9d3750>]],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXcAAAEICAYAAACktLTqAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAVTUlEQVR4nO3df5BlZX3n8fdHQFQGGVh0wi8drGItUSpEe5VdU6aRRAV2HUypi1KKv2qsXd3VDVXJqFulu1m2JlvBJFQSzSQY0RVHRFxZ0EQW7aX4AxVYguJIHGWEAQR/ANKjqwt+94/7DF6GvtO/bk9PP/1+VXXdc57znHOe89Tpzz33uefem6pCktSXJyx3AyRJ42e4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA4Z7lpVknw0yX9p05NJdi53m3ZL8oUk545hO1NJ3jaONmnlOnC5GyBpoKpOX+42qB9euasbSfbbi5X9uW3qk+GufSbJc9qQwQNJbk3yyiSnJPl+kgOG6r0qyS1t+glJNiX5TpIfJbk0yRFt2fokleStSe4AvtTKP922+WCSa5M8d5Ht/oMkdyV5KMltSU5r5Y8O8bT5xwzzJNnR1r0F2JXkPya5bI9t/1mSC9v0VJK3JTm49dHzhuo9LcnPkjw9yeFJrkzygyT3t+ljF3OM6o/hrn0iyUHA/wS+CDwd+HfAJ4D7gV3AS4eqvx64pE3/e+As4LeAo1v9v9hj878FPAd4eZv/AnBC289NbT8LbfezgXcC/6yqDm372DGPTbwOOBNYC3wcOCPJU9u2DwBey6+OFYCq+jlweVt3t9cC/7uq7mPwf/u3wDOBZwA/A/58vsemvhnu2ldOAdYAm6vqF1X1JeBKBgH2yfZIkkOBM1oZwNuB91XVzhZ6HwBevccwxweqaldV/Qygqj5SVQ8N1f/1JIctsN2PAAcDJyY5qKp2VNV35rH+hVV1Z1X9rKq+x+DJ5qy27KXAT6vq+hnWu4THhvujT3hV9aOq+kxV/bSqHgLOZ/AEJz3KcNe+cjRwZ1X9cqjse8AxDELrd5McDPwucFMLQhhcnX62DVM8AGxjELjrhrZz5+6JJAck2dyGcX7Cr66yj1xIo6tqO/BuBk8S9yXZmuToeWzizj3mh0N7+BXKnr4EPDnJi5I8EzgZ+CxAkqck+ask32vHeC2wdnhoSzLcta/cDRyXZPicewZwV1V9k0HQn87jA+9O4PSqWjv096SqumuozvD3Vr8e2AD8NnAYsL6VZ6ENr6pLquo3GTzRFPBHbdEu4ClDVX9tptX3mP80MNnGyF/FiHBvT4KXMngieD1wZbtKBzgPeDbwoqp6KvCSVr7gY1R/DHftK19hEIa/n+SgJJPAvwK2tuWXMBhffwmDANztw8D57ep19xuLG/ayn0OBnwM/YhC8/3UxjU7y7CQvba8q/i+D8e1H2uKbGYyhH5Hk1xhc4e9VVf0AmGIwZn57VW3bS/VLgH8NnMNjnwQObe14oL25/P75HZVWA8Nd+0RV/QJ4JYOr8x8Cfwm8saq+1ap8EpgEvlRVPxxa9c+AK4AvJnkIuB540V529TEGrwLuAr7Z6i/GwcDm1ubvM3iT9r1t2ceBf2Aw9PNF4FNz3OYlDF5ZjBqSAaCqdj8hHs3gTeLd/hR4cmvT9cDfzXG/WkXiLzFJUn+8cpekDvmpOa16SZ7BYAhnJidW1R37sj3SODgsI0kd2i+u3I888shav379vNfbtWsXhxxyyPgb1AH7ZjT7ZjT7ZrT9sW9uvPHGH1bV02Zatl+E+/r167nhhhvmvd7U1BSTk5Pjb1AH7JvR7JvR7JvR9se+SfK9Uct8Q1WSOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA4Z7pLUIcNdkjq0X3xCdTmt33TVgtfdsfnMMbZEksbHK3dJ6pDhLkkdMtwlqUOGuyR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDs0a7kmOS/LlJNuS3JrkXa38A0nuSnJz+ztjaJ33JNme5LYkL1/KA5AkPd6Bc6jzMHBeVd2U5FDgxiRXt2V/UlV/PFw5yYnA2cBzgaOB/5Xkn1bVI+NsuCRptFmv3Kvqnqq6qU0/BGwDjtnLKhuArVX186q6HdgOvHAcjZUkzU2qau6Vk/XAtcDzgN8D3gT8BLiBwdX9/Un+HLi+qv57W+ci4AtVddke29oIbARYt27dC7Zu3Trvxk9PT7NmzZp5rzfs63c9uOB1TzrmsEXteymNo296Zd+MZt+Mtj/2zamnnnpjVU3MtGwuwzIAJFkDfAZ4d1X9JMmHgD8Eqj1eALwFyAyrP+4ZpKq2AFsAJiYmanJycq5NedTU1BQLWW/YmzZdteB1d5yzuH0vpXH0Ta/sm9Hsm9FWWt/M6W6ZJAcxCPZPVNXlAFV1b1U9UlW/BP6aXw297ASOG1r9WODu8TVZkjSbudwtE+AiYFtVfXCo/Kihaq8CvtGmrwDOTnJwkuOBE4Cvjq/JkqTZzGVY5sXAG4CvJ7m5lb0XeF2SkxkMuewA3g5QVbcmuRT4JoM7bd7hnTKStG/NGu5VdR0zj6N/fi/rnA+cv4h2SZIWwU+oSlKHDHdJ6tCcb4XcX61fxK2MktQrr9wlqUOGuyR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdmjXckxyX5MtJtiW5Ncm7WvkRSa5O8u32eHgrT5ILk2xPckuS5y/1QUiSHmsuV+4PA+dV1XOAU4B3JDkR2ARcU1UnANe0eYDTgRPa30bgQ2NvtSRpr2YN96q6p6puatMPAduAY4ANwMWt2sXAWW16A/CxGrgeWJvkqLG3XJI0Uqpq7pWT9cC1wPOAO6pq7dCy+6vq8CRXApur6rpWfg3wB1V1wx7b2sjgyp5169a9YOvWrfNu/PT0NLc/+Mi81xuXk445bNn2PZvp6WnWrFmz3M3YL9k3o9k3o+2PfXPqqafeWFUTMy07cK4bSbIG+Azw7qr6SZKRVWcoe9wzSFVtAbYATExM1OTk5Fyb8qipqSkuuG7XvNcblx3nTC7bvmczNTXFQvp0NbBvRrNvRltpfTOnu2WSHMQg2D9RVZe34nt3D7e0x/ta+U7guKHVjwXuHk9zJUlzMZe7ZQJcBGyrqg8OLboCOLdNnwt8bqj8je2umVOAB6vqnjG2WZI0i7kMy7wYeAPw9SQ3t7L3ApuBS5O8FbgDeE1b9nngDGA78FPgzWNtsSRpVrOGe3tjdNQA+2kz1C/gHYtslyRpEfyEqiR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nq0Jx/Zk+Pt37TVQted8fmM8fYEkl6LK/cJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOjRruCf5SJL7knxjqOwDSe5KcnP7O2No2XuSbE9yW5KXL1XDJUmjzeXK/aPAK2Yo/5OqOrn9fR4gyYnA2cBz2zp/meSAcTVWkjQ3s4Z7VV0L/HiO29sAbK2qn1fV7cB24IWLaJ8kaQEW82Md70zyRuAG4Lyquh84Brh+qM7OVvY4STYCGwHWrVvH1NTUvBswPT3NeSc9Mu/19gcLOd75mJ6eXvJ9rFT2zWj2zWgrrW8WGu4fAv4QqPZ4AfAWIDPUrZk2UFVbgC0AExMTNTk5Oe9GTE1NccF1u+a93v5gxzmTS7r9qakpFtKnq4F9M5p9M9pK65sF3S1TVfdW1SNV9Uvgr/nV0MtO4LihqscCdy+uiZKk+VpQuCc5amj2VcDuO2muAM5OcnCS44ETgK8uromSpPmadVgmySeBSeDIJDuB9wOTSU5mMOSyA3g7QFXdmuRS4JvAw8A7qmplDopL0go2a7hX1etmKL5oL/XPB85fTKMkSYvjJ1QlqUOGuyR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA7NGu5JPpLkviTfGCo7IsnVSb7dHg9v5UlyYZLtSW5J8vylbLwkaWZzuXL/KPCKPco2AddU1QnANW0e4HTghPa3EfjQeJopSZqPWcO9qq4FfrxH8Qbg4jZ9MXDWUPnHauB6YG2So8bVWEnS3By4wPXWVdU9AFV1T5Knt/JjgDuH6u1sZffsuYEkGxlc3bNu3Tqmpqbm3Yjp6WnOO+mRea+3P1jI8c7H9PT0ku9jpbJvRrNvRltpfbPQcB8lM5TVTBWraguwBWBiYqImJyfnvbOpqSkuuG7XvNfbH+w4Z3JJtz81NcVC+nQ1sG9Gs29GW2l9s9BwvzfJUe2q/Sjgvla+EzhuqN6xwN2LaWCv1m+6asHr7th85hhbIqlHC70V8grg3DZ9LvC5ofI3trtmTgEe3D18I0nad2a9ck/ySWASODLJTuD9wGbg0iRvBe4AXtOqfx44A9gO/BR48xK0WZI0i1nDvapeN2LRaTPULeAdi22UJGlx/ISqJHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI6NOsPZKs/6zddteB1d2w+c4wtkbRUvHKXpA555b4CzeXK+7yTHuZNi7hCl7SyeeUuSR0y3CWpQ4a7JHXIcJekDi3qDdUkO4CHgEeAh6tqIskRwKeA9cAO4LVVdf/imilJmo9xXLmfWlUnV9VEm98EXFNVJwDXtHlJ0j60FMMyG4CL2/TFwFlLsA9J0l6kqha+cnI7cD9QwF9V1ZYkD1TV2qE691fV4TOsuxHYCLBu3boXbN26dd77n56e5vYHH1lw+3u27slw78/Gv92Tjjls/Bvdx6anp1mzZs1yN2O/ZN+Mtj/2zamnnnrj0KjJYyz2Q0wvrqq7kzwduDrJt+a6YlVtAbYATExM1OTk5Lx3PjU1xQXX7Zr3eqvBeSc9zAVfH/9n1HacMzn2be5rU1NTLOR8Ww3sm9FWWt8salimqu5uj/cBnwVeCNyb5CiA9njfYhspSZqfBV/aJTkEeEJVPdSmXwb8Z+AK4Fxgc3v83Dgaqv2DXzomrQyLed2+Dvhskt3buaSq/i7J14BLk7wVuAN4zeKbKUmajwWHe1V9F/j1Gcp/BJy2mEZJkhbHT6hKUocMd0nqkOEuSR3yxzq0z3injbTveOUuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQP9ah7u35IyHnnfQwb5rHD4f4QyFaiQx3rQiL+RUnaTVyWEaSOmS4S1KHDHdJ6pDhLkkdMtwlqUPeLSMtocXc5eMtmFqMJbtyT/KKJLcl2Z5k01LtR5L0eEty5Z7kAOAvgN8BdgJfS3JFVX1zKfYnLaXVdo/9Yo93Ma84luuVzlz2O98Pv83VUr1CW6phmRcC26vquwBJtgIbAMNdmqPleFI576SHWWwsLNeT4Wp7Ep5Nqmr8G01eDbyiqt7W5t8AvKiq3jlUZyOwsc0+G7htAbs6EvjhIpvbK/tmNPtmNPtmtP2xb55ZVU+bacFSXblnhrLHPItU1RZgy6J2ktxQVROL2Uav7JvR7JvR7JvRVlrfLNUbqjuB44bmjwXuXqJ9SZL2sFTh/jXghCTHJ3kicDZwxRLtS5K0hyUZlqmqh5O8E/h74ADgI1V16xLsalHDOp2zb0azb0azb0ZbUX2zJG+oSpKWl18/IEkdMtwlqUMrNtxX49cbJDkuyZeTbEtya5J3tfIjklyd5Nvt8fBWniQXtj66Jcnzh7Z1bqv/7STnLtcxjVOSA5L8nyRXtvnjk3ylHeOn2pv7JDm4zW9vy9cPbeM9rfy2JC9fniMZvyRrk1yW5Fvt/PnnnjeQ5D+0/6VvJPlkkid1c95U1Yr7Y/Am7XeAZwFPBP4BOHG527UPjvso4Plt+lDgH4ETgf8GbGrlm4A/atNnAF9g8LmDU4CvtPIjgO+2x8Pb9OHLfXxj6J/fAy4BrmzzlwJnt+kPA/+mTf9b4MNt+mzgU236xHYuHQwc386xA5b7uMbUNxcDb2vTTwTWrvbzBjgGuB148tD58qZezpuVeuX+6NcbVNUvgN1fb9C1qrqnqm5q0w8B2xicoBsY/PPSHs9q0xuAj9XA9cDaJEcBLweurqofV9X9wNXAK/bhoYxdkmOBM4G/afMBXgpc1qrs2S+7++sy4LRWfwOwtap+XlW3A9sZnGsrWpKnAi8BLgKoql9U1QN43sDgjsEnJzkQeApwD52cNys13I8B7hya39nKVo32kvA3gK8A66rqHhg8AQBPb9VG9VOP/fenwO8Dv2zz/wR4oKoebvPDx/jo8bflD7b6PfYLDF7h/gD42zZs9TdJDmGVnzdVdRfwx8AdDEL9QeBGOjlvVmq4z/r1Bj1Lsgb4DPDuqvrJ3qrOUFZ7KV+RkvxL4L6qunG4eIaqNcuyrvplyIHA84EPVdVvALsYDMOMsir6p73HsIHBUMrRwCHA6TNUXZHnzUoN91X79QZJDmIQ7J+oqstb8b3tZTPt8b5WPqqfeuu/FwOvTLKDwRDdSxlcya9tL7fhscf46PG35YcBP6a/ftltJ7Czqr7S5i9jEPar/bz5beD2qvpBVf0/4HLgX9DJebNSw31Vfr1BG9+7CNhWVR8cWnQFsPvOhXOBzw2Vv7Hd/XAK8GB7+f33wMuSHN6uXl7WylakqnpPVR1bVesZnAtfqqpzgC8Dr27V9uyX3f316la/WvnZ7a6I44ETgK/uo8NYMlX1feDOJM9uRacx+PrtVX3eMBiOOSXJU9r/1u5+6eO8We53dBf6x+Ad/X9k8M70+5a7PfvomH+Twcu9W4Cb298ZDMb9rgG+3R6PaPXD4EdTvgN8HZgY2tZbGLzxsx1483If2xj7aJJf3S3zLAb/ZNuBTwMHt/IntfntbfmzhtZ/X+uv24DTl/t4xtgvJwM3tHPnfzC422XVnzfAfwK+BXwD+DiDO166OG/8+gFJ6tBKHZaRJO2F4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI69P8BzwCQNgI3T9oAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "cleanClinicalData.hist(column='overall_survival', bins=20)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7ff46a628850>"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZQAAAD4CAYAAADLhBA1AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3de3hcV3nv8e+r0V3WXfLdjp3YIbEDIcGEck8baBwoMS2hMW1poElTIDkcoKdtUnrSNiVA4LQUCimEJjQEihPCpaYEQiiXUgqJnatjGyeKL7F8lS1ZkiWNpJHe88fessfySBpJs2eP5N/nefRoz5q117x7NNKrtdfaa5u7IyIiMl1FcQcgIiKzgxKKiIjkhBKKiIjkhBKKiIjkhBKKiIjkRHHcAcSpqanJly1bFncYIiIzymOPPXbE3ZtHl5/RCWXZsmVs3rw57jBERGYUM9uTqVynvEREJCeUUEREJCeUUEREJCciTShmttbMdphZi5ndlOH5MjO7L3z+ETNblvbczWH5DjO7fKI2LXCbmT1rZtvN7P1RHpuIiJwqskF5M0sAnwPeCLQCm8xso7tvS6t2LdDh7ivMbD1wO3C1ma0C1gOrgYXAD83s3HCfsdp8F7AEOM/dh81sblTHJiIip4uyh3IJ0OLuO919ANgArBtVZx1wT7j9AHCZmVlYvsHd+919F9AStjdem+8FbnX3YQB3PxzhsYmIyChRJpRFwN60x61hWcY67p4COoHGcfYdr81zCHo3m83se2a2MlNQZnZ9WGdzW1vblA5MREROF2VCsQxlo9fKH6vOZMsByoCku68BvgjcnSkod7/T3de4+5rm5tOuy5GYJAeH+Mov97DjYHfcoYjIFEWZUFoJxjRGLAb2j1XHzIqBWqB9nH3Ha7MV+Ea4/S3gJdM+AsmbTz60g7/69jO844u/pG9gKO5wRGQKokwom4CVZrbczEoJBtk3jqqzEbgm3L4K+JEHd/zaCKwPZ4EtB1YCj07Q5reB3wi3Xw88G9FxSY6lhob55uOtNFeX0d4zwHe3HIg7JBGZgsgSSjgmciPwELAduN/dt5rZrWZ2ZVjtLqDRzFqADwE3hftuBe4HtgHfB25w96Gx2gzb+jjwNjPbAnwMuC6qY5Pc2ryng47eQf72ytUsbajke0ooIjNSpGt5ufuDwIOjym5J204Cbx9j39uA27JpMyw/Brx5miFLDJ544RgArz6niVee3cj3njnA8LBTVJRpyExECpWulJfYPbO/kyUNFdRWlnDJ8ga6kil2HNLgvMhMo4Qisdu6r5MLFtYC8JLFwfftB7riDElEpkAJRWLVnxrihfZeVs6dA8CypipKEqYeisgMpIQisWrt6GPYg0QCUJIo4pzmOTx36HjMkYnIZCmhSKz2HO0B4KzGqhNlL5pfrQscRWYgJRSJ1Z6jvQAsa6w8UXbuvGr2HeujOzkYV1giMgVKKBKrve19VJYmaKgqPVG2IhxP2XWkJ66wRGQKlFAkVoe6ksyvLSdYZDqwtCHorbzQ3htXWCIyBUooEquDXUnmVZefUqaEIjIzKaFIrEZ6KOmqyoppmlPKC0eVUERmEiUUiY27c7irn3k15ac9t7ShUj0UkRlGCUVi094zwMDQMPNqyk57bmlD5YkZYCIyMyihSGwOdfUDMD9TD6WxigOdfQykhvMdlohMkRKKxOZQVxKAebWnJ5TFdRUM+8k6IlL4lFAkNicSSoYeyoK6oGzfsb68xiQiU6eEIrFp6w5OeTXNKT3tuYV1FQAc6FRCEZkplFAkNu29A1SXFVNWnDjtuYW1QULZf0ynvERmCiUUiU17zwD1Vaf3TgAqShPUVZaohyIygyihSGzaewZOWcNrtAW1FRxQD0VkxlBCkdh09I6fUBbWlrO/UwlFZKZQQpHYtB8foL5ynB5KXblOeYnMIEooEpv23gEaM8zwGrGwroJjvYP0DqTyGJWITJUSisSib2CI5ODwuD0UzfQSmVmUUCQWR3uCa1AaqkrGrLMgvIJep71EZoZIE4qZrTWzHWbWYmY3ZXi+zMzuC59/xMyWpT13c1i+w8wun6hNM/tXM9tlZk+GXy+N8thkejp6gtv7NlSdvjDkiBMXN6qHIjIjFEfVsJklgM8BbwRagU1mttHdt6VVuxbocPcVZrYeuB242sxWAeuB1cBC4Idmdm64z3ht/pm7PxDVMUnutPcOAOP3UObVlGMG+9VDEZkRouyhXAK0uPtOdx8ANgDrRtVZB9wTbj8AXGbBvWDXARvcvd/ddwEtYXvZtCkzQHt4ymu8MZTS4iKa5pSphyIyQ0SZUBYBe9Met4ZlGeu4ewroBBrH2XeiNm8zs6fN7FNmlvFcipldb2abzWxzW1vb5I9KcqI9POXVOM4pLwjGUQ5qxWGRGSHKhGIZyjzLOpMtB7gZOA94OdAA/EWmoNz9Tndf4+5rmpubM1WRPOjoGSBRZFSXj3/WdX5NOQd1caPIjBBlQmkFlqQ9XgzsH6uOmRUDtUD7OPuO2aa7H/BAP/AlgtNjUqCO9gxQX1lCUVGm/xFOWlCrixtFZoooE8omYKWZLTezUoJB9o2j6mwErgm3rwJ+5O4elq8PZ4EtB1YCj47XppktCL8b8FbgmQiPTabpWO8AdeOMn4yYX1tBVzJFT78ubhQpdJHN8nL3lJndCDwEJIC73X2rmd0KbHb3jcBdwL1m1kLQM1kf7rvVzO4HtgEp4AZ3HwLI1Gb4kl81s2aC02JPAu+J6thk+rqSg9RWjD3Da8TItSgHu5Kc0zwn6rBEZBoiSygA7v4g8OCoslvStpPA28fY9zbgtmzaDMt/Y7rxSv50J1PjLgw5Yn6YUA51KqGIFDpdKS+x6OobpKZ84h7K/JqRq+U1MC9S6JRQJBbdydSEM7zgZA9FU4dFCp8SiuSdu9OVHKQmizGU8pIE9bpzo8iMoIQiedefGmZwyLPqoUAw00vXoogUPiUUybuuvuAq+WzGUGDkWhQlFJFCp4QiedeVDK4pyb6HUs4hjaGIFDwlFMm7rmTYQ8liDAWCmV5Hjg/QnxqKMiwRmSYlFMm77rCHUjOJHgrA4a7+yGISkelTQpG8m8oYCuhaFJFCp4Qiedd9YgxlsglFU4dFCpkSiuTdyBjKZKYNA5o6LFLglFAk77qTgySKjMrSRFb155QVU11WrKvlRQqcEorkXVdfsOxKcKeB7Myr1Y22RAqdEorkXXcyu4Uh0+niRpHCp4QieZftwpDpdCtgkcKnhCJ51zXFHsrh7iSpoeGIohKR6VJCkbybUg+ltoJhh7bjurhRpFApoUjedfVlt3R9Ol3cKFL4lFAk76bSQ5lXc/JWwCJSmJRQJK+Ghp3u/tSUxlBAPRSRQqaEInl1vH9yS9ePqKssoay4SBc3ihQwJRTJqxMLQ05yDMXMdC2KSIFTQpG8muzS9enm15ZzUAtEihQsJRTJqxM315rkGArAgtoK9VBEClikCcXM1prZDjNrMbObMjxfZmb3hc8/YmbL0p67OSzfYWaXT6LNfzKz41Edk0zPZJeuTzevppzDXf0MD3uuwxKRHIgsoZhZAvgccAWwCniHma0aVe1aoMPdVwCfAm4P910FrAdWA2uBO8wsMVGbZrYGqIvqmGT6To6hTP6U14LacgaGhmnvHch1WCKSA1H2UC4BWtx9p7sPABuAdaPqrAPuCbcfAC6zYAnadcAGd+93911AS9jemG2GyeaTwJ9HeEwyTd0n7oUy+R7KyK2AtaaXSGGKMqEsAvamPW4NyzLWcfcU0Ak0jrPveG3eCGx09wPjBWVm15vZZjPb3NbWNqkDkunrSk5t2jDoWhSRQhdlQsl0s4vRJ7/HqjOpcjNbCLwd+KeJgnL3O919jbuvaW5unqi65Fh3cpCKkgQlicl/9E72UDTTS6QQRZlQWoElaY8XA/vHqmNmxUAt0D7OvmOVXwSsAFrMbDdQaWYtuToQyZ2pLLsyoqmqjOIiUw9FpEBFmVA2ASvNbLmZlRIMsm8cVWcjcE24fRXwI3f3sHx9OAtsObASeHSsNt39u+4+392XufsyoDcc6JcC05Wc/MKQI4qKjHk15bpaXqRATe1fxSy4e8rMbgQeAhLA3e6+1cxuBTa7+0bgLuDesDfRTpAgCOvdD2wDUsAN7j4EkKnNqI5Bcm86PRQYubhRCUWkEEWWUADc/UHgwVFlt6RtJwnGPjLtextwWzZtZqgzZyrxSvS6+gapqyyd8v7za8vZvr8rhxGJSK7oSnnJq+n2UBbUBOt5BWdGRaSQKKFIXk1nDAWCHkrf4BBdfakcRiUiuaCEInnVNd0eSm0FAAe6NHVYpNAooUjeJAeHGEgNT2lhyBHza8sAXS0vUoiUUCRvprN0/Yj5YQ9FCUWk8CihSN6cWLp+GmMoc6vLKDLYd0ynvEQKjRKK5E33NNbxGlGSKGJBbQWtHUooIoVGCUXy5sTS9dMYQwFYXF9Ba0dvLkISkRxSQpG8mc7NtdItaahkb7t6KCKFRglF8ubkGMr0FmhYXF/Boe4k/amhXIQlIjmihCJ5M52ba6VbUl+JO+w/ppleIoVECUXypqsvRZFBVWliWu0srg+mDu9t1ziKSCFRQpG86U4OUl1eQnCX56lb0lAJoJleIgUmq4RiZt8wszebmRKQTNl0F4YcMa+mnJKEsVczvUQKSrYJ4p+B3wOeM7OPm9l5EcYks1RXcnDaU4YBEkXGwjpdiyJSaLJKKO7+Q3f/feBiYDfwsJn9j5m928ym/xdCzgjTXRgy3eL6Co2hiBSYrE9hmVkj8C7gOuAJ4NMECebhSCKTWaerb3pL16dbXFepHopIgcnq30Uz+yZwHnAv8BZ3PxA+dZ+ZbY4qOJldcjWGArCkoYIjx/vpGxiiYpqzxkQkN7L97f6X8Na7J5hZmbv3u/uaCOKSWShXYygAi+tHZnr1snJedU7aFJHpyfaU10cylP0il4HI7DY87BzvT01r6fp0SxrCa1E000ukYIz7221m84FFQIWZXQSMXEBQA1RGHJvMIscHUrhPb+n6dGc1VgGw64gSikihmOjfxcsJBuIXA/+QVt4N/GVEMckslIul69M1VpVSXVbM7iM9OWlPRKZv3N9ud78HuMfM3ubu38hTTDIL5Wrp+hFmxvLmKnYfVUIRKRQTnfL6A3f/CrDMzD40+nl3/4cMu4mcJldL16db1ljF4y905Kw9EZmeiQblq8Lvc4DqDF8iWTnRQ5nm0vXpljVVsf9Yn5axFykQE53y+kL4/W+n0riZrSW4ADJBMPX446OeLwO+DLwMOApc7e67w+duBq4FhoD3u/tD47VpZncBawgmDjwLvMvdj08lbsm97v7cLF2fbnlTJcMerDq8Yq7+vxGJW7aLQ37CzGrMrMTM/tPMjpjZH0ywTwL4HHAFsAp4h5mtGlXtWqDD3VcAnwJuD/ddBawHVgNrgTvMLDFBmx909wvd/SXAC8CN2Ryb5EdXX3DKK1fThiE45QWa6SVSKLK9DuU33b0L+C2gFTgX+LMJ9rkEaHH3ne4+AGwA1o2qsw64J9x+ALjMgrXN1wEbwgsndwEtYXtjthnGR7h/BeBZHpvkQa5urpVueVOQUDTTS6QwZJtQRv4KvAn4mru3Z7HPImBv2uPWsCxjHXdPAZ1A4zj7jtummX0JOEiwTMw/ZQrKzK43s81mtrmtrS2Lw5Bc6EqmKC8porQ4d3dAqKsspb6yhF2a6SVSELL97f6Omf2KYIziP82sGZjo/quZ7qI0utcwVp3Jlgcb7u8GFgLbgaszBeXud7r7Gndf09zcnKmKRGDk5lq5tqypSj0UkQKR7fL1NwGvBNa4+yDQw+mnr0ZrBZakPV4M7B+rjpkVA7VA+zj7Ttimuw8B9wFvm+i4JH9yuXR9uuWNSigihWIy5x/OB642sz8ErgJ+c4L6m4CVZrbczEoJBtk3jqqzEbgm3L4K+JG7e1i+3szKzGw5sBJ4dKw2LbACToyhvAX41SSOTSLW1Ze7hSHTLWuqYn9nkr4BTR0WiVu2y9ffC5wDPEkwjReCU01fHmsfd0+Z2Y3AQwRTfO92961mdiuw2d03AncB95pZC0HPZH2471Yzux/YBqSAG8KeB2O0WURwRX8NwWmxp4D3TuJ9kIjlcun6dGc3BwPzO48cZ/XC2py3LyLZy/Y3fA2wKuw9ZC1c8v7BUWW3pG0ngbePse9twG1ZtjkMvHoysUl+dSUHWVRfkfN2V4bXn7QcVkIRiVu2p7yeAeZHGYjMbt3J3C1dn255UxXFRcZzh3QNq0jcsv0NbwK2mdmjQP9IobtfGUlUMut09UUzy6u0uIhlTVU8e6g7522LyORkm1D+JsogZHZLDg7RnxqmNkf3Qhlt5dw5/OqgEopI3LKdNvxTYDdQEm5vAh6PMC6ZRUZWGs7VzbVGWzmvmj1He0gOaqaXSJyyXcvrjwmWRvlCWLQI+HZUQcns0pUcuRdK7sdQIOihDDvsbNP1KCJxynZQ/gaCWVRdAO7+HDA3qqBkduk8sXR9ND2Uc+cFM72eO6zTXiJxyjah9IeLMQInrmrX4ouSlVzfrXG05U1VJDTTSyR22SaUn5rZXwIVZvZG4OvAd6ILS2aTrnAMpTaHN9dKV1pcxPKmKn51sCuS9kUkO9kmlJuANmAL8CcEFxb+VVRByewSdQ8FYPXCGrbuV0IRiVNW/zK6+7CZfRv4trtrzXeZlBOD8hGNoUCQUP79yf209wzQUFUa2euIyNjG7aGEiy7+jZkdIVhscYeZtZnZLePtJ5Kuqy9FaXER5SWJyF5jZNmVrfs7I3sNERnfRKe8PkAwu+vl7t7o7g3AK4BXm9kHI49OZoXOiFYaTrd6YQ2ATnuJxGiihPKHwDvC2/AC4O47gT8InxOZUFdykJqIBuRH1FWWsqiuQglFJEYTJZQSdz8yujAcR4n2X06ZNaK6F8powcC8TnmJxGWihDIwxedETuhKpiIdkB+xemEtu4700NOfivy1ROR0EyWUC82sK8NXN/DifAQoM19332Bky66ke/HiGtxhyz71UkTiMG5CcfeEu9dk+Kp2d53ykqx0JQcjW2k43UuX1APw+Asdkb+WiJxuMveUF5k0dw9meeUhoTRUlXJ2UxWP7zkW+WuJyOmUUCRSycFhBoc8L4PyABctreeJFzqY5N2qRSQHlFAkUievko9+DAXgZWfVc7RngD1He/PyeiJykhKKRCof63ilu/isOkDjKCJxUEKRSOVjHa90K+dWM6esmMf2KKGI5JsSikSqq29k6fr8JJREkfGys+p5ZFd7Xl5PRE5SQpFIRX3730xes6KJlsPHOdiZzNtrikjECcXM1prZDjNrMbObMjxfZmb3hc8/YmbL0p67OSzfYWaXT9SmmX01LH/GzO42M10nUwCivv1vJq9a0QjAz1tOWzVIRCIUWUIxswTwOeAKYBXwDjNbNaratUCHu68APgXcHu67ClgPrAbWAneYWWKCNr8KnEdwBX8FcF1UxybZ6+zN76A8wPnza2ioKlVCEcmzKHsolwAt7r4zvB/9BmDdqDrrgHvC7QeAy8zMwvIN7t4frnTcErY3Zpvu/qCHgEeBxREem2Spo3eQqtIEpcX5O7taVGS86pxGfv78EV2PIpJHUf6WLwL2pj1uDcsy1nH3FNAJNI6z74Rthqe63gl8P1NQZna9mW02s81tbbr5ZNSO9Q1QV5n/Oyi+ekUTh7r6ee7w8by/tsiZKsqEYhnKRv+7OFadyZanuwP4L3f/Waag3P1Od1/j7muam5szVZEcOtY7SF1l/oezLn1R8LN9eNuhvL+2yJkqyoTSCixJe7wY2D9WHTMrBmqB9nH2HbdNM/troBn4UE6OQKbtWO8A9TH0UBbUVnDhkjoe2now768tcqaKMqFsAlaa2XIzKyUYZN84qs5G4Jpw+yrgR+EYyEZgfTgLbDmwkmBcZMw2zew64HKCO0wOR3hcMgnHegepjaGHAnD56nk83drJvmN9sby+yJkmsoQSjoncCDwEbAfud/etZnarmV0ZVrsLaDSzFoJexU3hvluB+4FtBGMhN7j70Fhthm19HpgH/MLMnjSzW6I6NsleR+8A9TEllLWr5wPwA/VSRPIi0qvN3P1B4MFRZbekbSeBt4+x723Abdm0GZbn78o5ycrwcLB0fRynvADObp7DufPm8OCWA7z71ctjiUHkTKIr5SUy3ckUw56/ZVcyeetFi9i0u4OdbZrtJRI1JRSJzLG+AYDYeigAb7t4MUUGDzzWGlsMImcKJRSJTEd4lXwc04ZHzKsp59IXzeWBx1oZSGmuhkiUlFAkMh29QQ8ljgsb073zlWdxuLuf7zw1eta6iOSSEopEZmQdr7hmeY249NxmXjSvmi/81/NaikUkQkooEplC6aGYGe+59GyePXSc/3j6QKyxiMxmSigSmZExlDhneY248sJFnL+ghtu//yuSg0NxhyMyKymhSGQ6eweoKS8mUZRpCbb8ShQZ//fN59Pa0cfnftwSdzgis5ISikSmo3eQ+qp4T3ele9WKJt528WLu+MnzPPGC7jkvkmtKKBKZjpgWhhzPX1+5ivk15dzw1cc53K1bBIvkkhKKRKatu5+mOWVxh3GKmvISvvDOl9HRO8if3PuYxlNEckgJRSJztGeApjmF1UMBuGBRLZ+6+kKeeOEYf3r/UwwPayqxSC4ooUgkhoed9p4BGgswoQCsvWABH37T+Xx3ywE+9r3tcYcjMitohV6JRGffIEPDTmNVYZ3ySnfda5fT2tHLF3+2i0V1FbxLKxKLTIsSikTiaE8/QMH2UCC44PGWt6xmf2eSv/2PbSyoq+Dy8B4qIjJ5OuUlkThyPLhKvrnABuVHSxQZn1l/ES9ZXMf7v/YEj2s6sciUKaFIJI6GCaWxwBMKQEVpgruuWcO8mnLe95XH6egZiDskkRlJCUUiceR44Z/yStc0p4w7fv9ijvb08+ffeFqLSIpMgRKKROLo8X7M4r251mRdsKiWv1h7Hg9vO8RXfrkn7nBEZhwlFInEkZ4BGipLC2Idr8n4o1cv59IXNfN3393O9gNdcYcjMqMooUgkjh4vvKvks1FUZPy/t19IbUUJ/+trT9A7kIo7JJEZQwlFInH0eOFe1DiRpjllfOp3X8rzbce59Tvb4g5HZMZQQpFIHO0ZmBEzvMbympVNvO/Sc9iwaS8bdetgkawooUjOuTuHupIFfw3KRD7whnN52Vn13PSNp9nS2hl3OCIFL9KEYmZrzWyHmbWY2U0Zni8zs/vC5x8xs2Vpz90clu8ws8snatPMbgzL3MyaojwuGV93f4regSHm187shFKSKOKff/9i6itLefe/bmJve2/cIYkUtMgSipklgM8BVwCrgHeY2apR1a4FOtx9BfAp4PZw31XAemA1sBa4w8wSE7T5c+ANgOZ7xuxQZ3CfkXk15TFHMn1za8q5549ezkBqiN/7l1+y+0hP3CGJFKwoeyiXAC3uvtPdB4ANwLpRddYB94TbDwCXmZmF5Rvcvd/ddwEtYXtjtunuT7j77giPR7J0sCtIKPNnQUIBWDG3mnuvfQXHkymu+vwv2Lpfp79EMokyoSwC9qY9bg3LMtZx9xTQCTSOs282bUrMDoY9lPm1syOhAFy4pI6vv+eVlCaM3/38L/jxjsNxhyRScKJMKJmuaBu9nsVYdSZbnn1QZteb2WYz29zW1jaZXSVLh7pmzymvdCvmVvPN972asxqruO6ezfzbIy/EHZJIQYkyobQCS9IeLwZGz788UcfMioFaoH2cfbNpc1zufqe7r3H3Nc3NzZPZVbJ0sCtJbUUJ5SWJuEPJufm15dz/nlfyupVN/OW3tvCx723Xul8ioSgTyiZgpZktN7NSgkH2jaPqbASuCbevAn7kwW/nRmB9OAtsObASeDTLNiVmBzv7Z834SSZzyor54h+u4fdfsZQv/HQnf/+DZ+MOSaQgRHaDLXdPmdmNwENAArjb3bea2a3AZnffCNwF3GtmLQQ9k/XhvlvN7H5gG5ACbnD3IQimB49uMyx/P/DnwHzgaTN70N2vi+r4ZGyHupLMm0XjJ5kUJ4r4yFsvYGjY+eyPW1jaWMnvrlky8Y4is5idyd31NWvW+ObNm+MOY9a56NYfcMWLF/DR335x3KFEbnBomHd/aRObdrfz3fe/hhVzq+MOSSRyZvaYu68ZXa4r5SWnjven6OgdZHF9Rdyh5EVJooh/uPpCqsqK+cB9TzKQGo47JJHYKKFITrV2BFeTL6mvjDmS/JlbXc7HfufFPLOvi7t/vivucERio4QiObW3vQ+AJQ1nTkIBuHz1fN5w/jw+85/PnbgOR+RMo4QiOXWyh3JmnPJKd8tvrSI17Hz0we1xhyISCyUUyam97X1UlCRoqJqZ90KZjqWNlbzn9eew8an9PLqrPe5wRPJOCUVyam9HL0saKgiWZDvzvPf157Cgtpy/+49tDA+fuTMo5cykhCI5tbPtOMsaq+IOIzYVpQn+Yu15bNnXyTceb407HJG8UkKRnBkcGmbP0V5WzJ0TdyixuvLChbx0SR2feGgHPf26J72cOZRQJGf2HO0lNexnfEIpKjJuecsq2rr7ueMnLXGHI5I3SiiSMy2HjwNwTvOZnVAALl5az1tfupAv/myX7vQoZwwlFMmZ59vChHKG91BG/Pna8ygy+Pj3fhV3KCJ5oYQiOfPcoW4W1JYzpyyyNUdnlIV1Fbz39Sv47pYD/GDrwbjDEYmcEorkzDP7u1i9sCbuMArKey89h9ULa7jpm1s43K0r6GV2U0KRnDjen+L5tuO8eFFd3KEUlNLiIj69/qX09Ke48atP0J8aijskkcgooUhObN3XiTu8ZHFt3KEUnBVzq/nk2y/k0d3tfOi+pxgc0orEMjvpZLfkxJZ9nQBcsEgJJZMrL1zI4a4kH/nudnoGUvzj1S+lrvLMW55GZjf1UCQnHt3VzuL6Cpqry+IOpWBd99qz+djvvJiftxzhik//jJ/sOBx3SCI5pYQi05YaGuYXO4/ymhVNcYdS8N5xyVK+8d5XUVGS4F1f2sS7vvQoLYe74w5LJCeUUGTatuzrpDuZ4lVKKFl5yeI6vveB10GvmoQAAAujSURBVPLhN53PY3s6uPwff8ZffXsLR473xx2ayLQooci0/WRHG2bwqnMa4w5lxigrTvDHrzubn/7Zr/MHr1jK1x7dy6Wf/Al3/KRFM8FkxlJCkWlxd77z1H4uWdZA0xyNn0xWQ1Upf7vuAn7wwdfxa2c38onv7+CKf/wZP3uuLe7QRCZNCUWmZcu+TnYe6eGtFy2KO5QZ7ZzmOfzLNWv413e/nGF33nnXo9zw1cc50NkXd2giWVNCkWm56793UVma4E0XLIg7lFnh0hfN5fsfeB0feuO5/HD7IS77+59y5389r2tXZEZQQpEpaznczXee2s87f+0saitL4g5n1igvSfD+y1by8Adfz6+d3chHH/wVb/7Mz3jgsVaO6/4qUsDM/cy9TemaNWt88+bNcYcxIw0ODfP2z/+C3Ud7+OGHXq/xk4i4Oz/cfpiPPridXUd6SBQZ582vZkl9JU3VpdSUlzCnvJjqsmKqy0uoLi9mWVMVyxqrSBSdmbdhluiZ2WPuvmZ0eaRXypvZWuDTQAL4F3f/+Kjny4AvAy8DjgJXu/vu8LmbgWuBIeD97v7QeG2a2XJgA9AAPA68090Hojy+M1XvQIo/vf8pntx7jM/+3kVKJhEyM964ah5vOH8uj+3p4Cc72niq9RjPtx3nkV39dCdTpDLcu76yNMF586u5YFEtFyysZfWiGs6dV01JYuonJQZSw3QnBxkYdfrNOJm4bFQOszEe2KnPnLKfnVJ+8tGwO6khZ3BomMGhYVLDjjvUVBRTW1FCRUnilPqSf5H1UMwsATwLvBFoBTYB73D3bWl13ge8xN3fY2brgd9296vNbBXwNeASYCHwQ+DccLeMbZrZ/cA33X2DmX0eeMrd/3m8GNVDyV7fwBB72nv47+eOcM8vdrO3vY+/evP5XPfas+MO7Yzm7vSnhulOpuhODtLZN0jL4eNsO9DF1n1dbDvQdeI0WWmiiLMaK5lbU8bc6nIqSxOUJIooLS5icGiY5OAQfQND9A4M0Z1Mcbw/aPN4f4quZIqBVGGP41SWJlhSX8mShkqWNlSytKGChXXB6g0jX2XFibjDnBXi6KFcArS4+84wgA3AOmBbWp11wN+E2w8An7XgX4x1wAZ37wd2mVlL2B6Z2jSz7cBvAL8X1rknbHfchDJVH/7WFh7Z1c5IMj4lJfsp306rk56/PSwdKcuU28fa39Ne1U97zVHBZKxzeuwn4zj9NdPP3V+4pI7b3/YSXnWOLmSMm5lRXpKgvCRxYtmbi5bWn3h+eNjZ097LM/s6eWZfJ3uO9nK4O8mm3e30DQwxMDTMQGqY0kQR5aUJKkqCr+ryYprmlLKsqYrqE6fUgtNqpcVFJ3oRmT4/Jx5n+PyN3mf0k+O1B1CSKKI4YZQkjOKiIsygO5niWO8gh7uT7G3vo7Wjl/95/gi9A6dfz1OSMMqKE5SXFFFWnKA4ERzJyPGYpfWdbIzyWeKua17O0sbKnLYZZUJZBOxNe9wKvGKsOu6eMrNOoDEs/+WofUfmpWZqsxE45u6pDPVPYWbXA9cDLF26dHJHFFpYV8GL5lWHDZ7ybeQ1TimzUXXSu+U2amPkY5vpFICNV2dUQ6Nfc7z9xzrFkF6naU4Zi+oqeNlZ9SxpyO2HUKJTVGQsb6pieVMVb7lwYdzh5I27c7RngIOdSQ53J2nr7qetu5/egSGSg8P0p4LvQ8PDp/zjlP4PV/r/ZX56GpzxSotzPycryoSSKaGP/qmMVWes8kzvwHj1Ty90vxO4E4JTXpnqTOSGX18xld1EJE/MjKY5ZeH4nlbAzpcopw23AkvSHi8G9o9Vx8yKCX7y7ePsO1b5EaAubGOs1xIRkQhFmVA2ASvNbLmZlQLrgY2j6mwErgm3rwJ+5MEJ/I3AejMrC2dvrQQeHavNcJ8fh20QtvnvER6biIiMEtkpr3BM5EbgIYIpvne7+1YzuxXY7O4bgbuAe8NB93aCBEFY736CAfwUcIO7DwFkajN8yb8ANpjZR4AnwrZFRCRPdGGjpg2LiEzKWNOGtfSKiIjkhBKKiIjkhBKKiIjkhBKKiIjkxBk9KG9mbcAeoIngWpaZYqbFCzMvZsUbvZkWs+I96Sx3bx5deEYnlBFmtjnTjIVCNdPihZkXs+KN3kyLWfFOTKe8REQkJ5RQREQkJ5RQAnfGHcAkzbR4YebFrHijN9NiVrwT0BiKiIjkhHooIiKSE0ooIiKSE7M+oZjZ35jZPjN7Mvx6U9pzN5tZi5ntMLPL08rXhmUtZnZTWvlyM3vEzJ4zs/vCJfSjiPmTZvYrM3vazL5lZnVh+TIz60s7ls+n7fMyM9sSxvyZ8FbKmFmDmT0cxvywmdWP9boRHUvG9zLfzGyJmf3YzLab2VYz+99hec4+HxHFvTv8uT5pZpvDsow/Uwt8JozraTO7OK2da8L6z5nZNWO93jRjfVHa+/ikmXWZ2QcK7T02s7vN7LCZPZNWlrP3dKzfxRzHW5h/I9x9Vn8R3Fv+/2QoXwU8BZQBy4HnCZbET4TbZwOlYZ1V4T73A+vD7c8D740o5t8EisPt24Hbw+1lwDNj7PMo8EqCu1d+D7giLP8EcFO4fdNIW3l678d8L2P4HCwALg63q4Fnw89Azj4fEcW9G2gaVZbxZwq8KfzZG/BrwCNheQOwM/xeH27X5+FnfxA4q9DeY+B1wMXpv0u5fE/H+l3McbwF+Tdi1vdQxrEO2ODu/e6+C2gBLgm/Wtx9p7sPABuAdWE2/w3ggXD/e4C3RhGYu//A3VPhw18S3IFyTGa2AKhx91948Kn4clps68JYIcKYx5Dxvczj65/g7gfc/fFwuxvYDiwaZ5dJfT6ijT5jbJl+puuAL3vglwR3MV0AXA487O7t7t4BPAysjTjGy4Dn3X3POHVieY/d/b8I7r80OpZpv6cT/C7mLN5C/RtxpiSUG8Ou4d1p3blFwN60Oq1h2VjljcCxtB/iSHnU/ojgv4kRy83sCTP7qZm9NixbFMYzIj22ee5+AII/qsDcqANOM9Z7GSszWwZcBDwSFuXi8xEVB35gZo+Z2fVh2Vg/00KJGYKb5X0t7XEhv8eQu/d0vN/FqBTM34hZkVDM7Idm9kyGr3XAPwPnAC8FDgB/P7JbhqZ8CuVRxDxS58MEd6z8alh0AFjq7hcBHwL+zcxqch1bDhVcXGY2B/gG8AF37yJ3n4+ovNrdLwauAG4ws9eNU7cgYrZgbPFK4OthUaG/x+OJ9e/ERArtb0RktwDOJ3d/Qzb1zOyLwH+ED1uBJWlPLwb2h9uZyo8QdHeLw15Kev2cxxwO8v0WcFnYRcXd+4H+cPsxM3seODc8lvQub3psh8xsgbsfCLu9h6ca8xSM9x7nnZmVECSTr7r7NwHc/VDa89P5fETC3feH3w+b2bcITgeN9TMdK+ZW4NJR5T+JKmaC5Pf4yHtb6O9xKFfv6Xi/izlVkH8jpjNYNBO+gAVp2x8kOGcLsJpTBwR3EgwGFofbyzk5ILg63OfrnDoo/76IYl4LbAOaR5U3A4lw+2xgH9AQPt5EMGg4MuD2prD8k5w64PaJPL73Y76XMXwOjOC88T9G9fmIIOYqoDpt+3/Cz0bGnynwZk4dQH40LG8AdhEMHteH2w0RvtcbgHcX8nvMqMHrXL6nY/0u5jjegvwbkfdf7Hx/AfcCW4CngY2jPtwfJphNsoO0mRgEMzueDZ/7cFr52QQzJVoIkktZRDG3EJyffTL8+nxY/jZga/gL9jjwlrR91gDPhDF/lpOrIDQC/wk8F36P7A/JGMeS8b2M4XPwGoIu/tNp7+ubcvn5iCDms8Of9VPhz/3D4/1Mwz8Unwvj2gKsSWvrj8LPVQtpf+wjiLkSOArUppUV1HtMMLZzABgk+M/92ly+p2P9LuY43oL8G6GlV0REJCdmxaC8iIjETwlFRERyQglFRERyQglFRERyQglFRERyQglFRERyQglFRERy4v8DtJQa1TB3AN8AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "cleanClinicalData['overall_survival'].plot.kde()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>MIR7-1</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4548</td>\n",
       "      <td>0.2310</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>-0.4737</td>\n",
       "      <td>-0.4817</td>\n",
       "      <td>0.4142</td>\n",
       "      <td>-0.4366</td>\n",
       "      <td>-0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>0.3420</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4047.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>-0.4313</td>\n",
       "      <td>-0.4692</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.3972</td>\n",
       "      <td>-0.2806</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>0.4373</td>\n",
       "      <td>-0.4527</td>\n",
       "      <td>-0.4643</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>0.3857</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4005.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.3C.AALJ</td>\n",
       "      <td>-0.4285</td>\n",
       "      <td>-0.4697</td>\n",
       "      <td>0.2664</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>-0.0637</td>\n",
       "      <td>-0.4783</td>\n",
       "      <td>0.3738</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4416</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>0.3319</td>\n",
       "      <td>0.1929</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1474.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.3C.AALK</td>\n",
       "      <td>-0.4043</td>\n",
       "      <td>-0.4714</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>-0.0440</td>\n",
       "      <td>-0.4827</td>\n",
       "      <td>0.3704</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4558</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.3457</td>\n",
       "      <td>0.2445</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1448.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.4H.AAAK</td>\n",
       "      <td>-0.4083</td>\n",
       "      <td>-0.4651</td>\n",
       "      <td>0.1717</td>\n",
       "      <td>0.3406</td>\n",
       "      <td>-0.0880</td>\n",
       "      <td>-0.4788</td>\n",
       "      <td>0.4143</td>\n",
       "      <td>-0.4442</td>\n",
       "      <td>-0.4675</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.3542</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>NaN</td>\n",
       "      <td>348.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>762</td>\n",
       "      <td>TCGA.WT.AB44</td>\n",
       "      <td>-0.3903</td>\n",
       "      <td>-0.4683</td>\n",
       "      <td>0.2731</td>\n",
       "      <td>0.3682</td>\n",
       "      <td>-0.1018</td>\n",
       "      <td>-0.4842</td>\n",
       "      <td>0.3842</td>\n",
       "      <td>-0.4408</td>\n",
       "      <td>-0.4601</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.3322</td>\n",
       "      <td>0.2746</td>\n",
       "      <td>-0.0284</td>\n",
       "      <td>-0.0284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>883.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>763</td>\n",
       "      <td>TCGA.XX.A899</td>\n",
       "      <td>-0.4500</td>\n",
       "      <td>-0.4721</td>\n",
       "      <td>0.2597</td>\n",
       "      <td>0.4152</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4211</td>\n",
       "      <td>-0.4391</td>\n",
       "      <td>-0.4622</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.3908</td>\n",
       "      <td>0.4161</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>-0.2598</td>\n",
       "      <td>467.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>764</td>\n",
       "      <td>TCGA.XX.A89A</td>\n",
       "      <td>-0.4237</td>\n",
       "      <td>-0.4649</td>\n",
       "      <td>0.1951</td>\n",
       "      <td>0.3586</td>\n",
       "      <td>-0.4071</td>\n",
       "      <td>-0.4862</td>\n",
       "      <td>0.4494</td>\n",
       "      <td>-0.4609</td>\n",
       "      <td>-0.4639</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.4054</td>\n",
       "      <td>0.4089</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>-0.2470</td>\n",
       "      <td>488.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>765</td>\n",
       "      <td>TCGA.Z7.A8R5</td>\n",
       "      <td>-0.4197</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>0.1128</td>\n",
       "      <td>0.3803</td>\n",
       "      <td>-0.1014</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>-0.4283</td>\n",
       "      <td>-0.4676</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.3571</td>\n",
       "      <td>0.2107</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3287.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>766</td>\n",
       "      <td>TCGA.Z7.A8R6</td>\n",
       "      <td>-0.4682</td>\n",
       "      <td>-0.4576</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3621</td>\n",
       "      <td>-0.4837</td>\n",
       "      <td>-0.4846</td>\n",
       "      <td>0.4343</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>-0.4747</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.3632</td>\n",
       "      <td>0.4219</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>-0.4183</td>\n",
       "      <td>3256.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>767 rows × 20109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      attrib_name    RBL2   VDAC3   ACTN1  ATP2A1   SFRP1   NIPA2  MAN1B1  \\\n",
       "0    TCGA.3C.AAAU -0.4444 -0.4548  0.2310  0.3568 -0.4737 -0.4817  0.4142   \n",
       "1    TCGA.3C.AALI -0.4313 -0.4692  0.2859  0.3972 -0.2806 -0.4740  0.4373   \n",
       "2    TCGA.3C.AALJ -0.4285 -0.4697  0.2664  0.3482 -0.0637 -0.4783  0.3738   \n",
       "3    TCGA.3C.AALK -0.4043 -0.4714  0.2174  0.3533 -0.0440 -0.4827  0.3704   \n",
       "4    TCGA.4H.AAAK -0.4083 -0.4651  0.1717  0.3406 -0.0880 -0.4788  0.4143   \n",
       "..            ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "762  TCGA.WT.AB44 -0.3903 -0.4683  0.2731  0.3682 -0.1018 -0.4842  0.3842   \n",
       "763  TCGA.XX.A899 -0.4500 -0.4721  0.2597  0.4152  0.0244 -0.4843  0.4211   \n",
       "764  TCGA.XX.A89A -0.4237 -0.4649  0.1951  0.3586 -0.4071 -0.4862  0.4494   \n",
       "765  TCGA.Z7.A8R5 -0.4197 -0.4688  0.1128  0.3803 -0.1014 -0.4843  0.4244   \n",
       "766  TCGA.Z7.A8R6 -0.4682 -0.4576  0.3352  0.3621 -0.4837 -0.4846  0.4343   \n",
       "\n",
       "     TSEN34  LRRC16A  ...   BAGE2   BAGE4   BAGE3  MIR637  LOC100130932  \\\n",
       "0   -0.4366  -0.4610  ... -0.0420 -0.0420 -0.0420  0.3420        0.2383   \n",
       "1   -0.4527  -0.4643  ... -0.0640 -0.0640 -0.0640  0.3857        0.3783   \n",
       "2   -0.4430  -0.4416  ... -0.0873 -0.0873 -0.0873  0.3319        0.1929   \n",
       "3   -0.4430  -0.4558  ...  0.1426  0.1426  0.1426  0.3457        0.2445   \n",
       "4   -0.4442  -0.4675  ...  0.0218  0.0218  0.0218  0.3542        0.3128   \n",
       "..      ...      ...  ...     ...     ...     ...     ...           ...   \n",
       "762 -0.4408  -0.4601  ...  0.2111  0.2111  0.2111  0.3322        0.2746   \n",
       "763 -0.4391  -0.4622  ...  0.1998  0.1998  0.1998  0.3908        0.4161   \n",
       "764 -0.4609  -0.4639  ...  0.1236  0.1236  0.1236  0.4054        0.4089   \n",
       "765 -0.4283  -0.4676  ...  0.2220  0.2220  0.2220  0.3571        0.2107   \n",
       "766 -0.4688  -0.4747  ... -0.0479 -0.0479 -0.0479  0.3632        0.4219   \n",
       "\n",
       "     GOLGA8F  GOLGA8G  MIR7-1  overall_survival  status  \n",
       "0    -0.1168  -0.1168     NaN            4047.0     0.0  \n",
       "1    -0.1271  -0.1271     NaN            4005.0     0.0  \n",
       "2     0.0066   0.0066     NaN            1474.0     0.0  \n",
       "3    -0.0071  -0.0071     NaN            1448.0     0.0  \n",
       "4    -0.0929  -0.0929     NaN             348.0     0.0  \n",
       "..       ...      ...     ...               ...     ...  \n",
       "762  -0.0284  -0.0284     NaN             883.0     0.0  \n",
       "763  -0.0702  -0.0702 -0.2598             467.0     0.0  \n",
       "764  -0.1210  -0.1210 -0.2470             488.0     0.0  \n",
       "765   0.0495   0.0495     NaN            3287.0     0.0  \n",
       "766  -0.1819  -0.1819 -0.4183            3256.0     0.0  \n",
       "\n",
       "[767 rows x 20109 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylationData.merge(cleanClinicalData[['attrib_name','overall_survival', 'status']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Taking transpose of  RNA Seq data\n",
    "\n",
    "pd.read_csv('RNASeq.csv').set_index('attrib_name').T.to_csv('RNASeqoutput.csv',index='none')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2BP1</th>\n",
       "      <th>A2LD1</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAA1</th>\n",
       "      <th>...</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>psiTPTE22</th>\n",
       "      <th>tAKR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>7.6300</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.6999</td>\n",
       "      <td>1.2501</td>\n",
       "      <td>12.5017</td>\n",
       "      <td>6.1136</td>\n",
       "      <td>3.2655</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>...</td>\n",
       "      <td>7.0289</td>\n",
       "      <td>9.9784</td>\n",
       "      <td>10.6965</td>\n",
       "      <td>8.0196</td>\n",
       "      <td>10.2400</td>\n",
       "      <td>11.7765</td>\n",
       "      <td>10.8887</td>\n",
       "      <td>10.2064</td>\n",
       "      <td>1.4454</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>7.8971</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.1672</td>\n",
       "      <td>2.4196</td>\n",
       "      <td>12.8866</td>\n",
       "      <td>7.3101</td>\n",
       "      <td>0.6265</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.9264</td>\n",
       "      <td>8.8125</td>\n",
       "      <td>10.3924</td>\n",
       "      <td>7.6401</td>\n",
       "      <td>9.2398</td>\n",
       "      <td>12.4267</td>\n",
       "      <td>10.3659</td>\n",
       "      <td>8.6715</td>\n",
       "      <td>9.8573</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.3C.AALJ</td>\n",
       "      <td>8.7287</td>\n",
       "      <td>0.9310</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>7.3422</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>13.1101</td>\n",
       "      <td>9.1671</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.1842</td>\n",
       "      <td>9.0634</td>\n",
       "      <td>9.5884</td>\n",
       "      <td>8.3786</td>\n",
       "      <td>9.0585</td>\n",
       "      <td>12.4146</td>\n",
       "      <td>9.8825</td>\n",
       "      <td>8.9958</td>\n",
       "      <td>5.1842</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.3C.AALK</td>\n",
       "      <td>7.5851</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>5.9888</td>\n",
       "      <td>1.4087</td>\n",
       "      <td>13.4201</td>\n",
       "      <td>8.9870</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.8080</td>\n",
       "      <td>8.7772</td>\n",
       "      <td>9.7564</td>\n",
       "      <td>7.4629</td>\n",
       "      <td>9.2488</td>\n",
       "      <td>12.4748</td>\n",
       "      <td>9.6113</td>\n",
       "      <td>9.4551</td>\n",
       "      <td>6.0792</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.4H.AAAK</td>\n",
       "      <td>8.0762</td>\n",
       "      <td>0.5115</td>\n",
       "      <td>2.2720</td>\n",
       "      <td>7.2796</td>\n",
       "      <td>2.1389</td>\n",
       "      <td>13.2268</td>\n",
       "      <td>8.4226</td>\n",
       "      <td>0.5115</td>\n",
       "      <td>0.5115</td>\n",
       "      <td>...</td>\n",
       "      <td>5.6420</td>\n",
       "      <td>8.7322</td>\n",
       "      <td>10.0373</td>\n",
       "      <td>3.9110</td>\n",
       "      <td>9.6013</td>\n",
       "      <td>11.9811</td>\n",
       "      <td>9.7020</td>\n",
       "      <td>9.7858</td>\n",
       "      <td>7.5564</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>TCGA.5L.AAT0</td>\n",
       "      <td>7.6779</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.6659</td>\n",
       "      <td>6.8142</td>\n",
       "      <td>1.1198</td>\n",
       "      <td>13.5902</td>\n",
       "      <td>8.5358</td>\n",
       "      <td>2.3523</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.1217</td>\n",
       "      <td>9.3320</td>\n",
       "      <td>10.1250</td>\n",
       "      <td>4.0737</td>\n",
       "      <td>9.2934</td>\n",
       "      <td>12.0130</td>\n",
       "      <td>9.8504</td>\n",
       "      <td>10.0536</td>\n",
       "      <td>6.4272</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>TCGA.5L.AAT1</td>\n",
       "      <td>7.7299</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.6861</td>\n",
       "      <td>2.3434</td>\n",
       "      <td>13.9114</td>\n",
       "      <td>8.0457</td>\n",
       "      <td>0.8600</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.5181</td>\n",
       "      <td>9.1524</td>\n",
       "      <td>9.9507</td>\n",
       "      <td>5.7758</td>\n",
       "      <td>9.0069</td>\n",
       "      <td>12.1095</td>\n",
       "      <td>10.4641</td>\n",
       "      <td>9.1545</td>\n",
       "      <td>5.8793</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>TCGA.5T.A9QA</td>\n",
       "      <td>8.3001</td>\n",
       "      <td>0.6069</td>\n",
       "      <td>0.6069</td>\n",
       "      <td>7.8742</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>11.6364</td>\n",
       "      <td>8.6886</td>\n",
       "      <td>2.2207</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>4.8956</td>\n",
       "      <td>8.1653</td>\n",
       "      <td>9.6629</td>\n",
       "      <td>1.6286</td>\n",
       "      <td>8.3338</td>\n",
       "      <td>11.6541</td>\n",
       "      <td>10.1242</td>\n",
       "      <td>8.7876</td>\n",
       "      <td>2.2207</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>TCGA.A1.A0SB</td>\n",
       "      <td>5.6496</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.4607</td>\n",
       "      <td>5.2816</td>\n",
       "      <td>2.0543</td>\n",
       "      <td>14.9489</td>\n",
       "      <td>8.4027</td>\n",
       "      <td>0.9268</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.5993</td>\n",
       "      <td>9.0236</td>\n",
       "      <td>10.4685</td>\n",
       "      <td>4.3485</td>\n",
       "      <td>10.4153</td>\n",
       "      <td>12.5952</td>\n",
       "      <td>10.9161</td>\n",
       "      <td>10.4890</td>\n",
       "      <td>9.1120</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>TCGA.A1.A0SD</td>\n",
       "      <td>7.1629</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.4123</td>\n",
       "      <td>6.0846</td>\n",
       "      <td>3.1058</td>\n",
       "      <td>14.5383</td>\n",
       "      <td>8.5628</td>\n",
       "      <td>0.7326</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.6039</td>\n",
       "      <td>9.1781</td>\n",
       "      <td>10.2606</td>\n",
       "      <td>5.0910</td>\n",
       "      <td>9.7634</td>\n",
       "      <td>11.7979</td>\n",
       "      <td>10.3219</td>\n",
       "      <td>10.2247</td>\n",
       "      <td>6.4432</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>TCGA.A1.A0SE</td>\n",
       "      <td>7.5986</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.1685</td>\n",
       "      <td>0.9907</td>\n",
       "      <td>13.9412</td>\n",
       "      <td>8.0619</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.5915</td>\n",
       "      <td>9.5070</td>\n",
       "      <td>9.9941</td>\n",
       "      <td>5.8726</td>\n",
       "      <td>9.9093</td>\n",
       "      <td>11.5550</td>\n",
       "      <td>9.8570</td>\n",
       "      <td>10.0721</td>\n",
       "      <td>9.7601</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>TCGA.A1.A0SF</td>\n",
       "      <td>8.3532</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.5144</td>\n",
       "      <td>7.3429</td>\n",
       "      <td>2.1464</td>\n",
       "      <td>13.9279</td>\n",
       "      <td>8.7887</td>\n",
       "      <td>0.8928</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.2310</td>\n",
       "      <td>9.0617</td>\n",
       "      <td>10.0423</td>\n",
       "      <td>6.5801</td>\n",
       "      <td>9.7850</td>\n",
       "      <td>12.3838</td>\n",
       "      <td>9.8694</td>\n",
       "      <td>8.9918</td>\n",
       "      <td>5.7351</td>\n",
       "      <td>0.8928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>TCGA.A1.A0SG</td>\n",
       "      <td>7.4080</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.4114</td>\n",
       "      <td>6.8659</td>\n",
       "      <td>1.4854</td>\n",
       "      <td>13.6066</td>\n",
       "      <td>9.2774</td>\n",
       "      <td>0.7824</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.6427</td>\n",
       "      <td>8.7377</td>\n",
       "      <td>10.1119</td>\n",
       "      <td>1.2869</td>\n",
       "      <td>9.4589</td>\n",
       "      <td>11.5121</td>\n",
       "      <td>10.7278</td>\n",
       "      <td>9.1561</td>\n",
       "      <td>4.7312</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>TCGA.A1.A0SH</td>\n",
       "      <td>7.5024</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.4449</td>\n",
       "      <td>7.0378</td>\n",
       "      <td>1.2896</td>\n",
       "      <td>13.7730</td>\n",
       "      <td>8.4492</td>\n",
       "      <td>1.0590</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.4718</td>\n",
       "      <td>9.1851</td>\n",
       "      <td>9.6482</td>\n",
       "      <td>4.3829</td>\n",
       "      <td>10.4033</td>\n",
       "      <td>12.4163</td>\n",
       "      <td>10.3595</td>\n",
       "      <td>9.9688</td>\n",
       "      <td>8.8353</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>TCGA.A1.A0SI</td>\n",
       "      <td>7.0163</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.6678</td>\n",
       "      <td>5.7506</td>\n",
       "      <td>2.5857</td>\n",
       "      <td>13.2612</td>\n",
       "      <td>8.0688</td>\n",
       "      <td>1.3054</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.1386</td>\n",
       "      <td>8.5373</td>\n",
       "      <td>10.4975</td>\n",
       "      <td>7.5716</td>\n",
       "      <td>9.9338</td>\n",
       "      <td>12.1409</td>\n",
       "      <td>9.9824</td>\n",
       "      <td>9.9075</td>\n",
       "      <td>4.9821</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>TCGA.A1.A0SJ</td>\n",
       "      <td>7.8315</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>7.0437</td>\n",
       "      <td>0.7921</td>\n",
       "      <td>14.4327</td>\n",
       "      <td>9.3913</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.4054</td>\n",
       "      <td>11.4160</td>\n",
       "      <td>9.7063</td>\n",
       "      <td>5.7063</td>\n",
       "      <td>10.0949</td>\n",
       "      <td>11.5666</td>\n",
       "      <td>9.9060</td>\n",
       "      <td>9.4914</td>\n",
       "      <td>5.4767</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>TCGA.A1.A0SK</td>\n",
       "      <td>6.4536</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>8.6624</td>\n",
       "      <td>4.6947</td>\n",
       "      <td>12.0332</td>\n",
       "      <td>6.3351</td>\n",
       "      <td>0.7051</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.1297</td>\n",
       "      <td>9.2554</td>\n",
       "      <td>10.7949</td>\n",
       "      <td>7.3176</td>\n",
       "      <td>10.4322</td>\n",
       "      <td>9.2100</td>\n",
       "      <td>10.8137</td>\n",
       "      <td>10.3844</td>\n",
       "      <td>5.6576</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>TCGA.A1.A0SM</td>\n",
       "      <td>8.2691</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.3958</td>\n",
       "      <td>6.6966</td>\n",
       "      <td>0.3958</td>\n",
       "      <td>13.1295</td>\n",
       "      <td>8.2285</td>\n",
       "      <td>1.1780</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.9450</td>\n",
       "      <td>9.0956</td>\n",
       "      <td>9.9876</td>\n",
       "      <td>5.3383</td>\n",
       "      <td>10.4625</td>\n",
       "      <td>12.1583</td>\n",
       "      <td>10.2088</td>\n",
       "      <td>10.5077</td>\n",
       "      <td>10.1929</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>TCGA.A1.A0SN</td>\n",
       "      <td>6.4689</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.1080</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>13.4736</td>\n",
       "      <td>9.1908</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.5001</td>\n",
       "      <td>7.9669</td>\n",
       "      <td>9.9755</td>\n",
       "      <td>7.6680</td>\n",
       "      <td>9.4086</td>\n",
       "      <td>11.5063</td>\n",
       "      <td>10.8129</td>\n",
       "      <td>9.5356</td>\n",
       "      <td>4.7876</td>\n",
       "      <td>0.4013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>TCGA.A1.A0SO</td>\n",
       "      <td>8.2001</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.6860</td>\n",
       "      <td>4.5674</td>\n",
       "      <td>11.9996</td>\n",
       "      <td>5.2494</td>\n",
       "      <td>1.0576</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.1828</td>\n",
       "      <td>9.1351</td>\n",
       "      <td>8.5514</td>\n",
       "      <td>6.7736</td>\n",
       "      <td>10.8081</td>\n",
       "      <td>10.3904</td>\n",
       "      <td>9.6831</td>\n",
       "      <td>9.8119</td>\n",
       "      <td>8.7992</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows × 20156 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     attrib_name    A1BG    A1CF   A2BP1   A2LD1   A2ML1      A2M  A4GALT  \\\n",
       "0   TCGA.3C.AAAU  7.6300  0.0000  0.0000  6.6999  1.2501  12.5017  6.1136   \n",
       "1   TCGA.3C.AALI  7.8971  0.0000  0.0000  6.1672  2.4196  12.8866  7.3101   \n",
       "2   TCGA.3C.AALJ  8.7287  0.9310  0.0000  7.3422  0.0000  13.1101  9.1671   \n",
       "3   TCGA.3C.AALK  7.5851  0.0000  0.0000  5.9888  1.4087  13.4201  8.9870   \n",
       "4   TCGA.4H.AAAK  8.0762  0.5115  2.2720  7.2796  2.1389  13.2268  8.4226   \n",
       "5   TCGA.5L.AAT0  7.6779  0.0000  0.6659  6.8142  1.1198  13.5902  8.5358   \n",
       "6   TCGA.5L.AAT1  7.7299  0.0000  0.0000  6.6861  2.3434  13.9114  8.0457   \n",
       "7   TCGA.5T.A9QA  8.3001  0.6069  0.6069  7.8742  0.0000  11.6364  8.6886   \n",
       "8   TCGA.A1.A0SB  5.6496  0.0000  2.4607  5.2816  2.0543  14.9489  8.4027   \n",
       "9   TCGA.A1.A0SD  7.1629  0.0000  0.4123  6.0846  3.1058  14.5383  8.5628   \n",
       "10  TCGA.A1.A0SE  7.5986  0.0000  0.0000  6.1685  0.9907  13.9412  8.0619   \n",
       "11  TCGA.A1.A0SF  8.3532  0.0000  0.5144  7.3429  2.1464  13.9279  8.7887   \n",
       "12  TCGA.A1.A0SG  7.4080  0.0000  2.4114  6.8659  1.4854  13.6066  9.2774   \n",
       "13  TCGA.A1.A0SH  7.5024  0.0000  0.4449  7.0378  1.2896  13.7730  8.4492   \n",
       "14  TCGA.A1.A0SI  7.0163  0.0000  0.6678  5.7506  2.5857  13.2612  8.0688   \n",
       "15  TCGA.A1.A0SJ  7.8315  0.0000  0.0000  7.0437  0.7921  14.4327  9.3913   \n",
       "16  TCGA.A1.A0SK  6.4536  0.0000  0.0000  8.6624  4.6947  12.0332  6.3351   \n",
       "17  TCGA.A1.A0SM  8.2691  0.0000  0.3958  6.6966  0.3958  13.1295  8.2285   \n",
       "18  TCGA.A1.A0SN  6.4689  0.0000  0.0000  6.1080  0.0000  13.4736  9.1908   \n",
       "19  TCGA.A1.A0SO  8.2001  0.0000  0.0000  6.6860  4.5674  11.9996  5.2494   \n",
       "\n",
       "     A4GNT    AAA1  ...    ZXDA     ZXDB     ZXDC  ZYG11A   ZYG11B      ZYX  \\\n",
       "0   3.2655  0.4273  ...  7.0289   9.9784  10.6965  8.0196  10.2400  11.7765   \n",
       "1   0.6265  0.0000  ...  5.9264   8.8125  10.3924  7.6401   9.2398  12.4267   \n",
       "2   0.0000  0.0000  ...  5.1842   9.0634   9.5884  8.3786   9.0585  12.4146   \n",
       "3   0.0000  0.0000  ...  5.8080   8.7772   9.7564  7.4629   9.2488  12.4748   \n",
       "4   0.5115  0.5115  ...  5.6420   8.7322  10.0373  3.9110   9.6013  11.9811   \n",
       "5   2.3523  0.0000  ...  6.1217   9.3320  10.1250  4.0737   9.2934  12.0130   \n",
       "6   0.8600  0.0000  ...  5.5181   9.1524   9.9507  5.7758   9.0069  12.1095   \n",
       "7   2.2207  0.0000  ...  4.8956   8.1653   9.6629  1.6286   8.3338  11.6541   \n",
       "8   0.9268  0.0000  ...  6.5993   9.0236  10.4685  4.3485  10.4153  12.5952   \n",
       "9   0.7326  0.0000  ...  6.6039   9.1781  10.2606  5.0910   9.7634  11.7979   \n",
       "10  0.0000  0.0000  ...  6.5915   9.5070   9.9941  5.8726   9.9093  11.5550   \n",
       "11  0.8928  0.0000  ...  6.2310   9.0617  10.0423  6.5801   9.7850  12.3838   \n",
       "12  0.7824  0.0000  ...  5.6427   8.7377  10.1119  1.2869   9.4589  11.5121   \n",
       "13  1.0590  0.0000  ...  6.4718   9.1851   9.6482  4.3829  10.4033  12.4163   \n",
       "14  1.3054  0.0000  ...  6.1386   8.5373  10.4975  7.5716   9.9338  12.1409   \n",
       "15  0.0000  0.0000  ...  6.4054  11.4160   9.7063  5.7063  10.0949  11.5666   \n",
       "16  0.7051  0.0000  ...  6.1297   9.2554  10.7949  7.3176  10.4322   9.2100   \n",
       "17  1.1780  0.0000  ...  5.9450   9.0956   9.9876  5.3383  10.4625  12.1583   \n",
       "18  0.0000  0.0000  ...  5.5001   7.9669   9.9755  7.6680   9.4086  11.5063   \n",
       "19  1.0576  0.0000  ...  6.1828   9.1351   8.5514  6.7736  10.8081  10.3904   \n",
       "\n",
       "      ZZEF1     ZZZ3  psiTPTE22    tAKR  \n",
       "0   10.8887  10.2064     1.4454  0.0000  \n",
       "1   10.3659   8.6715     9.8573  0.0000  \n",
       "2    9.8825   8.9958     5.1842  0.0000  \n",
       "3    9.6113   9.4551     6.0792  0.0000  \n",
       "4    9.7020   9.7858     7.5564  0.0000  \n",
       "5    9.8504  10.0536     6.4272  0.0000  \n",
       "6   10.4641   9.1545     5.8793  0.0000  \n",
       "7   10.1242   8.7876     2.2207  0.0000  \n",
       "8   10.9161  10.4890     9.1120  0.0000  \n",
       "9   10.3219  10.2247     6.4432  0.0000  \n",
       "10   9.8570  10.0721     9.7601  0.0000  \n",
       "11   9.8694   8.9918     5.7351  0.8928  \n",
       "12  10.7278   9.1561     4.7312  0.0000  \n",
       "13  10.3595   9.9688     8.8353  0.0000  \n",
       "14   9.9824   9.9075     4.9821  0.0000  \n",
       "15   9.9060   9.4914     5.4767  0.0000  \n",
       "16  10.8137  10.3844     5.6576  0.0000  \n",
       "17  10.2088  10.5077    10.1929  0.0000  \n",
       "18  10.8129   9.5356     4.7876  0.4013  \n",
       "19   9.6831   9.8119     8.7992  0.0000  \n",
       "\n",
       "[20 rows x 20156 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read transpose of  RNA Seq data\n",
    "rnaSeqData = pd.read_csv('RNASeqoutput.csv')\n",
    "rnaSeqData[0:20]\n",
    "\n",
    "rnaSeqData = rnaSeqData.rename(columns = {\"Unnamed: 0\": \"attrib_name\"}) \n",
    "rnaSeqData[0:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>hsa-let-7a-1</th>\n",
       "      <th>hsa-let-7a-2</th>\n",
       "      <th>hsa-let-7a-3</th>\n",
       "      <th>hsa-let-7b</th>\n",
       "      <th>hsa-let-7c</th>\n",
       "      <th>hsa-let-7d</th>\n",
       "      <th>hsa-let-7e</th>\n",
       "      <th>hsa-let-7f-1</th>\n",
       "      <th>hsa-let-7f-2</th>\n",
       "      <th>...</th>\n",
       "      <th>hsa-mir-940</th>\n",
       "      <th>hsa-mir-941-1</th>\n",
       "      <th>hsa-mir-942</th>\n",
       "      <th>hsa-mir-943</th>\n",
       "      <th>hsa-mir-944</th>\n",
       "      <th>hsa-mir-95</th>\n",
       "      <th>hsa-mir-96</th>\n",
       "      <th>hsa-mir-98</th>\n",
       "      <th>hsa-mir-99a</th>\n",
       "      <th>hsa-mir-99b</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>13.1299</td>\n",
       "      <td>14.1180</td>\n",
       "      <td>13.1479</td>\n",
       "      <td>14.5952</td>\n",
       "      <td>8.4191</td>\n",
       "      <td>8.6695</td>\n",
       "      <td>10.5228</td>\n",
       "      <td>3.9742</td>\n",
       "      <td>11.8252</td>\n",
       "      <td>...</td>\n",
       "      <td>2.8825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.7072</td>\n",
       "      <td>0.2479</td>\n",
       "      <td>1.6146</td>\n",
       "      <td>2.3574</td>\n",
       "      <td>6.9187</td>\n",
       "      <td>5.7812</td>\n",
       "      <td>7.0356</td>\n",
       "      <td>15.5065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>12.9183</td>\n",
       "      <td>13.9224</td>\n",
       "      <td>12.9134</td>\n",
       "      <td>14.5127</td>\n",
       "      <td>9.6483</td>\n",
       "      <td>9.0065</td>\n",
       "      <td>9.1343</td>\n",
       "      <td>4.4543</td>\n",
       "      <td>12.6791</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0712</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.1985</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.7042</td>\n",
       "      <td>1.1753</td>\n",
       "      <td>5.9366</td>\n",
       "      <td>6.4437</td>\n",
       "      <td>7.8914</td>\n",
       "      <td>13.6263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.3C.AALJ</td>\n",
       "      <td>13.0122</td>\n",
       "      <td>14.0101</td>\n",
       "      <td>13.0287</td>\n",
       "      <td>13.4197</td>\n",
       "      <td>9.3147</td>\n",
       "      <td>9.2793</td>\n",
       "      <td>11.3962</td>\n",
       "      <td>5.3505</td>\n",
       "      <td>13.5304</td>\n",
       "      <td>...</td>\n",
       "      <td>1.2114</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.2073</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.8217</td>\n",
       "      <td>1.4031</td>\n",
       "      <td>6.6184</td>\n",
       "      <td>6.8905</td>\n",
       "      <td>7.5882</td>\n",
       "      <td>15.0139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.3C.AALK</td>\n",
       "      <td>13.1449</td>\n",
       "      <td>14.1418</td>\n",
       "      <td>13.1514</td>\n",
       "      <td>14.6673</td>\n",
       "      <td>11.5119</td>\n",
       "      <td>8.3891</td>\n",
       "      <td>10.3701</td>\n",
       "      <td>4.2378</td>\n",
       "      <td>12.6528</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9625</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.1630</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.1630</td>\n",
       "      <td>1.4538</td>\n",
       "      <td>6.2011</td>\n",
       "      <td>5.4122</td>\n",
       "      <td>10.0330</td>\n",
       "      <td>14.5548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.4H.AAAK</td>\n",
       "      <td>13.4118</td>\n",
       "      <td>14.4136</td>\n",
       "      <td>13.4206</td>\n",
       "      <td>14.4386</td>\n",
       "      <td>11.6944</td>\n",
       "      <td>8.4579</td>\n",
       "      <td>10.7422</td>\n",
       "      <td>4.5572</td>\n",
       "      <td>13.0097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4053</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.6361</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.0944</td>\n",
       "      <td>0.8567</td>\n",
       "      <td>4.3887</td>\n",
       "      <td>5.1449</td>\n",
       "      <td>10.0795</td>\n",
       "      <td>14.6504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>TCGA.5L.AAT0</td>\n",
       "      <td>13.3164</td>\n",
       "      <td>14.3110</td>\n",
       "      <td>13.3273</td>\n",
       "      <td>14.5766</td>\n",
       "      <td>11.1391</td>\n",
       "      <td>7.6537</td>\n",
       "      <td>9.7151</td>\n",
       "      <td>4.3607</td>\n",
       "      <td>12.8746</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8640</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.3887</td>\n",
       "      <td>0.1848</td>\n",
       "      <td>1.8467</td>\n",
       "      <td>1.4006</td>\n",
       "      <td>6.5431</td>\n",
       "      <td>5.3830</td>\n",
       "      <td>9.3659</td>\n",
       "      <td>14.1468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>TCGA.5L.AAT1</td>\n",
       "      <td>13.4454</td>\n",
       "      <td>14.4486</td>\n",
       "      <td>13.4469</td>\n",
       "      <td>14.6112</td>\n",
       "      <td>11.2850</td>\n",
       "      <td>8.3013</td>\n",
       "      <td>10.4853</td>\n",
       "      <td>5.0527</td>\n",
       "      <td>13.4510</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7741</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.8961</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.2751</td>\n",
       "      <td>1.2028</td>\n",
       "      <td>4.8486</td>\n",
       "      <td>5.9911</td>\n",
       "      <td>9.3093</td>\n",
       "      <td>14.5058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>TCGA.5T.A9QA</td>\n",
       "      <td>13.7280</td>\n",
       "      <td>14.7146</td>\n",
       "      <td>13.7370</td>\n",
       "      <td>15.0988</td>\n",
       "      <td>9.2000</td>\n",
       "      <td>8.9298</td>\n",
       "      <td>11.1385</td>\n",
       "      <td>4.8882</td>\n",
       "      <td>12.9272</td>\n",
       "      <td>...</td>\n",
       "      <td>1.4064</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.3791</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.8682</td>\n",
       "      <td>2.8734</td>\n",
       "      <td>5.7052</td>\n",
       "      <td>5.3635</td>\n",
       "      <td>6.9476</td>\n",
       "      <td>15.1009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>TCGA.A1.A0SB</td>\n",
       "      <td>13.6016</td>\n",
       "      <td>14.6088</td>\n",
       "      <td>13.6132</td>\n",
       "      <td>16.5058</td>\n",
       "      <td>13.3923</td>\n",
       "      <td>10.4487</td>\n",
       "      <td>9.3709</td>\n",
       "      <td>4.2779</td>\n",
       "      <td>12.4457</td>\n",
       "      <td>...</td>\n",
       "      <td>2.5892</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.2269</td>\n",
       "      <td>0.5040</td>\n",
       "      <td>4.2779</td>\n",
       "      <td>1.6279</td>\n",
       "      <td>5.6049</td>\n",
       "      <td>6.0218</td>\n",
       "      <td>12.0124</td>\n",
       "      <td>15.8454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>TCGA.A1.A0SD</td>\n",
       "      <td>13.5989</td>\n",
       "      <td>14.6070</td>\n",
       "      <td>13.6063</td>\n",
       "      <td>15.6389</td>\n",
       "      <td>11.4203</td>\n",
       "      <td>9.0489</td>\n",
       "      <td>10.5510</td>\n",
       "      <td>4.6218</td>\n",
       "      <td>13.2264</td>\n",
       "      <td>...</td>\n",
       "      <td>1.1490</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.9716</td>\n",
       "      <td>0.3144</td>\n",
       "      <td>1.5598</td>\n",
       "      <td>1.7804</td>\n",
       "      <td>4.5029</td>\n",
       "      <td>5.6975</td>\n",
       "      <td>10.1494</td>\n",
       "      <td>15.6171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>TCGA.A1.A0SF</td>\n",
       "      <td>12.3430</td>\n",
       "      <td>13.3412</td>\n",
       "      <td>12.3516</td>\n",
       "      <td>14.3223</td>\n",
       "      <td>11.9218</td>\n",
       "      <td>8.2020</td>\n",
       "      <td>9.3039</td>\n",
       "      <td>3.4391</td>\n",
       "      <td>12.0122</td>\n",
       "      <td>...</td>\n",
       "      <td>2.3840</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.7001</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.2670</td>\n",
       "      <td>1.5450</td>\n",
       "      <td>5.5751</td>\n",
       "      <td>5.3718</td>\n",
       "      <td>10.9691</td>\n",
       "      <td>15.2262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>TCGA.A1.A0SG</td>\n",
       "      <td>13.0403</td>\n",
       "      <td>14.0595</td>\n",
       "      <td>13.0644</td>\n",
       "      <td>15.2934</td>\n",
       "      <td>11.0065</td>\n",
       "      <td>9.4127</td>\n",
       "      <td>10.0053</td>\n",
       "      <td>4.3052</td>\n",
       "      <td>12.4670</td>\n",
       "      <td>...</td>\n",
       "      <td>3.3273</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.5524</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.9912</td>\n",
       "      <td>5.4627</td>\n",
       "      <td>6.0966</td>\n",
       "      <td>9.3541</td>\n",
       "      <td>16.5759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>TCGA.A1.A0SI</td>\n",
       "      <td>12.4060</td>\n",
       "      <td>13.4118</td>\n",
       "      <td>12.4280</td>\n",
       "      <td>14.8164</td>\n",
       "      <td>11.5133</td>\n",
       "      <td>8.3363</td>\n",
       "      <td>9.1036</td>\n",
       "      <td>2.6697</td>\n",
       "      <td>11.7792</td>\n",
       "      <td>...</td>\n",
       "      <td>2.1045</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.2366</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.8680</td>\n",
       "      <td>4.9596</td>\n",
       "      <td>4.5409</td>\n",
       "      <td>10.4735</td>\n",
       "      <td>15.0163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>TCGA.A1.A0SN</td>\n",
       "      <td>11.8468</td>\n",
       "      <td>12.8271</td>\n",
       "      <td>11.8313</td>\n",
       "      <td>13.5496</td>\n",
       "      <td>10.4015</td>\n",
       "      <td>8.2158</td>\n",
       "      <td>8.0903</td>\n",
       "      <td>3.5868</td>\n",
       "      <td>11.4282</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0105</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.6795</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.0714</td>\n",
       "      <td>3.6827</td>\n",
       "      <td>5.0176</td>\n",
       "      <td>8.9271</td>\n",
       "      <td>14.1129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>TCGA.A1.A0SQ</td>\n",
       "      <td>13.8134</td>\n",
       "      <td>14.8228</td>\n",
       "      <td>13.8248</td>\n",
       "      <td>17.1428</td>\n",
       "      <td>9.7178</td>\n",
       "      <td>9.2139</td>\n",
       "      <td>9.7305</td>\n",
       "      <td>4.4592</td>\n",
       "      <td>11.7823</td>\n",
       "      <td>...</td>\n",
       "      <td>2.2343</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.4122</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.5706</td>\n",
       "      <td>6.9814</td>\n",
       "      <td>5.6377</td>\n",
       "      <td>7.9579</td>\n",
       "      <td>16.2743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>TCGA.A2.A04N</td>\n",
       "      <td>11.2948</td>\n",
       "      <td>12.2896</td>\n",
       "      <td>11.3159</td>\n",
       "      <td>13.2788</td>\n",
       "      <td>9.8753</td>\n",
       "      <td>6.8906</td>\n",
       "      <td>8.4705</td>\n",
       "      <td>1.8638</td>\n",
       "      <td>11.6533</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0060</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4616</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.4616</td>\n",
       "      <td>2.7347</td>\n",
       "      <td>3.0294</td>\n",
       "      <td>9.6316</td>\n",
       "      <td>15.5480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>TCGA.A2.A04U</td>\n",
       "      <td>12.8271</td>\n",
       "      <td>13.8125</td>\n",
       "      <td>12.8277</td>\n",
       "      <td>13.9926</td>\n",
       "      <td>9.8608</td>\n",
       "      <td>10.4524</td>\n",
       "      <td>9.9146</td>\n",
       "      <td>4.7107</td>\n",
       "      <td>14.1508</td>\n",
       "      <td>...</td>\n",
       "      <td>4.8107</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.7312</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.4604</td>\n",
       "      <td>2.5574</td>\n",
       "      <td>4.9400</td>\n",
       "      <td>7.4731</td>\n",
       "      <td>8.2778</td>\n",
       "      <td>15.7213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>TCGA.A2.A04W</td>\n",
       "      <td>11.9105</td>\n",
       "      <td>12.9028</td>\n",
       "      <td>11.9400</td>\n",
       "      <td>13.6782</td>\n",
       "      <td>11.9034</td>\n",
       "      <td>8.7605</td>\n",
       "      <td>9.9147</td>\n",
       "      <td>3.3232</td>\n",
       "      <td>11.3483</td>\n",
       "      <td>...</td>\n",
       "      <td>2.5641</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.3490</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>1.7895</td>\n",
       "      <td>6.0258</td>\n",
       "      <td>5.5754</td>\n",
       "      <td>10.4454</td>\n",
       "      <td>16.2797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>TCGA.A2.A0CK</td>\n",
       "      <td>13.5031</td>\n",
       "      <td>14.5018</td>\n",
       "      <td>13.5061</td>\n",
       "      <td>15.4973</td>\n",
       "      <td>11.5980</td>\n",
       "      <td>8.0621</td>\n",
       "      <td>9.2273</td>\n",
       "      <td>4.0570</td>\n",
       "      <td>12.2441</td>\n",
       "      <td>...</td>\n",
       "      <td>1.3161</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.5280</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.0969</td>\n",
       "      <td>2.3707</td>\n",
       "      <td>6.1382</td>\n",
       "      <td>5.2175</td>\n",
       "      <td>10.7712</td>\n",
       "      <td>14.0020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>TCGA.A2.A0CL</td>\n",
       "      <td>12.6012</td>\n",
       "      <td>13.5889</td>\n",
       "      <td>12.6128</td>\n",
       "      <td>13.8688</td>\n",
       "      <td>10.9182</td>\n",
       "      <td>8.8473</td>\n",
       "      <td>9.2771</td>\n",
       "      <td>5.3000</td>\n",
       "      <td>13.2447</td>\n",
       "      <td>...</td>\n",
       "      <td>3.5132</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.2598</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.1313</td>\n",
       "      <td>1.8793</td>\n",
       "      <td>4.4087</td>\n",
       "      <td>6.3143</td>\n",
       "      <td>9.4620</td>\n",
       "      <td>14.9758</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows × 824 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     attrib_name  hsa-let-7a-1  hsa-let-7a-2  hsa-let-7a-3  hsa-let-7b  \\\n",
       "0   TCGA.3C.AAAU       13.1299       14.1180       13.1479     14.5952   \n",
       "1   TCGA.3C.AALI       12.9183       13.9224       12.9134     14.5127   \n",
       "2   TCGA.3C.AALJ       13.0122       14.0101       13.0287     13.4197   \n",
       "3   TCGA.3C.AALK       13.1449       14.1418       13.1514     14.6673   \n",
       "4   TCGA.4H.AAAK       13.4118       14.4136       13.4206     14.4386   \n",
       "5   TCGA.5L.AAT0       13.3164       14.3110       13.3273     14.5766   \n",
       "6   TCGA.5L.AAT1       13.4454       14.4486       13.4469     14.6112   \n",
       "7   TCGA.5T.A9QA       13.7280       14.7146       13.7370     15.0988   \n",
       "8   TCGA.A1.A0SB       13.6016       14.6088       13.6132     16.5058   \n",
       "9   TCGA.A1.A0SD       13.5989       14.6070       13.6063     15.6389   \n",
       "10  TCGA.A1.A0SF       12.3430       13.3412       12.3516     14.3223   \n",
       "11  TCGA.A1.A0SG       13.0403       14.0595       13.0644     15.2934   \n",
       "12  TCGA.A1.A0SI       12.4060       13.4118       12.4280     14.8164   \n",
       "13  TCGA.A1.A0SN       11.8468       12.8271       11.8313     13.5496   \n",
       "14  TCGA.A1.A0SQ       13.8134       14.8228       13.8248     17.1428   \n",
       "15  TCGA.A2.A04N       11.2948       12.2896       11.3159     13.2788   \n",
       "16  TCGA.A2.A04U       12.8271       13.8125       12.8277     13.9926   \n",
       "17  TCGA.A2.A04W       11.9105       12.9028       11.9400     13.6782   \n",
       "18  TCGA.A2.A0CK       13.5031       14.5018       13.5061     15.4973   \n",
       "19  TCGA.A2.A0CL       12.6012       13.5889       12.6128     13.8688   \n",
       "\n",
       "    hsa-let-7c  hsa-let-7d  hsa-let-7e  hsa-let-7f-1  hsa-let-7f-2  ...  \\\n",
       "0       8.4191      8.6695     10.5228        3.9742       11.8252  ...   \n",
       "1       9.6483      9.0065      9.1343        4.4543       12.6791  ...   \n",
       "2       9.3147      9.2793     11.3962        5.3505       13.5304  ...   \n",
       "3      11.5119      8.3891     10.3701        4.2378       12.6528  ...   \n",
       "4      11.6944      8.4579     10.7422        4.5572       13.0097  ...   \n",
       "5      11.1391      7.6537      9.7151        4.3607       12.8746  ...   \n",
       "6      11.2850      8.3013     10.4853        5.0527       13.4510  ...   \n",
       "7       9.2000      8.9298     11.1385        4.8882       12.9272  ...   \n",
       "8      13.3923     10.4487      9.3709        4.2779       12.4457  ...   \n",
       "9      11.4203      9.0489     10.5510        4.6218       13.2264  ...   \n",
       "10     11.9218      8.2020      9.3039        3.4391       12.0122  ...   \n",
       "11     11.0065      9.4127     10.0053        4.3052       12.4670  ...   \n",
       "12     11.5133      8.3363      9.1036        2.6697       11.7792  ...   \n",
       "13     10.4015      8.2158      8.0903        3.5868       11.4282  ...   \n",
       "14      9.7178      9.2139      9.7305        4.4592       11.7823  ...   \n",
       "15      9.8753      6.8906      8.4705        1.8638       11.6533  ...   \n",
       "16      9.8608     10.4524      9.9146        4.7107       14.1508  ...   \n",
       "17     11.9034      8.7605      9.9147        3.3232       11.3483  ...   \n",
       "18     11.5980      8.0621      9.2273        4.0570       12.2441  ...   \n",
       "19     10.9182      8.8473      9.2771        5.3000       13.2447  ...   \n",
       "\n",
       "    hsa-mir-940  hsa-mir-941-1  hsa-mir-942  hsa-mir-943  hsa-mir-944  \\\n",
       "0        2.8825            0.0       2.7072       0.2479       1.6146   \n",
       "1        1.0712            0.0       3.1985       0.0000       0.7042   \n",
       "2        1.2114            0.0       2.2073       0.0000       0.8217   \n",
       "3        0.9625            0.0       2.1630       0.0000       2.1630   \n",
       "4        0.4053            0.0       1.6361       0.0000       1.0944   \n",
       "5        0.8640            0.0       2.3887       0.1848       1.8467   \n",
       "6        0.7741            0.0       1.8961       0.0000       1.2751   \n",
       "7        1.4064            0.0       4.3791       0.0000       0.8682   \n",
       "8        2.5892            0.0       3.2269       0.5040       4.2779   \n",
       "9        1.1490            0.0       1.9716       0.3144       1.5598   \n",
       "10       2.3840            0.0       2.7001       0.0000       1.2670   \n",
       "11       3.3273            0.0       2.5524       0.0000       0.0000   \n",
       "12       2.1045            0.0       2.2366       0.0000       0.0000   \n",
       "13       2.0105            0.0       1.6795       0.0000       0.0000   \n",
       "14       2.2343            0.0       2.4122       0.0000       0.0000   \n",
       "15       2.0060            0.0       0.4616       0.0000       0.0000   \n",
       "16       4.8107            0.0       4.7312       0.0000       0.4604   \n",
       "17       2.5641            0.0       2.3490       0.0000       0.3482   \n",
       "18       1.3161            0.0       2.5280       0.0000       2.0969   \n",
       "19       3.5132            0.0       4.2598       0.0000       1.1313   \n",
       "\n",
       "    hsa-mir-95  hsa-mir-96  hsa-mir-98  hsa-mir-99a  hsa-mir-99b  \n",
       "0       2.3574      6.9187      5.7812       7.0356      15.5065  \n",
       "1       1.1753      5.9366      6.4437       7.8914      13.6263  \n",
       "2       1.4031      6.6184      6.8905       7.5882      15.0139  \n",
       "3       1.4538      6.2011      5.4122      10.0330      14.5548  \n",
       "4       0.8567      4.3887      5.1449      10.0795      14.6504  \n",
       "5       1.4006      6.5431      5.3830       9.3659      14.1468  \n",
       "6       1.2028      4.8486      5.9911       9.3093      14.5058  \n",
       "7       2.8734      5.7052      5.3635       6.9476      15.1009  \n",
       "8       1.6279      5.6049      6.0218      12.0124      15.8454  \n",
       "9       1.7804      4.5029      5.6975      10.1494      15.6171  \n",
       "10      1.5450      5.5751      5.3718      10.9691      15.2262  \n",
       "11      2.9912      5.4627      6.0966       9.3541      16.5759  \n",
       "12      0.8680      4.9596      4.5409      10.4735      15.0163  \n",
       "13      1.0714      3.6827      5.0176       8.9271      14.1129  \n",
       "14      2.5706      6.9814      5.6377       7.9579      16.2743  \n",
       "15      0.4616      2.7347      3.0294       9.6316      15.5480  \n",
       "16      2.5574      4.9400      7.4731       8.2778      15.7213  \n",
       "17      1.7895      6.0258      5.5754      10.4454      16.2797  \n",
       "18      2.3707      6.1382      5.2175      10.7712      14.0020  \n",
       "19      1.8793      4.4087      6.3143       9.4620      14.9758  \n",
       "\n",
       "[20 rows x 824 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read transpose of  miRNA data\n",
    "miRNAData = pd.read_csv('miRNATranspose.csv')\n",
    "miRNAData[0:20]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cheking common patients across all datasets - clinical and molecular"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.3C.AALJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.3C.AALK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.4H.AAAK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>607</td>\n",
       "      <td>TCGA.WT.AB44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>608</td>\n",
       "      <td>TCGA.XX.A899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>609</td>\n",
       "      <td>TCGA.XX.A89A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>610</td>\n",
       "      <td>TCGA.Z7.A8R5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>611</td>\n",
       "      <td>TCGA.Z7.A8R6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>612 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      attrib_name\n",
       "0    TCGA.3C.AAAU\n",
       "1    TCGA.3C.AALI\n",
       "2    TCGA.3C.AALJ\n",
       "3    TCGA.3C.AALK\n",
       "4    TCGA.4H.AAAK\n",
       "..            ...\n",
       "607  TCGA.WT.AB44\n",
       "608  TCGA.XX.A899\n",
       "609  TCGA.XX.A89A\n",
       "610  TCGA.Z7.A8R5\n",
       "611  TCGA.Z7.A8R6\n",
       "\n",
       "[612 rows x 1 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylationData[['attrib_name']].merge(cleanClinicalData[['attrib_name']], on='attrib_name', how='inner').merge(rnaSeqData[['attrib_name']], on='attrib_name', how='inner').merge(miRNAData[['attrib_name']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylClinical = methylationData.merge(cleanClinicalData[['attrib_name','overall_survival', 'status']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>MIR7-1</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4548</td>\n",
       "      <td>0.2310</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>-0.4737</td>\n",
       "      <td>-0.4817</td>\n",
       "      <td>0.4142</td>\n",
       "      <td>-0.4366</td>\n",
       "      <td>-0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>0.3420</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4047.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>-0.4313</td>\n",
       "      <td>-0.4692</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.3972</td>\n",
       "      <td>-0.2806</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>0.4373</td>\n",
       "      <td>-0.4527</td>\n",
       "      <td>-0.4643</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>0.3857</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4005.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.3C.AALJ</td>\n",
       "      <td>-0.4285</td>\n",
       "      <td>-0.4697</td>\n",
       "      <td>0.2664</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>-0.0637</td>\n",
       "      <td>-0.4783</td>\n",
       "      <td>0.3738</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4416</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>0.3319</td>\n",
       "      <td>0.1929</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1474.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 20109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    attrib_name    RBL2   VDAC3   ACTN1  ATP2A1   SFRP1   NIPA2  MAN1B1  \\\n",
       "0  TCGA.3C.AAAU -0.4444 -0.4548  0.2310  0.3568 -0.4737 -0.4817  0.4142   \n",
       "1  TCGA.3C.AALI -0.4313 -0.4692  0.2859  0.3972 -0.2806 -0.4740  0.4373   \n",
       "2  TCGA.3C.AALJ -0.4285 -0.4697  0.2664  0.3482 -0.0637 -0.4783  0.3738   \n",
       "\n",
       "   TSEN34  LRRC16A  ...   BAGE2   BAGE4   BAGE3  MIR637  LOC100130932  \\\n",
       "0 -0.4366  -0.4610  ... -0.0420 -0.0420 -0.0420  0.3420        0.2383   \n",
       "1 -0.4527  -0.4643  ... -0.0640 -0.0640 -0.0640  0.3857        0.3783   \n",
       "2 -0.4430  -0.4416  ... -0.0873 -0.0873 -0.0873  0.3319        0.1929   \n",
       "\n",
       "   GOLGA8F  GOLGA8G  MIR7-1  overall_survival  status  \n",
       "0  -0.1168  -0.1168     NaN            4047.0     0.0  \n",
       "1  -0.1271  -0.1271     NaN            4005.0     0.0  \n",
       "2   0.0066   0.0066     NaN            1474.0     0.0  \n",
       "\n",
       "[3 rows x 20109 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylClinical[0:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaSeqlClinical = rnaSeqData.merge(cleanClinicalData[['attrib_name','overall_survival', 'status']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2BP1</th>\n",
       "      <th>A2LD1</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAA1</th>\n",
       "      <th>...</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>psiTPTE22</th>\n",
       "      <th>tAKR</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>7.6300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.6999</td>\n",
       "      <td>1.2501</td>\n",
       "      <td>12.5017</td>\n",
       "      <td>6.1136</td>\n",
       "      <td>3.2655</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>...</td>\n",
       "      <td>10.6965</td>\n",
       "      <td>8.0196</td>\n",
       "      <td>10.2400</td>\n",
       "      <td>11.7765</td>\n",
       "      <td>10.8887</td>\n",
       "      <td>10.2064</td>\n",
       "      <td>1.4454</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4047.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>7.8971</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.1672</td>\n",
       "      <td>2.4196</td>\n",
       "      <td>12.8866</td>\n",
       "      <td>7.3101</td>\n",
       "      <td>0.6265</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>10.3924</td>\n",
       "      <td>7.6401</td>\n",
       "      <td>9.2398</td>\n",
       "      <td>12.4267</td>\n",
       "      <td>10.3659</td>\n",
       "      <td>8.6715</td>\n",
       "      <td>9.8573</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4005.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 20158 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    attrib_name    A1BG  A1CF  A2BP1   A2LD1   A2ML1      A2M  A4GALT   A4GNT  \\\n",
       "0  TCGA.3C.AAAU  7.6300   0.0    0.0  6.6999  1.2501  12.5017  6.1136  3.2655   \n",
       "1  TCGA.3C.AALI  7.8971   0.0    0.0  6.1672  2.4196  12.8866  7.3101  0.6265   \n",
       "\n",
       "     AAA1  ...     ZXDC  ZYG11A   ZYG11B      ZYX    ZZEF1     ZZZ3  \\\n",
       "0  0.4273  ...  10.6965  8.0196  10.2400  11.7765  10.8887  10.2064   \n",
       "1  0.0000  ...  10.3924  7.6401   9.2398  12.4267  10.3659   8.6715   \n",
       "\n",
       "   psiTPTE22  tAKR  overall_survival  status  \n",
       "0     1.4454   0.0            4047.0     0.0  \n",
       "1     9.8573   0.0            4005.0     0.0  \n",
       "\n",
       "[2 rows x 20158 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rnaSeqlClinical[0:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "miRNAClinical = miRNAData.merge(cleanClinicalData[['attrib_name','overall_survival', 'status']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>hsa-let-7a-1</th>\n",
       "      <th>hsa-let-7a-2</th>\n",
       "      <th>hsa-let-7a-3</th>\n",
       "      <th>hsa-let-7b</th>\n",
       "      <th>hsa-let-7c</th>\n",
       "      <th>hsa-let-7d</th>\n",
       "      <th>hsa-let-7e</th>\n",
       "      <th>hsa-let-7f-1</th>\n",
       "      <th>hsa-let-7f-2</th>\n",
       "      <th>...</th>\n",
       "      <th>hsa-mir-942</th>\n",
       "      <th>hsa-mir-943</th>\n",
       "      <th>hsa-mir-944</th>\n",
       "      <th>hsa-mir-95</th>\n",
       "      <th>hsa-mir-96</th>\n",
       "      <th>hsa-mir-98</th>\n",
       "      <th>hsa-mir-99a</th>\n",
       "      <th>hsa-mir-99b</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>13.1299</td>\n",
       "      <td>14.1180</td>\n",
       "      <td>13.1479</td>\n",
       "      <td>14.5952</td>\n",
       "      <td>8.4191</td>\n",
       "      <td>8.6695</td>\n",
       "      <td>10.5228</td>\n",
       "      <td>3.9742</td>\n",
       "      <td>11.8252</td>\n",
       "      <td>...</td>\n",
       "      <td>2.7072</td>\n",
       "      <td>0.2479</td>\n",
       "      <td>1.6146</td>\n",
       "      <td>2.3574</td>\n",
       "      <td>6.9187</td>\n",
       "      <td>5.7812</td>\n",
       "      <td>7.0356</td>\n",
       "      <td>15.5065</td>\n",
       "      <td>4047.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>12.9183</td>\n",
       "      <td>13.9224</td>\n",
       "      <td>12.9134</td>\n",
       "      <td>14.5127</td>\n",
       "      <td>9.6483</td>\n",
       "      <td>9.0065</td>\n",
       "      <td>9.1343</td>\n",
       "      <td>4.4543</td>\n",
       "      <td>12.6791</td>\n",
       "      <td>...</td>\n",
       "      <td>3.1985</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.7042</td>\n",
       "      <td>1.1753</td>\n",
       "      <td>5.9366</td>\n",
       "      <td>6.4437</td>\n",
       "      <td>7.8914</td>\n",
       "      <td>13.6263</td>\n",
       "      <td>4005.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 826 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    attrib_name  hsa-let-7a-1  hsa-let-7a-2  hsa-let-7a-3  hsa-let-7b  \\\n",
       "0  TCGA.3C.AAAU       13.1299       14.1180       13.1479     14.5952   \n",
       "1  TCGA.3C.AALI       12.9183       13.9224       12.9134     14.5127   \n",
       "\n",
       "   hsa-let-7c  hsa-let-7d  hsa-let-7e  hsa-let-7f-1  hsa-let-7f-2  ...  \\\n",
       "0      8.4191      8.6695     10.5228        3.9742       11.8252  ...   \n",
       "1      9.6483      9.0065      9.1343        4.4543       12.6791  ...   \n",
       "\n",
       "   hsa-mir-942  hsa-mir-943  hsa-mir-944  hsa-mir-95  hsa-mir-96  hsa-mir-98  \\\n",
       "0       2.7072       0.2479       1.6146      2.3574      6.9187      5.7812   \n",
       "1       3.1985       0.0000       0.7042      1.1753      5.9366      6.4437   \n",
       "\n",
       "   hsa-mir-99a  hsa-mir-99b  overall_survival  status  \n",
       "0       7.0356      15.5065            4047.0     0.0  \n",
       "1       7.8914      13.6263            4005.0     0.0  \n",
       "\n",
       "[2 rows x 826 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miRNAClinical[0:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Taking rows common across all molecular datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylClinicalCommon = methylClinical.merge(rnaSeqlClinical[['attrib_name']], on='attrib_name', how='inner').merge(miRNAClinical[['attrib_name']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>MIR7-1</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4548</td>\n",
       "      <td>0.2310</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>-0.4737</td>\n",
       "      <td>-0.4817</td>\n",
       "      <td>0.4142</td>\n",
       "      <td>-0.4366</td>\n",
       "      <td>-0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.042</td>\n",
       "      <td>-0.042</td>\n",
       "      <td>-0.042</td>\n",
       "      <td>0.3420</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4047.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>-0.4313</td>\n",
       "      <td>-0.4692</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.3972</td>\n",
       "      <td>-0.2806</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>0.4373</td>\n",
       "      <td>-0.4527</td>\n",
       "      <td>-0.4643</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.064</td>\n",
       "      <td>-0.064</td>\n",
       "      <td>-0.064</td>\n",
       "      <td>0.3857</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4005.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 20109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    attrib_name    RBL2   VDAC3   ACTN1  ATP2A1   SFRP1   NIPA2  MAN1B1  \\\n",
       "0  TCGA.3C.AAAU -0.4444 -0.4548  0.2310  0.3568 -0.4737 -0.4817  0.4142   \n",
       "1  TCGA.3C.AALI -0.4313 -0.4692  0.2859  0.3972 -0.2806 -0.4740  0.4373   \n",
       "\n",
       "   TSEN34  LRRC16A  ...  BAGE2  BAGE4  BAGE3  MIR637  LOC100130932  GOLGA8F  \\\n",
       "0 -0.4366  -0.4610  ... -0.042 -0.042 -0.042  0.3420        0.2383  -0.1168   \n",
       "1 -0.4527  -0.4643  ... -0.064 -0.064 -0.064  0.3857        0.3783  -0.1271   \n",
       "\n",
       "   GOLGA8G  MIR7-1  overall_survival  status  \n",
       "0  -0.1168     NaN            4047.0     0.0  \n",
       "1  -0.1271     NaN            4005.0     0.0  \n",
       "\n",
       "[2 rows x 20109 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylClinicalCommon[0:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaSeqlClinicalCommon =rnaSeqlClinical.merge(methylClinicalCommon[['attrib_name']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "miRNAClinicalCommon =miRNAClinical.merge(methylClinicalCommon[['attrib_name']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>MIR7-1</th>\n",
       "      <th>overall_survival</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4548</td>\n",
       "      <td>0.2310</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>-0.4737</td>\n",
       "      <td>-0.4817</td>\n",
       "      <td>0.4142</td>\n",
       "      <td>-0.4366</td>\n",
       "      <td>-0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>0.3420</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4047.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>-0.4313</td>\n",
       "      <td>-0.4692</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.3972</td>\n",
       "      <td>-0.2806</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>0.4373</td>\n",
       "      <td>-0.4527</td>\n",
       "      <td>-0.4643</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>0.3857</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4005.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.3C.AALJ</td>\n",
       "      <td>-0.4285</td>\n",
       "      <td>-0.4697</td>\n",
       "      <td>0.2664</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>-0.0637</td>\n",
       "      <td>-0.4783</td>\n",
       "      <td>0.3738</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4416</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>0.3319</td>\n",
       "      <td>0.1929</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1474.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.3C.AALK</td>\n",
       "      <td>-0.4043</td>\n",
       "      <td>-0.4714</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>-0.0440</td>\n",
       "      <td>-0.4827</td>\n",
       "      <td>0.3704</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4558</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.3457</td>\n",
       "      <td>0.2445</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1448.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.4H.AAAK</td>\n",
       "      <td>-0.4083</td>\n",
       "      <td>-0.4651</td>\n",
       "      <td>0.1717</td>\n",
       "      <td>0.3406</td>\n",
       "      <td>-0.0880</td>\n",
       "      <td>-0.4788</td>\n",
       "      <td>0.4143</td>\n",
       "      <td>-0.4442</td>\n",
       "      <td>-0.4675</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.3542</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>NaN</td>\n",
       "      <td>348.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>607</td>\n",
       "      <td>TCGA.WT.AB44</td>\n",
       "      <td>-0.3903</td>\n",
       "      <td>-0.4683</td>\n",
       "      <td>0.2731</td>\n",
       "      <td>0.3682</td>\n",
       "      <td>-0.1018</td>\n",
       "      <td>-0.4842</td>\n",
       "      <td>0.3842</td>\n",
       "      <td>-0.4408</td>\n",
       "      <td>-0.4601</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.3322</td>\n",
       "      <td>0.2746</td>\n",
       "      <td>-0.0284</td>\n",
       "      <td>-0.0284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>883.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>608</td>\n",
       "      <td>TCGA.XX.A899</td>\n",
       "      <td>-0.4500</td>\n",
       "      <td>-0.4721</td>\n",
       "      <td>0.2597</td>\n",
       "      <td>0.4152</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4211</td>\n",
       "      <td>-0.4391</td>\n",
       "      <td>-0.4622</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.3908</td>\n",
       "      <td>0.4161</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>-0.2598</td>\n",
       "      <td>467.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>609</td>\n",
       "      <td>TCGA.XX.A89A</td>\n",
       "      <td>-0.4237</td>\n",
       "      <td>-0.4649</td>\n",
       "      <td>0.1951</td>\n",
       "      <td>0.3586</td>\n",
       "      <td>-0.4071</td>\n",
       "      <td>-0.4862</td>\n",
       "      <td>0.4494</td>\n",
       "      <td>-0.4609</td>\n",
       "      <td>-0.4639</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.4054</td>\n",
       "      <td>0.4089</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>-0.2470</td>\n",
       "      <td>488.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>610</td>\n",
       "      <td>TCGA.Z7.A8R5</td>\n",
       "      <td>-0.4197</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>0.1128</td>\n",
       "      <td>0.3803</td>\n",
       "      <td>-0.1014</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>-0.4283</td>\n",
       "      <td>-0.4676</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.3571</td>\n",
       "      <td>0.2107</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3287.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>611</td>\n",
       "      <td>TCGA.Z7.A8R6</td>\n",
       "      <td>-0.4682</td>\n",
       "      <td>-0.4576</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3621</td>\n",
       "      <td>-0.4837</td>\n",
       "      <td>-0.4846</td>\n",
       "      <td>0.4343</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>-0.4747</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.3632</td>\n",
       "      <td>0.4219</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>-0.4183</td>\n",
       "      <td>3256.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>612 rows × 20108 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      attrib_name    RBL2   VDAC3   ACTN1  ATP2A1   SFRP1   NIPA2  MAN1B1  \\\n",
       "0    TCGA.3C.AAAU -0.4444 -0.4548  0.2310  0.3568 -0.4737 -0.4817  0.4142   \n",
       "1    TCGA.3C.AALI -0.4313 -0.4692  0.2859  0.3972 -0.2806 -0.4740  0.4373   \n",
       "2    TCGA.3C.AALJ -0.4285 -0.4697  0.2664  0.3482 -0.0637 -0.4783  0.3738   \n",
       "3    TCGA.3C.AALK -0.4043 -0.4714  0.2174  0.3533 -0.0440 -0.4827  0.3704   \n",
       "4    TCGA.4H.AAAK -0.4083 -0.4651  0.1717  0.3406 -0.0880 -0.4788  0.4143   \n",
       "..            ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "607  TCGA.WT.AB44 -0.3903 -0.4683  0.2731  0.3682 -0.1018 -0.4842  0.3842   \n",
       "608  TCGA.XX.A899 -0.4500 -0.4721  0.2597  0.4152  0.0244 -0.4843  0.4211   \n",
       "609  TCGA.XX.A89A -0.4237 -0.4649  0.1951  0.3586 -0.4071 -0.4862  0.4494   \n",
       "610  TCGA.Z7.A8R5 -0.4197 -0.4688  0.1128  0.3803 -0.1014 -0.4843  0.4244   \n",
       "611  TCGA.Z7.A8R6 -0.4682 -0.4576  0.3352  0.3621 -0.4837 -0.4846  0.4343   \n",
       "\n",
       "     TSEN34  LRRC16A  ...    BAGE   BAGE2   BAGE4   BAGE3  MIR637  \\\n",
       "0   -0.4366  -0.4610  ... -0.0420 -0.0420 -0.0420 -0.0420  0.3420   \n",
       "1   -0.4527  -0.4643  ... -0.0640 -0.0640 -0.0640 -0.0640  0.3857   \n",
       "2   -0.4430  -0.4416  ... -0.0873 -0.0873 -0.0873 -0.0873  0.3319   \n",
       "3   -0.4430  -0.4558  ...  0.1426  0.1426  0.1426  0.1426  0.3457   \n",
       "4   -0.4442  -0.4675  ...  0.0218  0.0218  0.0218  0.0218  0.3542   \n",
       "..      ...      ...  ...     ...     ...     ...     ...     ...   \n",
       "607 -0.4408  -0.4601  ...  0.2111  0.2111  0.2111  0.2111  0.3322   \n",
       "608 -0.4391  -0.4622  ...  0.1998  0.1998  0.1998  0.1998  0.3908   \n",
       "609 -0.4609  -0.4639  ...  0.1236  0.1236  0.1236  0.1236  0.4054   \n",
       "610 -0.4283  -0.4676  ...  0.2220  0.2220  0.2220  0.2220  0.3571   \n",
       "611 -0.4688  -0.4747  ... -0.0479 -0.0479 -0.0479 -0.0479  0.3632   \n",
       "\n",
       "     LOC100130932  GOLGA8F  GOLGA8G  MIR7-1  overall_survival  \n",
       "0          0.2383  -0.1168  -0.1168     NaN            4047.0  \n",
       "1          0.3783  -0.1271  -0.1271     NaN            4005.0  \n",
       "2          0.1929   0.0066   0.0066     NaN            1474.0  \n",
       "3          0.2445  -0.0071  -0.0071     NaN            1448.0  \n",
       "4          0.3128  -0.0929  -0.0929     NaN             348.0  \n",
       "..            ...      ...      ...     ...               ...  \n",
       "607        0.2746  -0.0284  -0.0284     NaN             883.0  \n",
       "608        0.4161  -0.0702  -0.0702 -0.2598             467.0  \n",
       "609        0.4089  -0.1210  -0.1210 -0.2470             488.0  \n",
       "610        0.2107   0.0495   0.0495     NaN            3287.0  \n",
       "611        0.4219  -0.1819  -0.1819 -0.4183            3256.0  \n",
       "\n",
       "[612 rows x 20108 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#removing status column which is the predictor\n",
    "\n",
    "methylClinicalCommon.loc[:, methylClinicalCommon.columns != 'status']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylClinicalNoAttr = methylClinicalCommon.drop(['attrib_name'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>CNBP</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>MIR7-1</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>count</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>573.000000</td>\n",
       "      <td>612.000000</td>\n",
       "      <td>612.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mean</td>\n",
       "      <td>-0.438864</td>\n",
       "      <td>-0.445767</td>\n",
       "      <td>0.228004</td>\n",
       "      <td>0.383736</td>\n",
       "      <td>-0.183418</td>\n",
       "      <td>-0.485252</td>\n",
       "      <td>0.431599</td>\n",
       "      <td>-0.451468</td>\n",
       "      <td>-0.469642</td>\n",
       "      <td>-0.406238</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054123</td>\n",
       "      <td>0.054123</td>\n",
       "      <td>0.054123</td>\n",
       "      <td>0.390465</td>\n",
       "      <td>0.420950</td>\n",
       "      <td>-0.061686</td>\n",
       "      <td>-0.061686</td>\n",
       "      <td>-0.358640</td>\n",
       "      <td>1172.183007</td>\n",
       "      <td>0.129085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>std</td>\n",
       "      <td>0.042837</td>\n",
       "      <td>0.091793</td>\n",
       "      <td>0.079596</td>\n",
       "      <td>0.041210</td>\n",
       "      <td>0.208140</td>\n",
       "      <td>0.002545</td>\n",
       "      <td>0.021719</td>\n",
       "      <td>0.018178</td>\n",
       "      <td>0.008477</td>\n",
       "      <td>0.078671</td>\n",
       "      <td>...</td>\n",
       "      <td>0.125952</td>\n",
       "      <td>0.125952</td>\n",
       "      <td>0.125952</td>\n",
       "      <td>0.034417</td>\n",
       "      <td>0.045036</td>\n",
       "      <td>0.093724</td>\n",
       "      <td>0.093724</td>\n",
       "      <td>0.074457</td>\n",
       "      <td>982.438069</td>\n",
       "      <td>0.335568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>min</td>\n",
       "      <td>-0.481500</td>\n",
       "      <td>-0.485300</td>\n",
       "      <td>-0.127200</td>\n",
       "      <td>0.076600</td>\n",
       "      <td>-0.488100</td>\n",
       "      <td>-0.490800</td>\n",
       "      <td>0.279500</td>\n",
       "      <td>-0.480800</td>\n",
       "      <td>-0.487600</td>\n",
       "      <td>-0.468100</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.340800</td>\n",
       "      <td>-0.340800</td>\n",
       "      <td>-0.340800</td>\n",
       "      <td>0.003500</td>\n",
       "      <td>0.182700</td>\n",
       "      <td>-0.349800</td>\n",
       "      <td>-0.349800</td>\n",
       "      <td>-0.470200</td>\n",
       "      <td>34.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25%</td>\n",
       "      <td>-0.457100</td>\n",
       "      <td>-0.473025</td>\n",
       "      <td>0.185750</td>\n",
       "      <td>0.365600</td>\n",
       "      <td>-0.395225</td>\n",
       "      <td>-0.487100</td>\n",
       "      <td>0.421200</td>\n",
       "      <td>-0.464225</td>\n",
       "      <td>-0.475100</td>\n",
       "      <td>-0.443700</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039725</td>\n",
       "      <td>-0.039725</td>\n",
       "      <td>-0.039725</td>\n",
       "      <td>0.376700</td>\n",
       "      <td>0.415875</td>\n",
       "      <td>-0.122650</td>\n",
       "      <td>-0.122650</td>\n",
       "      <td>-0.414700</td>\n",
       "      <td>477.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50%</td>\n",
       "      <td>-0.447700</td>\n",
       "      <td>-0.466200</td>\n",
       "      <td>0.234650</td>\n",
       "      <td>0.389350</td>\n",
       "      <td>-0.142850</td>\n",
       "      <td>-0.485400</td>\n",
       "      <td>0.435100</td>\n",
       "      <td>-0.454000</td>\n",
       "      <td>-0.470500</td>\n",
       "      <td>-0.429250</td>\n",
       "      <td>...</td>\n",
       "      <td>0.063500</td>\n",
       "      <td>0.063500</td>\n",
       "      <td>0.063500</td>\n",
       "      <td>0.392350</td>\n",
       "      <td>0.432600</td>\n",
       "      <td>-0.065500</td>\n",
       "      <td>-0.065500</td>\n",
       "      <td>-0.382800</td>\n",
       "      <td>784.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75%</td>\n",
       "      <td>-0.434125</td>\n",
       "      <td>-0.455875</td>\n",
       "      <td>0.279700</td>\n",
       "      <td>0.409200</td>\n",
       "      <td>-0.008700</td>\n",
       "      <td>-0.483675</td>\n",
       "      <td>0.445700</td>\n",
       "      <td>-0.443525</td>\n",
       "      <td>-0.465800</td>\n",
       "      <td>-0.404600</td>\n",
       "      <td>...</td>\n",
       "      <td>0.142825</td>\n",
       "      <td>0.142825</td>\n",
       "      <td>0.142825</td>\n",
       "      <td>0.409225</td>\n",
       "      <td>0.443525</td>\n",
       "      <td>-0.003400</td>\n",
       "      <td>-0.003400</td>\n",
       "      <td>-0.318900</td>\n",
       "      <td>1605.750000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>max</td>\n",
       "      <td>-0.103500</td>\n",
       "      <td>0.421300</td>\n",
       "      <td>0.423800</td>\n",
       "      <td>0.478300</td>\n",
       "      <td>0.242300</td>\n",
       "      <td>-0.474000</td>\n",
       "      <td>0.473500</td>\n",
       "      <td>-0.322400</td>\n",
       "      <td>-0.400100</td>\n",
       "      <td>0.209300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.362300</td>\n",
       "      <td>0.362300</td>\n",
       "      <td>0.362300</td>\n",
       "      <td>0.460000</td>\n",
       "      <td>0.469400</td>\n",
       "      <td>0.344500</td>\n",
       "      <td>0.344500</td>\n",
       "      <td>-0.070900</td>\n",
       "      <td>6593.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 20108 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             RBL2       VDAC3       ACTN1      ATP2A1       SFRP1       NIPA2  \\\n",
       "count  612.000000  612.000000  612.000000  612.000000  612.000000  612.000000   \n",
       "mean    -0.438864   -0.445767    0.228004    0.383736   -0.183418   -0.485252   \n",
       "std      0.042837    0.091793    0.079596    0.041210    0.208140    0.002545   \n",
       "min     -0.481500   -0.485300   -0.127200    0.076600   -0.488100   -0.490800   \n",
       "25%     -0.457100   -0.473025    0.185750    0.365600   -0.395225   -0.487100   \n",
       "50%     -0.447700   -0.466200    0.234650    0.389350   -0.142850   -0.485400   \n",
       "75%     -0.434125   -0.455875    0.279700    0.409200   -0.008700   -0.483675   \n",
       "max     -0.103500    0.421300    0.423800    0.478300    0.242300   -0.474000   \n",
       "\n",
       "           MAN1B1      TSEN34     LRRC16A        CNBP  ...       BAGE2  \\\n",
       "count  612.000000  612.000000  612.000000  612.000000  ...  612.000000   \n",
       "mean     0.431599   -0.451468   -0.469642   -0.406238  ...    0.054123   \n",
       "std      0.021719    0.018178    0.008477    0.078671  ...    0.125952   \n",
       "min      0.279500   -0.480800   -0.487600   -0.468100  ...   -0.340800   \n",
       "25%      0.421200   -0.464225   -0.475100   -0.443700  ...   -0.039725   \n",
       "50%      0.435100   -0.454000   -0.470500   -0.429250  ...    0.063500   \n",
       "75%      0.445700   -0.443525   -0.465800   -0.404600  ...    0.142825   \n",
       "max      0.473500   -0.322400   -0.400100    0.209300  ...    0.362300   \n",
       "\n",
       "            BAGE4       BAGE3      MIR637  LOC100130932     GOLGA8F  \\\n",
       "count  612.000000  612.000000  612.000000    612.000000  612.000000   \n",
       "mean     0.054123    0.054123    0.390465      0.420950   -0.061686   \n",
       "std      0.125952    0.125952    0.034417      0.045036    0.093724   \n",
       "min     -0.340800   -0.340800    0.003500      0.182700   -0.349800   \n",
       "25%     -0.039725   -0.039725    0.376700      0.415875   -0.122650   \n",
       "50%      0.063500    0.063500    0.392350      0.432600   -0.065500   \n",
       "75%      0.142825    0.142825    0.409225      0.443525   -0.003400   \n",
       "max      0.362300    0.362300    0.460000      0.469400    0.344500   \n",
       "\n",
       "          GOLGA8G      MIR7-1  overall_survival      status  \n",
       "count  612.000000  573.000000        612.000000  612.000000  \n",
       "mean    -0.061686   -0.358640       1172.183007    0.129085  \n",
       "std      0.093724    0.074457        982.438069    0.335568  \n",
       "min     -0.349800   -0.470200         34.000000    0.000000  \n",
       "25%     -0.122650   -0.414700        477.000000    0.000000  \n",
       "50%     -0.065500   -0.382800        784.000000    0.000000  \n",
       "75%     -0.003400   -0.318900       1605.750000    0.000000  \n",
       "max      0.344500   -0.070900       6593.000000    1.000000  \n",
       "\n",
       "[8 rows x 20108 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylClinicalNoAttr.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2079"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(methylClinicalNoAttr.columns[methylClinicalNoAttr.isna().any()].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "612"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(methylClinicalNoAttr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NPHP4,4\n",
      "TMEM182,1\n",
      "SMARCC2,6\n",
      "AGA,2\n",
      "OR1F2P,7\n",
      "MTHFD2L,1\n",
      "PANX1,2\n",
      "ROD1,5\n",
      "OAZ1,1\n",
      "PELI2,4\n",
      "FCGR1C,3\n",
      "PRKCH,3\n",
      "STC1,5\n",
      "ARHGEF18,5\n",
      "PLEC1,41\n",
      "NKX6-2,31\n",
      "LTBP4,2\n",
      "UNC84A,9\n",
      "FEV,1\n",
      "GBF1,6\n",
      "ARL6IP5,31\n",
      "COX5A,2\n",
      "PADI3,1\n",
      "C6orf10,9\n",
      "OSBPL1A,3\n",
      "ITPKB,1\n",
      "SNX7,12\n",
      "TGIF1,5\n",
      "CROT,16\n",
      "ATP11A,1\n",
      "MIR1977,8\n",
      "ABI2,25\n",
      "ATXN7,7\n",
      "DGKQ,1\n",
      "OR10G8,1\n",
      "EXOC4,22\n",
      "SEMA6B,7\n",
      "TANK,1\n",
      "LCN1,1\n",
      "DLGAP1,2\n",
      "CD164,20\n",
      "EPHA7,1\n",
      "POP5,5\n",
      "RUNX1T1,2\n",
      "MIR526B,1\n",
      "DACH1,17\n",
      "LOC25845,1\n",
      "CSDE1,5\n",
      "RNF216,23\n",
      "ZNF627,2\n",
      "BRUNOL5,3\n",
      "NUBP1,1\n",
      "C1D,22\n",
      "RIMBP2,21\n",
      "GRM7,1\n",
      "SLITRK3,29\n",
      "C6orf173,6\n",
      "KIAA1688,1\n",
      "C15orf61,1\n",
      "FAM66B,1\n",
      "CHD3,17\n",
      "ZNF407,1\n",
      "OPN3,3\n",
      "METTL6,1\n",
      "CDH10,1\n",
      "SASH1,8\n",
      "RHOBTB2,12\n",
      "FARP1,29\n",
      "MOCS2,2\n",
      "KIAA0146,1\n",
      "ALG1,1\n",
      "ZNF740,12\n",
      "CHD4,3\n",
      "SAMD14,1\n",
      "DLL1,1\n",
      "BTBD12,6\n",
      "TBC1D14,3\n",
      "ARHGEF2,9\n",
      "MAX,11\n",
      "MARVELD3,9\n",
      "NMI,2\n",
      "TMEM136,10\n",
      "UBE3A,3\n",
      "PDLIM5,1\n",
      "RPTN,5\n",
      "MAP2K1,1\n",
      "WDR51A,2\n",
      "LOC440839,1\n",
      "ZNF462,1\n",
      "CUGBP2,34\n",
      "KIAA1026,5\n",
      "MAGI2,9\n",
      "CCDC48,4\n",
      "CUL1,2\n",
      "TOPORS,1\n",
      "MRPL22,5\n",
      "SNCA,33\n",
      "NFKB1,17\n",
      "NDUFA3,2\n",
      "WSB1,1\n",
      "BRMS1L,1\n",
      "UBE2QP1,2\n",
      "C14orf102,7\n",
      "SMAP1,15\n",
      "EPHA5,13\n",
      "PRMT3,1\n",
      "FKBP5,9\n",
      "CCDC109B,1\n",
      "CAMLG,11\n",
      "REEP5,1\n",
      "MYH1,1\n",
      "GRIK2,3\n",
      "KLHL18,1\n",
      "SNTG1,1\n",
      "EPB41L2,29\n",
      "SLC6A9,1\n",
      "LHX8,5\n",
      "FAM5C,3\n",
      "KRR1,4\n",
      "KIF2C,1\n",
      "MRPL3,1\n",
      "LCA5,15\n",
      "MED13L,2\n",
      "ALMS1P,1\n",
      "ELMO1,24\n",
      "WDFY3,20\n",
      "NEB,18\n",
      "PDE7A,1\n",
      "B3GALNT1,1\n",
      "NAE1,1\n",
      "OSGIN2,4\n",
      "GPR78,6\n",
      "BRUNOL4,6\n",
      "NFIA,11\n",
      "ENAH,13\n",
      "RAB5B,13\n",
      "PERP,9\n",
      "C3orf24,4\n",
      "FMO9P,17\n",
      "PPFIA1,1\n",
      "PCDH21,2\n",
      "ZNF527,2\n",
      "BTBD11,10\n",
      "NUDT9,1\n",
      "BUB3,1\n",
      "SUSD4,1\n",
      "C4orf22,14\n",
      "EEF1E1,15\n",
      "FAM160B1,2\n",
      "LHX4,4\n",
      "GLG1,8\n",
      "FAM72B,3\n",
      "LOC388965,1\n",
      "USP46,3\n",
      "MSI2,1\n",
      "WDFY2,3\n",
      "POLR2K,1\n",
      "PASK,1\n",
      "ZC3HAV1,1\n",
      "DSC2,1\n",
      "IMPAD1,1\n",
      "USP32,9\n",
      "C16orf68,1\n",
      "THSD1P,1\n",
      "OR13C8,12\n",
      "FAR2,4\n",
      "ZNF423,4\n",
      "FLJ44606,2\n",
      "DTD1,1\n",
      "KIAA0174,1\n",
      "ANP32A,1\n",
      "KRT6A,5\n",
      "ZC3HC1,2\n",
      "C6orf165,2\n",
      "ZNF629,1\n",
      "ZNF10,6\n",
      "EI24,4\n",
      "SEMA3D,1\n",
      "RPL23P8,1\n",
      "IRX3,5\n",
      "FNBP4,5\n",
      "KIAA0319L,1\n",
      "LASS5,1\n",
      "CLPTM1,1\n",
      "KIAA0430,1\n",
      "GPR177,3\n",
      "RPS6,16\n",
      "CD200R1,3\n",
      "FLJ32810,4\n",
      "STK24,7\n",
      "HIST1H3H,17\n",
      "GAK,1\n",
      "C17orf50,3\n",
      "LOC100130017,6\n",
      "RAB11FIP1,2\n",
      "ZDHHC5,1\n",
      "EYS,6\n",
      "KLF5,1\n",
      "ZNF566,2\n",
      "CDC73,14\n",
      "TGM3,6\n",
      "QKI,5\n",
      "C18orf25,1\n",
      "CAST,10\n",
      "KIF3B,3\n",
      "CCDC41,5\n",
      "LCOR,12\n",
      "NAGA,12\n",
      "Magmas,1\n",
      "EFHD1,5\n",
      "PPP2R5D,1\n",
      "FGF14,15\n",
      "SCFD2,9\n",
      "ZDHHC8P,9\n",
      "TBRG1,6\n",
      "CLDN15,1\n",
      "EVPL,1\n",
      "LOC440910,3\n",
      "SNORD115-8,1\n",
      "LTBP1,17\n",
      "KIAA1033,30\n",
      "CHCHD1,7\n",
      "RUNX1,15\n",
      "COX4I1,9\n",
      "CAMTA1,2\n",
      "HCRTR2,1\n",
      "FAM98A,1\n",
      "NMD3,1\n",
      "ARHGAP11A,14\n",
      "ICA1L,23\n",
      "ZNF274,1\n",
      "STT3A,1\n",
      "LRRC32,1\n",
      "SCRT1,1\n",
      "SLC45A3,1\n",
      "MTMR9,2\n",
      "TLR3,1\n",
      "BSND,1\n",
      "CAV1,5\n",
      "HTR2A,1\n",
      "HSD17B11,1\n",
      "R3HDM1,1\n",
      "NAV1,9\n",
      "C21orf2,1\n",
      "DCUN1D5,1\n",
      "PARN,15\n",
      "METT5D1,35\n",
      "MYOM2,1\n",
      "CD81,1\n",
      "ANXA3,1\n",
      "DUSP26,6\n",
      "SLC9A8,3\n",
      "ETS1,1\n",
      "DRAM2,2\n",
      "EIF4E,36\n",
      "DLAT,5\n",
      "SKAP2,1\n",
      "C17orf97,8\n",
      "VCAN,1\n",
      "PCDH24,7\n",
      "WASL,4\n",
      "MC2R,2\n",
      "ELL2,7\n",
      "RPS27L,2\n",
      "RASL11B,1\n",
      "ISPD,7\n",
      "NHP2L1,1\n",
      "VSIG10,16\n",
      "MMP10,3\n",
      "C22orf41,6\n",
      "HERC4,1\n",
      "KDM3A,4\n",
      "CASQ2,1\n",
      "POT1,27\n",
      "NCKAP1,10\n",
      "RSRC2,3\n",
      "KISS1,1\n",
      "MOG,6\n",
      "ASB4,10\n",
      "MLL,1\n",
      "TFIP11,1\n",
      "CD300C,1\n",
      "MYB,12\n",
      "CHRNB3,3\n",
      "HCCA2,19\n",
      "PPP2R3A,1\n",
      "GTPBP1,1\n",
      "NRG4,37\n",
      "FRYL,16\n",
      "ACSL1,1\n",
      "KIAA1310,9\n",
      "ANKRD28,38\n",
      "N6AMT1,5\n",
      "ABCB9,1\n",
      "LINGO3,1\n",
      "HAS2,3\n",
      "SNORA76,2\n",
      "CFI,4\n",
      "SETD1B,1\n",
      "ALCAM,5\n",
      "APBA2,2\n",
      "ZNF815,1\n",
      "ATP10D,13\n",
      "FAM59B,17\n",
      "ARHGAP28,1\n",
      "YWHAZ,29\n",
      "TMEM167B,4\n",
      "SFRS16,1\n",
      "OCLN,1\n",
      "HCG4P6,5\n",
      "MTL5,1\n",
      "TBCC,3\n",
      "HIST1H1E,23\n",
      "TMTC2,2\n",
      "PHTF1,7\n",
      "CYP2A7,4\n",
      "DMXL1,18\n",
      "OR52K1,1\n",
      "NUP43,1\n",
      "TTLL13,4\n",
      "C8orf41,5\n",
      "MYH10,1\n",
      "KDM3B,1\n",
      "HIST1H4F,7\n",
      "RESP18,23\n",
      "ZNF704,1\n",
      "RBPMS,13\n",
      "CALCRL,6\n",
      "SPAG5,3\n",
      "ZFAND1,4\n",
      "PXMP3,1\n",
      "ZC3H13,3\n",
      "GABRA2,28\n",
      "NKX2-8,1\n",
      "SDC3,5\n",
      "IFI44L,4\n",
      "ATF7IP2,8\n",
      "LOC641518,9\n",
      "SIPA1L2,2\n",
      "ZNF549,5\n",
      "GRIA2,1\n",
      "KIR3DP1,6\n",
      "DDR1,1\n",
      "FRMD4B,4\n",
      "TSPYL4,7\n",
      "PDE1A,1\n",
      "ASCC3,3\n",
      "DSTN,4\n",
      "ZNF30,18\n",
      "C7orf20,2\n",
      "MYH15,1\n",
      "ATN1,2\n",
      "BTLA,6\n",
      "C2orf29,3\n",
      "ZNF235,8\n",
      "SCAI,1\n",
      "NAGS,6\n",
      "SPATS2,7\n",
      "KBTBD6,5\n",
      "FIGN,7\n",
      "TMPRSS8,2\n",
      "TEPP,2\n",
      "CEP70,1\n",
      "BAIAP3,3\n",
      "NLGN1,2\n",
      "SP1,13\n",
      "AIRE,1\n",
      "TLL1,1\n",
      "MED12L,9\n",
      "RAB3IP,4\n",
      "VTRNA1-2,20\n",
      "STAT6,4\n",
      "PET117,1\n",
      "PSMD7,15\n",
      "LOC642846,8\n",
      "CTXN2,23\n",
      "NDST3,4\n",
      "SCN2A,22\n",
      "CCDC162,2\n",
      "FAM71B,4\n",
      "ZNF273,8\n",
      "ZNF530,3\n",
      "ZNF443,1\n",
      "ZDHHC17,1\n",
      "HDGF2,29\n",
      "C20orf108,9\n",
      "EVI5,1\n",
      "KIF11,35\n",
      "PIK3C3,1\n",
      "BAZ1B,17\n",
      "LOC157627,4\n",
      "C5orf37,5\n",
      "FANCA,16\n",
      "INTS6,40\n",
      "POLS,2\n",
      "GTF3C6,3\n",
      "LGI1,1\n",
      "C1orf26,12\n",
      "UNC80,11\n",
      "TMEM167A,17\n",
      "PSKH1,1\n",
      "ZNF20,31\n",
      "HLA-DRB6,24\n",
      "SNORD114-22,24\n",
      "WHSC1L1,29\n",
      "LMNB2,11\n",
      "ARHGEF7,7\n",
      "C20orf95,5\n",
      "NAMPT,7\n",
      "LATS2,1\n",
      "CHUK,2\n",
      "NUP133,1\n",
      "SESN1,7\n",
      "EFCAB4B,17\n",
      "CRLF1,14\n",
      "CRTC2,4\n",
      "HP1BP3,8\n",
      "ATF4,8\n",
      "C11orf60,7\n",
      "ALG9,1\n",
      "REG1B,1\n",
      "TFCP2L1,7\n",
      "PFN1,5\n",
      "OSBPL11,10\n",
      "EIF4G3,34\n",
      "NCEH1,1\n",
      "FNDC3A,9\n",
      "SSBP2,11\n",
      "SLC25A26,1\n",
      "MIR453,1\n",
      "ZBTB24,3\n",
      "PIGG,34\n",
      "NEIL2,2\n",
      "TAF12,1\n",
      "POM121,8\n",
      "ZNF783,4\n",
      "HSF2,9\n",
      "ANO4,3\n",
      "MGST1,7\n",
      "NDFIP1,2\n",
      "ASAH1,2\n",
      "C8orf46,1\n",
      "PRKCQ,1\n",
      "LOC80154,26\n",
      "RBM9,4\n",
      "CHN2,8\n",
      "TMEM97,3\n",
      "ELF1,30\n",
      "PFN2,1\n",
      "MSH6,1\n",
      "PCBD2,3\n",
      "SNORD93,6\n",
      "LYRM7,1\n",
      "NR3C1,4\n",
      "GFI1B,8\n",
      "FES,3\n",
      "TFAMP1,9\n",
      "FANCI,1\n",
      "OR14J1,5\n",
      "LOC157381,1\n",
      "FAM118A,1\n",
      "C6orf35,1\n",
      "SIX4,7\n",
      "YBX1,2\n",
      "SLC16A12,1\n",
      "TCF4,35\n",
      "LOC400043,1\n",
      "BARHL1,14\n",
      "ZNF107,1\n",
      "HAGH,1\n",
      "DEFB133,10\n",
      "PQLC3,1\n",
      "PAWR,5\n",
      "BDH2,9\n",
      "LOC404266,21\n",
      "PSMC6,1\n",
      "RAP1GDS1,3\n",
      "CAP1,1\n",
      "FAAH,1\n",
      "PCNXL2,3\n",
      "USP53,7\n",
      "VAMP3,2\n",
      "RBM27,31\n",
      "DEFA10P,4\n",
      "NCK2,3\n",
      "STK17B,17\n",
      "NDUFB7,20\n",
      "USP24,17\n",
      "SIK1,1\n",
      "LOC100130093,1\n",
      "MIR184,5\n",
      "FERMT2,1\n",
      "TWISTNB,1\n",
      "C12orf67,1\n",
      "INO80,3\n",
      "BATF,14\n",
      "INO80D,3\n",
      "CUEDC1,1\n",
      "TUBB4Q,1\n",
      "XBP1,3\n",
      "SCN5A,18\n",
      "LRRFIP2,5\n",
      "SEMA3C,9\n",
      "PPP4R4,1\n",
      "ATP2A2,2\n",
      "KIF14,24\n",
      "SLC12A6,4\n",
      "IPCEF1,1\n",
      "WDR51B,3\n",
      "PLCH1,3\n",
      "OTUD1,1\n",
      "PGBD2,1\n",
      "TMEM69,7\n",
      "KIAA1432,18\n",
      "C14orf143,23\n",
      "KAT2B,5\n",
      "TMPRSS11D,1\n",
      "UBE2MP1,6\n",
      "BCAP29,1\n",
      "C3orf25,5\n",
      "C14orf179,3\n",
      "FLJ41350,1\n",
      "HIPK1,1\n",
      "SNX9,1\n",
      "SNORD113-5,4\n",
      "RAGE,10\n",
      "ISCA1L,11\n",
      "SOX4,1\n",
      "PHACTR4,11\n",
      "BDP1,1\n",
      "OXSR1,1\n",
      "ASCL1,1\n",
      "RPL9,1\n",
      "NDRG1,9\n",
      "FAM82A2,1\n",
      "C21orf59,2\n",
      "NAA30,33\n",
      "FBXO16,1\n",
      "PITRM1,17\n",
      "NUDT19,1\n",
      "GRID2,2\n",
      "ZNF284,1\n",
      "BMI1,21\n",
      "MTMR10,10\n",
      "MAN1A1,2\n",
      "ASXL2,4\n",
      "LANCL2,2\n",
      "TNFAIP3,1\n",
      "COQ3,2\n",
      "PSMD14,10\n",
      "ARAP2,4\n",
      "CYB5R2,1\n",
      "KIAA0895L,3\n",
      "EID2,3\n",
      "MIR802,4\n",
      "GULP1,1\n",
      "COPS7B,2\n",
      "C7orf42,6\n",
      "NEK5,17\n",
      "ZFP1,10\n",
      "SF3B1,32\n",
      "OXA1L,6\n",
      "MNT,1\n",
      "SLC4A7,32\n",
      "USP10,4\n",
      "CDKN2C,3\n",
      "MIR34A,1\n",
      "SNORD114-6,6\n",
      "CLP1,20\n",
      "PYROXD1,1\n",
      "TMCO6,6\n",
      "MEGF9,1\n",
      "SPRR1A,26\n",
      "PRPF18,19\n",
      "CDKN1A,1\n",
      "RNASE8,7\n",
      "GOLSYN,21\n",
      "LYPD1,2\n",
      "SNIP1,1\n",
      "CLMN,2\n",
      "OTUD7B,14\n",
      "TIMM50,1\n",
      "RNF135,32\n",
      "STX7,7\n",
      "MANBA,8\n",
      "ENOSF1,1\n",
      "ANKRD52,9\n",
      "CCDC9,1\n",
      "CLEC12B,19\n",
      "SOD1,1\n",
      "HECTD1,10\n",
      "MYH4,6\n",
      "SNORD115-7,1\n",
      "SCML4,21\n",
      "MLL5,13\n",
      "LAMP1,2\n",
      "PLEKHM2,1\n",
      "GRP,1\n",
      "DKFZp434L192,1\n",
      "SF3A1,1\n",
      "TMCC1,2\n",
      "MDN1,3\n",
      "IRF6,3\n",
      "GRLF1,1\n",
      "C8orf59,2\n",
      "CMTM6,1\n",
      "YOD1,9\n",
      "OR6C1,33\n",
      "PHKB,1\n",
      "MBNL2,10\n",
      "UGDH,10\n",
      "FGF20,28\n",
      "SMEK1,1\n",
      "DEFB135,1\n",
      "PIK3R3,14\n",
      "SNORD115-39,1\n",
      "CCDC99,5\n",
      "CEP120,4\n",
      "NPTN,14\n",
      "TASP1,1\n",
      "ZNF2,1\n",
      "GALC,8\n",
      "PLIN1,1\n",
      "FAM178A,6\n",
      "USP29,2\n",
      "C2orf3,1\n",
      "LRIT1,11\n",
      "RIMS1,2\n",
      "LGALS3,3\n",
      "PPP2R5A,1\n",
      "HSD17B7,6\n",
      "PRKG2,23\n",
      "MIR17HG,12\n",
      "KIAA1244,2\n",
      "ZNF238,4\n",
      "RSF1,9\n",
      "FAM175B,1\n",
      "FAM69A,9\n",
      "IL20RA,4\n",
      "NDUFA5,16\n",
      "MAS1L,5\n",
      "HIF1AN,2\n",
      "SIK3,2\n",
      "MYBL2,1\n",
      "ARFGEF1,2\n",
      "CNST,4\n",
      "CNOT1,16\n",
      "PCDHB4,32\n",
      "THBD,3\n",
      "NALCN,12\n",
      "MIR548N,39\n",
      "SLC25A36,21\n",
      "TEX14,12\n",
      "ATP6V1E2,1\n",
      "CAMK2A,1\n",
      "FOXK2,9\n",
      "ZNF638,1\n",
      "SIRT1,1\n",
      "CSGALNACT1,1\n",
      "MIR494,23\n",
      "STXBP5,1\n",
      "SLC22A9,4\n",
      "GOLPH3L,5\n",
      "STARD9,1\n",
      "COMMD4,13\n",
      "CEP164,1\n",
      "EPB41L5,1\n",
      "CNTNAP4,2\n",
      "SETD6,1\n",
      "DDX42,2\n",
      "NTSR2,5\n",
      "ATP6V0A2,1\n",
      "MIR216B,22\n",
      "ZNF24,1\n",
      "MIR2117,1\n",
      "SNORD89,4\n",
      "MFN2,1\n",
      "ZSWIM6,5\n",
      "MALAT1,1\n",
      "LOX,2\n",
      "LOC100130872-SPON2,5\n",
      "SOX5,6\n",
      "GCN1L1,21\n",
      "RWDD3,3\n",
      "ARPP-21,3\n",
      "PRC1,1\n",
      "ATG10,26\n",
      "ZNF259,1\n",
      "CASP6,4\n",
      "RNF26,4\n",
      "C10orf84,1\n",
      "RAB11FIP2,1\n",
      "C20orf20,1\n",
      "AGPAT6,1\n",
      "PCDHB7,4\n",
      "SYDE2,2\n",
      "C5orf43,1\n",
      "RRM1,1\n",
      "GSTT1,37\n",
      "TPBG,2\n",
      "SNRPN,10\n",
      "PTPN13,2\n",
      "HSPB3,1\n",
      "DEPDC5,8\n",
      "RAB30,3\n",
      "USE1,1\n",
      "IL1RL1,4\n",
      "PDRG1,1\n",
      "LOC646405,4\n",
      "C9orf103,26\n",
      "MYL6,1\n",
      "SLC7A8,12\n",
      "C11orf91,1\n",
      "SPOPL,1\n",
      "PAK6,1\n",
      "MIR548H4,1\n",
      "BTC,1\n",
      "TP53INP1,1\n",
      "KRT222,5\n",
      "TRPC1,1\n",
      "IFIT5,1\n",
      "SRI,3\n",
      "CLEC12A,6\n",
      "C2orf56,1\n",
      "USP47,35\n",
      "SLC22A10,11\n",
      "H2AFJ,1\n",
      "PROC,1\n",
      "MAP3K1,1\n",
      "SCRG1,1\n",
      "ZNF266,5\n",
      "ZZZ3,2\n",
      "CSGALNACT2,1\n",
      "DEFB122,2\n",
      "MANEA,1\n",
      "HOXA9,23\n",
      "TRPS1,20\n",
      "C21orf99,22\n",
      "STX8,21\n",
      "HEATR5A,13\n",
      "EPHA3,17\n",
      "TSG101,5\n",
      "MAP3K7IP2,3\n",
      "RHOA,1\n",
      "BRUNOL6,1\n",
      "TMEM86A,1\n",
      "TIMM23,3\n",
      "ANAPC5,9\n",
      "PSMD6,1\n",
      "ZNF227,2\n",
      "ZNF829,1\n",
      "EML5,3\n",
      "SUNC1,1\n",
      "CDC37L1,30\n",
      "GPBP1,11\n",
      "SPINK5L3,11\n",
      "SFRS18,2\n",
      "TRNP1,2\n",
      "PRSS37,2\n",
      "ZNF593,1\n",
      "ZBTB41,24\n",
      "LOC100133091,6\n",
      "TBC1D19,3\n",
      "SMEK2,27\n",
      "DLEU2,13\n",
      "PHF12,5\n",
      "MIR30B,17\n",
      "CAB39L,18\n",
      "NDRG3,10\n",
      "DCAF8,7\n",
      "DBT,20\n",
      "PSEN1,6\n",
      "SCN4A,8\n",
      "ERMP1,21\n",
      "SIRT4,24\n",
      "DPP8,1\n",
      "SYNGR1,3\n",
      "C2orf64,1\n",
      "LGI2,2\n",
      "TNFSF13B,6\n",
      "RBM4B,1\n",
      "OR5T1,6\n",
      "DEFB134,1\n",
      "HAVCR1,1\n",
      "DMBT1,24\n",
      "FAM86C,1\n",
      "GSK3B,21\n",
      "GJA1,4\n",
      "PDAP1,1\n",
      "BRPF3,29\n",
      "CDC5L,2\n",
      "GJC1,1\n",
      "C6orf89,5\n",
      "OR6F1,23\n",
      "AURKAIP1,5\n",
      "MIR129-2,22\n",
      "SENP6,11\n",
      "ARF3,1\n",
      "IGSF22,1\n",
      "IDH3A,4\n",
      "FAM108A1,5\n",
      "RGS17,2\n",
      "PSMB6,2\n",
      "AIG1,1\n",
      "MACROD2,7\n",
      "LOC222699,1\n",
      "SNORD115-38,1\n",
      "MADD,1\n",
      "EHD2,1\n",
      "SETDB2,1\n",
      "HAND2,8\n",
      "DYNC2H1,1\n",
      "ACTL6A,1\n",
      "PREX2,1\n",
      "OGDH,1\n",
      "TC2N,3\n",
      "MLX,1\n",
      "NACAP1,1\n",
      "LYPLA2,2\n",
      "C6orf52,13\n",
      "CTAGE5,3\n",
      "IGLON5,4\n",
      "NRARP,18\n",
      "DHRS4L2,5\n",
      "CDK17,8\n",
      "CALM2,11\n",
      "CD74,1\n",
      "PPIL5,2\n",
      "FCRL4,1\n",
      "THUMPD1,1\n",
      "SNORD1C,19\n",
      "CHPF2,1\n",
      "OR2T33,2\n",
      "MIR1260,1\n",
      "YTHDF2,4\n",
      "FAHD1,1\n",
      "SNX26,3\n",
      "BRF2,1\n",
      "ZSCAN22,1\n",
      "FIBIN,1\n",
      "VAMP4,1\n",
      "TRIM3,1\n",
      "RAPGEF6,28\n",
      "CCNDBP1,1\n",
      "BCO2,1\n",
      "ZNF217,27\n",
      "ACAD8,1\n",
      "NUDT13,3\n",
      "HBA1,3\n",
      "BNIP3L,1\n",
      "C14orf106,41\n",
      "MIR548H3,12\n",
      "GOPC,2\n",
      "VSTM1,1\n",
      "C6orf72,1\n",
      "B4GALNT1,1\n",
      "MRFAP1,5\n",
      "FAM12B,1\n",
      "FAM65B,1\n",
      "RBM43,3\n",
      "FEM1A,1\n",
      "OR10K2,1\n",
      "AP1AR,1\n",
      "MTMR15,1\n",
      "ATG16L1,2\n",
      "C20orf202,1\n",
      "OR14A16,3\n",
      "ADAMDEC1,1\n",
      "TRPC4,1\n",
      "COPS8,1\n",
      "SNORD114-28,7\n",
      "LOC729082,2\n",
      "ZNF397,8\n",
      "SDAD1,3\n",
      "SLC38A11,6\n",
      "TMEM165,1\n",
      "METTL11A,4\n",
      "SENP7,3\n",
      "LCP1,1\n",
      "PRRC1,2\n",
      "STMN1,1\n",
      "LOC344967,2\n",
      "CHD1,5\n",
      "BTNL8,18\n",
      "CLINT1,9\n",
      "LOC389458,1\n",
      "C7orf36,10\n",
      "C4orf41,1\n",
      "SFRS13A,1\n",
      "MYOF,4\n",
      "PKP2,5\n",
      "CCNA2,2\n",
      "SUPT6H,1\n",
      "LOC552889,14\n",
      "FLJ40125,17\n",
      "KBTBD10,4\n",
      "TFF1,2\n",
      "FAM63B,1\n",
      "SOCS3,1\n",
      "LOC100128977,1\n",
      "DNAH9,1\n",
      "ZNF649,1\n",
      "SPIRE1,1\n",
      "MAP3K7IP1,12\n",
      "ZBTB39,1\n",
      "FAM138D,1\n",
      "OR2W1,32\n",
      "C3orf23,23\n",
      "LRRC28,2\n",
      "IL18R1,7\n",
      "IL11,1\n",
      "SBDSP,27\n",
      "ADAMTS12,16\n",
      "COPG2,2\n",
      "CDGAP,4\n",
      "AKD1,4\n",
      "C8orf83,10\n",
      "PTPRK,12\n",
      "NOVA1,6\n",
      "RAB18,10\n",
      "BTBD8,32\n",
      "KRTAP1-1,1\n",
      "MC4R,1\n",
      "ACIN1,2\n",
      "INSIG2,1\n",
      "MPP5,1\n",
      "C16orf62,1\n",
      "SP110,4\n",
      "TMED10,1\n",
      "CAND1,2\n",
      "ZNF501,1\n",
      "CMC1,1\n",
      "LIAS,1\n",
      "GC,3\n",
      "HIST4H4,5\n",
      "AQR,1\n",
      "SDCBP,7\n",
      "CASP12,3\n",
      "NSMCE1,6\n",
      "PDGFD,18\n",
      "TMEM156,1\n",
      "FAM18B2,26\n",
      "ARL6,12\n",
      "MIR548G,13\n",
      "ABCF2,1\n",
      "CCDC14,31\n",
      "FOXE1,1\n",
      "GALNT11,2\n",
      "RNF8,2\n",
      "KCTD17,4\n",
      "MIR300,11\n",
      "6-Mar,1\n",
      "DOPEY1,1\n",
      "FAM179A,1\n",
      "LRRC6,8\n",
      "FAM49A,1\n",
      "PSMD12,1\n",
      "SLC24A2,2\n",
      "LMF2,16\n",
      "ANUBL1,5\n",
      "MIR124-2,12\n",
      "ATAD5,1\n",
      "ZNF248,1\n",
      "MIR192,1\n",
      "DIP2A,7\n",
      "ATPGD1,1\n",
      "SYVN1,1\n",
      "C17orf58,27\n",
      "LYSMD3,15\n",
      "ADK,1\n",
      "TBL2,15\n",
      "ZNF684,3\n",
      "DERL3,2\n",
      "ABL2,1\n",
      "NPS,16\n",
      "NKAIN4,3\n",
      "RASA1,1\n",
      "SNX6,23\n",
      "CDC45L,16\n",
      "MIR518A2,11\n",
      "C19orf62,4\n",
      "RAB33B,15\n",
      "SFXN5,8\n",
      "SNX13,5\n",
      "LOC144776,2\n",
      "EPCAM,5\n",
      "CASD1,12\n",
      "GNMT,7\n",
      "LOC285550,3\n",
      "LOC731275,5\n",
      "VPS4B,1\n",
      "ZBTB3,2\n",
      "CRABP2,1\n",
      "FEM1C,1\n",
      "MAST2,3\n",
      "PDE11A,1\n",
      "CASC1,3\n",
      "LRRC66,32\n",
      "MRPS27,37\n",
      "OR8H3,15\n",
      "LIN28,5\n",
      "NDE1,1\n",
      "JAK2,1\n",
      "POPDC3,3\n",
      "ERCC8,3\n",
      "LILRB4,1\n",
      "SLC22A1,2\n",
      "CEP152,1\n",
      "IFNAR2,5\n",
      "CDC42EP3,1\n",
      "CFD,1\n",
      "MEP1A,1\n",
      "OR5L2,5\n",
      "C10orf88,1\n",
      "ATP9B,5\n",
      "BMPR1B,1\n",
      "OR4K5,8\n",
      "CD1B,4\n",
      "GBP4,2\n",
      "C1orf77,11\n",
      "ANXA5,9\n",
      "SLC6A15,1\n",
      "KRTAP13-2,1\n",
      "GRPEL1,27\n",
      "PRPF40A,14\n",
      "OR5H1,3\n",
      "MIR21,7\n",
      "SEC31A,1\n",
      "FBXO25,16\n",
      "DNAJC9,4\n",
      "ZDHHC11,1\n",
      "IL22,3\n",
      "ATP13A3,14\n",
      "UBE2L3,17\n",
      "DUSP19,16\n",
      "C18orf32,18\n",
      "MIR493,1\n",
      "STAM,3\n",
      "GOLM1,18\n",
      "IDH2,5\n",
      "ZNF250,2\n",
      "NPVF,1\n",
      "NOMO3,1\n",
      "LRRC55,1\n",
      "ARL17A,30\n",
      "GLRX2,4\n",
      "SNORD113-2,12\n",
      "C5orf30,14\n",
      "LOC100190940,4\n",
      "TRAF6,2\n",
      "ATRN,7\n",
      "RPL32P3,3\n",
      "MBOAT2,29\n",
      "GABRA6,30\n",
      "FIP1L1,20\n",
      "RMST,1\n",
      "GRK4,1\n",
      "NAV3,2\n",
      "FUT10,5\n",
      "EIF3H,5\n",
      "WNK1,10\n",
      "RSL1D1,15\n",
      "UNK,2\n",
      "DEFB131,4\n",
      "GUCY2G,42\n",
      "MAP2K4,1\n",
      "SNORD115-48,1\n",
      "TPH2,5\n",
      "HISPPD1,18\n",
      "GMNN,1\n",
      "LARS,8\n",
      "MIR568,6\n",
      "CISD3,2\n",
      "MYCT1,9\n",
      "SKIL,12\n",
      "PPP1R9A,10\n",
      "C13orf18,1\n",
      "UQCRH,1\n",
      "PHF20L1,1\n",
      "TCEB3,35\n",
      "RORB,1\n",
      "FOLH1B,2\n",
      "EDEM3,2\n",
      "KITLG,5\n",
      "CAP2,1\n",
      "ABCC11,2\n",
      "LOC148189,1\n",
      "ALB,34\n",
      "ZMYM6,2\n",
      "HCP5,15\n",
      "ZNF221,1\n",
      "RCOR3,2\n",
      "OR51L1,4\n",
      "RICS,24\n",
      "PRKCI,1\n",
      "OR4K1,1\n",
      "TMEM98,4\n",
      "SNORD113-9,1\n",
      "C4orf33,1\n",
      "ATXN10,14\n",
      "RAB3C,17\n",
      "OGFRL1,1\n",
      "FANCE,1\n",
      "TMPRSS11E,16\n",
      "FICD,1\n",
      "RNF146,7\n",
      "LOC729384,13\n",
      "ALKBH8,5\n",
      "LOC388796,1\n",
      "EYA1,3\n",
      "CRNKL1,4\n",
      "DRGX,8\n",
      "C11orf61,4\n",
      "F13B,12\n",
      "TAS2R10,23\n",
      "CDON,29\n",
      "ZNF445,1\n",
      "FLJ36031,7\n",
      "OR5D14,1\n",
      "ITFG1,1\n",
      "TXNDC12,1\n",
      "SAP30,2\n",
      "C7orf38,1\n",
      "SCN3A,22\n",
      "RNF149,1\n",
      "OSTN,1\n",
      "BZW1,11\n",
      "CCDC148,4\n",
      "C1orf53,6\n",
      "OR6Y1,2\n",
      "GEMIN6,1\n",
      "MAP9,28\n",
      "SNX24,10\n",
      "C19orf2,1\n",
      "ATL2,13\n",
      "OR1A2,2\n",
      "ARMC1,1\n",
      "SCAMP2,10\n",
      "GPBP1L1,1\n",
      "LOC100101938,8\n",
      "MSGN1,1\n",
      "PCF11,16\n",
      "GALNTL6,2\n",
      "MIR146A,1\n",
      "OR52M1,10\n",
      "BCL2L15,1\n",
      "N4BP2L2,3\n",
      "MRPL40,1\n",
      "C6orf58,7\n",
      "TMEM2,1\n",
      "SAP30L,1\n",
      "FAM186A,1\n",
      "ATF6,15\n",
      "CDS1,4\n",
      "RIMKLB,1\n",
      "ZNF367,1\n",
      "MIR29B2,8\n",
      "FAM131C,1\n",
      "ZC3H14,23\n",
      "GABRG1,2\n",
      "TRAM1,3\n",
      "CGGBP1,1\n",
      "ART3,1\n",
      "KTELC1,7\n",
      "LOC100133161,1\n",
      "CD300E,1\n",
      "MIER3,10\n",
      "MTMR3,3\n",
      "SIGLEC5,34\n",
      "NCKIPSD,1\n",
      "C10orf104,27\n",
      "RICTOR,2\n",
      "ZNF281,21\n",
      "USP25,15\n",
      "ETAA1,1\n",
      "CUGBP1,1\n",
      "PCDHB15,2\n",
      "FAM12A,1\n",
      "C2orf86,8\n",
      "MORF4L1,19\n",
      "DSG2,12\n",
      "MIR187,13\n",
      "PHC3,40\n",
      "INADL,3\n",
      "PSG11,1\n",
      "CD163,15\n",
      "MAP3K4,14\n",
      "DYNC1I2,1\n",
      "NBR1,4\n",
      "PSMG1,2\n",
      "CBFA2T2,13\n",
      "CCDC102B,3\n",
      "GLTPD2,1\n",
      "ZNF25,2\n",
      "P11,5\n",
      "ARL6IP6,2\n",
      "OR1C1,1\n",
      "GANAB,1\n",
      "NFE2L3,1\n",
      "MIPEP,1\n",
      "BIRC6,14\n",
      "DYRK3,4\n",
      "YIPF7,11\n",
      "SLC17A8,17\n",
      "C6orf103,7\n",
      "GLYATL2,5\n",
      "SLC17A1,7\n",
      "MACC1,4\n",
      "ALKBH4,1\n",
      "LOC441208,31\n",
      "TFAM,1\n",
      "EXOSC2,1\n",
      "FAM134A,2\n",
      "GREM1,2\n",
      "LVRN,2\n",
      "MIER1,1\n",
      "NPHP3,1\n",
      "SLCO1B3,2\n",
      "RPL34,1\n",
      "ZDHHC4,1\n",
      "TSEN15,1\n",
      "KLKBL4,1\n",
      "SF3B4,1\n",
      "NUP98,31\n",
      "SNORD115-6,1\n",
      "RCN2,10\n",
      "IGJ,6\n",
      "ZNF618,25\n",
      "CPEB2,21\n",
      "RPL7,1\n",
      "KRTAP20-2,6\n",
      "GRPEL2,4\n",
      "CDKN2A,1\n",
      "GSR,2\n",
      "PSG7,1\n",
      "SECISBP2,22\n",
      "DIRC1,6\n",
      "EXTL2,1\n",
      "TP53TG1,16\n",
      "INTS3,1\n",
      "AQP12A,4\n",
      "ATP6V0A1,1\n",
      "LIN28B,24\n",
      "OR4C15,17\n",
      "CASP1,1\n",
      "PRDM10,6\n",
      "PLG,5\n",
      "ASF1A,3\n",
      "PRRG2,1\n",
      "CRISP1,1\n",
      "STX3,1\n",
      "TMX2,1\n",
      "SMC4,28\n",
      "LOC391322,5\n",
      "CALB2,11\n",
      "C6orf225,1\n",
      "REV1,10\n",
      "ITGB8,1\n",
      "SNORD113-6,9\n",
      "GSX2,1\n",
      "HIST1H2AI,4\n",
      "MIR137,6\n",
      "TTC28,1\n",
      "ZNF576,1\n",
      "FRRS1,7\n",
      "BTF3L4,1\n",
      "OR4S2,1\n",
      "ASB7,1\n",
      "C15orf54,1\n",
      "PHF10,1\n",
      "OR5K1,1\n",
      "ZNF708,6\n",
      "ABCA6,19\n",
      "IL8,3\n",
      "DHX15,20\n",
      "GPR89B,5\n",
      "OR5K4,21\n",
      "UBN1,15\n",
      "TYW1B,27\n",
      "HSD17B4,1\n",
      "DEFB121,1\n",
      "CDH12,1\n",
      "CELA2A,1\n",
      "SNRPC,2\n",
      "MIR548A2,7\n",
      "ZNRF2,1\n",
      "SEMG1,2\n",
      "MIR759,6\n",
      "ODAM,3\n",
      "MIR519D,3\n",
      "RPSAP52,6\n",
      "FLJ41856,5\n",
      "DYTN,1\n",
      "LOC93432,2\n",
      "OR4S1,1\n",
      "COL24A1,6\n",
      "PHF5A,1\n",
      "TSC1,2\n",
      "FREQ,1\n",
      "TXN,6\n",
      "GTPBP10,3\n",
      "GUF1,3\n",
      "SC5DL,1\n",
      "FBXO8,1\n",
      "RIT2,1\n",
      "MARK4,1\n",
      "ATP5I,1\n",
      "CLRN3,2\n",
      "AP3M1,1\n",
      "LOC100130581,1\n",
      "SNORD113-4,8\n",
      "ZNF434,3\n",
      "GNAT3,1\n",
      "OR2T11,15\n",
      "CREG2,2\n",
      "MXD1,23\n",
      "KLF17,1\n",
      "GOLGA6B,7\n",
      "PKD2L2,25\n",
      "CNGA1,1\n",
      "TSPO2,2\n",
      "LOC93622,22\n",
      "POSTN,1\n",
      "AHR,4\n",
      "SNORD114-20,2\n",
      "TMEM19,3\n",
      "SLC26A3,3\n",
      "CYP2J2,1\n",
      "OR52B2,1\n",
      "MIR1179,6\n",
      "OR5B12,4\n",
      "NT5C3,7\n",
      "ANKRD30A,31\n",
      "OR1Q1,1\n",
      "TXNDC8,1\n",
      "RPL23,30\n",
      "GYPB,1\n",
      "GUSBL1,3\n",
      "SNORD114-13,27\n",
      "TMEM104,1\n",
      "ANKRD40,3\n",
      "MIR1185-1,23\n",
      "FLJ13197,1\n",
      "PRPF39,19\n",
      "SUPT16H,1\n",
      "TEP1,5\n",
      "OR4F4,7\n",
      "DNAJB4,11\n",
      "MPP7,5\n",
      "DDX5,3\n",
      "JMJD1C,2\n",
      "FCRL3,24\n",
      "TMEM9,1\n",
      "SNORD114-31,28\n",
      "SGOL2,22\n",
      "C7orf45,28\n",
      "PLA2G12A,1\n",
      "ADNP2,10\n",
      "BEST3,2\n",
      "AMPD1,1\n",
      "KRTAP4-8,3\n",
      "NEUROD6,1\n",
      "OR8H2,17\n",
      "LMAN1,22\n",
      "SPAG11A,4\n",
      "TIGD6,1\n",
      "RMND1,1\n",
      "CASP4,10\n",
      "LOC284379,7\n",
      "OR5B17,4\n",
      "MTX3,1\n",
      "OR52E4,3\n",
      "C11orf52,3\n",
      "PUM2,3\n",
      "13-Sep,25\n",
      "C6orf217,2\n",
      "KLKB1,9\n",
      "OR6B3,2\n",
      "PECI,3\n",
      "KCTD19,2\n",
      "HEPACAM,5\n",
      "NOX3,1\n",
      "PDCL,1\n",
      "C9orf100,4\n",
      "HIST1H2AB,1\n",
      "PLAC1L,1\n",
      "COLEC12,15\n",
      "SNORD114-29,17\n",
      "LRRC3,1\n",
      "APOL2,5\n",
      "SCG2,1\n",
      "MCM9,7\n",
      "IPW,1\n",
      "WEE1,13\n",
      "C8orf44,7\n",
      "SLC38A2,2\n",
      "YES1,1\n",
      "RP9P,10\n",
      "KRTAP2-2,4\n",
      "LNX2,31\n",
      "CFH,8\n",
      "TMEM133,1\n",
      "EML2,1\n",
      "TMPRSS11B,1\n",
      "TMEM232,2\n",
      "ZNF285A,9\n",
      "C6orf48,30\n",
      "SLC26A7,2\n",
      "ARHGAP19,12\n",
      "TNP2,1\n",
      "OR6X1,1\n",
      "PSG4,1\n",
      "MIR548F5,32\n",
      "KRTAP2-1,1\n",
      "PTAR1,2\n",
      "KIAA0776,4\n",
      "MIR1323,2\n",
      "KIFAP3,2\n",
      "UGT2A1,8\n",
      "RNU6ATAC,11\n",
      "CDH9,1\n",
      "C11orf82,1\n",
      "SPPL2A,1\n",
      "KIAA1712,1\n",
      "NDUFV2,3\n",
      "DDI2,6\n",
      "LOC727924,1\n",
      "ZNF329,4\n",
      "MIR516B1,11\n",
      "CFHR5,5\n",
      "FAM153A,1\n",
      "OR5AS1,11\n",
      "OR4N4,10\n",
      "DNAJC12,12\n",
      "VPS13A,1\n",
      "PNLIPRP3,2\n",
      "SPTA1,6\n",
      "C3orf63,12\n",
      "TMX1,1\n",
      "SULT1B1,9\n",
      "MIR200B,3\n",
      "CHRM5,3\n",
      "SMR3B,3\n",
      "FLJ26850,2\n",
      "DGCR10,1\n",
      "ANKRD31,8\n",
      "TMEM50A,4\n",
      "CFHR1,31\n",
      "GKN1,1\n",
      "NOC3L,15\n",
      "ATPIF1,1\n",
      "C13orf30,1\n",
      "C17orf57,4\n",
      "CYP2B6,3\n",
      "TTC23,3\n",
      "SNORD114-10,18\n",
      "FAM159B,6\n",
      "CD163L1,2\n",
      "RNASE3,1\n",
      "CRYBG3,1\n",
      "OR51A2,1\n",
      "LOC389765,1\n",
      "ZNF174,3\n",
      "EPGN,6\n",
      "C18orf10,7\n",
      "CNTN3,4\n",
      "HMBOX1,1\n",
      "EPHX2,1\n",
      "NAIP,1\n",
      "OR52E8,6\n",
      "C11orf40,23\n",
      "OR8K3,1\n",
      "CASC2,3\n",
      "DYDC2,4\n",
      "SAMD8,1\n",
      "HAUS6,2\n",
      "HSD17B14,1\n",
      "HIST1H2BE,1\n",
      "YEATS4,7\n",
      "OR6C74,2\n",
      "SCYL1,1\n",
      "C11orf36,2\n",
      "CHIA,2\n",
      "RAD23B,1\n",
      "ACO2,1\n",
      "CHTF8,1\n",
      "VIP,1\n",
      "HTN3,4\n",
      "GPR15,1\n",
      "OR7G1,1\n",
      "HBII-52-46,5\n",
      "TAS2R8,1\n",
      "TOB1,25\n",
      "TRIM64B,8\n",
      "FGL1,1\n",
      "TBP,3\n",
      "POTEH,1\n",
      "ARRDC1,1\n",
      "ZNF876P,3\n",
      "CLEC2B,14\n",
      "TFPI,3\n",
      "CCDC47,2\n",
      "CXADRP2,1\n",
      "LOC100287834,18\n",
      "SAMD9,1\n",
      "EEF1AL7,17\n",
      "CENPN,22\n",
      "TMEM99,18\n",
      "CSN1S1,6\n",
      "GLIPR1,2\n",
      "C21orf66,2\n",
      "PRR24,1\n",
      "TNFAIP1,1\n",
      "MIR1283-1,1\n",
      "ILF2,16\n",
      "UBAP2L,24\n",
      "XRCC4,17\n",
      "LOC400759,21\n",
      "IL2,2\n",
      "SPAG11B,21\n",
      "LOC100302652,7\n",
      "EXOSC7,5\n",
      "SNORD115-4,1\n",
      "TAS2R38,1\n",
      "LOC541471,19\n",
      "ALPP,2\n",
      "PRCP,1\n",
      "MIR1279,4\n",
      "OR4A15,1\n",
      "KRTAP19-2,1\n",
      "TMPO,17\n",
      "KERA,5\n",
      "SLC38A6,37\n",
      "CREG1,15\n",
      "C14orf133,1\n",
      "FAM75A6,24\n",
      "KLRC4,1\n",
      "FUT8,24\n",
      "SNORD114-14,24\n",
      "TNP1,2\n",
      "MIR133B,1\n",
      "GLYATL3,3\n",
      "VN1R2,1\n",
      "OR6C3,1\n",
      "MYL1,1\n",
      "UBA2,1\n",
      "ZNF22,1\n",
      "POU1F1,8\n",
      "OR6C76,16\n",
      "AGR3,2\n",
      "DENND4C,27\n",
      "PHF7,1\n",
      "NBPF14,1\n",
      "GSTTP1,5\n",
      "OR8H1,34\n",
      "OR8J3,9\n",
      "CCDC107,13\n",
      "SERPINI2,8\n",
      "FAM90A20,1\n",
      "FLJ39582,10\n",
      "LOC286367,9\n",
      "GNRHR,19\n",
      "PTPRQ,19\n",
      "MYH8,2\n",
      "MIR1305,2\n",
      "GBP1,2\n",
      "INTS9,1\n",
      "LOC100128542,1\n",
      "OR10A7,27\n",
      "BCL8,6\n",
      "C1orf25,12\n",
      "CHRFAM7A,15\n",
      "MIR2054,2\n",
      "BZW2,2\n",
      "NSUN5B,1\n",
      "GVIN1,2\n",
      "PPBPL1,2\n",
      "MED19,1\n",
      "SNRPD1,2\n",
      "TMEM87A,2\n",
      "INTS12,2\n",
      "KDM5A,1\n",
      "S100PBP,7\n",
      "SNORD114-16,26\n",
      "OR8U8,36\n",
      "OR51V1,5\n",
      "CLEC4E,1\n",
      "C6orf218,6\n",
      "OR5D16,21\n",
      "PWRN1,9\n",
      "C6orf211,1\n",
      "SNORD115-10,3\n",
      "DNAI1,1\n",
      "OR5P2,6\n",
      "ZCCHC7,1\n",
      "SNAPC3,6\n",
      "ZNF189,4\n",
      "PARG,1\n",
      "RBM18,1\n",
      "EPYC,7\n",
      "KRTAP3-2,6\n",
      "SDF2,1\n",
      "ZNF326,5\n",
      "C9orf79,1\n",
      "MIR217,2\n",
      "GBP2,4\n",
      "SNORD114-8,2\n",
      "NIPSNAP3A,7\n",
      "MRRF,4\n",
      "RWDD4A,1\n",
      "FRMPD1,21\n",
      "LOC201651,4\n",
      "GOLGA2,38\n",
      "OR5W2,1\n",
      "CDH19,5\n",
      "SNHG5,11\n",
      "CPT1C,2\n",
      "EFCAB9,7\n",
      "SGTB,1\n",
      "SNORD114-5,26\n",
      "CYLC2,12\n",
      "LOC646214,2\n",
      "ARFIP1,1\n",
      "SENP8,1\n",
      "OR2A25,3\n",
      "GEN1,1\n",
      "LOC100272217,4\n",
      "OR7A10,4\n",
      "TGS1,15\n",
      "CSTA,1\n",
      "OR4A16,1\n",
      "KDM4D,4\n",
      "LRRIQ3,6\n",
      "LRIT3,1\n",
      "WDR87,7\n",
      "TRIM77,2\n",
      "MBD2,2\n",
      "ACP1,1\n",
      "FAM35B2,2\n",
      "C12orf24,7\n",
      "C10orf57,1\n",
      "SDCCAG10,12\n",
      "GLUD1,5\n",
      "OR11H12,6\n",
      "SNORD56B,2\n",
      "OR5AN1,2\n",
      "DCAF17,9\n",
      "GBP3,6\n",
      "GYPE,1\n",
      "MIR30A,9\n",
      "MYO9A,1\n",
      "TMEM116,1\n",
      "MIR1283-2,1\n",
      "MTTP,4\n",
      "KRTAP4-5,2\n",
      "OR52J3,2\n",
      "KRTAP19-6,1\n",
      "ATG12,1\n",
      "OR10AG1,1\n",
      "HIST1H3J,1\n",
      "C12orf50,14\n",
      "SNORD113-7,34\n",
      "CBWD1,26\n",
      "LOC285735,2\n",
      "C11orf67,9\n",
      "MIR519A2,2\n",
      "MTERFD1,3\n",
      "SULT1E1,1\n",
      "OR4C46,12\n",
      "DKFZP434L187,1\n",
      "OR4K14,6\n",
      "HEATR5B,5\n",
      "FLJ16779,3\n",
      "ZNF705G,4\n",
      "MIR1973,27\n",
      "C6orf201,3\n",
      "ZNF826,1\n",
      "MIR548F1,3\n",
      "OR2W3,5\n",
      "TEX15,3\n",
      "MORG1,1\n",
      "THYN1,1\n",
      "OR5T2,3\n",
      "MAP3K2,1\n",
      "NFKBIL1,25\n",
      "OR4L1,1\n",
      "HNF4G,11\n",
      "OR8J1,28\n",
      "OR4A5,13\n",
      "OR8K1,3\n",
      "KBTBD3,3\n",
      "OR10Z1,2\n",
      "KIAA1328,7\n",
      "MIR181D,6\n",
      "ZBTB11,26\n",
      "LOC286016,1\n",
      "HTN1,7\n",
      "CSN2,10\n",
      "LOC284441,1\n",
      "UGT2B15,5\n",
      "AMY2A,5\n",
      "TRIM48,1\n",
      "PLSCR5,10\n",
      "OR2T10,33\n",
      "HSPE1,1\n",
      "PAK1IP1,8\n",
      "KRTAP21-2,6\n",
      "FLJ41562,1\n",
      "SNORD114-17,14\n",
      "LINS1,1\n",
      "KLRC2,14\n",
      "SERPINB4,1\n",
      "MIR181C,6\n",
      "C1orf180,1\n",
      "LOC100216001,5\n",
      "C19orf57,1\n",
      "LOC644669,31\n",
      "C19orf42,22\n",
      "SNORD116-21,1\n",
      "OR4P4,6\n",
      "ZNF479,25\n",
      "C21orf131,1\n",
      "SNORD114-25,10\n",
      "SCYL2,19\n",
      "CCDC45,9\n",
      "SPATA1,14\n",
      "KDELC1,2\n",
      "MIR1251,1\n",
      "MIR1537,1\n",
      "BZW1L1,11\n",
      "PCDHGA5,12\n",
      "PCDHGA2,3\n",
      "MIR938,15\n",
      "NME2,1\n",
      "PCDHGC3,1\n",
      "PCDHGB6,1\n",
      "SNORD84,23\n",
      "SNORD76,16\n",
      "SNORD75,16\n",
      "LOC100288730,5\n",
      "MRPL43,1\n",
      "PRH1,12\n",
      "SNORA7A,1\n",
      "PIH1D2,1\n",
      "ZFP91-CNTF,1\n",
      "PCDHA10,8\n",
      "PCDHA5,34\n",
      "MIR1915,7\n",
      "LPAR6,11\n",
      "PEX13,6\n",
      "SNORD116-17,3\n",
      "RSPH10B,2\n",
      "SNORD43,1\n",
      "SNORD27,17\n",
      "SNHG1,1\n",
      "SNORD28,17\n",
      "SNORD25,1\n",
      "SNORD26,17\n",
      "RNU5E,5\n",
      "RNU5D,5\n",
      "SNORD115-11,8\n",
      "SNORD115-15,10\n",
      "SNORD115-21,10\n",
      "SNORD115-43,8\n",
      "SNORD115-36,8\n",
      "SNORD115-29,8\n",
      "LOC285456,1\n",
      "MIR1292,10\n",
      "MIR1469,30\n",
      "TRIM6-TRIM34,15\n",
      "EAF2,2\n",
      "JRKL,14\n",
      "C4B,9\n",
      "FLRT3,7\n",
      "EID1,2\n",
      "MIR484,1\n",
      "MIR191,8\n",
      "MIR425,8\n",
      "ZFP91,1\n",
      "MIR1234,2\n",
      "CDC2,4\n",
      "C1QTNF5,2\n",
      "GTF2H2D,5\n",
      "MIR125B2,1\n",
      "DYDC1,4\n",
      "KRTAP9-9,1\n",
      "OR4M2,1\n",
      "C2orf60,4\n",
      "SNORD23,1\n",
      "MIR124-1,13\n",
      "SNORD103A,3\n",
      "SNORD100,1\n",
      "SNORA33,1\n",
      "SNORD104,9\n",
      "SNORD115-5,3\n",
      "SNORD115-12,3\n",
      "SNORD115-9,3\n",
      "SNORD83B,7\n",
      "LOC145663,2\n",
      "CATSPER2P1,1\n",
      "DBF4,2\n",
      "C14orf45,1\n",
      "MIR597,1\n",
      "MGC26647,1\n",
      "MIR638,1\n",
      "MIR218-1,1\n",
      "KRTAP9-3,4\n",
      "HIST2H2AA4,1\n",
      "MIR1908,1\n",
      "MIR636,1\n",
      "SNORD34,4\n",
      "RPL13AP5,30\n",
      "SNORD35A,25\n",
      "LOC219347,1\n",
      "CREB3,3\n",
      "MTPN,26\n",
      "SNORD88B,1\n",
      "GHRLOS,3\n",
      "MIRLET7I,8\n",
      "MIR1181,5\n",
      "LOC100130148,1\n",
      "SNORD101,1\n",
      "LOC492303,6\n",
      "CHMP1B,13\n",
      "NOL8,1\n",
      "PPWD1,24\n",
      "MIR762,1\n",
      "SNORD114-18,5\n",
      "SNORD114-19,5\n",
      "MIR449B,11\n",
      "MIR449A,15\n",
      "SNHG9,1\n",
      "SNORD114-23,26\n",
      "SNORD114-24,21\n",
      "SNORD58A,1\n",
      "SNORD58B,1\n",
      "RBM12,1\n",
      "LOC100289511,2\n",
      "MIR485,1\n",
      "MIR548F3,9\n",
      "SNORD2,3\n",
      "CEP290,17\n",
      "LOC100129794,3\n",
      "C16orf10,1\n",
      "C10orf25,1\n",
      "LOC728661,21\n",
      "MIR34B,17\n",
      "MIR34C,17\n",
      "MIR654,3\n",
      "MIR376B,3\n",
      "MIR128-1,1\n",
      "SNORD110,2\n",
      "HIST1H2AA,1\n",
      "HIST1H2BL,17\n",
      "SNORD117,3\n",
      "SNORD29,15\n",
      "KIR2DL1,10\n",
      "LOC100128191,17\n",
      "IFT80,1\n",
      "LOC100130557,7\n",
      "COX15,1\n",
      "RMRP,13\n",
      "KRT10,18\n",
      "SNURF,10\n",
      "SDHD,1\n",
      "SNORD36B,3\n",
      "SNORD24,3\n",
      "SNORD50A,26\n",
      "SNORD12C,2\n",
      "FAM90A14,3\n",
      "FAM90A13,3\n",
      "TSPAN19,3\n",
      "LRRIQ1,3\n",
      "MRGPRG,1\n",
      "ARGFXP2,3\n",
      "MIR1306,9\n",
      "SNORD114-26,10\n",
      "SNORD52,1\n",
      "MIR598,1\n",
      "TRIM34,15\n",
      "SNORA70B,4\n",
      "RG9MTD2,4\n",
      "MIR1276,4\n",
      "C21orf49,2\n",
      "MIR1539,19\n",
      "SNORD115-19,1\n",
      "SNORD115-17,1\n",
      "SNORD115-18,1\n",
      "LOC100128003,1\n",
      "UBA5,2\n",
      "HAS2AS,3\n",
      "RNU12,13\n",
      "MIR103-2,4\n",
      "HNRPA1L-2,1\n",
      "MIR592,4\n",
      "MIR125B1,21\n",
      "FLJ31306,5\n",
      "MIR1322,1\n",
      "UGT2A2,2\n",
      "WDR78,1\n",
      "MIR19A,15\n",
      "MIR92A1,5\n",
      "MIR20A,15\n",
      "MIR17,15\n",
      "MIR19B1,15\n",
      "MIR18A,15\n",
      "C14orf178,17\n",
      "C1orf133,1\n",
      "SNORD48,45\n",
      "MIR136,1\n",
      "MIR432,1\n",
      "SNORD114-21,5\n",
      "ZNF568,1\n",
      "MIR617,1\n",
      "MIR22,1\n",
      "MIR618,2\n",
      "LOC100133315,1\n",
      "FANCD2,1\n",
      "MIR574,4\n",
      "MIR346,2\n",
      "SNORD33,4\n",
      "MIR449C,1\n",
      "MIR621,1\n",
      "SNORD66,22\n",
      "PLGLB1,1\n",
      "MIR1296,6\n",
      "NCAPG,15\n",
      "15-Sep,7\n",
      "MIR302A,9\n",
      "MIR302D,9\n",
      "MIR367,9\n",
      "FLJ35776,2\n",
      "KCNRG,3\n",
      "C1orf43,24\n",
      "SNORD30,1\n",
      "SNORD31,1\n",
      "SCARNA13,9\n",
      "KCTD4,1\n",
      "SNORD78,26\n",
      "SNORD44,26\n",
      "SNORD80,20\n",
      "SNORD79,20\n",
      "SCARNA6,2\n",
      "LRRC37A2,20\n",
      "SNORD114-15,23\n",
      "MIR489,1\n",
      "BLID,5\n",
      "SNORA54,1\n",
      "SNORD59A,4\n",
      "SNORD50B,23\n",
      "COL8A1,4\n",
      "SNORD124,1\n",
      "MIR320A,3\n",
      "MIR26B,4\n",
      "MIR103-1,1\n",
      "SNORD70,4\n",
      "MIR632,24\n",
      "SNORD6,11\n",
      "ANKRD18A,1\n",
      "SNORD77,16\n",
      "MIR549,3\n",
      "RPS28,2\n",
      "MIRLET7A2,16\n",
      "MIR609,2\n",
      "MIR515-2,4\n",
      "MIR515-1,4\n",
      "MIR941-1,8\n",
      "MIR941-3,8\n",
      "MIR941-2,8\n",
      "OR5M9,4\n",
      "MIR423,22\n",
      "MIR30E,6\n",
      "NCRNA00110,2\n",
      "MIR107,19\n",
      "C14orf33,2\n",
      "LOC728190,3\n",
      "LOC439994,3\n",
      "SNORD114-12,23\n",
      "RPL35A,1\n",
      "MIR591,3\n",
      "DEFA3,28\n",
      "DEFA1,28\n",
      "LOC729338,23\n",
      "TNXA,11\n",
      "HBII-52-27,1\n",
      "SNORD42B,2\n",
      "SNORD114-27,5\n",
      "SPATA12,2\n",
      "MIR376A1,11\n",
      "GPR171,6\n",
      "MIR26A1,33\n",
      "MIR335,2\n",
      "MIR1278,5\n",
      "SNORD45C,6\n",
      "SNORA71B,1\n",
      "SCARNA3,1\n",
      "COL10A1,1\n",
      "MIR604,2\n",
      "KRTAP22-1,7\n",
      "SNORD111B,12\n",
      "CARD16,1\n",
      "PDC,1\n",
      "SNORA60,3\n",
      "SNORD116-9,1\n",
      "SNORD116-3,1\n",
      "SNORD115-32,1\n",
      "MIR211,3\n",
      "LOC440926,17\n",
      "SNORD32A,4\n",
      "KRTAP12-1,2\n",
      "SNORD115-31,7\n",
      "MIR1-2,9\n",
      "OR1D5,1\n",
      "OR1D4,1\n",
      "FAM45B,4\n",
      "FAM45A,4\n",
      "MIR133A1,6\n",
      "MIR511-1,1\n",
      "MIR511-2,1\n",
      "MRC1L1,1\n",
      "SNORD97,38\n",
      "MIR920,2\n",
      "LOC100286948,1\n",
      "SNORD114-4,9\n",
      "MIR643,1\n",
      "SNORD45B,2\n",
      "SNHG3,4\n",
      "MIR16-2,39\n",
      "MIR517B,1\n",
      "SNORD59B,4\n",
      "SNORD115-33,1\n",
      "MIR628,6\n",
      "MIR599,16\n",
      "MIR875,16\n",
      "SNORD127,19\n",
      "FAM122A,1\n",
      "SNORA37,2\n",
      "OR2AK2,7\n",
      "MIR302B,9\n",
      "MIR302C,9\n",
      "IAPP,10\n",
      "MIR1295,1\n",
      "LOC100093631,2\n",
      "MIR15B,31\n",
      "PRR20A,1\n",
      "PRR20D,1\n",
      "PRR20C,1\n",
      "PRR20E,1\n",
      "PRR20B,1\n",
      "OR8U1,12\n",
      "MIRLET7C,5\n",
      "SNORA19,32\n",
      "MIR155,13\n",
      "SNORA53,2\n",
      "LOC100286938,2\n",
      "SNORD19,8\n",
      "KCNJ13,10\n",
      "OR5AR1,6\n",
      "SNORD115-34,1\n",
      "OR2L8,5\n",
      "C10orf68,13\n",
      "LOC728407,5\n",
      "SNORD114-9,12\n",
      "OR5M3,17\n",
      "LOC100132724,31\n",
      "MIR100,4\n",
      "PLN,23\n",
      "TAS2R46,2\n",
      "UGT2B17,7\n",
      "MIR16-1,17\n",
      "MIR15A,17\n",
      "GPR22,2\n",
      "MIR99A,4\n",
      "MIR520D,1\n",
      "MIR194-1,2\n",
      "MIR215,2\n",
      "SNORD108,1\n",
      "OCLM,2\n",
      "MIR7-1,39\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "2079"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#checking which columns have NA in Methyl dataset and how many\n",
    "colNames=[]\n",
    "for column in methylClinicalNoAttr:\n",
    "    if methylClinicalNoAttr[column].isnull().sum() >= 1:\n",
    "       print('{0},{1}'.format(column, methylClinicalNoAttr[column].isnull().sum()))\n",
    "       colNames.append(column)\n",
    "len(colNames)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(rnaSeqlClinicalCommon.columns[rnaSeqlClinicalCommon.isna().any()].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(miRNAClinicalCommon.columns[miRNAClinicalCommon.isna().any()].tolist())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First dropping columns with more than 10 missing values.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylClinicalNoAttr.drop(methylClinicalNoAttr.columns[methylClinicalNoAttr.isna().sum()>10],axis = 1,inplace = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now row numbers remain intact.Column numbers dropped to 19661.Now dropping all columns which have more than 1 NAs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylClinicalNoAttr.drop(methylClinicalNoAttr.columns[methylClinicalNoAttr.isna().sum()>1],axis = 1,inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>CNBP</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4548</td>\n",
       "      <td>0.2310</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>-0.4737</td>\n",
       "      <td>-0.4817</td>\n",
       "      <td>0.4142</td>\n",
       "      <td>-0.4366</td>\n",
       "      <td>-0.4610</td>\n",
       "      <td>-0.4151</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>0.3420</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>4047.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>-0.4313</td>\n",
       "      <td>-0.4692</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.3972</td>\n",
       "      <td>-0.2806</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>0.4373</td>\n",
       "      <td>-0.4527</td>\n",
       "      <td>-0.4643</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>0.3857</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>4005.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>-0.4285</td>\n",
       "      <td>-0.4697</td>\n",
       "      <td>0.2664</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>-0.0637</td>\n",
       "      <td>-0.4783</td>\n",
       "      <td>0.3738</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4416</td>\n",
       "      <td>-0.4185</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>0.3319</td>\n",
       "      <td>0.1929</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>1474.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>-0.4043</td>\n",
       "      <td>-0.4714</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>-0.0440</td>\n",
       "      <td>-0.4827</td>\n",
       "      <td>0.3704</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4558</td>\n",
       "      <td>-0.3999</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.3457</td>\n",
       "      <td>0.2445</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>1448.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>-0.4083</td>\n",
       "      <td>-0.4651</td>\n",
       "      <td>0.1717</td>\n",
       "      <td>0.3406</td>\n",
       "      <td>-0.0880</td>\n",
       "      <td>-0.4788</td>\n",
       "      <td>0.4143</td>\n",
       "      <td>-0.4442</td>\n",
       "      <td>-0.4675</td>\n",
       "      <td>-0.3977</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.3542</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>348.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>607</td>\n",
       "      <td>-0.3903</td>\n",
       "      <td>-0.4683</td>\n",
       "      <td>0.2731</td>\n",
       "      <td>0.3682</td>\n",
       "      <td>-0.1018</td>\n",
       "      <td>-0.4842</td>\n",
       "      <td>0.3842</td>\n",
       "      <td>-0.4408</td>\n",
       "      <td>-0.4601</td>\n",
       "      <td>-0.3462</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.3322</td>\n",
       "      <td>0.2746</td>\n",
       "      <td>-0.0284</td>\n",
       "      <td>-0.0284</td>\n",
       "      <td>883.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>608</td>\n",
       "      <td>-0.4500</td>\n",
       "      <td>-0.4721</td>\n",
       "      <td>0.2597</td>\n",
       "      <td>0.4152</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4211</td>\n",
       "      <td>-0.4391</td>\n",
       "      <td>-0.4622</td>\n",
       "      <td>-0.4471</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.3908</td>\n",
       "      <td>0.4161</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>467.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>609</td>\n",
       "      <td>-0.4237</td>\n",
       "      <td>-0.4649</td>\n",
       "      <td>0.1951</td>\n",
       "      <td>0.3586</td>\n",
       "      <td>-0.4071</td>\n",
       "      <td>-0.4862</td>\n",
       "      <td>0.4494</td>\n",
       "      <td>-0.4609</td>\n",
       "      <td>-0.4639</td>\n",
       "      <td>-0.4376</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.4054</td>\n",
       "      <td>0.4089</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>488.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>610</td>\n",
       "      <td>-0.4197</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>0.1128</td>\n",
       "      <td>0.3803</td>\n",
       "      <td>-0.1014</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>-0.4283</td>\n",
       "      <td>-0.4676</td>\n",
       "      <td>-0.3551</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.3571</td>\n",
       "      <td>0.2107</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>3287.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>611</td>\n",
       "      <td>-0.4682</td>\n",
       "      <td>-0.4576</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3621</td>\n",
       "      <td>-0.4837</td>\n",
       "      <td>-0.4846</td>\n",
       "      <td>0.4343</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>-0.4747</td>\n",
       "      <td>-0.4326</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.3632</td>\n",
       "      <td>0.4219</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>3256.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>612 rows × 18762 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       RBL2   VDAC3   ACTN1  ATP2A1   SFRP1   NIPA2  MAN1B1  TSEN34  LRRC16A  \\\n",
       "0   -0.4444 -0.4548  0.2310  0.3568 -0.4737 -0.4817  0.4142 -0.4366  -0.4610   \n",
       "1   -0.4313 -0.4692  0.2859  0.3972 -0.2806 -0.4740  0.4373 -0.4527  -0.4643   \n",
       "2   -0.4285 -0.4697  0.2664  0.3482 -0.0637 -0.4783  0.3738 -0.4430  -0.4416   \n",
       "3   -0.4043 -0.4714  0.2174  0.3533 -0.0440 -0.4827  0.3704 -0.4430  -0.4558   \n",
       "4   -0.4083 -0.4651  0.1717  0.3406 -0.0880 -0.4788  0.4143 -0.4442  -0.4675   \n",
       "..      ...     ...     ...     ...     ...     ...     ...     ...      ...   \n",
       "607 -0.3903 -0.4683  0.2731  0.3682 -0.1018 -0.4842  0.3842 -0.4408  -0.4601   \n",
       "608 -0.4500 -0.4721  0.2597  0.4152  0.0244 -0.4843  0.4211 -0.4391  -0.4622   \n",
       "609 -0.4237 -0.4649  0.1951  0.3586 -0.4071 -0.4862  0.4494 -0.4609  -0.4639   \n",
       "610 -0.4197 -0.4688  0.1128  0.3803 -0.1014 -0.4843  0.4244 -0.4283  -0.4676   \n",
       "611 -0.4682 -0.4576  0.3352  0.3621 -0.4837 -0.4846  0.4343 -0.4688  -0.4747   \n",
       "\n",
       "       CNBP  ...    BAGE   BAGE2   BAGE4   BAGE3  MIR637  LOC100130932  \\\n",
       "0   -0.4151  ... -0.0420 -0.0420 -0.0420 -0.0420  0.3420        0.2383   \n",
       "1    0.2093  ... -0.0640 -0.0640 -0.0640 -0.0640  0.3857        0.3783   \n",
       "2   -0.4185  ... -0.0873 -0.0873 -0.0873 -0.0873  0.3319        0.1929   \n",
       "3   -0.3999  ...  0.1426  0.1426  0.1426  0.1426  0.3457        0.2445   \n",
       "4   -0.3977  ...  0.0218  0.0218  0.0218  0.0218  0.3542        0.3128   \n",
       "..      ...  ...     ...     ...     ...     ...     ...           ...   \n",
       "607 -0.3462  ...  0.2111  0.2111  0.2111  0.2111  0.3322        0.2746   \n",
       "608 -0.4471  ...  0.1998  0.1998  0.1998  0.1998  0.3908        0.4161   \n",
       "609 -0.4376  ...  0.1236  0.1236  0.1236  0.1236  0.4054        0.4089   \n",
       "610 -0.3551  ...  0.2220  0.2220  0.2220  0.2220  0.3571        0.2107   \n",
       "611 -0.4326  ... -0.0479 -0.0479 -0.0479 -0.0479  0.3632        0.4219   \n",
       "\n",
       "     GOLGA8F  GOLGA8G  overall_survival  status  \n",
       "0    -0.1168  -0.1168            4047.0     0.0  \n",
       "1    -0.1271  -0.1271            4005.0     0.0  \n",
       "2     0.0066   0.0066            1474.0     0.0  \n",
       "3    -0.0071  -0.0071            1448.0     0.0  \n",
       "4    -0.0929  -0.0929             348.0     0.0  \n",
       "..       ...      ...               ...     ...  \n",
       "607  -0.0284  -0.0284             883.0     0.0  \n",
       "608  -0.0702  -0.0702             467.0     0.0  \n",
       "609  -0.1210  -0.1210             488.0     0.0  \n",
       "610   0.0495   0.0495            3287.0     0.0  \n",
       "611  -0.1819  -0.1819            3256.0     0.0  \n",
       "\n",
       "[612 rows x 18762 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylClinicalNoAttr\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>CNBP</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>-0.4151</td>\n",
       "      <td>-0.4685</td>\n",
       "      <td>0.2419</td>\n",
       "      <td>0.3673</td>\n",
       "      <td>-0.3054</td>\n",
       "      <td>-0.4827</td>\n",
       "      <td>0.4148</td>\n",
       "      <td>-0.4482</td>\n",
       "      <td>-0.4606</td>\n",
       "      <td>-0.4075</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.0954</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>0.2846</td>\n",
       "      <td>-0.0858</td>\n",
       "      <td>-0.0858</td>\n",
       "      <td>1477.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4811</td>\n",
       "      <td>0.2578</td>\n",
       "      <td>0.3802</td>\n",
       "      <td>-0.1725</td>\n",
       "      <td>-0.4875</td>\n",
       "      <td>0.4135</td>\n",
       "      <td>-0.4769</td>\n",
       "      <td>-0.4785</td>\n",
       "      <td>-0.4183</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>0.3908</td>\n",
       "      <td>0.4529</td>\n",
       "      <td>-0.1917</td>\n",
       "      <td>-0.1917</td>\n",
       "      <td>434.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>-0.4487</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>0.2003</td>\n",
       "      <td>0.3522</td>\n",
       "      <td>-0.0607</td>\n",
       "      <td>-0.4835</td>\n",
       "      <td>0.4514</td>\n",
       "      <td>-0.4674</td>\n",
       "      <td>-0.4744</td>\n",
       "      <td>-0.4618</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.0678</td>\n",
       "      <td>0.3959</td>\n",
       "      <td>0.4546</td>\n",
       "      <td>-0.1176</td>\n",
       "      <td>-0.1176</td>\n",
       "      <td>635.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>-0.4522</td>\n",
       "      <td>-0.4729</td>\n",
       "      <td>0.2267</td>\n",
       "      <td>0.4081</td>\n",
       "      <td>-0.0196</td>\n",
       "      <td>-0.4876</td>\n",
       "      <td>0.4171</td>\n",
       "      <td>-0.4648</td>\n",
       "      <td>-0.4760</td>\n",
       "      <td>-0.4442</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>0.3983</td>\n",
       "      <td>0.4621</td>\n",
       "      <td>0.0284</td>\n",
       "      <td>0.0284</td>\n",
       "      <td>1196.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>-0.4649</td>\n",
       "      <td>-0.4500</td>\n",
       "      <td>0.2753</td>\n",
       "      <td>0.4081</td>\n",
       "      <td>-0.0237</td>\n",
       "      <td>-0.4888</td>\n",
       "      <td>0.4390</td>\n",
       "      <td>-0.4808</td>\n",
       "      <td>-0.4771</td>\n",
       "      <td>-0.4400</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0260</td>\n",
       "      <td>-0.0260</td>\n",
       "      <td>-0.0260</td>\n",
       "      <td>-0.0260</td>\n",
       "      <td>0.3927</td>\n",
       "      <td>0.4362</td>\n",
       "      <td>-0.2403</td>\n",
       "      <td>-0.2403</td>\n",
       "      <td>554.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>591</td>\n",
       "      <td>-0.4412</td>\n",
       "      <td>-0.4661</td>\n",
       "      <td>0.1129</td>\n",
       "      <td>0.4037</td>\n",
       "      <td>-0.3849</td>\n",
       "      <td>-0.4835</td>\n",
       "      <td>0.4386</td>\n",
       "      <td>-0.4543</td>\n",
       "      <td>-0.4653</td>\n",
       "      <td>-0.4291</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1371</td>\n",
       "      <td>0.1371</td>\n",
       "      <td>0.1371</td>\n",
       "      <td>0.1371</td>\n",
       "      <td>0.4018</td>\n",
       "      <td>0.4311</td>\n",
       "      <td>-0.0653</td>\n",
       "      <td>-0.0653</td>\n",
       "      <td>302.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>601</td>\n",
       "      <td>-0.4653</td>\n",
       "      <td>-0.4668</td>\n",
       "      <td>0.2579</td>\n",
       "      <td>0.4160</td>\n",
       "      <td>-0.0130</td>\n",
       "      <td>-0.4841</td>\n",
       "      <td>0.4723</td>\n",
       "      <td>-0.4748</td>\n",
       "      <td>-0.4770</td>\n",
       "      <td>-0.4453</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2894</td>\n",
       "      <td>0.2894</td>\n",
       "      <td>0.2894</td>\n",
       "      <td>0.2894</td>\n",
       "      <td>0.3851</td>\n",
       "      <td>0.4385</td>\n",
       "      <td>-0.0920</td>\n",
       "      <td>-0.0920</td>\n",
       "      <td>424.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>604</td>\n",
       "      <td>-0.4425</td>\n",
       "      <td>-0.4666</td>\n",
       "      <td>0.2475</td>\n",
       "      <td>0.3613</td>\n",
       "      <td>-0.1268</td>\n",
       "      <td>-0.4847</td>\n",
       "      <td>0.4131</td>\n",
       "      <td>-0.4520</td>\n",
       "      <td>-0.4719</td>\n",
       "      <td>-0.4456</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.1040</td>\n",
       "      <td>-0.1040</td>\n",
       "      <td>-0.1040</td>\n",
       "      <td>-0.1040</td>\n",
       "      <td>0.3807</td>\n",
       "      <td>0.4283</td>\n",
       "      <td>-0.0982</td>\n",
       "      <td>-0.0982</td>\n",
       "      <td>2033.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>608</td>\n",
       "      <td>-0.4500</td>\n",
       "      <td>-0.4721</td>\n",
       "      <td>0.2597</td>\n",
       "      <td>0.4152</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4211</td>\n",
       "      <td>-0.4391</td>\n",
       "      <td>-0.4622</td>\n",
       "      <td>-0.4471</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.3908</td>\n",
       "      <td>0.4161</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>467.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>611</td>\n",
       "      <td>-0.4682</td>\n",
       "      <td>-0.4576</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3621</td>\n",
       "      <td>-0.4837</td>\n",
       "      <td>-0.4846</td>\n",
       "      <td>0.4343</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>-0.4747</td>\n",
       "      <td>-0.4326</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.3632</td>\n",
       "      <td>0.4219</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>3256.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 18762 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       RBL2   VDAC3   ACTN1  ATP2A1   SFRP1   NIPA2  MAN1B1  TSEN34  LRRC16A  \\\n",
       "5   -0.4151 -0.4685  0.2419  0.3673 -0.3054 -0.4827  0.4148 -0.4482  -0.4606   \n",
       "10  -0.4444 -0.4811  0.2578  0.3802 -0.1725 -0.4875  0.4135 -0.4769  -0.4785   \n",
       "11  -0.4487 -0.4688  0.2003  0.3522 -0.0607 -0.4835  0.4514 -0.4674  -0.4744   \n",
       "12  -0.4522 -0.4729  0.2267  0.4081 -0.0196 -0.4876  0.4171 -0.4648  -0.4760   \n",
       "13  -0.4649 -0.4500  0.2753  0.4081 -0.0237 -0.4888  0.4390 -0.4808  -0.4771   \n",
       "..      ...     ...     ...     ...     ...     ...     ...     ...      ...   \n",
       "591 -0.4412 -0.4661  0.1129  0.4037 -0.3849 -0.4835  0.4386 -0.4543  -0.4653   \n",
       "601 -0.4653 -0.4668  0.2579  0.4160 -0.0130 -0.4841  0.4723 -0.4748  -0.4770   \n",
       "604 -0.4425 -0.4666  0.2475  0.3613 -0.1268 -0.4847  0.4131 -0.4520  -0.4719   \n",
       "608 -0.4500 -0.4721  0.2597  0.4152  0.0244 -0.4843  0.4211 -0.4391  -0.4622   \n",
       "611 -0.4682 -0.4576  0.3352  0.3621 -0.4837 -0.4846  0.4343 -0.4688  -0.4747   \n",
       "\n",
       "       CNBP  ...    BAGE   BAGE2   BAGE4   BAGE3  MIR637  LOC100130932  \\\n",
       "5   -0.4075  ...  0.0954  0.0954  0.0954  0.0954  0.3482        0.2846   \n",
       "10  -0.4183  ... -0.0925 -0.0925 -0.0925 -0.0925  0.3908        0.4529   \n",
       "11  -0.4618  ...  0.0678  0.0678  0.0678  0.0678  0.3959        0.4546   \n",
       "12  -0.4442  ...  0.1269  0.1269  0.1269  0.1269  0.3983        0.4621   \n",
       "13  -0.4400  ... -0.0260 -0.0260 -0.0260 -0.0260  0.3927        0.4362   \n",
       "..      ...  ...     ...     ...     ...     ...     ...           ...   \n",
       "591 -0.4291  ...  0.1371  0.1371  0.1371  0.1371  0.4018        0.4311   \n",
       "601 -0.4453  ...  0.2894  0.2894  0.2894  0.2894  0.3851        0.4385   \n",
       "604 -0.4456  ... -0.1040 -0.1040 -0.1040 -0.1040  0.3807        0.4283   \n",
       "608 -0.4471  ...  0.1998  0.1998  0.1998  0.1998  0.3908        0.4161   \n",
       "611 -0.4326  ... -0.0479 -0.0479 -0.0479 -0.0479  0.3632        0.4219   \n",
       "\n",
       "     GOLGA8F  GOLGA8G  overall_survival  status  \n",
       "5    -0.0858  -0.0858            1477.0     0.0  \n",
       "10   -0.1917  -0.1917             434.0     0.0  \n",
       "11   -0.1176  -0.1176             635.0     0.0  \n",
       "12    0.0284   0.0284            1196.0     0.0  \n",
       "13   -0.2403  -0.2403             554.0     0.0  \n",
       "..       ...      ...               ...     ...  \n",
       "591  -0.0653  -0.0653             302.0     0.0  \n",
       "601  -0.0920  -0.0920             424.0     0.0  \n",
       "604  -0.0982  -0.0982            2033.0     0.0  \n",
       "608  -0.0702  -0.0702             467.0     0.0  \n",
       "611  -0.1819  -0.1819            3256.0     0.0  \n",
       "\n",
       "[433 rows x 18762 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "testm = methylClinicalNoAttr\n",
    "testm.dropna()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Above shows if we just delete the rows which have single NA, total numbers of rows are being reduced by a big number making total no. of rows at 433.Since we have many features, it is better to drop the features with NAs altogether."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylClinicalNoAttr.drop(methylClinicalNoAttr.columns[methylClinicalNoAttr.isna().sum()>0],axis = 1,inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RBL2</th>\n",
       "      <th>VDAC3</th>\n",
       "      <th>ACTN1</th>\n",
       "      <th>ATP2A1</th>\n",
       "      <th>SFRP1</th>\n",
       "      <th>NIPA2</th>\n",
       "      <th>MAN1B1</th>\n",
       "      <th>TSEN34</th>\n",
       "      <th>LRRC16A</th>\n",
       "      <th>CNBP</th>\n",
       "      <th>...</th>\n",
       "      <th>BAGE</th>\n",
       "      <th>BAGE2</th>\n",
       "      <th>BAGE4</th>\n",
       "      <th>BAGE3</th>\n",
       "      <th>MIR637</th>\n",
       "      <th>LOC100130932</th>\n",
       "      <th>GOLGA8F</th>\n",
       "      <th>GOLGA8G</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>-0.4444</td>\n",
       "      <td>-0.4548</td>\n",
       "      <td>0.2310</td>\n",
       "      <td>0.3568</td>\n",
       "      <td>-0.4737</td>\n",
       "      <td>-0.4817</td>\n",
       "      <td>0.4142</td>\n",
       "      <td>-0.4366</td>\n",
       "      <td>-0.4610</td>\n",
       "      <td>-0.4151</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>-0.0420</td>\n",
       "      <td>0.3420</td>\n",
       "      <td>0.2383</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>-0.1168</td>\n",
       "      <td>4047.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>-0.4313</td>\n",
       "      <td>-0.4692</td>\n",
       "      <td>0.2859</td>\n",
       "      <td>0.3972</td>\n",
       "      <td>-0.2806</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>0.4373</td>\n",
       "      <td>-0.4527</td>\n",
       "      <td>-0.4643</td>\n",
       "      <td>0.2093</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>0.3857</td>\n",
       "      <td>0.3783</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>-0.1271</td>\n",
       "      <td>4005.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>-0.4285</td>\n",
       "      <td>-0.4697</td>\n",
       "      <td>0.2664</td>\n",
       "      <td>0.3482</td>\n",
       "      <td>-0.0637</td>\n",
       "      <td>-0.4783</td>\n",
       "      <td>0.3738</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4416</td>\n",
       "      <td>-0.4185</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>-0.0873</td>\n",
       "      <td>0.3319</td>\n",
       "      <td>0.1929</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>1474.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>-0.4043</td>\n",
       "      <td>-0.4714</td>\n",
       "      <td>0.2174</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>-0.0440</td>\n",
       "      <td>-0.4827</td>\n",
       "      <td>0.3704</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>-0.4558</td>\n",
       "      <td>-0.3999</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.1426</td>\n",
       "      <td>0.3457</td>\n",
       "      <td>0.2445</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>1448.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>-0.4083</td>\n",
       "      <td>-0.4651</td>\n",
       "      <td>0.1717</td>\n",
       "      <td>0.3406</td>\n",
       "      <td>-0.0880</td>\n",
       "      <td>-0.4788</td>\n",
       "      <td>0.4143</td>\n",
       "      <td>-0.4442</td>\n",
       "      <td>-0.4675</td>\n",
       "      <td>-0.3977</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.0218</td>\n",
       "      <td>0.3542</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>-0.0929</td>\n",
       "      <td>348.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>607</td>\n",
       "      <td>-0.3903</td>\n",
       "      <td>-0.4683</td>\n",
       "      <td>0.2731</td>\n",
       "      <td>0.3682</td>\n",
       "      <td>-0.1018</td>\n",
       "      <td>-0.4842</td>\n",
       "      <td>0.3842</td>\n",
       "      <td>-0.4408</td>\n",
       "      <td>-0.4601</td>\n",
       "      <td>-0.3462</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>0.3322</td>\n",
       "      <td>0.2746</td>\n",
       "      <td>-0.0284</td>\n",
       "      <td>-0.0284</td>\n",
       "      <td>883.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>608</td>\n",
       "      <td>-0.4500</td>\n",
       "      <td>-0.4721</td>\n",
       "      <td>0.2597</td>\n",
       "      <td>0.4152</td>\n",
       "      <td>0.0244</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4211</td>\n",
       "      <td>-0.4391</td>\n",
       "      <td>-0.4622</td>\n",
       "      <td>-0.4471</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.1998</td>\n",
       "      <td>0.3908</td>\n",
       "      <td>0.4161</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>-0.0702</td>\n",
       "      <td>467.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>609</td>\n",
       "      <td>-0.4237</td>\n",
       "      <td>-0.4649</td>\n",
       "      <td>0.1951</td>\n",
       "      <td>0.3586</td>\n",
       "      <td>-0.4071</td>\n",
       "      <td>-0.4862</td>\n",
       "      <td>0.4494</td>\n",
       "      <td>-0.4609</td>\n",
       "      <td>-0.4639</td>\n",
       "      <td>-0.4376</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.1236</td>\n",
       "      <td>0.4054</td>\n",
       "      <td>0.4089</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>488.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>610</td>\n",
       "      <td>-0.4197</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>0.1128</td>\n",
       "      <td>0.3803</td>\n",
       "      <td>-0.1014</td>\n",
       "      <td>-0.4843</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>-0.4283</td>\n",
       "      <td>-0.4676</td>\n",
       "      <td>-0.3551</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.3571</td>\n",
       "      <td>0.2107</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>0.0495</td>\n",
       "      <td>3287.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>611</td>\n",
       "      <td>-0.4682</td>\n",
       "      <td>-0.4576</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>0.3621</td>\n",
       "      <td>-0.4837</td>\n",
       "      <td>-0.4846</td>\n",
       "      <td>0.4343</td>\n",
       "      <td>-0.4688</td>\n",
       "      <td>-0.4747</td>\n",
       "      <td>-0.4326</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.3632</td>\n",
       "      <td>0.4219</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>-0.1819</td>\n",
       "      <td>3256.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>612 rows × 18029 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       RBL2   VDAC3   ACTN1  ATP2A1   SFRP1   NIPA2  MAN1B1  TSEN34  LRRC16A  \\\n",
       "0   -0.4444 -0.4548  0.2310  0.3568 -0.4737 -0.4817  0.4142 -0.4366  -0.4610   \n",
       "1   -0.4313 -0.4692  0.2859  0.3972 -0.2806 -0.4740  0.4373 -0.4527  -0.4643   \n",
       "2   -0.4285 -0.4697  0.2664  0.3482 -0.0637 -0.4783  0.3738 -0.4430  -0.4416   \n",
       "3   -0.4043 -0.4714  0.2174  0.3533 -0.0440 -0.4827  0.3704 -0.4430  -0.4558   \n",
       "4   -0.4083 -0.4651  0.1717  0.3406 -0.0880 -0.4788  0.4143 -0.4442  -0.4675   \n",
       "..      ...     ...     ...     ...     ...     ...     ...     ...      ...   \n",
       "607 -0.3903 -0.4683  0.2731  0.3682 -0.1018 -0.4842  0.3842 -0.4408  -0.4601   \n",
       "608 -0.4500 -0.4721  0.2597  0.4152  0.0244 -0.4843  0.4211 -0.4391  -0.4622   \n",
       "609 -0.4237 -0.4649  0.1951  0.3586 -0.4071 -0.4862  0.4494 -0.4609  -0.4639   \n",
       "610 -0.4197 -0.4688  0.1128  0.3803 -0.1014 -0.4843  0.4244 -0.4283  -0.4676   \n",
       "611 -0.4682 -0.4576  0.3352  0.3621 -0.4837 -0.4846  0.4343 -0.4688  -0.4747   \n",
       "\n",
       "       CNBP  ...    BAGE   BAGE2   BAGE4   BAGE3  MIR637  LOC100130932  \\\n",
       "0   -0.4151  ... -0.0420 -0.0420 -0.0420 -0.0420  0.3420        0.2383   \n",
       "1    0.2093  ... -0.0640 -0.0640 -0.0640 -0.0640  0.3857        0.3783   \n",
       "2   -0.4185  ... -0.0873 -0.0873 -0.0873 -0.0873  0.3319        0.1929   \n",
       "3   -0.3999  ...  0.1426  0.1426  0.1426  0.1426  0.3457        0.2445   \n",
       "4   -0.3977  ...  0.0218  0.0218  0.0218  0.0218  0.3542        0.3128   \n",
       "..      ...  ...     ...     ...     ...     ...     ...           ...   \n",
       "607 -0.3462  ...  0.2111  0.2111  0.2111  0.2111  0.3322        0.2746   \n",
       "608 -0.4471  ...  0.1998  0.1998  0.1998  0.1998  0.3908        0.4161   \n",
       "609 -0.4376  ...  0.1236  0.1236  0.1236  0.1236  0.4054        0.4089   \n",
       "610 -0.3551  ...  0.2220  0.2220  0.2220  0.2220  0.3571        0.2107   \n",
       "611 -0.4326  ... -0.0479 -0.0479 -0.0479 -0.0479  0.3632        0.4219   \n",
       "\n",
       "     GOLGA8F  GOLGA8G  overall_survival  status  \n",
       "0    -0.1168  -0.1168            4047.0     0.0  \n",
       "1    -0.1271  -0.1271            4005.0     0.0  \n",
       "2     0.0066   0.0066            1474.0     0.0  \n",
       "3    -0.0071  -0.0071            1448.0     0.0  \n",
       "4    -0.0929  -0.0929             348.0     0.0  \n",
       "..       ...      ...               ...     ...  \n",
       "607  -0.0284  -0.0284             883.0     0.0  \n",
       "608  -0.0702  -0.0702             467.0     0.0  \n",
       "609  -0.1210  -0.1210             488.0     0.0  \n",
       "610   0.0495   0.0495            3287.0     0.0  \n",
       "611  -0.1819  -0.1819            3256.0     0.0  \n",
       "\n",
       "[612 rows x 18029 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methylClinicalNoAttr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Row numbers remain 612, column numbers reduced to 18029.Checking if methylClinicalNoAttr has any more NAs.Seems there is none."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(methylClinicalNoAttr.columns[methylClinicalNoAttr.isna().any()].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Splitting the Methylation dataset into the Training set and Test set\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "methyl1_train, methyl1_test, methyl2_train, methyl2_test= train_test_split(methylClinicalNoAttr.loc[:, methylClinicalNoAttr.columns != 'status'],methylClinicalNoAttr['status'],test_size=0.5, random_state=0)\n",
    "                                                                         "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((306, 18028), (306,), (306, 18028), (306,))"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methyl1_train.shape,methyl2_train.shape,methyl1_test.shape,methyl2_test.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "579    1.0\n",
       "52     1.0\n",
       "534    0.0\n",
       "348    0.0\n",
       "55     0.0\n",
       "      ... \n",
       "276    0.0\n",
       "284    0.0\n",
       "487    0.0\n",
       "587    0.0\n",
       "281    0.0\n",
       "Name: status, Length: 306, dtype: float64"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methyl2_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#trying SVM on Methylation dataset\n",
    "\n",
    "from sklearn import svm, metrics\n",
    "\n",
    "classifierS = svm.SVC(gamma=0.001)\n",
    "classifierS.fit(methyl1_train, methyl2_train)\n",
    "classifierS.score(methyl1_test, methyl2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Using KNeighborsClassifier Method of neighbors class to use Nearest Neighbor algorithm\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=10)\n",
    "classifierK.fit(methyl1_train, methyl2_train)\n",
    "classifierK.score(methyl1_test, methyl2_test)                                                                       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/f19/stud05/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.\n",
      "  \"10 in version 0.20 to 100 in 0.22.\", FutureWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.ensemble import RandomForestClassifier\n",
    "# Create a classifier: a random forest classifier\n",
    "\n",
    "# random forest model creation\n",
    "classifierR = RandomForestClassifier()\n",
    "classifierR.fit(methyl1_train, methyl2_train)\n",
    "classifierR.score(methyl1_test, methyl2_test)     "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8137254901960784"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Trying gradient boosting\n",
    "from sklearn.ensemble import GradientBoostingClassifier  #GBM algorithm\n",
    "\n",
    "classifierXGB = GradientBoostingClassifier(random_state=10)\n",
    "classifierXGB.fit(methyl1_train, methyl2_train)\n",
    "classifierXGB.score(methyl1_test, methyl2_test)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO: run PCA on the Methylation data and project the dataset down to 10 dimensions\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scaler = StandardScaler()\n",
    "# Fit on training set only.\n",
    "scaler.fit(methyl1_train)\n",
    "\n",
    "# Apply transform to both the training set and the test set.\n",
    "methyl1_train = scaler.transform(methyl1_train)\n",
    "methyl1_test = scaler.transform(methyl1_test)\n",
    "\n",
    "pca = PCA(n_components = 50)\n",
    "# actually run the fit algorithm\n",
    "eigenbasis = pca.fit(methyl1_train)\n",
    "# transform our data using the learned transform\n",
    "Methyl_Trainpca = eigenbasis.transform(methyl1_train)\n",
    "Methyl_Testpca = eigenbasis.transform(methyl1_test)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8398692810457516"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#trying KNN\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=5)\n",
    "classifierK.fit(Methyl_Trainpca, methyl2_train)\n",
    "classifierK.score(Methyl_Testpca, methyl2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create a classifier: a support vector classifier\n",
    "classifierS = svm.SVC(gamma=0.001)\n",
    "classifierS.fit(Methyl_Trainpca, methyl2_train)\n",
    "classifierS.score(Methyl_Testpca, methyl2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/f19/stud05/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.\n",
      "  \"10 in version 0.20 to 100 in 0.22.\", FutureWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.8431372549019608"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# random forest model creation\n",
    "classifierR = RandomForestClassifier()\n",
    "classifierR.fit(Methyl_Trainpca, methyl2_train)\n",
    "classifierR.score(Methyl_Testpca, methyl2_test)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "principal components:\n",
      " [[ 0.00623084  0.0039654   0.01192871 ... -0.00290976 -0.00290976\n",
      "   0.00071914]\n",
      " [ 0.01562579  0.00027695 -0.00351952 ...  0.019142    0.019142\n",
      "   0.00047944]\n",
      " [-0.00167167  0.00705425 -0.00036759 ... -0.00673502 -0.00673502\n",
      "  -0.00422324]\n",
      " ...\n",
      " [-0.00037656 -0.01323052  0.00753636 ...  0.00711173  0.00711173\n",
      "   0.02342058]\n",
      " [-0.00537334  0.00444833  0.01169757 ... -0.00028909 -0.00028909\n",
      "   0.01232487]\n",
      " [ 0.0032968  -0.00627711 -0.00810636 ... -0.00390603 -0.00390603\n",
      "   0.00042573]]\n",
      "percentage of variance explained by first 2 principal components: [0.11527744 0.08399751 0.06376843 0.04003597 0.03802553 0.02836492\n",
      " 0.02363141 0.01880561 0.01771327 0.01582277 0.01370953 0.01287884\n",
      " 0.01122172 0.01001714 0.00995656 0.00937141 0.00896863 0.00808908\n",
      " 0.00743392 0.00699191 0.00691119 0.00651404 0.00637962 0.00626723\n",
      " 0.00589032 0.00557789 0.00541951 0.00510194 0.00500691 0.0049849\n",
      " 0.00479562 0.00473538 0.00453574 0.00444757 0.0042447  0.0042136\n",
      " 0.00398703 0.00397503 0.00386548 0.00373627 0.00363486 0.00356298\n",
      " 0.00345968 0.00333062 0.00331004 0.00319796 0.00316856 0.00309306\n",
      " 0.00302819 0.00298311]\n"
     ]
    }
   ],
   "source": [
    "# the actual components\n",
    "print(\"principal components:\\n\", pca.components_)\n",
    "# let's also look at how much of the total variance we were able to cover with 2 dimensions\n",
    "print('percentage of variance explained by first 2 principal components:', pca.explained_variance_ratio_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<matplotlib.lines.Line2D at 0x7ff46a728810>]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXoAAAD7CAYAAABkO19ZAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAfsUlEQVR4nO3de3RedZ3v8ff3ueR50tx6SdokbaEttJSC3IzAGkQRBiw6UuGAUnUd5iyWzHGGGT3qKLpceuQsZ4Y5jnCcwVnDEs7gjYsc0ao4DAp455JSbm0pDaW0Ib2k6SVNm+S5fc8feyc8hLR92iR9kv18Xmtl5dl7/54n3w3pZ+/89m//trk7IiISXbFyFyAiIhNLQS8iEnEKehGRiFPQi4hEnIJeRCTiFPQiIhFXUtCb2XIz22BmHWZ20yjb32Vmz5hZzsyuLlp/lpn90czWmtnzZvbh8SxeRESOzI40jt7M4sDLwKVAJ/A0sNLd1xW1WQDUA58FVrn7A+H6JYC7+0YzawVWA6e6+97x3xURERlNooQ25wId7r4JwMzuBVYAw0Hv7pvDbYXiN7r7y0Wvu8xsJ9AEHDLoGxsbfcGCBaXvgYiIsHr16l3u3jTatlKCfi6wtWi5EzjvaIsws3OBKuCVw7VbsGAB7e3tR/vxIiIVzcxeO9S2UvrobZR1RzVvgpm1AN8F/pu7F0bZfoOZtZtZe3d399F8tIiIHEEpQd8JzC9angd0lfoDzKwe+DnwJXd/YrQ27n6Hu7e5e1tT06h/eYiIyDEqJeifBhab2UIzqwKuBVaV8uFh+weB77j7D4+9TBEROVZHDHp3zwE3Ag8D64H73X2tmd1sZlcAmNk7zKwTuAb4NzNbG779Q8C7gD83s2fDr7MmZE9ERGRURxxeeby1tbW5LsaKiBwdM1vt7m2jbdOdsSIiEaegFxGJuMgE/b7+LLf98mWe26qbbkVEikUm6M3gtl9u5KlXd5e7FBGRSSUyQV+XSlBTFadrX3+5SxERmVQiE/RmRsv0arbtHSh3KSIik0pkgh6gpSHNNp3Ri4i8SaSCvrWhmq59OqMXESkWqaBvmZ5mV98gmdxb5k0TEalYkQr61oZq3GFHr87qRUSGRCroW6anAdim7hsRkWHRCvqGagBdkBURKRKxoA/O6Ls0xFJEZFikgr4mlaA+ndAZvYhIkUgFPUDr9Gqd0YuIFIlc0OumKRGRN4te0E+v1qgbEZEikQv61oY0uw9kGMjmy12KiMikELmgHxpiuV1n9SIiQBSDPrxpStMVi4gEohf0QzdNaeSNiAgQyaAfmgZBZ/QiIhDBoE8n48ysqdJ0xSIiocgFPYRj6ffqjF5EBCIb9BpLLyIyJJJB3zo9TZfO6EVEgBKD3syWm9kGM+sws5tG2f4uM3vGzHJmdvWIbdeZ2cbw67rxKvxwWhqq6R3IcWAwdzx+nIjIpHbEoDezOHA7cDmwDFhpZstGNNsC/DnwgxHvnQl8BTgPOBf4ipnNGHvZh9eqB5CIiAwr5Yz+XKDD3Te5ewa4F1hR3MDdN7v788DIh7W+F3jE3Xe7+x7gEWD5ONR9WM31GmIpIjKklKCfC2wtWu4M15ViLO89Zq3TddOUiMiQUoLeRlnnJX5+Se81sxvMrN3M2ru7u0v86EObU5/GTNMgiIhAaUHfCcwvWp4HdJX4+SW9193vcPc2d29ramoq8aMPrSoRo7E2pTN6ERFKC/qngcVmttDMqoBrgVUlfv7DwGVmNiO8CHtZuG7CtTakdUYvIkIJQe/uOeBGgoBeD9zv7mvN7GYzuwLAzN5hZp3ANcC/mdna8L27gf9FcLB4Grg5XDfhdNOUiEggUUojd38IeGjEui8XvX6aoFtmtPfeBdw1hhqPScv0NL/d2I27YzbapQIRkcoQyTtjIZjv5kAmz37dNCUiFS7CQa8hliIiEOGgb9WTpkREgAgHvc7oRUQCkQ362XUpYqZpEEREIhv0iXiMOfVpunRGLyIVLrJBD+GTpnRGLyIVLtpBP103TYmIRDvo64MzevdS52ATEYmeaAf99GoGsgX2HsyWuxQRkbKJdNC3NmgsvYhIpIO+RQ8gERGJdtAPndFr5I2IVLJIB31jbYpk3OjSyBsRqWCRDvpYzJhTn2bbXp3Ri0jlinTQA7Q2VOuMXkQqWuSDvrkhzXYFvYhUsMgHfcv0IOgLBd00JSKVKfJB39pQTSZfoOdAptyliIiUReSDvkVDLEWkwkU+6FvDm6Y0XbGIVKrIB73O6EWk0kU+6GfWVJFKxDRdsYhUrMgHvZkxd0Y1W3oOlrsUEZGyiHzQAyyZXcfLO/aXuwwRkbKojKBvrmNzzwEGsvlylyIictxVRNAvba6j4NCxs6/cpYiIHHclBb2ZLTezDWbWYWY3jbI9ZWb3hdufNLMF4fqkmd1tZi+Y2Xoz+8L4ll+aU5rrAHhpu7pvRKTyHDHozSwO3A5cDiwDVprZshHNrgf2uPvJwK3ALeH6a4CUu78NeDvwF0MHgePpxJnTqErE2LC993j/aBGRsivljP5coMPdN7l7BrgXWDGizQrg7vD1A8AlZmaAAzVmlgCqgQxw3NM2EY+xeHYtG3ao60ZEKk8pQT8X2Fq03BmuG7WNu+eAfcAsgtA/AGwDtgBfd/fdI3+Amd1gZu1m1t7d3X3UO1GKU5rrdEYvIhWplKC3UdaNnAryUG3OBfJAK7AQ+IyZLXpLQ/c73L3N3duamppKKOnonTKnjh29g+w9qMnNRKSylBL0ncD8ouV5QNeh2oTdNA3AbuAjwH+4e9bddwK/B9rGWvSxGLogu0EXZEWkwpQS9E8Di81soZlVAdcCq0a0WQVcF76+GnjU3Z2gu+ZiC9QA5wMvjU/pR2dpcz0AG3TjlIhUmCMGfdjnfiPwMLAeuN/d15rZzWZ2RdjsTmCWmXUAnwaGhmDeDtQCLxIcMP6vuz8/zvtQkjn1KerTCQ2xFJGKkyilkbs/BDw0Yt2Xi14PEAylHPm+vtHWl4OZsbS5npcV9CJSYSriztghpzTXsWHHfoJeJRGRylBRQb+kuY79Azm6NGWxiFSQigr6peHIG3XfiEglqaigXzJHc96ISOWpqKBvqE7S0pDWHbIiUlEqKuhh6IKs5rwRkcpRkUH/ys4+svlCuUsRETkuKi/o59SRyRfYvOtAuUsRETkuKi/oh+a80VQIIlIhKi7oT2qqJR4zTW4mIhWj4oI+nYyzYNY0DbEUkYpRcUEPwUyWL6vrRkQqREUG/SnNdWzZfZCDmVy5SxERmXAVGfRL5tThDi9rPL2IVICKDHrNeSMilaQig/6EmdNIJ2O6ICsiFaEigz4WM5bMqWPDDs15IyLRV5FBD8Edshu2q49eRKKvcoO+uY5dfYP09A2WuxQRkQlV0UEP6A5ZEYk8Bb1unBKRiKvYoG+qTTGzpkpn9CISeRUb9GbGkjm1GmIpIpFXsUEPsHh2HZu6+3D3cpciIjJhKjroFzbW0DuQY/eBTLlLERGZMJUd9E01ALyqp02JSISVFPRmttzMNphZh5ndNMr2lJndF25/0swWFG07w8z+aGZrzewFM0uPX/ljs6gxCPpN3Qp6EYmuIwa9mcWB24HLgWXASjNbNqLZ9cAedz8ZuBW4JXxvAvge8N/d/TTgIiA7btWP0dzp1STjxiad0YtIhJVyRn8u0OHum9w9A9wLrBjRZgVwd/j6AeASMzPgMuB5d38OwN173D0/PqWPXSIe48RZNby6S1MhiEh0lRL0c4GtRcud4bpR27h7DtgHzAKWAG5mD5vZM2b2ubGXPL4WNtaoj15EIq2UoLdR1o0cj3ioNgngncBHw+9Xmtklb/kBZjeYWbuZtXd3d5dQ0vhZ1FjD5p6D5AsaYiki0VRK0HcC84uW5wFdh2oT9ss3ALvD9b92913ufhB4CDhn5A9w9zvcvc3d25qamo5+L8ZgYWMNmVyBrr39x/XniogcL6UE/dPAYjNbaGZVwLXAqhFtVgHXha+vBh714C6kh4EzzGxaeAB4N7BufEofHwsbNcRSRKLtiEEf9rnfSBDa64H73X2tmd1sZleEze4EZplZB/Bp4KbwvXuAbxAcLJ4FnnH3n4//bhw7jaUXkahLlNLI3R8i6HYpXvflotcDwDWHeO/3CIZYTkpNtSlqUwkFvYhEVkXfGQvB5GYLG2s0ll5EIqvigx6GhlhqLL2IRJOCniDoO/f0M5ibNPdyiYiMGwU9sKipBnfY0nOw3KWIiIw7BT1vDLFUP72IRJGCHligsfQiEmEKeqA+naSxNsWmbl2QFZHoUdCHFmlyMxGJKAV9aFGTgl5EoklBH1rYWMOuvgz7+ifNc1FERMaFgj40NPJms87qRSRiFPShRZrcTEQiSkEfmj9zGjHTWHoRiR4FfSiViDNvxjSd0YtI5Cjoi2hyMxGJIgV9kYWNNbzafYDg4VgiItGgoC+yqKmGA5k83fsHy12KiMi4UdAX0eRmIhJFCvoielC4iESRgr5Ia0M1VYmYgl5EIkVBXyQWMxbOqmFTt4JeRKJDQT9C8KBwDbEUkehQ0I+wqKmGLT0HyeUL5S5FRGRcKOhHWNhYQ67gdO7pL3cpIiLjQkE/giY3E5GoUdCPsLCxFtBYehGJjpKC3syWm9kGM+sws5tG2Z4ys/vC7U+a2YIR208wsz4z++z4lD1xZkxL0lCd1Jw3IhIZRwx6M4sDtwOXA8uAlWa2bESz64E97n4ycCtwy4jttwK/GHu5E8/MwsnNdEYvItFQyhn9uUCHu29y9wxwL7BiRJsVwN3h6weAS8zMAMzsg8AmYO34lDzxFoWTm4mIREEpQT8X2Fq03BmuG7WNu+eAfcAsM6sBPg98deylHj8nza6la98A3/zVRvoz+XKXIyIyJqUEvY2ybuQ8vodq81XgVnc/bIe3md1gZu1m1t7d3V1CSRPrY+edyPve1sw3HnmZi//pcX7y7OuaulhEpqxSgr4TmF+0PA/oOlQbM0sADcBu4DzgH81sM/Ap4ItmduPIH+Dud7h7m7u3NTU1HfVOjLeGaUm+9dG3c98N5zOzpopP3vssV/3rH3hmy55ylyYictRKCfqngcVmttDMqoBrgVUj2qwCrgtfXw086oEL3X2Buy8AbgP+zt3/ZZxqn3DnLZrFT298J//76jPo3NPPVd/6A5+6dw0DWXXniMjUccSgD/vcbwQeBtYD97v7WjO72cyuCJvdSdAn3wF8GnjLEMypKhYzrmmbz2OfvYhPXHQSP362ix+u7ix3WSIiJbPJ1vfc1tbm7e3t5S5jVO7O8tt+y7RUnAf/8oJylyMiMszMVrt722jbdGfsUTAzrjxnLmu27NU4exGZMhT0R+mDZ83FDB5c83q5SxERKYmC/ig1N6S54KRGHlzTqSGXIjIlKOiPwVXnzGXr7n7aX9NwSxGZ/BT0x+C9pzVTnYzzo2c0+kZEJj8F/TGoSSW4/PRmfvb8No2pF5FJT0F/jK48Zy77B3L8av3OcpciInJYCvpj9CcnNTKnPsWDa9R9IyKTm4L+GMVjxgfPmsvjG7rp6RssdzkiIoekoB+Dq86ZR67g/PS5kXO8iYhMHgr6MTiluY5lLfX8SDdPicgkpqAfo6vOmcvznfvo2Lm/3KWIiIxKQT9GV5zVSszgR8/orF5EJicF/RjNrkvzriVN/HjN6xQKmhJBRCYfBf04uPLsuXTtG+CJTT3lLkVE5C0U9OPgsmXNNNZWcfPP1ulOWRGZdBT046C6Ks4/fegsXtq+n6/+dG25yxEReRMF/Th595Im/vKik7jnqa385FldmBWRyUNBP44+fekS2k6cwRd/9AKbuvvKXY6ICKCgH1eJeIxvrjybZCLGjT9Yo/56EZkUFPTjrHV6Nd/40Jms29bL136+vtzliIgo6CfCxUvn8PELF/LdJ17joRe2lbscEalwCvoJ8rnlSzlr/nQ+/8DzvNZzoNzliEgFU9BPkGQ8xj+vPBsz+Ot71pDJFcpdkohUKAX9BJo/cxr/ePWZPN+5j6//54ZylyMiFUpBP8GWn97Mx84/gTt+s4nHNuixgyJy/JUU9Ga23Mw2mFmHmd00yvaUmd0Xbn/SzBaE6y81s9Vm9kL4/eLxLX9q+NL7l7G0uY7P3v8cO3sHyl2OiFSYIwa9mcWB24HLgWXASjNbNqLZ9cAedz8ZuBW4JVy/C/iAu78NuA747ngVPpWkk3H+eeXZHMjk+B/3P6tZLkXkuCrljP5coMPdN7l7BrgXWDGizQrg7vD1A8AlZmbuvsbdh56ztxZIm1lqPAqfahbPqeN/fuA0ft/Rw7/++pVylyMiFaSUoJ8LbC1a7gzXjdrG3XPAPmDWiDb/BVjj7hX7JO0Pv2M+7z+jhW888jKrX9tT7nJEpEKUEvQ2yrqRfQ+HbWNmpxF05/zFqD/A7AYzazez9u7u7hJKmprMjL+/6m20NKT5m3vWsK8/W+6SRKQClBL0ncD8ouV5QNeh2phZAmgAdofL84AHgf/q7qP2Wbj7He7e5u5tTU1NR7cHU0x9Osk3V57N9t4BbvhOO4+s20F/RnPiiMjESZTQ5mlgsZktBF4HrgU+MqLNKoKLrX8ErgYedXc3s+nAz4EvuPvvx6/sqe2cE2Zw84rT+IdfvMTHv9NOOhnjwsVNXLpsDpcsnc2s2oq8jCEiE8TcjzwCxMzeB9wGxIG73P1rZnYz0O7uq8wsTTCi5myCM/lr3X2TmX0J+AKwsejjLnP3Qw4ob2tr8/b29mPfoykkkyvw5Ks9PLJuB4+s28G2fQPEDM5bOIu/vvhk/uTkxnKXKCJThJmtdve2UbeVEvTHUyUFfTF358XXe3lk3XZ+uLqTbfsGuHBxI59fvpTT5zaUuzwRmeQU9FPMQDbPd//4Grc/3sHeg1muOLOVz1y2hBNn1ZS7NBGZpBT0U9S+/ix3/OYV7vzdq+TyzkfOO4FP/ekSZtZUlbs0EZlkDhf0mutmEmuoTvK3713Kr//2PXzoHfP5/pNbeM/XH+d7T7xGXnfXikiJFPRTwJz6NH935dv4xScv5NSWOr704xdZcfvvWLNFN12JyJEp6KeQJXPquOfj5/PNlWfTvX+QK7/1Bz7/wPP09FXszcYiUoJSxtHLJGJmXHFmKxcvnc03f7WRu373Kr94cRtXnTOPtgUzeMeCmcypT5e7TBGZRHQxdorbuGM/t/zHS/y+o4f+bHCH7Qkzpw2H/kWnNNHSUF3mKkVkomnUTQXI5gus6+rl6c27eXrzbto376HnQIZEzLjirFY+8e6TWDynrtxlisgEUdBXIHfnle4+vv/kFu59aiv92TyXLpvDJy46iXNOmFHu8kRknCnoK9zuAxnu/sNm/v0Pm9nXn+X8RTP5+IWLeNeSJpJxXY8XiQIFvQBwYDDHPU9t4du/fZXtvQPMmJZk+ektfOCMFs5bNIt4bLTZpkVkKlDQy5tkcgUe37CTnz2/jV+u38HBTJ6muhTvO72ZS5c101CdJB4zEnEjETMSsRhViRhNdSkdDEQmKQW9HFJ/Js9jG3by0+e6ePSlnQzmCodsm0rEOKmplsVzalkyp46TZ9eyeHYt82dOUxeQSJkp6KUkfYM51mzZw2C2QK5QIFdwcnknV3AGsnle6znAxp19bNzRx+t7+4ffZwaz61K0NFTTOj0dfq/mpKYa2hbMpDal2zVEJtrhgl7/AmVYbSrBhYtLe8JX32COV3b28fKO/XTu6Wfbvn669g7w0vb9PPZS9/CY/njMOH1uA+cvmsn5i2bRduIM6tLJidwNERlBZ/Qy7tydvQezrO3q5clXe3hiUw/Pbt1LNu/EY8aylvrgq7WeU1vqWdpSR73CX2RM1HUjZdefyfPMlj08uamH1Vv2sK6rlz0H33g4+vyZ1ZzaXM+Js6Yxb8Y05s2oZt6MacydUa2uH5ESqOtGyq66Ks4FJzdyQfh4RHdnR+8g67f1sm5bL+u6enlpey+/frn7LReEp09LUlOVoCoRIxkPRgElEzGq4sb0aVW0NqRpmV5NS0NwfaClIU1jbYp0MoaZRgmJKOilLMyM5oY0zQ1p3rN09vB6d2dXX4bOPQfp3NNP555+Xt97kP5MgWz+ja9M3snk8mzpOcgTm3rYP5Ab9eekkzGqk3HSyTjVyTipZJx0MkYqESOViJNKxEiH65rqUjTXp2luqKa5Ps2chhSNNSliGlIqU5yCXiYVM6OpLkVTXYqzj2Kqhr7BHNvDC8Lb9vXTcyDDQLbAQDbPQDZPfyZPf/h6MFdgMFdgb3+WwXD5YCbHrr7MWx7okogZjbUpZtVW0VibCr7qqmiqTVFfnWRaVZyaqgTVRd/r0wlm1lSR0JBTmSQU9BIJtakEJ8+u4+TZxz5xW77g9PQNsm3fANt7B9gefu/eP0hP3yC7+jJs3LGfXX0ZMvlD328AwZDTGdOqaCw6QMysqSKVjFEVD78Sb3zFzTALDnQxMwyIxaA6GacmlaA2laAunRh+nU7GScRMXVNSEgW9SCgeM2bXp5ldn+bMw7Rzd3oHcvT2Z+nP5jmYyXNwMMfBTJ4DmRy9Azl27R9kV9/QV4Znt+5lz8EMg7kCmcPclHYsNcdjRjL8XpWIU5cODgpDB4faVJK69BvXOKricZIJGz7Y1FQlwvckh99bl05SlYgRM4iFB5+h10MHJJk6FPQiR8nMaKhO0lB9bENC3Z1s3snkg9DP5gsU3Cl4sM0d3KHgTn82z4HBHPsHc/QN5OgbzLF/IBve1ObDN7blwxvbBnN59g+3y9Gz6yD7B7L0DebI5Atk8z7m5w0n40Z9Otj/uvC/Q314YAnyPzwYEPxlE7PgoJJKvnFdJJWIkUoGr4euoaSKrqcMbR96X1U8eI8OMMdGQS9ynJkZVQmjKhGD1PH/+fmCk80Xhv+6ODAYHBh6B7LBQWIgOJhk807BnXx48CkUgoNRfzZP70CW3v4svQM59vVn6dx9kL7BHE5wkILwgEVwwMqE10XG4yCTiMVIxI1kPEYiFnxPxo1UIk7V8EFk6OAQXGhPhweQoQvvxV1fBsSGu80gEX/zxfpU2D4Zj73pr5qYBX9FxYzhkWDJsJ7i+uKToItNQS9SYYLuniD0AJrqjt/RJpcvkMkXGMwWGMjlGcwWhi+S92eD5TcumOeHDxBDB6VMvkAu/MskVyiQzTnZQrCcKW6fLdDbn2Mwlx++KD/0+Ue6vjIRkvGhLrbgIDDU5RY3IxYLJg+MhTcT/stHzhn3n6+gF5HjJhGPkYjHmFZVvhry4dxN+eFuMh/uKit4sH0wFxxshg862QKZfD5sF7Txofbu5PIFcmF33NCBKJMP/oLJ5Ye62YK/pHL54K+kQriuUAiWcwXnhJnTJmSfSwp6M1sO/B8gDnzb3f9hxPYU8B3g7UAP8GF33xxu+wJwPZAH/sbdHx636kVEjlI8ZtRU2N3WRxzoa2Zx4HbgcmAZsNLMlo1odj2wx91PBm4Fbgnfuwy4FjgNWA58K/w8ERE5Tkq5o+NcoMPdN7l7BrgXWDGizQrg7vD1A8AlFlx9WAHc6+6D7v4q0BF+noiIHCelBP1cYGvRcme4btQ27p4D9gGzSnyviIhMoFKCfrRxQSPHSB2qTSnvxcxuMLN2M2vv7u4uoSQRESlVKUHfCcwvWp4HdB2qjZklgAZgd4nvxd3vcPc2d29rairtwRciIlKaUoL+aWCxmS00syqCi6urRrRZBVwXvr4aeNSDie5XAdeaWcrMFgKLgafGp3QRESnFEccYuXvOzG4EHiYYXnmXu681s5uBdndfBdwJfNfMOgjO5K8N37vWzO4H1gE54K/cPT9B+yIiIqPQE6ZERCJgSj1K0My6gdfG8BGNwK5xKmcq0X5XFu13ZSllv09091Evck66oB8rM2s/1FEtyrTflUX7XVnGut96BI6ISMQp6EVEIi6KQX9HuQsoE+13ZdF+V5Yx7Xfk+uhFROTNonhGLyIiRSIT9Ga23Mw2mFmHmd1U7nomkpndZWY7zezFonUzzewRM9sYfp9RzhrHm5nNN7PHzGy9ma01s0+G66O+32kze8rMngv3+6vh+oVm9mS43/eFd61HjpnFzWyNmf0sXK6U/d5sZi+Y2bNm1h6uO+bf9UgEfYlz5kfJvxPM71/sJuBX7r4Y+FW4HCU54DPufipwPvBX4f/jqO/3IHCxu58JnAUsN7PzCZ75cGu433sIngkRRZ8E1hctV8p+A7zH3c8qGlZ5zL/rkQh6SpszPzLc/TcEU00UK34mwN3AB49rURPM3be5+zPh6/0E//jnEv39dnfvCxeT4ZcDFxM8+wEiuN8AZjYPeD/w7XDZqID9Poxj/l2PStBr3nuY4+7bIAhFYHaZ65kwZrYAOBt4kgrY77D74llgJ/AI8AqwN3z2A0T39/024HPA0NO8Z1EZ+w3Bwfw/zWy1md0Qrjvm3/WoPDixpHnvZeozs1rg/wGfcvfe4CQv2sKJAM8ys+nAg8CpozU7vlVNLDP7M2Cnu682s4uGVo/SNFL7XeQCd+8ys9nAI2b20lg+LCpn9CXNex9xO8ysBSD8vrPM9Yw7M0sShPz33f1H4erI7/cQd98LPE5wjWJ6+OwHiObv+wXAFWa2maAr9mKCM/yo7zcA7t4Vft9JcHA/lzH8rkcl6EuZMz/qip8JcB3wkzLWMu7C/tk7gfXu/o2iTVHf76bwTB4zqwb+lOD6xGMEz36ACO63u3/B3ee5+wKCf8+PuvtHifh+A5hZjZnVDb0GLgNeZAy/65G5YcrM3kdwxB+aM/9rZS5pwpjZPcBFBDPa7QC+AvwYuB84AdgCXOPuIy/YTllm9k7gt8ALvNFn+0WCfvoo7/cZBBfe4gQnZve7+81mtojgTHcmsAb4mLsPlq/SiRN23XzW3f+sEvY73McHw8UE8AN3/5qZzeIYf9cjE/QiIjK6qHTdiIjIISjoRUQiTkEvIhJxCnoRkYhT0IuIRJyCXkQk4hT0IiIRp6AXEYm4/w/Zd+5XOxLZmwAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.plot(pca.explained_variance_ratio_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Trying SVD to see feature selection."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import TruncatedSVD\n",
    "\n",
    "# svd\n",
    "svd = TruncatedSVD(n_components=300)\n",
    "\n",
    "# actually run the fit algorithm\n",
    "eigenbasis = svd.fit(methyl1_train)\n",
    "\n",
    "# transform our data using the learned transform\n",
    "Methyl_Trainsvd = eigenbasis.transform(methyl1_train)\n",
    "Methyl_Testsvd = eigenbasis.transform(methyl1_test)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8366013071895425"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#trying KNN\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=5)\n",
    "classifierK.fit(Methyl_Trainsvd, methyl2_train)\n",
    "classifierK.score(Methyl_Testsvd, methyl2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create a classifier: a support vector classifier\n",
    "classifierS = svm.SVC(gamma=0.001)\n",
    "classifierS.fit(Methyl_Trainsvd, methyl2_train)\n",
    "classifierS.score(Methyl_Testsvd, methyl2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/f19/stud05/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.\n",
      "  \"10 in version 0.20 to 100 in 0.22.\", FutureWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# random forest model creation\n",
    "classifierR = RandomForestClassifier()\n",
    "classifierR.fit(Methyl_Trainsvd, methyl2_train)\n",
    "classifierR.score(Methyl_Testsvd, methyl2_test) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Trying forward feature selection algorithom."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "from sklearn.metrics import accuracy_score as acc\n",
    "from mlxtend.feature_selection import SequentialFeatureSelector as sfs\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "# Build RF classifier to use in feature selection\n",
    "clf = RandomForestClassifier(n_estimators=100, n_jobs=-1)\n",
    "\n",
    "#methyl1_train, methyl1_test, methyl2_train, methyl2_test\n",
    "\n",
    "# Build step forward feature selection\n",
    "sfs1 = sfs(clf,\n",
    "           k_features=10,\n",
    "           forward=True,\n",
    "           floating=False,\n",
    "           verbose=2,\n",
    "           scoring='accuracy',\n",
    "           cv=5)\n",
    "\n",
    "# Perform SFFS\n",
    "sfs1 = sfs1.fit(methyl1_train, methyl2_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>attrib_name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2BP1</th>\n",
       "      <th>A2LD1</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAA1</th>\n",
       "      <th>...</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>psiTPTE22</th>\n",
       "      <th>tAKR</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>TCGA.3C.AAAU</td>\n",
       "      <td>7.6300</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.6999</td>\n",
       "      <td>1.2501</td>\n",
       "      <td>12.5017</td>\n",
       "      <td>6.1136</td>\n",
       "      <td>3.2655</td>\n",
       "      <td>0.4273</td>\n",
       "      <td>...</td>\n",
       "      <td>10.6965</td>\n",
       "      <td>8.0196</td>\n",
       "      <td>10.2400</td>\n",
       "      <td>11.7765</td>\n",
       "      <td>10.8887</td>\n",
       "      <td>10.2064</td>\n",
       "      <td>1.4454</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>4047.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>TCGA.3C.AALI</td>\n",
       "      <td>7.8971</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.1672</td>\n",
       "      <td>2.4196</td>\n",
       "      <td>12.8866</td>\n",
       "      <td>7.3101</td>\n",
       "      <td>0.6265</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>10.3924</td>\n",
       "      <td>7.6401</td>\n",
       "      <td>9.2398</td>\n",
       "      <td>12.4267</td>\n",
       "      <td>10.3659</td>\n",
       "      <td>8.6715</td>\n",
       "      <td>9.8573</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>4005.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>TCGA.3C.AALJ</td>\n",
       "      <td>8.7287</td>\n",
       "      <td>0.9310</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>7.3422</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>13.1101</td>\n",
       "      <td>9.1671</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>9.5884</td>\n",
       "      <td>8.3786</td>\n",
       "      <td>9.0585</td>\n",
       "      <td>12.4146</td>\n",
       "      <td>9.8825</td>\n",
       "      <td>8.9958</td>\n",
       "      <td>5.1842</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1474.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>TCGA.3C.AALK</td>\n",
       "      <td>7.5851</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>5.9888</td>\n",
       "      <td>1.4087</td>\n",
       "      <td>13.4201</td>\n",
       "      <td>8.9870</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>9.7564</td>\n",
       "      <td>7.4629</td>\n",
       "      <td>9.2488</td>\n",
       "      <td>12.4748</td>\n",
       "      <td>9.6113</td>\n",
       "      <td>9.4551</td>\n",
       "      <td>6.0792</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1448.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>TCGA.4H.AAAK</td>\n",
       "      <td>8.0762</td>\n",
       "      <td>0.5115</td>\n",
       "      <td>2.2720</td>\n",
       "      <td>7.2796</td>\n",
       "      <td>2.1389</td>\n",
       "      <td>13.2268</td>\n",
       "      <td>8.4226</td>\n",
       "      <td>0.5115</td>\n",
       "      <td>0.5115</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0373</td>\n",
       "      <td>3.9110</td>\n",
       "      <td>9.6013</td>\n",
       "      <td>11.9811</td>\n",
       "      <td>9.7020</td>\n",
       "      <td>9.7858</td>\n",
       "      <td>7.5564</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>348.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>607</td>\n",
       "      <td>TCGA.WT.AB44</td>\n",
       "      <td>8.8835</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>5.9711</td>\n",
       "      <td>2.9100</td>\n",
       "      <td>12.4015</td>\n",
       "      <td>8.4828</td>\n",
       "      <td>0.8596</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>9.5961</td>\n",
       "      <td>6.6374</td>\n",
       "      <td>8.3047</td>\n",
       "      <td>13.4171</td>\n",
       "      <td>9.6468</td>\n",
       "      <td>8.7947</td>\n",
       "      <td>9.5036</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>883.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>608</td>\n",
       "      <td>TCGA.XX.A899</td>\n",
       "      <td>7.8088</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.4789</td>\n",
       "      <td>7.0469</td>\n",
       "      <td>0.4789</td>\n",
       "      <td>14.3127</td>\n",
       "      <td>8.9845</td>\n",
       "      <td>0.8379</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>10.3078</td>\n",
       "      <td>4.4764</td>\n",
       "      <td>9.4288</td>\n",
       "      <td>12.3217</td>\n",
       "      <td>10.9180</td>\n",
       "      <td>9.3903</td>\n",
       "      <td>6.6355</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>467.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>609</td>\n",
       "      <td>TCGA.XX.A89A</td>\n",
       "      <td>8.0008</td>\n",
       "      <td>1.7492</td>\n",
       "      <td>1.2733</td>\n",
       "      <td>6.3386</td>\n",
       "      <td>5.8263</td>\n",
       "      <td>14.0614</td>\n",
       "      <td>9.2679</td>\n",
       "      <td>1.9389</td>\n",
       "      <td>0.9595</td>\n",
       "      <td>...</td>\n",
       "      <td>10.4268</td>\n",
       "      <td>7.3697</td>\n",
       "      <td>9.5465</td>\n",
       "      <td>12.4196</td>\n",
       "      <td>10.4903</td>\n",
       "      <td>9.8989</td>\n",
       "      <td>7.8841</td>\n",
       "      <td>0.9595</td>\n",
       "      <td>488.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>610</td>\n",
       "      <td>TCGA.Z7.A8R5</td>\n",
       "      <td>8.7831</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.6756</td>\n",
       "      <td>6.3628</td>\n",
       "      <td>1.7610</td>\n",
       "      <td>15.1690</td>\n",
       "      <td>8.8372</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>9.6388</td>\n",
       "      <td>5.7270</td>\n",
       "      <td>8.9856</td>\n",
       "      <td>12.7049</td>\n",
       "      <td>9.5611</td>\n",
       "      <td>9.5543</td>\n",
       "      <td>7.9065</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3287.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>611</td>\n",
       "      <td>TCGA.Z7.A8R6</td>\n",
       "      <td>7.9619</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>4.7108</td>\n",
       "      <td>2.3718</td>\n",
       "      <td>12.8416</td>\n",
       "      <td>9.5881</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>9.8901</td>\n",
       "      <td>7.1348</td>\n",
       "      <td>9.1661</td>\n",
       "      <td>11.7328</td>\n",
       "      <td>9.1432</td>\n",
       "      <td>8.8552</td>\n",
       "      <td>4.4514</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3256.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>612 rows × 20158 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      attrib_name    A1BG    A1CF   A2BP1   A2LD1   A2ML1      A2M  A4GALT  \\\n",
       "0    TCGA.3C.AAAU  7.6300  0.0000  0.0000  6.6999  1.2501  12.5017  6.1136   \n",
       "1    TCGA.3C.AALI  7.8971  0.0000  0.0000  6.1672  2.4196  12.8866  7.3101   \n",
       "2    TCGA.3C.AALJ  8.7287  0.9310  0.0000  7.3422  0.0000  13.1101  9.1671   \n",
       "3    TCGA.3C.AALK  7.5851  0.0000  0.0000  5.9888  1.4087  13.4201  8.9870   \n",
       "4    TCGA.4H.AAAK  8.0762  0.5115  2.2720  7.2796  2.1389  13.2268  8.4226   \n",
       "..            ...     ...     ...     ...     ...     ...      ...     ...   \n",
       "607  TCGA.WT.AB44  8.8835  0.0000  0.0000  5.9711  2.9100  12.4015  8.4828   \n",
       "608  TCGA.XX.A899  7.8088  0.0000  0.4789  7.0469  0.4789  14.3127  8.9845   \n",
       "609  TCGA.XX.A89A  8.0008  1.7492  1.2733  6.3386  5.8263  14.0614  9.2679   \n",
       "610  TCGA.Z7.A8R5  8.7831  0.0000  0.6756  6.3628  1.7610  15.1690  8.8372   \n",
       "611  TCGA.Z7.A8R6  7.9619  0.0000  0.0000  4.7108  2.3718  12.8416  9.5881   \n",
       "\n",
       "      A4GNT    AAA1  ...     ZXDC  ZYG11A   ZYG11B      ZYX    ZZEF1     ZZZ3  \\\n",
       "0    3.2655  0.4273  ...  10.6965  8.0196  10.2400  11.7765  10.8887  10.2064   \n",
       "1    0.6265  0.0000  ...  10.3924  7.6401   9.2398  12.4267  10.3659   8.6715   \n",
       "2    0.0000  0.0000  ...   9.5884  8.3786   9.0585  12.4146   9.8825   8.9958   \n",
       "3    0.0000  0.0000  ...   9.7564  7.4629   9.2488  12.4748   9.6113   9.4551   \n",
       "4    0.5115  0.5115  ...  10.0373  3.9110   9.6013  11.9811   9.7020   9.7858   \n",
       "..      ...     ...  ...      ...     ...      ...      ...      ...      ...   \n",
       "607  0.8596  0.0000  ...   9.5961  6.6374   8.3047  13.4171   9.6468   8.7947   \n",
       "608  0.8379  0.0000  ...  10.3078  4.4764   9.4288  12.3217  10.9180   9.3903   \n",
       "609  1.9389  0.9595  ...  10.4268  7.3697   9.5465  12.4196  10.4903   9.8989   \n",
       "610  0.0000  0.0000  ...   9.6388  5.7270   8.9856  12.7049   9.5611   9.5543   \n",
       "611  0.0000  0.0000  ...   9.8901  7.1348   9.1661  11.7328   9.1432   8.8552   \n",
       "\n",
       "     psiTPTE22    tAKR  overall_survival  status  \n",
       "0       1.4454  0.0000            4047.0     0.0  \n",
       "1       9.8573  0.0000            4005.0     0.0  \n",
       "2       5.1842  0.0000            1474.0     0.0  \n",
       "3       6.0792  0.0000            1448.0     0.0  \n",
       "4       7.5564  0.0000             348.0     0.0  \n",
       "..         ...     ...               ...     ...  \n",
       "607     9.5036  0.0000             883.0     0.0  \n",
       "608     6.6355  0.0000             467.0     0.0  \n",
       "609     7.8841  0.9595             488.0     0.0  \n",
       "610     7.9065  0.0000            3287.0     0.0  \n",
       "611     4.4514  0.0000            3256.0     0.0  \n",
       "\n",
       "[612 rows x 20158 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rnaSeqlClinicalCommon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "rnaSeqlClinicalNoAttr = rnaSeqlClinicalCommon.drop(['attrib_name'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna1_train, rna1_test, rna2_train, rna2_test= train_test_split(rnaSeqlClinicalNoAttr.loc[:, rnaSeqlClinicalNoAttr.columns != 'status'],rnaSeqlClinicalNoAttr['status'],test_size=0.5, random_state=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#trying SVM on rnaSeq dataset\n",
    "\n",
    "from sklearn import svm, metrics\n",
    "\n",
    "classifierS = svm.SVC(gamma=0.001)\n",
    "classifierS.fit(rna1_train, rna2_train)\n",
    "classifierS.score(rna1_test, rna2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Using KNeighborsClassifier Method of neighbors class to use Nearest Neighbor algorithm\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=10)\n",
    "classifierK.fit(rna1_train, rna2_train)\n",
    "classifierK.score(rna1_test, rna2_test) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/f19/stud05/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.\n",
      "  \"10 in version 0.20 to 100 in 0.22.\", FutureWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.8431372549019608"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.ensemble import RandomForestClassifier\n",
    "# Create a classifier: a random forest classifier\n",
    "\n",
    "# random forest model creation\n",
    "classifierR = RandomForestClassifier()\n",
    "classifierR.fit(rna1_train, rna2_train)\n",
    "classifierR.score(rna1_test, rna2_test) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8235294117647058"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Trying gradient boosting\n",
    "from sklearn.ensemble import GradientBoostingClassifier  #GBM algorithm\n",
    "\n",
    "classifierXGB = GradientBoostingClassifier(random_state=10)\n",
    "classifierXGB.fit(rna1_train, rna2_train)\n",
    "classifierXGB.score(rna1_test, rna2_test)  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Starting processing of miRNAClinicalCommon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "miRNAClinicalNoAttr = miRNAClinicalCommon.drop(['attrib_name'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "miRna1_train, miRna1_test, miRna2_train, miRna2_test= train_test_split(miRNAClinicalNoAttr.loc[:, miRNAClinicalNoAttr.columns != 'status'],miRNAClinicalNoAttr['status'],test_size=0.5, random_state=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#trying SVM on miRna dataset\n",
    "\n",
    "from sklearn import svm, metrics\n",
    "\n",
    "classifierS = svm.SVC(gamma=0.001)\n",
    "classifierS.fit(miRna1_train, miRna2_train)\n",
    "classifierS.score(miRna1_test, miRna2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Using KNeighborsClassifier Method of neighbors class to use Nearest Neighbor algorithm\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=10)\n",
    "classifierK.fit(miRna1_train, miRna2_train)\n",
    "classifierK.score(miRna1_test, miRna2_test) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/f19/stud05/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.\n",
      "  \"10 in version 0.20 to 100 in 0.22.\", FutureWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.8464052287581699"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.ensemble import RandomForestClassifier\n",
    "# Create a classifier: a random forest classifier\n",
    "\n",
    "# random forest model creation\n",
    "classifierR = RandomForestClassifier()\n",
    "classifierR.fit(miRna1_train, miRna2_train)\n",
    "classifierR.score(miRna1_test, miRna2_test) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "from sklearn.preprocessing import StandardScaler\n",
    "scaler = StandardScaler()\n",
    "\n",
    "methyl1_train1 = scaler.fit_transform(methyl1_train)\n",
    "\n",
    "lr_scores = cross_val_score(lr,\n",
    "                            methyl1_train, methyl2_train,\n",
    "                            cv=5,\n",
    "                            scoring='roc_auc')\n",
    "print('LR Scores: ', lr_scores)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "lr_scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(306, 18028)"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn import feature_selection\n",
    "\n",
    "sel = feature_selection.VarianceThreshold()\n",
    "train_variance = sel.fit_transform(methyl1_train)\n",
    "train_variance.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(306, 18028)"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn import feature_selection\n",
    "from sklearn.feature_selection import VarianceThreshold\n",
    "\n",
    "sel = feature_selection.VarianceThreshold()\n",
    "\n",
    "thresholder = VarianceThreshold(threshold=0.01)\n",
    "\n",
    "# Conduct variance thresholding\n",
    "X_high_variance = thresholder.fit_transform(methyl1_train)\n",
    "\n",
    "X_high_variance.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Now to get all the features that are not constant, we can use the get_support() method of the filter that we created\n",
    "len(methyl1_train.columns[thresholder.get_support()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'numpy.ndarray' object has no attribute 'columns'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-79-e1929848827a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m#Similarly, you can find the number of constant features with the help of the following script:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m constant_columns = [column for column in methyl1_train.columns\n\u001b[0m\u001b[1;32m      3\u001b[0m                     if column not in methyl1_train.columns[thresholder.get_support()]]\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mconstant_columns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'numpy.ndarray' object has no attribute 'columns'"
     ]
    }
   ],
   "source": [
    "#Similarly, you can find the number of constant features with the help of the following script:\n",
    "constant_columns = [column for column in methyl1_train.columns\n",
    "                    if column not in methyl1_train.columns[thresholder.get_support()]]\n",
    "\n",
    "print(len(constant_columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for column in constant_columns:\n",
    "    print(column)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.feature_selection import SelectKBest, f_classif\n",
    "# feature extraction\n",
    "k_best = SelectKBest(score_func=f_classif, k=100)\n",
    "# fit on train set\n",
    "fit = k_best.fit(methyl1_train, methyl2_train)\n",
    "# transform train set\n",
    "univariate_features = fit.transform(methyl1_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "univariate_features.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.discriminant_analysis import LinearDiscriminantAnalysis\n",
    "# Create and run an LDA\n",
    "lda = LinearDiscriminantAnalysis(n_components=None)\n",
    "methyl_lda = lda.fit(methyl1_train, methyl2_train)\n",
    "\n",
    "# Create array of explained variance ratios\n",
    "lda_var_ratios = lda.explained_variance_ratio_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lda_var_ratios"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create Function Calculating Number Of Components Required To Pass Threshold\n",
    "# Create a function\n",
    "def select_n_components(var_ratio, goal_var: float) -> int:\n",
    "    # Set initial variance explained so far\n",
    "    total_variance = 0.0\n",
    "    \n",
    "    # Set initial number of features\n",
    "    n_components = 0\n",
    "    \n",
    "    # For the explained variance of each feature:\n",
    "    for explained_variance in var_ratio:\n",
    "        \n",
    "        # Add the explained variance to the total\n",
    "        total_variance += explained_variance\n",
    "        \n",
    "        # Add one to the number of components\n",
    "        n_components += 1\n",
    "        \n",
    "        # If we reach our goal level of explained variance\n",
    "        if total_variance >= goal_var:\n",
    "            # End the loop\n",
    "            break\n",
    "            \n",
    "    # Return the number of components\n",
    "    return n_components"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Run function\n",
    "select_n_components(lda_var_ratios, 0.25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create correlation matrix\n",
    "corr_matrix = methylClinicalNoAttr.corr().abs()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_matrix.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Select upper triangle of correlation matrix\n",
    "upper = corr_matrix.where(np.triu(np.ones(corr_matrix.shape), k=1).astype(np.bool))\n",
    "\n",
    "# Find index of feature columns with correlation greater than 0.95\n",
    "to_drop = [column for column in upper.columns if any(upper[column] > 0.95)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "to_drop[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_matrix['MORN1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(corr_matrix['overall_survival'].sort_values(ascending=False).head(10))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Next, we will loop through all the columns in the correlation_matrix and will add the columns with a correlation value of 0.8 to the correlated_features set as shown below. \n",
    "correlated_features = set()\n",
    "for i in range(len(corr_matrix.columns)):\n",
    "    for j in range(i):\n",
    "        if abs(corr_matrix.iloc[i, j]) > 0.8:\n",
    "            colname = corr_matrix.columns[i]\n",
    "            correlated_features.add(colname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import PCA\n",
    "# pca - keep 90% of variance\n",
    "pca = PCA(0.90)\n",
    "principal_components = pca.fit_transform(methyl1_train)\n",
    "principal_df = pd.DataFrame(data = principal_components)\n",
    "print(principal_df)\n",
    "\n",
    "#We can see that we are left with 1 feature that explain 90% of the variance in our data."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "from sklearn.feature_selection import RFE\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "rfc = RandomForestClassifier(n_estimators=100)\n",
    "# feature extraction\n",
    "rfe = RFE(rfc, n_features_to_select=100)\n",
    "# fit on train set\n",
    "fit = rfe.fit(methyl1_train, methyl2_train)\n",
    "# transform train set\n",
    "recursive_features = fit.transform(methyl1_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'keras'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-82-416930c810d3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mkeras\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmodels\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mSequential\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mkeras\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlayers\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mDense\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;31m# define the keras model\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mmodel\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSequential\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'keras'"
     ]
    }
   ],
   "source": [
    "\n",
    "from keras.models import Sequential\n",
    "from keras.layers import Dense\n",
    "\n",
    "# define the keras model\n",
    "model = Sequential()\n",
    "model.add(Dense(100, input_dim=18028, activation='relu'))\n",
    "model.add(Dense(10, activation='relu'))\n",
    "model.add(Dense(2, activation='sigmoid'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Changing Methylation pandas dataframe to numpy array\n",
    "X =methylClinicalNoAttr.loc[:, methylClinicalNoAttr.columns != 'status']\n",
    "y = methylClinicalNoAttr['status'].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Normalizing the data\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "sc = StandardScaler()\n",
    "X = sc.fit_transform(X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import OneHotEncoder\n",
    "ohe = OneHotEncoder(categories='auto')\n",
    "y = ohe.fit_transform(y.reshape(-1, 1)).toarray()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "X_train,X_test,y_train,y_test = train_test_split(X,y,test_size = 0.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Dependencies\n",
    "from tensorflow.python.keras.models import Sequential\n",
    "from tensorflow.python.keras.layers import Input, Dense,Dropout\n",
    "from tensorflow.python.keras.optimizers import SGD\n",
    "\n",
    "# Neural network\n",
    "model = Sequential()\n",
    "model.add(Dense(100, input_dim=18028, activation='relu'))\n",
    "model.add(Dense(10, activation='relu'))\n",
    "model.add(Dropout(0.5))\n",
    "model.add(Dense(2, activation='softmax'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "#sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)\n",
    "model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 489 samples\n",
      "Epoch 1/100\n",
      "489/489 [==============================] - 1s 3ms/sample - loss: 2.8229 - accuracy: 0.6892\n",
      "Epoch 2/100\n",
      "489/489 [==============================] - 0s 128us/sample - loss: 1.3202 - accuracy: 0.7710\n",
      "Epoch 3/100\n",
      "489/489 [==============================] - 0s 126us/sample - loss: 0.6258 - accuracy: 0.8671\n",
      "Epoch 4/100\n",
      "489/489 [==============================] - 0s 117us/sample - loss: 0.6963 - accuracy: 0.8753\n",
      "Epoch 5/100\n",
      "489/489 [==============================] - 0s 111us/sample - loss: 0.4923 - accuracy: 0.8691\n",
      "Epoch 6/100\n",
      "489/489 [==============================] - 0s 116us/sample - loss: 0.3911 - accuracy: 0.8834\n",
      "Epoch 7/100\n",
      "489/489 [==============================] - 0s 114us/sample - loss: 0.3713 - accuracy: 0.8916\n",
      "Epoch 8/100\n",
      "489/489 [==============================] - 0s 117us/sample - loss: 0.3677 - accuracy: 0.8978\n",
      "Epoch 9/100\n",
      "489/489 [==============================] - 0s 115us/sample - loss: 0.3964 - accuracy: 0.8978\n",
      "Epoch 10/100\n",
      "489/489 [==============================] - 0s 115us/sample - loss: 0.3295 - accuracy: 0.9059\n",
      "Epoch 11/100\n",
      "489/489 [==============================] - 0s 117us/sample - loss: 0.2775 - accuracy: 0.9141\n",
      "Epoch 12/100\n",
      "489/489 [==============================] - 0s 120us/sample - loss: 0.2832 - accuracy: 0.9080\n",
      "Epoch 13/100\n",
      "489/489 [==============================] - 0s 119us/sample - loss: 0.2481 - accuracy: 0.9018\n",
      "Epoch 14/100\n",
      "489/489 [==============================] - 0s 116us/sample - loss: 0.2130 - accuracy: 0.9223\n",
      "Epoch 15/100\n",
      "489/489 [==============================] - 0s 112us/sample - loss: 0.2348 - accuracy: 0.9202\n",
      "Epoch 16/100\n",
      "489/489 [==============================] - 0s 130us/sample - loss: 0.2080 - accuracy: 0.9243\n",
      "Epoch 17/100\n",
      "489/489 [==============================] - 0s 125us/sample - loss: 0.2188 - accuracy: 0.9182\n",
      "Epoch 18/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.2310 - accuracy: 0.9223\n",
      "Epoch 19/100\n",
      "489/489 [==============================] - 0s 110us/sample - loss: 0.1950 - accuracy: 0.9448\n",
      "Epoch 20/100\n",
      "489/489 [==============================] - 0s 110us/sample - loss: 0.2313 - accuracy: 0.9223\n",
      "Epoch 21/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.1946 - accuracy: 0.9366\n",
      "Epoch 22/100\n",
      "489/489 [==============================] - 0s 108us/sample - loss: 0.3265 - accuracy: 0.9223\n",
      "Epoch 23/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.2183 - accuracy: 0.9202\n",
      "Epoch 24/100\n",
      "489/489 [==============================] - 0s 108us/sample - loss: 0.2202 - accuracy: 0.9366\n",
      "Epoch 25/100\n",
      "489/489 [==============================] - 0s 111us/sample - loss: 0.2188 - accuracy: 0.9407\n",
      "Epoch 26/100\n",
      "489/489 [==============================] - 0s 104us/sample - loss: 0.2529 - accuracy: 0.9387\n",
      "Epoch 27/100\n",
      "489/489 [==============================] - 0s 104us/sample - loss: 0.2204 - accuracy: 0.9325\n",
      "Epoch 28/100\n",
      "489/489 [==============================] - 0s 107us/sample - loss: 0.2102 - accuracy: 0.9284\n",
      "Epoch 29/100\n",
      "489/489 [==============================] - 0s 113us/sample - loss: 0.2091 - accuracy: 0.9305\n",
      "Epoch 30/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.2024 - accuracy: 0.9264\n",
      "Epoch 31/100\n",
      "489/489 [==============================] - 0s 98us/sample - loss: 0.2174 - accuracy: 0.9243\n",
      "Epoch 32/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.1738 - accuracy: 0.9325\n",
      "Epoch 33/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.1529 - accuracy: 0.9530\n",
      "Epoch 34/100\n",
      "489/489 [==============================] - 0s 106us/sample - loss: 0.2520 - accuracy: 0.9325\n",
      "Epoch 35/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.2312 - accuracy: 0.9387\n",
      "Epoch 36/100\n",
      "489/489 [==============================] - 0s 101us/sample - loss: 0.1642 - accuracy: 0.9366\n",
      "Epoch 37/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.2390 - accuracy: 0.9264\n",
      "Epoch 38/100\n",
      "489/489 [==============================] - 0s 109us/sample - loss: 0.2388 - accuracy: 0.9346\n",
      "Epoch 39/100\n",
      "489/489 [==============================] - 0s 108us/sample - loss: 0.2282 - accuracy: 0.9346\n",
      "Epoch 40/100\n",
      "489/489 [==============================] - 0s 109us/sample - loss: 0.2584 - accuracy: 0.9202\n",
      "Epoch 41/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.2391 - accuracy: 0.9141\n",
      "Epoch 42/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.2081 - accuracy: 0.9346\n",
      "Epoch 43/100\n",
      "489/489 [==============================] - 0s 106us/sample - loss: 0.1974 - accuracy: 0.9162\n",
      "Epoch 44/100\n",
      "489/489 [==============================] - 0s 110us/sample - loss: 0.1930 - accuracy: 0.9182\n",
      "Epoch 45/100\n",
      "489/489 [==============================] - 0s 110us/sample - loss: 0.1698 - accuracy: 0.9284\n",
      "Epoch 46/100\n",
      "489/489 [==============================] - 0s 111us/sample - loss: 0.1814 - accuracy: 0.9223\n",
      "Epoch 47/100\n",
      "489/489 [==============================] - 0s 108us/sample - loss: 0.1565 - accuracy: 0.9202\n",
      "Epoch 48/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.1420 - accuracy: 0.9387\n",
      "Epoch 49/100\n",
      "489/489 [==============================] - 0s 98us/sample - loss: 0.1452 - accuracy: 0.9387\n",
      "Epoch 50/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.1385 - accuracy: 0.9346\n",
      "Epoch 51/100\n",
      "489/489 [==============================] - 0s 99us/sample - loss: 0.1511 - accuracy: 0.9305\n",
      "Epoch 52/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.1493 - accuracy: 0.9305\n",
      "Epoch 53/100\n",
      "489/489 [==============================] - 0s 91us/sample - loss: 0.2193 - accuracy: 0.9305\n",
      "Epoch 54/100\n",
      "489/489 [==============================] - 0s 98us/sample - loss: 0.1759 - accuracy: 0.9284\n",
      "Epoch 55/100\n",
      "489/489 [==============================] - 0s 95us/sample - loss: 0.1611 - accuracy: 0.9346\n",
      "Epoch 56/100\n",
      "489/489 [==============================] - 0s 104us/sample - loss: 0.1544 - accuracy: 0.9387\n",
      "Epoch 57/100\n",
      "489/489 [==============================] - 0s 104us/sample - loss: 0.1539 - accuracy: 0.9366\n",
      "Epoch 58/100\n",
      "489/489 [==============================] - 0s 104us/sample - loss: 0.2063 - accuracy: 0.9202\n",
      "Epoch 59/100\n",
      "489/489 [==============================] - 0s 100us/sample - loss: 0.1865 - accuracy: 0.9407\n",
      "Epoch 60/100\n",
      "489/489 [==============================] - 0s 97us/sample - loss: 0.1660 - accuracy: 0.9325\n",
      "Epoch 61/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.1846 - accuracy: 0.9284\n",
      "Epoch 62/100\n",
      "489/489 [==============================] - 0s 100us/sample - loss: 0.1700 - accuracy: 0.9305\n",
      "Epoch 63/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.1588 - accuracy: 0.9305\n",
      "Epoch 64/100\n",
      "489/489 [==============================] - 0s 116us/sample - loss: 0.1499 - accuracy: 0.9305\n",
      "Epoch 65/100\n",
      "489/489 [==============================] - 0s 110us/sample - loss: 0.1668 - accuracy: 0.9325\n",
      "Epoch 66/100\n",
      "489/489 [==============================] - 0s 107us/sample - loss: 0.1632 - accuracy: 0.9284\n",
      "Epoch 67/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.1743 - accuracy: 0.9325\n",
      "Epoch 68/100\n",
      "489/489 [==============================] - 0s 111us/sample - loss: 0.1684 - accuracy: 0.9202\n",
      "Epoch 69/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.1729 - accuracy: 0.9223\n",
      "Epoch 70/100\n",
      "489/489 [==============================] - 0s 109us/sample - loss: 0.1519 - accuracy: 0.9427\n",
      "Epoch 71/100\n",
      "489/489 [==============================] - 0s 119us/sample - loss: 0.1280 - accuracy: 0.9509\n",
      "Epoch 72/100\n",
      "489/489 [==============================] - 0s 110us/sample - loss: 0.1586 - accuracy: 0.9305\n",
      "Epoch 73/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.1490 - accuracy: 0.9305\n",
      "Epoch 74/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.1366 - accuracy: 0.9427\n",
      "Epoch 75/100\n",
      "489/489 [==============================] - 0s 108us/sample - loss: 0.1647 - accuracy: 0.9366\n",
      "Epoch 76/100\n",
      "489/489 [==============================] - 0s 109us/sample - loss: 0.1511 - accuracy: 0.9325\n",
      "Epoch 77/100\n",
      "489/489 [==============================] - 0s 111us/sample - loss: 0.1507 - accuracy: 0.9325\n",
      "Epoch 78/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.1274 - accuracy: 0.9489\n",
      "Epoch 79/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.1379 - accuracy: 0.9346\n",
      "Epoch 80/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.1362 - accuracy: 0.9346\n",
      "Epoch 81/100\n",
      "489/489 [==============================] - 0s 100us/sample - loss: 0.1348 - accuracy: 0.9325\n",
      "Epoch 82/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.1301 - accuracy: 0.9366\n",
      "Epoch 83/100\n",
      "489/489 [==============================] - 0s 104us/sample - loss: 0.1352 - accuracy: 0.9346\n",
      "Epoch 84/100\n",
      "489/489 [==============================] - 0s 111us/sample - loss: 0.1343 - accuracy: 0.9346\n",
      "Epoch 85/100\n",
      "489/489 [==============================] - 0s 104us/sample - loss: 0.1284 - accuracy: 0.9427\n",
      "Epoch 86/100\n",
      "489/489 [==============================] - 0s 102us/sample - loss: 0.1250 - accuracy: 0.9468\n",
      "Epoch 87/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.1168 - accuracy: 0.9530\n",
      "Epoch 88/100\n",
      "489/489 [==============================] - 0s 106us/sample - loss: 0.1363 - accuracy: 0.9346\n",
      "Epoch 89/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.1363 - accuracy: 0.9305\n",
      "Epoch 90/100\n",
      "489/489 [==============================] - 0s 110us/sample - loss: 0.1178 - accuracy: 0.9509\n",
      "Epoch 91/100\n",
      "489/489 [==============================] - 0s 108us/sample - loss: 0.1747 - accuracy: 0.9468\n",
      "Epoch 92/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.1500 - accuracy: 0.9448\n",
      "Epoch 93/100\n",
      "489/489 [==============================] - 0s 109us/sample - loss: 0.1367 - accuracy: 0.9489\n",
      "Epoch 94/100\n",
      "489/489 [==============================] - 0s 113us/sample - loss: 0.1522 - accuracy: 0.9489\n",
      "Epoch 95/100\n",
      "489/489 [==============================] - 0s 111us/sample - loss: 0.1506 - accuracy: 0.9325\n",
      "Epoch 96/100\n",
      "489/489 [==============================] - 0s 103us/sample - loss: 0.1661 - accuracy: 0.9489\n",
      "Epoch 97/100\n",
      "489/489 [==============================] - 0s 101us/sample - loss: 0.2097 - accuracy: 0.9387\n",
      "Epoch 98/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.1644 - accuracy: 0.9427\n",
      "Epoch 99/100\n",
      "489/489 [==============================] - 0s 110us/sample - loss: 0.1245 - accuracy: 0.9530\n",
      "Epoch 100/100\n",
      "489/489 [==============================] - 0s 105us/sample - loss: 0.1435 - accuracy: 0.9387\n"
     ]
    }
   ],
   "source": [
    "\n",
    "history = model.fit(X_train, y_train, epochs=100, batch_size=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "y_pred = model.predict(X_test)\n",
    "#Converting predictions to label\n",
    "pred = list()\n",
    "for i in range(len(y_pred)):\n",
    "    pred.append(np.argmax(y_pred[i]))\n",
    "#Converting one hot encoded test label to label\n",
    "test = list()\n",
    "for i in range(len(y_test)):\n",
    "    test.append(np.argmax(y_test[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy is: 92.6829268292683\n"
     ]
    }
   ],
   "source": [
    "from sklearn.metrics import accuracy_score\n",
    "a = accuracy_score(pred,test)\n",
    "print('Accuracy is:', a*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 489 samples, validate on 123 samples\n",
      "Epoch 1/100\n",
      "489/489 [==============================] - 0s 579us/sample - loss: 0.1295 - accuracy: 0.9489 - val_loss: 1.8567 - val_accuracy: 0.9268\n",
      "Epoch 2/100\n",
      "489/489 [==============================] - 0s 205us/sample - loss: 0.1503 - accuracy: 0.9305 - val_loss: 2.0430 - val_accuracy: 0.9106\n",
      "Epoch 3/100\n",
      "489/489 [==============================] - 0s 207us/sample - loss: 0.1446 - accuracy: 0.9407 - val_loss: 2.3035 - val_accuracy: 0.8780\n",
      "Epoch 4/100\n",
      "489/489 [==============================] - 0s 214us/sample - loss: 0.1166 - accuracy: 0.9571 - val_loss: 2.3675 - val_accuracy: 0.8862\n",
      "Epoch 5/100\n",
      "489/489 [==============================] - 0s 189us/sample - loss: 0.1481 - accuracy: 0.9366 - val_loss: 2.3905 - val_accuracy: 0.8862\n",
      "Epoch 6/100\n",
      "489/489 [==============================] - 0s 212us/sample - loss: 0.1323 - accuracy: 0.9489 - val_loss: 2.5598 - val_accuracy: 0.8780\n",
      "Epoch 7/100\n",
      "489/489 [==============================] - 0s 227us/sample - loss: 0.1351 - accuracy: 0.9489 - val_loss: 2.8272 - val_accuracy: 0.8618\n",
      "Epoch 8/100\n",
      "489/489 [==============================] - 0s 230us/sample - loss: 0.1312 - accuracy: 0.9489 - val_loss: 3.0861 - val_accuracy: 0.8780\n",
      "Epoch 9/100\n",
      "489/489 [==============================] - 0s 211us/sample - loss: 0.1492 - accuracy: 0.9346 - val_loss: 2.9658 - val_accuracy: 0.8780\n",
      "Epoch 10/100\n",
      "489/489 [==============================] - 0s 214us/sample - loss: 0.1026 - accuracy: 0.9673 - val_loss: 2.9772 - val_accuracy: 0.8780\n",
      "Epoch 11/100\n",
      "489/489 [==============================] - 0s 201us/sample - loss: 0.1079 - accuracy: 0.9550 - val_loss: 3.0000 - val_accuracy: 0.8780\n",
      "Epoch 12/100\n",
      "489/489 [==============================] - 0s 188us/sample - loss: 0.1084 - accuracy: 0.9550 - val_loss: 2.9568 - val_accuracy: 0.8780\n",
      "Epoch 13/100\n",
      "489/489 [==============================] - 0s 187us/sample - loss: 0.0888 - accuracy: 0.9611 - val_loss: 3.2274 - val_accuracy: 0.8537\n",
      "Epoch 14/100\n",
      "489/489 [==============================] - 0s 195us/sample - loss: 0.1572 - accuracy: 0.9468 - val_loss: 3.2168 - val_accuracy: 0.8618\n",
      "Epoch 15/100\n",
      "489/489 [==============================] - 0s 193us/sample - loss: 0.1559 - accuracy: 0.9489 - val_loss: 2.6847 - val_accuracy: 0.9106\n",
      "Epoch 16/100\n",
      "489/489 [==============================] - 0s 173us/sample - loss: 0.1241 - accuracy: 0.9448 - val_loss: 2.8806 - val_accuracy: 0.8862\n",
      "Epoch 17/100\n",
      "489/489 [==============================] - 0s 180us/sample - loss: 0.1198 - accuracy: 0.9509 - val_loss: 2.9396 - val_accuracy: 0.8780\n",
      "Epoch 18/100\n",
      "489/489 [==============================] - 0s 177us/sample - loss: 0.1121 - accuracy: 0.9468 - val_loss: 2.8266 - val_accuracy: 0.9268\n",
      "Epoch 19/100\n",
      "489/489 [==============================] - 0s 183us/sample - loss: 0.1134 - accuracy: 0.9468 - val_loss: 2.8381 - val_accuracy: 0.9187\n",
      "Epoch 20/100\n",
      "489/489 [==============================] - 0s 189us/sample - loss: 0.1355 - accuracy: 0.9325 - val_loss: 2.9137 - val_accuracy: 0.9268\n",
      "Epoch 21/100\n",
      "489/489 [==============================] - 0s 199us/sample - loss: 0.1139 - accuracy: 0.9530 - val_loss: 3.1221 - val_accuracy: 0.9268\n",
      "Epoch 22/100\n",
      "489/489 [==============================] - 0s 188us/sample - loss: 0.0932 - accuracy: 0.9530 - val_loss: 3.2344 - val_accuracy: 0.9268\n",
      "Epoch 23/100\n",
      "489/489 [==============================] - 0s 184us/sample - loss: 0.1188 - accuracy: 0.9427 - val_loss: 3.2848 - val_accuracy: 0.9268\n",
      "Epoch 24/100\n",
      "489/489 [==============================] - 0s 177us/sample - loss: 0.0937 - accuracy: 0.9591 - val_loss: 3.2782 - val_accuracy: 0.9268\n",
      "Epoch 25/100\n",
      "489/489 [==============================] - 0s 177us/sample - loss: 0.0925 - accuracy: 0.9550 - val_loss: 3.2717 - val_accuracy: 0.9268\n",
      "Epoch 26/100\n",
      "489/489 [==============================] - 0s 178us/sample - loss: 0.0842 - accuracy: 0.9591 - val_loss: 3.2723 - val_accuracy: 0.9268\n",
      "Epoch 27/100\n",
      "489/489 [==============================] - 0s 197us/sample - loss: 0.1127 - accuracy: 0.9468 - val_loss: 3.1959 - val_accuracy: 0.9268\n",
      "Epoch 28/100\n",
      "489/489 [==============================] - 0s 201us/sample - loss: 0.1044 - accuracy: 0.9468 - val_loss: 3.1651 - val_accuracy: 0.9268\n",
      "Epoch 29/100\n",
      "489/489 [==============================] - 0s 199us/sample - loss: 0.1014 - accuracy: 0.9489 - val_loss: 3.1207 - val_accuracy: 0.9350\n",
      "Epoch 30/100\n",
      "489/489 [==============================] - 0s 208us/sample - loss: 0.1001 - accuracy: 0.9550 - val_loss: 3.0508 - val_accuracy: 0.9187\n",
      "Epoch 31/100\n",
      "489/489 [==============================] - 0s 196us/sample - loss: 0.1002 - accuracy: 0.9591 - val_loss: 3.1412 - val_accuracy: 0.8943\n",
      "Epoch 32/100\n",
      "489/489 [==============================] - 0s 177us/sample - loss: 0.0941 - accuracy: 0.9611 - val_loss: 3.2653 - val_accuracy: 0.8537\n",
      "Epoch 33/100\n",
      "489/489 [==============================] - 0s 183us/sample - loss: 0.1065 - accuracy: 0.9448 - val_loss: 3.2685 - val_accuracy: 0.8618\n",
      "Epoch 34/100\n",
      "489/489 [==============================] - 0s 181us/sample - loss: 0.0954 - accuracy: 0.9571 - val_loss: 2.9130 - val_accuracy: 0.9106\n",
      "Epoch 35/100\n",
      "489/489 [==============================] - 0s 179us/sample - loss: 0.0852 - accuracy: 0.9571 - val_loss: 2.9521 - val_accuracy: 0.9187\n",
      "Epoch 36/100\n",
      "489/489 [==============================] - 0s 173us/sample - loss: 0.1292 - accuracy: 0.9693 - val_loss: 3.3309 - val_accuracy: 0.8699\n",
      "Epoch 37/100\n",
      "489/489 [==============================] - 0s 187us/sample - loss: 0.1611 - accuracy: 0.9407 - val_loss: 3.5537 - val_accuracy: 0.8862\n",
      "Epoch 38/100\n",
      "489/489 [==============================] - 0s 191us/sample - loss: 0.1591 - accuracy: 0.9489 - val_loss: 3.3981 - val_accuracy: 0.8699\n",
      "Epoch 39/100\n",
      "489/489 [==============================] - 0s 185us/sample - loss: 0.1993 - accuracy: 0.9530 - val_loss: 3.1553 - val_accuracy: 0.8374\n",
      "Epoch 40/100\n",
      "489/489 [==============================] - 0s 188us/sample - loss: 0.1144 - accuracy: 0.9509 - val_loss: 2.9532 - val_accuracy: 0.8862\n",
      "Epoch 41/100\n",
      "489/489 [==============================] - 0s 180us/sample - loss: 0.1321 - accuracy: 0.9673 - val_loss: 2.9544 - val_accuracy: 0.8780\n",
      "Epoch 42/100\n",
      "489/489 [==============================] - 0s 190us/sample - loss: 0.1472 - accuracy: 0.9448 - val_loss: 2.3856 - val_accuracy: 0.9106\n",
      "Epoch 43/100\n",
      "489/489 [==============================] - 0s 180us/sample - loss: 0.1100 - accuracy: 0.9571 - val_loss: 2.4969 - val_accuracy: 0.9106\n",
      "Epoch 44/100\n",
      "489/489 [==============================] - 0s 175us/sample - loss: 0.2069 - accuracy: 0.9509 - val_loss: 2.4637 - val_accuracy: 0.9024\n",
      "Epoch 45/100\n",
      "489/489 [==============================] - 0s 196us/sample - loss: 0.1852 - accuracy: 0.9632 - val_loss: 2.2361 - val_accuracy: 0.8943\n",
      "Epoch 46/100\n",
      "489/489 [==============================] - 0s 168us/sample - loss: 0.0978 - accuracy: 0.9571 - val_loss: 2.2013 - val_accuracy: 0.8780\n",
      "Epoch 47/100\n",
      "489/489 [==============================] - 0s 182us/sample - loss: 0.1095 - accuracy: 0.9550 - val_loss: 2.1621 - val_accuracy: 0.8862\n",
      "Epoch 48/100\n",
      "489/489 [==============================] - 0s 168us/sample - loss: 0.1148 - accuracy: 0.9530 - val_loss: 2.1674 - val_accuracy: 0.9106\n",
      "Epoch 49/100\n",
      "489/489 [==============================] - 0s 175us/sample - loss: 0.1064 - accuracy: 0.9489 - val_loss: 2.1793 - val_accuracy: 0.9106\n",
      "Epoch 50/100\n",
      "489/489 [==============================] - 0s 183us/sample - loss: 0.1007 - accuracy: 0.9550 - val_loss: 2.0291 - val_accuracy: 0.9187\n",
      "Epoch 51/100\n",
      "489/489 [==============================] - 0s 187us/sample - loss: 0.1008 - accuracy: 0.9530 - val_loss: 1.9276 - val_accuracy: 0.9187\n",
      "Epoch 52/100\n",
      "489/489 [==============================] - 0s 190us/sample - loss: 0.1136 - accuracy: 0.9550 - val_loss: 1.8500 - val_accuracy: 0.9024\n",
      "Epoch 53/100\n",
      "489/489 [==============================] - 0s 197us/sample - loss: 0.1028 - accuracy: 0.9550 - val_loss: 1.9961 - val_accuracy: 0.8943\n",
      "Epoch 54/100\n",
      "489/489 [==============================] - 0s 184us/sample - loss: 0.1170 - accuracy: 0.9571 - val_loss: 2.0001 - val_accuracy: 0.9187\n",
      "Epoch 55/100\n",
      "489/489 [==============================] - 0s 168us/sample - loss: 0.0805 - accuracy: 0.9652 - val_loss: 2.1036 - val_accuracy: 0.9106\n",
      "Epoch 56/100\n",
      "489/489 [==============================] - 0s 202us/sample - loss: 0.0952 - accuracy: 0.9591 - val_loss: 2.1605 - val_accuracy: 0.9106\n",
      "Epoch 57/100\n",
      "489/489 [==============================] - 0s 189us/sample - loss: 0.0857 - accuracy: 0.9632 - val_loss: 2.1835 - val_accuracy: 0.9106\n",
      "Epoch 58/100\n",
      "489/489 [==============================] - 0s 181us/sample - loss: 0.1033 - accuracy: 0.9591 - val_loss: 1.8145 - val_accuracy: 0.9106\n",
      "Epoch 59/100\n",
      "489/489 [==============================] - 0s 187us/sample - loss: 0.1025 - accuracy: 0.9571 - val_loss: 1.8589 - val_accuracy: 0.9106\n",
      "Epoch 60/100\n",
      "489/489 [==============================] - 0s 171us/sample - loss: 0.1097 - accuracy: 0.9571 - val_loss: 1.8863 - val_accuracy: 0.9187\n",
      "Epoch 61/100\n",
      "489/489 [==============================] - 0s 190us/sample - loss: 0.0745 - accuracy: 0.9755 - val_loss: 2.0042 - val_accuracy: 0.8943\n",
      "Epoch 62/100\n",
      "489/489 [==============================] - 0s 203us/sample - loss: 0.1073 - accuracy: 0.9571 - val_loss: 2.1132 - val_accuracy: 0.8943\n",
      "Epoch 63/100\n",
      "489/489 [==============================] - 0s 192us/sample - loss: 0.0843 - accuracy: 0.9693 - val_loss: 2.2050 - val_accuracy: 0.8943\n",
      "Epoch 64/100\n",
      "489/489 [==============================] - 0s 205us/sample - loss: 0.0916 - accuracy: 0.9632 - val_loss: 2.3001 - val_accuracy: 0.8862\n",
      "Epoch 65/100\n",
      "489/489 [==============================] - 0s 200us/sample - loss: 0.1076 - accuracy: 0.9571 - val_loss: 2.3486 - val_accuracy: 0.8862\n",
      "Epoch 66/100\n",
      "489/489 [==============================] - 0s 194us/sample - loss: 0.0878 - accuracy: 0.9652 - val_loss: 2.5907 - val_accuracy: 0.8943\n",
      "Epoch 67/100\n",
      "489/489 [==============================] - 0s 185us/sample - loss: 0.2831 - accuracy: 0.9571 - val_loss: 3.0070 - val_accuracy: 0.9106\n",
      "Epoch 68/100\n",
      "489/489 [==============================] - 0s 199us/sample - loss: 0.1617 - accuracy: 0.9652 - val_loss: 2.8343 - val_accuracy: 0.7967\n",
      "Epoch 69/100\n",
      "489/489 [==============================] - 0s 181us/sample - loss: 0.2280 - accuracy: 0.9448 - val_loss: 2.1082 - val_accuracy: 0.8943\n",
      "Epoch 70/100\n",
      "489/489 [==============================] - 0s 199us/sample - loss: 0.2993 - accuracy: 0.9509 - val_loss: 1.8381 - val_accuracy: 0.9024\n",
      "Epoch 71/100\n",
      "489/489 [==============================] - 0s 195us/sample - loss: 0.1777 - accuracy: 0.9611 - val_loss: 2.5331 - val_accuracy: 0.8293\n",
      "Epoch 72/100\n",
      "489/489 [==============================] - 0s 196us/sample - loss: 0.2378 - accuracy: 0.9448 - val_loss: 2.2622 - val_accuracy: 0.8537\n",
      "Epoch 73/100\n",
      "489/489 [==============================] - 0s 181us/sample - loss: 0.1321 - accuracy: 0.9550 - val_loss: 2.0022 - val_accuracy: 0.8699\n",
      "Epoch 74/100\n",
      "489/489 [==============================] - 0s 185us/sample - loss: 0.1209 - accuracy: 0.9509 - val_loss: 1.9652 - val_accuracy: 0.8780\n",
      "Epoch 75/100\n",
      "489/489 [==============================] - 0s 185us/sample - loss: 0.1118 - accuracy: 0.9591 - val_loss: 1.9065 - val_accuracy: 0.8862\n",
      "Epoch 76/100\n",
      "489/489 [==============================] - 0s 211us/sample - loss: 0.1217 - accuracy: 0.9448 - val_loss: 1.9760 - val_accuracy: 0.8862\n",
      "Epoch 77/100\n",
      "489/489 [==============================] - 0s 201us/sample - loss: 0.1072 - accuracy: 0.9591 - val_loss: 2.0156 - val_accuracy: 0.8537\n",
      "Epoch 78/100\n",
      "489/489 [==============================] - 0s 197us/sample - loss: 0.1046 - accuracy: 0.9571 - val_loss: 2.0862 - val_accuracy: 0.8618\n",
      "Epoch 79/100\n",
      "489/489 [==============================] - 0s 207us/sample - loss: 0.1012 - accuracy: 0.9550 - val_loss: 2.1271 - val_accuracy: 0.8537\n",
      "Epoch 80/100\n",
      "489/489 [==============================] - 0s 174us/sample - loss: 0.1153 - accuracy: 0.9509 - val_loss: 2.1156 - val_accuracy: 0.8618\n",
      "Epoch 81/100\n",
      "489/489 [==============================] - 0s 182us/sample - loss: 0.1071 - accuracy: 0.9530 - val_loss: 2.2681 - val_accuracy: 0.8455\n",
      "Epoch 82/100\n",
      "489/489 [==============================] - 0s 164us/sample - loss: 0.1126 - accuracy: 0.9509 - val_loss: 2.1859 - val_accuracy: 0.8537\n",
      "Epoch 83/100\n",
      "489/489 [==============================] - 0s 170us/sample - loss: 0.1158 - accuracy: 0.9468 - val_loss: 2.3451 - val_accuracy: 0.8374\n",
      "Epoch 84/100\n",
      "489/489 [==============================] - 0s 189us/sample - loss: 0.1088 - accuracy: 0.9509 - val_loss: 2.2907 - val_accuracy: 0.8537\n",
      "Epoch 85/100\n",
      "489/489 [==============================] - 0s 183us/sample - loss: 0.0825 - accuracy: 0.9652 - val_loss: 2.2782 - val_accuracy: 0.8699\n",
      "Epoch 86/100\n",
      "489/489 [==============================] - 0s 200us/sample - loss: 0.0867 - accuracy: 0.9591 - val_loss: 2.3830 - val_accuracy: 0.8780\n",
      "Epoch 87/100\n",
      "489/489 [==============================] - 0s 191us/sample - loss: 0.1146 - accuracy: 0.9468 - val_loss: 2.4994 - val_accuracy: 0.8862\n",
      "Epoch 88/100\n",
      "489/489 [==============================] - 0s 188us/sample - loss: 0.0828 - accuracy: 0.9611 - val_loss: 2.2786 - val_accuracy: 0.8862\n",
      "Epoch 89/100\n",
      "489/489 [==============================] - 0s 178us/sample - loss: 0.0940 - accuracy: 0.9611 - val_loss: 2.3149 - val_accuracy: 0.8862\n",
      "Epoch 90/100\n",
      "489/489 [==============================] - 0s 176us/sample - loss: 0.0751 - accuracy: 0.9714 - val_loss: 2.4484 - val_accuracy: 0.8780\n",
      "Epoch 91/100\n",
      "489/489 [==============================] - 0s 204us/sample - loss: 0.0782 - accuracy: 0.9714 - val_loss: 2.6419 - val_accuracy: 0.8374\n",
      "Epoch 92/100\n",
      "489/489 [==============================] - 0s 185us/sample - loss: 0.1123 - accuracy: 0.9550 - val_loss: 2.7570 - val_accuracy: 0.8293\n",
      "Epoch 93/100\n",
      "489/489 [==============================] - 0s 178us/sample - loss: 0.0948 - accuracy: 0.9591 - val_loss: 3.1320 - val_accuracy: 0.8293\n",
      "Epoch 94/100\n",
      "489/489 [==============================] - 0s 185us/sample - loss: 0.0783 - accuracy: 0.9693 - val_loss: 3.4081 - val_accuracy: 0.8130\n",
      "Epoch 95/100\n",
      "489/489 [==============================] - 0s 188us/sample - loss: 0.0679 - accuracy: 0.9714 - val_loss: 3.3544 - val_accuracy: 0.8455\n",
      "Epoch 96/100\n",
      "489/489 [==============================] - 0s 180us/sample - loss: 0.2246 - accuracy: 0.9796 - val_loss: 2.6309 - val_accuracy: 0.8537\n",
      "Epoch 97/100\n",
      "489/489 [==============================] - 0s 191us/sample - loss: 0.0719 - accuracy: 0.9755 - val_loss: 2.6417 - val_accuracy: 0.8537\n",
      "Epoch 98/100\n",
      "489/489 [==============================] - 0s 197us/sample - loss: 0.0776 - accuracy: 0.9714 - val_loss: 2.6648 - val_accuracy: 0.8537\n",
      "Epoch 99/100\n",
      "489/489 [==============================] - 0s 167us/sample - loss: 0.1952 - accuracy: 0.9530 - val_loss: 3.3067 - val_accuracy: 0.8618\n",
      "Epoch 100/100\n",
      "489/489 [==============================] - 0s 192us/sample - loss: 0.1413 - accuracy: 0.9714 - val_loss: 2.7344 - val_accuracy: 0.8293\n"
     ]
    }
   ],
   "source": [
    "history = model.fit(X_train, y_train,validation_data = (X_test,y_test), epochs=100, batch_size=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAY4AAAEWCAYAAABxMXBSAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nOydd3hb1d34P0fytuW9EjuJEydO4thJSJwwkxQIEMqGsPrSll3aUkoptLQvP0pp+0JbSlltKaWhUMpKWGEGCmSws5w4y07sOF7xtuU9ZJ3fH+deWZYlW3Isr9zP8+ixdJfOleTzPd8tpJQYGBgYGBh4i2m0B2BgYGBgML4wBIeBgYGBgU8YgsPAwMDAwCcMwWFgYGBg4BOG4DAwMDAw8AlDcBgYGBgY+IQhOAwMPCCESBNCSCFEgBfHXiuE+HQkxmVgMNoYgsNgQiCEKBZCdAkh4l2252qTf9rojMzAYOJhCA6DicRh4Gr9hRAiGwgdveGMDbzRmAwMfMEQHAYTiX8D33F6/V3gOecDhBBRQojnhBA1QogjQoh7hBAmbZ9ZCPGQEKJWCFEEnOfm3H8KIY4KIcqFEL8VQpi9GZgQYq0QolIIYRVCbBZCzHPaFyqE+JM2HqsQ4lMhRKi27zQhxOdCiEYhRKkQ4lpt+0YhxI1O1+hjKtO0rB8KIQ4CB7Vtj2rXaBJCbBdCLHM63iyE+KUQolAI0aztnyKE+IsQ4k8u9/KWEOJ2b+7bYGJiCA6DicSXQKQQYq42oV8JPO9yzONAFDADWIESNNdp+24CzgdOAHKA1S7nPgvYgJnaMWcDN+Id7wGzgERgB/Afp30PAYuBU4BY4GeAXQgxVTvvcSABWAjkevl+ABcDJwKZ2uut2jVigReAtUKIEG3fHSht7ZtAJHA90Kbd89VOwjUeOBN40YdxGEw0pJTGw3iM+wdQDKwE7gEeAFYBHwIBgATSADPQCWQ6nfc9YKP2/GPgFqd9Z2vnBgBJ2rmhTvuvBj7Rnl8LfOrlWKO160ahFm/twAI3x/0CeN3DNTYCNzq97vP+2vXPGGQcDfr7AvnARR6O2w+cpT2/FXh3tL9v4zG6D8P2aTDR+DewGZiOi5kKiAeCgCNO244AKdrzyUCpyz6daUAgcFQIoW8zuRzvFk37+R1wOUpzsDuNJxgIAQrdnDrFw3Zv6TM2IcRPURrSZJRgidTGMNh7PQtcgxLE1wCPHsOYDCYAhqnKYEIhpTyCcpJ/E3jNZXct0I0SAjpTgXLt+VHUBOq8T6cUpXHESymjtUeklHIeg/Mt4CKURhSF0n4AhDamDiDdzXmlHrYDtAJhTq+T3RzjKH2t+TN+DlwBxEgpowGrNobB3ut54CIhxAJgLvCGh+MMjhMMwWEwEbkBZaZpdd4opewBXgF+J4SwCCGmoWz7uh/kFeA2IUSqECIGuNvp3KPAB8CfhBCRQgiTECJdCLHCi/FYUEKnDjXZ/5/Tde3AGuBhIcRkzUl9shAiGOUHWSmEuEIIESCEiBNCLNROzQUuFUKECSFmavc82BhsQA0QIIS4F6Vx6DwN/EYIMUso5gsh4rQxlqH8I/8GXpVStntxzwYTGENwGEw4pJSFUsptHnb/CLVaLwI+RTmJ12j7/gFsAHahHNiuGst3UKaufSj/wDpgkhdDeg5l9irXzv3SZf+dQB5qcq4Hfg+YpJQlKM3pp9r2XGCBds6fgS6gCmVK+g8DswHlaC/QxtJBX1PWwyjB+QHQBPyTvqHMzwLZKOFhcJwjpDQaORkYGAyMEGI5SjNL07Qkg+MYQ+MwMDAYECFEIPBj4GlDaBiAITgMDAwGQAgxF2hEmeQeGeXhGIwRDFOVgYGBgYFPGBqHgYGBgYFPHBcJgPHx8TItLW20h2FgYGAwrti+fXutlDLBdftxITjS0tLYts1TdKaBgYGBgTuEEEfcbTdMVQYGBgYGPmEIDgMDAwMDnzAEh4GBgYGBTxwXPg53dHd3U1ZWRkdHx2gPZUQICQkhNTWVwMDA0R6KgYHBOOe4FRxlZWVYLBbS0tJwKpM9IZFSUldXR1lZGdOnTx/t4RgYGIxzjltTVUdHB3FxcRNeaAAIIYiLiztutCsDAwP/ctwKDuC4EBo6x9O9GhgY+JfjWnAYGBgYjBf2VTTx0f6q0R4GYAiOUaOuro6FCxeycOFCkpOTSUlJcbzu6ury6hrXXXcd+fn5fh6pgYHBWOB/38jj5n9vZ0dJw2gP5fh1jo82cXFx5ObmAnDfffcRERHBnXfe2ecYvTG8yeRevj/zzDN+H6eBgcHoU1rfxs6SRgDueDmXd3+8jLCg0Zu+/apxCCFWCSHyhRCHhBB3u9k/TQjxkRBitxBioxAiVdt+uhAi1+nRIYS4WNv3LyHEYad9C12vO545dOgQWVlZ3HLLLSxatIijR49y8803k5OTw7x587j//vsdx5522mnk5uZis9mIjo7m7rvvZsGCBZx88slUV1eP4l0YGBgMJ+t3VQDw0OULOFLfxu/e2T+q4/GbyBJCmIG/AGcBZcBWIcR6KeU+p8MeAp6TUj4rhDgDeAD4tpTyE2Chdp1Y4BCqpaXOXVLKdcM11l+/tZd9FU3DdTkAMidH8qsL5g3p3H379vHMM8/w5JNPAvDggw8SGxuLzWbj9NNPZ/Xq1WRmZvY5x2q1smLFCh588EHuuOMO1qxZw91395PVBgYG45C3dlWwaGo0qxenUlDVzFObi1g5N4nT5ySOynj8qXEsBQ5JKYuklF3AS8BFLsdkAh9pzz9xsx9gNfCelLLNbyMdY6Snp7NkyRLH6xdffJFFixaxaNEi9u/fz759+/qdExoayrnnngvA4sWLKS4uHqnhGoxTDlU3k3HPexRUNY/2UAwGoKCqmQOVzVy4YDIAPz07gznJFu5at5u6ls5RGZM/jWQpQKnT6zLgRJdjdgGXAY8ClwAWIUSclLLO6ZirgIddzvudEOJelNC5W0rZ79MTQtwM3AwwderUAQc6VM3AX4SHhzueHzx4kEcffZSvv/6a6OhorrnmGrf5GEFBQY7nZrMZm802ImM1GL/sONJIl83OjiMNZCRZRns4Bh5Yn1uBScB585XgCA4w8+crF3LRE5/xy9fzePKaxSMebu9PjcPdnbi2G7wTWCGE2AmsAMoBx4wnhJgEZAMbnM75BTAHWALEAj939+ZSyqeklDlSypyEhH7l5McNTU1NWCwWIiMjOXr0KBs2bBj8JAMDLyisaenz12DsIaVk/a4KTkmPJ8ES7Ng+d1IkPz07gw17q3h1R/mIj8ufGkcZMMXpdSpQ4XyAlLICuBRACBEBXCaltDodcgXwupSy2+mco9rTTiHEMyjhM2FZtGgRmZmZZGVlMWPGDE499dTRHpLBBKGwphWAIu2vwdhjV5mVkvo2bj19Zr99Ny6bwUcHqrlv/V5OnB7LlNiwERuX33qOCyECgALgTJQmsRX4lpRyr9Mx8UC9lNIuhPgd0COlvNdp/5fALzRnub5tkpTyqFC62Z+BDinlgF7gnJwc6drIaf/+/cydO/eY73M8cTzes4FnznhoI0W1raTFhbHxrtNHezgTkh675Lfv7KO6WVnTzUJwy4p0MidHDnqu3S65c90u3t51lK33rCQqtH+B0tL6Ns59dAtZKZG8eNNJfUxWuaWN/GNLEXevmjNkoSKE2C6lzHHd7jdTlZTSBtyKMjPtB16RUu4VQtwvhLhQO+wbQL4QogBIAn7nNOA0lMayyeXS/xFC5AF5QDzwW3/dg4HBRKW7x05JfRuBZkFpQzudtp7RHtKEZEdJA898VkxuSSMHjjbxTt5RXtlWOuh51U0dfGfN17y2o5xvnTjVrdAAmBIbxi0rZvBlUT31rX0Th3eXNfLO7qMEBQz/NO/XDBIp5bvAuy7b7nV6vg5wG1YrpSxGOdhdt58xvKM0mCh8uK+Kk9PjiAg28loH40hdGza7ZEVGApsKaiipa2OWk4P8/T1HOWVmPJEhRhn+Y2FzQQ0mAe/etoyosEAu/stn5FcOHMX26cFabntpJ21dNh64NJurlkwZ8Ph5KVEAFNW2EhfR6wcpqmklIjiARCffyHBhlBwxmBAUVDVz03PbeH3nyDsKxyO6Q/yszKQ+rwEOVbdwy/M7+NMGo5zNsbK5oIaFU6KJClMCeHaShfyqZjy5CKSU/PzV3USHBfL2j5Zx9dKpg0ZMpcdHAFBY3TfIobCmhRkJ4X6JuDIExxiiy2b3+IMyGJivD9cDUGltH9L51vZurG3dgx84QdAd4ivn6oKj10G+tVh9li9vKx21PIGJQH1rF7vLrSzP6I3qnJ1sob61i9oW9/XoCmtaKW9s5/pTpzMzMcKr90mJCSUowERRbd8gh8LqFtITvLuGrxiCY4xgt0sKqpo9/qAMBmb7EVX4rappaBPdHS/ncuuLO4ZzSGOawpoWEi3BJEeFkBwZ0kfj2FbcQFiQmY5uO899cWQURzm++fRQLVLCChfBAXg0V20uqAH6njMYZpNgelx4H42jrctGhbWDGfHhA5w5dAzBMUaw2e3YpaSl00jcGwrbjqhVsh694iu5pY2D2p4nEkWaGQMgPTG8T0jutiP1LJsVz8q5iTz7RTFtXcZvcihsLqghOiyQ+anRjm0OweEhW3/zwRpmxIf7HAWVnhjeR+PQv890L7UWXzEExyjhWlY9bdpUrjhnGeeuOInOTu8nvzVr1lBZWenHkY59qpo6KK1XJqrqJt+7HNY0d1LX2kV1cycd3WM/uuieN/L49j+/GrJZU0pJYU2rw4wxIz6CwpoWpJRUN3dwpK6NnGmx3LIinca2bl7e2j8KSErJ5U9+zh/eP9Bv38/W7eKOl3OHNLaxyPpdFZz64Mc+mTKllGwuqOHUmfGYTb0+hviIYOLCg8iv7F8br6O7hy+L6vqYtrxlRnwEJfVtdNnsAA4hoi8OhhtDcIwSeln13NxcbrnlFn5w6228smELr2zYgjR5HxVkCA5lWgGYNzlySBqHs6ZR0Tg0H8lI0dppY932MrYcrOUrza/jK/WtXVjbu5mhCY70hHCaO2zUtnSxXfssF6fFkJMWS860GJ7ecpjuHnufa+ytaGJrcQMvfl3SZ19jWxev7yzn88I6JgobD1RT3tjO8195b7Y7UNlMdXMnK2b1FwKzky3kV/XP1t9W3EBHt53lGfE+jzE9MZweu6SkXgmMwuoWhIC0OENwTGjsTqvHf655hqVLl7Jw4UJ+8IMfYLfbsdlsfPvb3yY7O5usrCwee+wxXn75ZXJzc7nyyit9agA10dh2pJ6QQBNnzEmkvrXLseryFmezQWmDd4JjR0kDt/x7+4hrKB/uq6Kj206Q2cTfNxX22ffO7qPc8UruoPevO8LTtdWoLkAKa1rYdqSB4AATWZNViOf3VqRT3tjOO7uP9rnGW1qZ74a2bj49WOvY/v6eSrp7JJVNHeNCe/OGvHJVzOKZzw57fU+6r2KZGyGQkWThYFUzdntfjXHzwRqCzCZOmhHn8xhnaJFVh6q1agC1rUyJCSMk0OzztbzBCHgHeO9uqMwb3msmZ8O5D3p9uP4jOlywn/Vvvsnnn39OQEAAN998My+99BLp6enU1taSl6fG2djYSHR0NI8//jhPPPEECxdOqLYkPrGtuIEFqdFMjg4FoLal0/HcG/Irmwgym+jqsVPW4F0R5td2lPH+3krWbS/jmpOmDWncQ2H9rgomRYVw9dKpPPxhAQcqm5iTHMnh2lbuXLuL9u4eUqJD+enZsz1eo0hzhOumKt0OXlTTyrYjDSyYEu1IGjtzTiIzEyN4clMhFy2cjBACu13y1q4KTpsZT165lfW7KhzlvfW+EQAl9W3jvnhiW5eNwpoWTpwey1eH673+vjcfrGF2koVJUf1/h3OSLbR19VDW0M7UuF5fxuaCGnLSYobUoEk3SRXVavXHqlv8ZqYCQ+MYM9ilKkew/Yst5O7cTk5ODgsXLmTTpk0UFhYyc+ZM8vPzueUHt/LGW+8QFRU12kP2K3vKrTy9pWhQO35rp419R5tYkhZLUqRKdKry0c+RX9XComnRBJoFZV5qHLp57B9biuixj0wIdUNrF5sLarhgwWS+c/I0woLM/H1TEbYeOz95OZegABNnZybxl08OOaLM3FFY00JwgMkhXCdFhhASaGJPhZW95VZypsU4jjWZBDcvn8GBymY2aavo7SUNVFg7uGxxCudmJfPB3krau3qoburgi6I6R0RQca3nGliHqlt47oviY/9Q/Mz+o03YpaoLtSA1yuP33dZl4/639nH3q7u5+9XdbD3c4NHklKE5yA84+TkqrR0cqGwekn8DwBISSFJkMIXVrdjtkqJa/4XigqFxKHzQDPxFj10SYBYEmgQXXfE//OXh3xNo7ivXt+/MZc1Lr/PIY4/x7ltv8tRTT43SaP2H3S55+tMi/rghn+4eyWmz4pmT7Lmuz67SRnrsksVpMSRoWbO++DnsdsnBqmauyJlCpbWD0vrBNQ5rezf5Vc1kp0SRV27l/T2VnDd/ktfvOVTe21OJzS65cMFkosOCuHrpVP71eTEhgWZySxt54lsnsCIjgVWPbOGOV3J597ZlhLvJoi+qaWV6fLjDaWsyCWbER/DO7qPY7JKctJg+x1+8MIWHPyjg75uK+MbsRNbnVhAcYOKszGSSLCG8tLWUjw9UU93cgZTwozNmqmz0AT7Lv248xGs7yrlg/mRiwoM8Hjfa5JUpM1V2ShS3rEjn+//Z4fb73pRfw5rPDhMfEYTZJEiKCuaihf0KXwA4tLCCqmbOnpcMKA0FfAvDdWVGfARFtS0cbeqgo9tuaBzHA3YpMZtMnHPOSj54+w1KypXDu66ujpKSEmpqaqhr7uCs8y7i+z/5BTt2qJwDi8VCc/PIh5F+VVRHQ2t/n8qecqvX5h5X6lo6+e4zX/N/7x5whDAOFiK77UgDQsCiqTGO0gq+RFaVNbTT1tXDnGQLqTFhXmkcO0sakBJ+tmo20+PDeXJT4Ygkbq7fVc6M+HDmaQXybjhtOgJ48esSLl44mfPnT8YSEsifr1xISX0b97+1z+24Cmv6r0ZnJIRjbVdRQ4um9hUcQQEmbjhtOl8U1bH9SAPv5h1l5dwkIoIDOHFGHAmWYNbvKmf9rgrmTookJy2WqNBAiuvcaxx2u2RzgfKLeApLdUVKycb8arer/e1H+tdpGi7yypuIjwgmKTKYs+cle/y+88qtBJgEn/78DL765Uq2/OwMslLcWwUiggNIjQnlgNNve3NBDYmWYOYkD920l56ocjn0fA5/ahyG4Bgj2CUEmAQ5Jyzk+z/5OReet4r58+dz9tlnU1VVRfGREi5YtZIrzlnGr352G/fd/xsArrvuOm688cYRdY63ddn4n6e/4nv/3t7nH/lQdQurn/ycX725d4Cz3WO3S257aSdfH67n/y7J5oWbTiTAJAbtTre1uJ6MRAtRoYHERQRjEr5pHLq5ICPZwpTYUK8Ex7biBswmwaKpMdy0bAZ55Va+8HMUUaW1g68O13PBgsmOEhKTo0P51olTSYsL49cXZTmOXTpdhdK+vK2Um57b3mdS7bT1UFLf5nCM6+iTTEZSBNFh/TWAq5ZOwRISwI9f2kldaxcXaN3ozCbBedmT+PhANTtLGh1d6qbFhXGkzv0CYn9lE7VaRrq33Qc/PlDNtc9sZcPevhGE1U0drH7yC855ZLPDIT2c7Cm3kpUSiRACs0lw3alp5JVbOehS3iOv3EpGksVrZ/ScZIvj3g9WNfPhvirOnJt4TOVBZsRH0NRhc2T++1PjMExVx4iUUjMzDV0G33fffeyraCLALDAJweorr+LSy6/sU3SuprmTl9/frK2K25gSo5xqV1xxBVdcccUx34cvHKxqwWaXfF1czz+2FHHLinS6e+zc8UouHd12thbXY7dLTCbv/wn+9Xkxnx2q48FLs7lqqerYOD0+nPzK/mGLxbWttHf3ICXkljRy4cLeSSw+IphqH7LH9X/ejCSlcdS2dNLe1UNokJoAumx22rt7+lQn3XaknsxJkYQHB3DpohQe/rCAv20q5JSZvodResvbuyuQEse96vz6wnnY7LKfWfOus2cTHxHM7987wKpHNvObi7OYGqs0KrvsjaTS0SeZxdNi3b6/JSSQb580jb9uLMQSHMA3ZveaVC5cOJl/fV4MwAULlAlnWlw4u0ob3V5L1zZCAk19Vt0D8WaucrrvKmvkm9m9ZqK8civ64v87a77mxtOmc9eq2QQHHHs0UXtXDwermzl7XpJj23ItvHZrcb3D5CSlZE+5lbMzk72+dkaShY35NbR22rj95VzCgwO44yzPAQ3eoAc5fLivCktIgMN06w8MjeMYqWvtIr+quU84ra8o4WMnwKS+jvCgADq67Y7VvF1Kals6CQ8OICYsEJMQtI9iqKNuPlo4JZo/fZDPvoomHv/4ELvLrJydmURTh41DPnSVO1jVzIPvH2Dl3ESudKoEquLd+yZKbSuu5xsPbeTcR7fwzce20NxpY+n03skuMTKYqmbvTVUHKptJjQl1mA8Ayht7V8p/3HCAlQ9vcmRPd/fYyS1tZLHmQA4JNHPDadPZcrCW+9/a55fy5B8fqOIvnxwiKyWyn/lBCNFPaIDyW9xw2nRe/+EpWEIC+N6/t3Puo1u46TnVl2a2i0lk7iRl/jpphnvBAXDdqdMJDjBxbnZyn5X1CVOimRYXxpK0GFK1Bc202DDKG9v75X+AMsvMSbYwPyWaAi8ER1uXjQ/3VQFKA3Amr9yKELDh9uV85+RpPP3pYS796+fD0tVwf6VyjDubnKbFhREfEeTIdwEob2ynoa2brJTBe2zozE62YLNLbn85l70VTTxwaXafDn9DQS8vcqCymRkJEX5tJ2toHMdIa6eNHruky2Yfcsy0zS6RQIBZfdFhwWZks6S9y0ZESCCNbd1099hJjQlFCEFIoJn2rlEUHFXNhASaePq7OZz76Ba+9/w2Kho7uHRRCredMYsP9lWxrdi7PtZdNju3v5yLJTiABy6d3+fHPjvJwtu7j9LaaXM4eT87VIcQ8NhVJxBoFgQHmFk2q3eln2QJ4ajVe8FRUNXssCvrgqO0oZ2ZiWrbloO11DR38srWUq49dTp7K5ro6LazJK13gr3htOlUNXWw5rPDfFFUx+NXL3Scfyx0dPfw4HsH+NfnxcydFMmjV53g8zXmTY7i7R8t47NDtdjsahKPDAl0CAqdjCQLb916msN/4o4ESzDv3HYaiZEhfbYLIfjPjScS5CTApsWF0WOXlDe0k+ZUL6m108a2I/Vcf+p02rp6eCO3HCnlgJPcR/urae/uYVZiBHvKm/ocv6fcSnpCBLHhQdx/URbLZyVw17pdnP/Yp/zqgkyuXDJlyBOoLqSynQSHEILF02LY5hS1ph/nyafhDl1wf7iviityUjlnnvfaiidSokMJDjDRabP3M0UON8e1xjEcDs3ObvXP6GvSmTO6ZhGgmXbCNDPJ4do28sqtlDe0ERJodvSZCAsya6Ya78c/nM7b/MpmMpIsxEcE84fV8ymtbyc5MoT7LpznWJFtK/Yuq/mpzYUeV1z6P5ezHXzbkXpmJ1m4YMFkVmVN4vQ5iX3MhImRwVR7qXF02ewU1bQ6BJxu/ivTooH06CmAf2w5jK3H7rgv58ijoAAT9104jzXX5lDV1MH5j3/KC1+VePzMrW3drHpkM/e/ta9PQtmu0kbO/vMmMu55j4x73iPrVxv41+fF3HDadN744SlDdnaGBplZmZnEqqxJrMqa5NGklp0aNah5cWaixW2PjtSYsD4CRRcWrg7yLwrr6O6RLM9IICPZQnOHrY+g31pcz/I/fMIhJx/C+l0VJEeG8N1T0rC2dzvKy4DSOJwn9pWZSbx/+3IWT4vh7tfyePC9/iVRvCWvzEpceBCTovoKyiVpsZTUtzmCMPaUN2E2iX7CeCBmxEcQaBZMiQ3l3gvmDXmMzphMguna5+5Pxzgcx4IjJCSEurq6Y5pQ7VLSqQmMLjcqubfYtHN1U1WAycSUmDDiLUHERwQRbwlmiqZtgDKPOL/3YEgpqaurIyQkZPCDvSC/qtkx2Z4+O5G//s8inr1+CZEhgW5XZJ5o67Lx9KeHWTk30RGW6Iyr4OixS3aWNPYLF3Um0RJCXWuX4zMdiKJa5avR3yc+IpigAJPDQb5Di566/tTpKns67yjbjzSQGhNKUmT/z/KMOUm8f/sylqTF8svX87jl+e1uI8+e/+oIByqbWfPZYS756+fkVzbz142HuOxvn9PSYeO6U9O4/tTp3LhsBi/ceCL/7/zMYbHZjyTTtCJ9riG5mw/WEBpoJictxqHpOUfOvZlbTkl9Gz95OZfuHjvW9m425ddw/vxJLNAi7fRM7urmDqqaOvut9JMiQ3ju+qVcvXQqf99c1Cez3Rfyyq1kpUT101h0M6X+G88rtzIrMcIni0NQgIk/X7mQNd9dMqyNx3Q/h781juPWVJWamkpZWRk1NUOPxOjusTvKeLdWBVATNrRuaW1dPSrypTHYrb0awNnNqL9vd12g11mmISEhpKamDnqcta2b21/eSWunWgkHBgh+fWGWozdAfWsXNc2dfcIGnZ2VoFZkG/ZWUd3U0c+s4cwrW0tpbOvm+99Id7t/SkwYoYFmhwP1QGUTLZ02cjw4cEFpHFJCbUsXyS4rRSklD7x3gNjwIG5aNsMxYemCw2QSpEaHUqqFE2/XoqfuODuDLQdr+NvGQmpbuvqYxvq9vyWEZ69byprPDvP79w9w7qNbeOGmEx3O6I7uHp757DDLMxK49pRp3LV2N+c8sln7HJN54JL5jqY/45kESzChgWaKa10ER0ENJ82IJTjATEZib6VYPfN8c0EtyZEh5JVbefyjg6TGhNHVY+fChZPJSFar9LxyK+fNn+TWlKRjMgnuPT+Trw/XcefaXWy4fTlRYYF0dPfw8IcFmITg9pWzPE72Hd09HKxu4cy5if32zZscRXCAiW3FDZyblcyecitnzOl/3GCcP3/y4Af5SHp83zIy/sKvgkMIsQp4FDADT0spH3TZPw1YAyQA9cA1UsoybV8Pqq84QImU8kJt+3TgJSAW2AF8W0rpc8IRvs8AACAASURBVBxqYGAg06dPH9J96byxs5zb1+cSHmRm6fRYnrlu/pCus+bTw9z/9hF2/r+zvEqGsvXYueK+DXxr6TTuvWDukN7TE1uL6/kkv4YFU6IJCzTzZVE9r+4o4+er5gC9q8OB/BfOKzJXoaLT3WPnH1sOkzMtxmMkj8kkyEiKcGgcejb04mkDaxygssddBceLX5fy1OYiQIV3psaEEqAlv+mkxvbmcmwtrmfe5EgiggO4efkM7lq3e9D318d947IZnDQjjmv++RU/eTmXdd8/hUCziVd3lFHb0sUtK2ZwSno87/14GX/YkM/S6bFcvjjVrw7NkUQIoYXk9pqqSuraKK5r49pT0gCICgtkUlSI4zdVXNtKSX0bv75wHrvLrDzxySGmxYWTFhdGtrbyz0iyOARGXlkTQkCmB79MaJCZP1+5kEv/+jn/7809/OD0dG57cScFWoHBjfnVPHb1CW5/y/uPNtFjl26FUlCAiYVTotl+pJ6j1g7qWrt88m/4kwsWTKa+rctvfTh0/GaqEkKYgb8A5wKZwNVCiEyXwx4CnpNSzgfuBx5w2tcupVyoPS502v574M9SyllAA3CDv+5hMPKrmgk0C06ZGc8RLzKOPVHX2onZJDw2pHclwGxi7qTIfhEm7iitb+O1HWVem+T0+3jm2iW8ePNJLJ4a0yc+Xi8HPVCikvOKzBPv7D5KeWM7t6xwr23oZCRZHCG5W4sbSI4McTix3aGXHXHN5SiubeU3b+/jtJnx/HH1fPaWW3ltRzkzEsIddZlAOcjLGtrpstnZVdYbPXXRwhSSNe1pIFOZM1kpUfzfJdnsKrPyxMeH6LFL/rG5iAWpUZysFbJLjAzhocsXcEXO0J24Y5VpcWF9/i82adnRzmU11PerBMdmp/2/ujCTSVGhHK5t5UKn3BU9W19KSV65lenx4QOaeuanRvPjM2exflcF5z/2KfWt3Tx3/VKeuW4JtS2dXPD4p6zd1r9s/GAO75y0GK1CcP2Ax400s5Is/Pbi7GNKD/AGf159KXBISlmkaQQvARe5HJMJfKQ9/8TN/j4I9es5A1inbXoWuHjYRuwj+ZXNpCdEkJ4QQWl925BrFtW1dBEbHuRT3kN2ShR7K6z9Kmy6cvdru7njlV19is8NxJG6ViwhKuwXYHlGPHsrmqjRJuL8qhaiwwIHDB3UV2R6cyVXpJQ8uamQWYkRg6r4s5Mt1LZ0UtfSyfbiehanxQw4wTprHDq2Hjs/eSWXQLPgj5fP5/KcKbxz2zJOnRnHedl9zQWpMaHUt3axtbiejm67wywWFGDip2dnsGhqtMPE4g3fzJ7EpSek8MQnh/jD+wcormvjeyvSJ5yQcEdaXDgl9W3Y7RIpJWu3lTIjIdzhwAW1ADlU04Ktx87mghqmxIaSFhdGpJYBnxYXxmWLe02sWSlRWNu7KWtoZ2+F1a1G4Mr3v5HOmXMSWTlX+aCWZyRw+uxE3vtxrxPdOeektdPGP7YcJi0ujBQPxTJzpsVis0ue//IIJgGZPjjGJwL+FBwpgLMoL9O2ObMLuEx7fglgEULoNYVDhBDbhBBfCiF04RAHNEop9ZZk7q4JgBDiZu38bUP1Y2w/0uAo7OYOPbooLS6M7h7Zp5dDd4+dD/ZW8vbuin6PD/dV9XHe1rZ0EedjvZ6slChau3ocDVsqGtsdVU918sqsfHaojvAgM/e8scerXhNH6tqYFhfmmNj01eGnh2q0e25idpJl0IlPX5G56x63saCGA5XN3Lx8xqDCUvc/fJJfQ4W1o08BPnfERwQhXLLH/7axkJ0ljfz2kmxHtdK0+HD+c+NJ/HjlrD7n65FVb+wsd9yHzuU5U3jtB6f6JOAB7rtoHsmRIfx9cxHT48OHJfRyPDA1Lowum53Kpg6+KKxjd5mVG0+b0ee3k5Fkoctm52B1C58XqgKJ+v6l02PZeNfpTHPqKaELio0FNRy1dnglOALMJv557RKe/PZi4p2S4hIswfztmsUkWoL5ySu5jhD3376zj9KGNv6weoHH3/miqTEIobTgWYkWR8Lo8YI/BYe7T9x1eXwnsEIIsRNYAZQD+kwzVUqZA3wLeEQIke7lNdVGKZ+SUuZIKXMSEoZWOOyJjw96DOdr7uimvLGd2ckWR2lk5wiS9bkV3Pzv7dz6ws5+j5ue28aGvVWOY2tbOn1O/tH/YfaUW1m3vYyVD2/ioic+o9xJODy5WWX5vvy9k+mxS+5cu2tQDeVIXWuff9SsyVHEhgexuaAWKSUFVS39ksfckTMtlh67JNcle/jN3HJue2EnKdGhHovAOaO/1wtaEx3n/Al3BJhNxIUHU6OF5Frbu3n8k0OcN3+SoxzGQOhmsPf3VDIl1n30lK9EhgTyx8vnE2Q2cevpM/t0hJvI6E2EiutaeXJzEfERwVy6qO933vv9ltDW1ePIzPbE7GQLASbBi1+VAMduIooKDeRPly+gqKaVB9/bz3/3VfHi16V8b3l6n8TSfueFBTo0z7FiphpJ/OkcLwOmOL1OBfrYS6SUFcClAEKICOAyKaXVaR9SyiIhxEbgBOBVIFoIEaBpHf2uOZykJ0TwRVGd2/IZuoNtdpKlzz/IqVqM/NeH64kKDWTtLSf3kXZ2CRc88Sm7yhodFTbrWjtJi/Otx/CsxAiCA0z89p191LZ0sSQthn0VTfz0lVxeuPEkShvaeC/vKDcvTycrJYp7z8/k7tfy+NumQscEagkJ6FOXyNZjp6yhvU/lT5NJsGxWPFsO1lDW0E5Lp80rwaEXytuUX8OUGJUM9vjHh3h1RxmLpkbz6FUn9PEteCIhIpiYsEB2lDQSFmT2qghcoiXYEe22YU8lXTY7Ny2bMeh5gCPzubnTxsrMpEGO9p5T0uPZee9ZbqvVTlSmab/p9/dUsrmghrvOmd0vimlmYgQmAWu3lxJgEpycPnATo5BAMxlJFvYdVb62gRIWveWUmfHccNp0/vnpYV7bWc7cSZH85KxZg563OC1Gq5J8fJmpwL+CYyswS4uCKgeuQmkPDoQQ8UC9lNIO/AIVYYUQIgZok1J2asecCvxBSimFEJ8Aq1E+k+8Cb/rrBmYkRNDRbafC2u6YUHScQzmTI0MICjBR4lTUbeuRehZPi3EbsTE32eIo1wzKxxHnY12ZALOJ7JQodpY2ctc5s7llRTqvbi/jZ6/uZs1nhzlc20qAycT1p6YBcOWSKfx3fxV/3JDPHzfkAxBoVtU89VV1RWMHNrtkWmzfiIzlsxJ4M7eC1zXzzWwvMsKjwgKZk2zh75uL+LsWyWQScNuZs7jtjJleO++EEMxOtvBlUT2LpsZ4dV6SUxLg+l0VTIsLY0Gqd6vC+IggQgJNyr/hpRPcW44noQEwKSqUQLPg+S+PEB5k5poT+zdACgk0kxYfTlFNK0unx2Jxk1zoSnZKFPuONjEjPtyr473hrnNms7mghiN1bTxy5UKv8mZOnB7LC1+VsGBK9LCMYTzht1+ylNImhLgV2IAKx10jpdwrhLgf2CalXA98A3hACCGBzcAPtdPnAn8XQthR5rQHpZT7tH0/B14SQvwW2An801/3oCfRFNa09hMcBVXNhAeZSYkOxWQSTI0Nc2TJ1rd2UVTTyurF7vMmslKiWL+rAikl7d09tHX1EBfhe0+CP1+5kE6b3ZFjcXlOKh/ur+IP7+eDgEsXpTjyKIQQPHb1CXy4r0rZna0d/OnDAnaWNLAqS2kY+vinuWg/evtLvfFOhpeln5/41gnsLOk1Vc2dFDkktX52khIcg4XB6iRaQthb0UR1cwefF9byg2/M9NoZLYQgNSaMQ9UtA+aLGAyO2SSYEhtGUU0r3zpxqsf8lNlJFopqWr3uRZGVGsXL20qH1UQUEmjmxZtPorqp0yuNGlQeRkJEMCdMHd4FxnjAr0sgKeW7wLsu2+51er6O3ggp52M+B7I9XLMIFbHld3pbarb0+1HnVzaTkWxxmLDSnMpI6/kGniae7JQo/vNVCUfq2hz27vghVLKcEtt3ghdC8MCl2ax6ZDN1rV3ctLyveSYsKMDhV+jo7uGRjw6SV251CI4jDsHRV+NItIQwd1Ik+482MTkqxG3JCXfMTLQMS82m2VojJ281gKTIYGpbOnlr11HsbirKDsaUmFCqmzqYlejfJKrjgbS4cErr27j+NM85U7OTLby3p9J7waGZp3wpKugN8RHBPv0fmk3CrxWRxzLHl+7sI3HhQUSGBPSrtCmlJL+qmbOdbOBTY8P57JAqYbKtuJ4gs4n5Hswj+kopr9zqcMbGD0HjcEd8RDD/um6p22Y9zoQEmh1F43SO1LUREmhyNERyZkVGAvuPNnm9GhtOzl8wCWt7tyP3YTASIkOwS6UhzUm2+Nz3+kdnzqK6qdPn6CmD/vzw9JlcckKK297bOlcumYIlJNBrf8WC1Gh++c05rF48ZfCDDfyCITgGQAhBemIERTV9C7XVtnRR39rVZxJNiw+jvbuHmuZOth1pICsl0mM5g4wkC0FmE3vKrYRqx8SFD1/t/KyUKK/U+OyUKD4+UO2oNlpc18a02HC3E+byjHie3FTotZlqOIkMCfRYlsQduuA7UtfGXef43uPAtQOewdBZPC1mUBPjpKhQbhhAI3FF9UH3/vdgMPwct0UOvWVGfEQ/jcPhGHdayermnYKqFvLKrOQMEDYaFGBidrKFPRVW6lpV9M9QfBzHSnZqFHWtXY7qpCX1rY7QYldypsVy0cLJnJ89/PV1hhvnEFpvQnANDAx8wxAcg5CeGE5VUyfNHd2ObXqpbWeNQ68G+vbuCrp67IMmqmWlRLGnvInaFlVmayg+jmPF2WRmt0uO1LV5DAsOCjDx6FUnkO1ldNJoomscJ0yN7ucHMjAwOHYMwTEIegG8w7W95qqdJQ0kRQb3CaFNiQnFbBK8s/soMHghvGytdMLOkkYiggOG3ATqWMicFInZJNhTbqWquYNOm52pcf4tjjYSJFqCSU8I57snp432UAwMJiSGj2MQZibqIbktzE+Npscu+fRQLWfO6ZscFmg2kRIdSkl9GzPiwwfNy9Azvz8vrD3mlpFDpddBbnVEhPmaiDgWCTCb+Oin3xjtYRgYTFgMjWMQpsaGYzYJh4M8r9xKY1s3yzP6h+Hp+Q/ehI3qvQXaunp8rlM1nGSlRJFX3kSxplGlTQCNw8DAwL8YgmMQggJMTI0NczjINxfUIAQsc1NTR590vUkcCw4wO8JER8O/oZM1OZLalk6+Lq4nwCT6tck0MDAwcMUQHF4wIz6cwmq1It9UUMP8FFX4zxW9XLS3iWq6ucrXciPDie7s/nBvFVNiw/xex9/AwGD8Y8wSXpCeGMHhulYaWrvILW3s04jGmSuWTOGZ65Z43bZRj2oaruS/oZA5KQqTUEX9po5UBFJ7IzT2b55jYGAwPjAEhxfMiA+ny2Zn7fZSeuzSo+CICA7g9Nne9x52aByj6OMIDTI7al2NmGN8/Y/g2fNH5r0MDAyGHUNweIFes+rZz49gCQ5g4TBVw8xKieL2lbMctaJGC13zca1R5RfaG6HgfWgohnbPrWUNDAzGLobg8AK98Xt5YzunzIwjcJj8AGaT4PaVGSSPskM62yE4RkDj2P8W9KikR6r2DXysgYHBmMQQHF4QGx5EtFYSekWG96ao8cLKuUmckh43MjWa8tZCmFassNoQHAYG4xFDcHiBEMJRadZd/sZ4Z0psGC/cdBIx/va1NFdC8RbIuQFCogzBYWAwTjEyx73kRK3/sGtDJwMf2Ps6SDtkr4biTw1TlYHBOMUQHF7ys1VzkFKO9jDGN3nrIDkbEmZDUibsXgtSgpfd+QwMDMYGhqnKB7xtP2rghvoiKN8G2Zer14lzodMK1rLRHZeBgYHP+FXjEEKsAh5F9Rx/Wkr5oMv+acAaIAGoB66RUpYJIRYCfwMigR7gd1LKl7Vz/gWsAKzaZa6VUub65Qby1kFnM+Rc55fLjzittXB4E8y7tO8qv6sVvn4KbJ3+e+8K7SvKukz9TZyn/lbvh2ijk5uBwXjCb4JDCGEG/gKcBZQBW4UQ66WUzobth4DnpJTPCiHOAB4Avg20Ad+RUh4UQkwGtgshNkgpG7Xz7tL6lfuXva9DzYGJIzh2vQgf3ANxM2HSgt7tO5+H/97n//effR5EparniXPV3+q9kHG2/9/bwMBg2PCnxrEUOCSlLAIQQrwEXAQ4C45M4Cfa80+ANwCklAX6AVLKCiFENUoraWQkSZoH+e9CdzsEeu6ZPG7QE+7y1vUVHHlrISkLbvl05MYSGg2RqYaD3MBgHOJPH0cK4FyQqEzb5swuQLNdcAlgEULEOR8ghFgKBAGFTpt/J4TYLYT4sxDCbYVAIcTNQohtQohtNTU1Q7uDxEwVBVSTP7TzxxodTervntfAblfP6w9D2VYV6SSE/x/OJM41QnINDMYh/hQc7jzJrmFJdwIrhBA7UX6LcsDmuIAQk4B/A9dJKbWZjl8Ac4AlQCzwc3dvLqV8SkqZI6XMSUhwX1tqUJJ0O/wEmdw6NcHRVAalX6rne15Vf3Xfw0iSlKmEck/34McaGBiMGfwpOMoAZ69nKlDhfICUskJKeamU8gTgf7VtVgAhRCTwDnCPlPJLp3OOSkUn8AzKJOYfYmeAORiq9vrtLUaUjiaImQ4BocpcBUpwTDkJoqeO/HgS54G9G+oKBz/WwMBgzOBPwbEVmCWEmC6ECAKuAtY7HyCEiBdC6GP4BSrCCu3411GO87Uu50zS/grgYmCP3+7AZFY5BxNJ44icDHO+qRz/R3epe8tePTrjScpUf6sniGA2MDhO8JvgkFLagFuBDcB+4BUp5V4hxP1CiAu1w74B5AshCoAk4Hfa9iuA5cC1Qohc7bFQ2/cfIUQekAfEA7/11z0Aylw1URy4HU0QHAlZq6G9Ht76MQgzzLtkdMYTn6Hef6J8vgYGxwl+zeOQUr4LvOuy7V6n5+uAfmG1Usrngec9XPOMYR7mwCRmqjDWtnoIG7wl7Jim0wohmTBzJYREQ8VO9Tx8lOpvBQSr0OCJotEZGBwnGJnjg+Ewp0yAyU3XOAKCIFNT+rJGyUylk5Q5dB9SZwu881NorhreMfmbHhusvw3+c7l6vHAllH498Dm7XoJdLx/7ezdVwNs/USHmBgZDxBAcg5GoCY7xbk6RUmXBh0Sq1yf9AOZeoB6jSXwGWEuHFlm19zXY+rSquDueKN4MO56FxhJorYHCj1US5kB89ii8f/exR6B9/RRsW6O0TQODIWIIjsGwTFJmnfHuwO1uA9kDwRb1OnEuXPk8BHvXH91vRKWqXJnmo76fq0eGdVgHPm6skfeq0vxu3gQ3b4QpJw6u0VrLlF+qaOPQ31dK9d769QwMhoghOAZDCOUgr94/2iM5NvTkv+DI0R2HK5FaTqivE1lzJRzerJ7r+Snjge4O2L9eaXqBWufHxEz1+9KTMl3psPbeY95a98d4Q+nXYC1Rzw3BYXAMGILDG/R/7PFcVl2feEKiRnccrkRpqT6+TmR7X8eRT9oxjgTHoQ/Vd+EcAp04F7paeid1V6zl6m94Aux/G7rahvbee9ZBQAgEWQzBYXBMGILDG5Iy1T+7tXTwY8cqY1XjiNI1Dh8/27y1qrdHaOz40jjy1ioBkLa8d1uSU6Vgd+iT/Enfh+5WKHjf9/ftsSlhm3EOxKYZgsPgmDAEhzdMBAd5p+YHCBljgiMoXE3+vkxk9UVQvl1FhIVEjh+No6MJCjaosvZmp0j4hDnqr6foMl2ozr8SIpJ7fTu+cHiTcsRnX660PENwGBwDhuDwBucS4OOVsapxgHKQ+zKR5TnV1wqOHD8ax4F3wNbRP1M/JBKipnp2kFvLwBQIlsnqng9+0Fvp2Fvy1qnPauZZvn/eBgYuGILDG0Ki1CptPDvIHT6OsSg4pvTa8QdDSmXumXqyagAVEjV+NI4961RNsNQl/fclZXrWaK1lqlSMyaSEjr0b9r/l/ft2t6vj516oHPJRqUoDHS+fm8GYw+g57i2JA/xjjwW+fBLqC+Gbf3S/f6xrHMVe9gKpOQC1+fDNh9Tr4EhoPDK84/n8cfj0kd7XkxfC/6zzvTd6cyWsWaXyZwDa6uC0n7i/TmImHPov2LpUgqYz1rLeBliTT1DFN9+9C/77a+/GYbdBVzNkaxWQ9Ws1lQ/fQuL9X0JEIpx2+/Bcz2BMYwgOb4meCqVfjfYoPHPovyqpy5Pg6GwCBASNct6GO6JStBWwdfCorwZNSExepP76w8dR+LEqcDnnfOVfOPgBVO7u2/zKGypyoeEwZF4MYXFgDoITv+f+2KR5aoKvO9jrLNexlsG0k9VzIeC8h33TOEA55KevUM+dI9l0M+yxICXsekF9J4bgOC4wBIe3hCdAR6P7FeFYoLMZ2mo9dyvUy42YxqB1Ul8BW8sHFxz66l1fKQdH9jr+h4vWGrWyP/9hVaPsoQxlHvNVcOhO7XN/D5bkgY/VJ/CqfX0Fh70Hmit6PyOA9NPVY6hEDjGSzRPNlcrn0tM1PNczGPOMwVlkjBKhNYNqqx3dcXhC92F48hV0No1N/wb4lsuhCwnd5BZsUcJkOHNsWmp6Cz+GxapCkM5dE71Fd2qHJw5+bNwsMAX0D8BoqVKaiLPgOFYsyaoq8XA5yPUxG4LjuMEQHN4SrgmO1iG2ofU3urnG0ypS1zjGIg6Nw4sVcIeLkz8kUpUs6WodnrHY7Wpx4DzZZ69W/oCSL3y7VlN5r1N7MAKCVN0uVz+aPrlHTel/zlAxmZXWMVyCQx+zITiOGwb9RQshbhVCxIzEYMY0Y11wONrCjkONIyJJrba90jia1LEBWrkOXRgOV0huR6Na4Yc7tRuefS4EhqmoKF+wlvk24Sdm9g/J1YXpcGoc+vWGTePQxmwzBMfxgjcaRzKwVQjxihBildZ57/hDn0haxqDgsNt7bf+eJoPOMaxxmMxqZe7NRKZrTvrPUBeGw+Ug1xcGzoIjKBxmfxP2vuFbdVrnaChvSJyrBIXzveifie6XGC78ITgMjeO4YVDBIaW8B5gF/BO4FjgohPg/IUS6n8c2thjLGkdXM466TQOZqsaqxgEQmepZW3LGVXMKjurdPhzo329EQt/t2Zer6rSFn3h3nR6b6n3hi+BwV3rEWqbucbi/u6hUNT57z7Fdx94DNfnquSE4jhu88nFIKSVQqT1sQAywTgjxBz+ObWwRbFHmkbEoONytUF0ZyxoHaCtgL30czvcx3BpHS7X6G+4iONLPUOX1va1O21Kpytj7pHG46cHuq9biLVGpKpFQv9+hUl+ksuHNwYMLDnuPYc6aIHjj47hNCLEd+APwGZAtpfw+sBi4bJBzVwkh8oUQh4QQd7vZP00I8ZEQYrcQYqMQItVp33eFEAe1x3edti8WQuRp13xsxExnQqjJZCwKDn21bQp0Lzik1CZcy8iOyxe8XQF3NvUN2XX4OIYpJLdVi5pzFRwBQTDvYlU2xBtHvB7d5ouPI3qquh/nJkvWUv8JDjh2c5VeXys5e3DB8fFv4O/Lju39DMYE3mgc8cClUspzpJRrpZTdAFJKO3C+p5OEEGbgL8C5QCZwtRAi0+Wwh4DnpJTzgfuBB7RzY4FfAScCS4FfOTno/wbcjDKfzQJWeXOjw0J4/NgUHPpqO2G2mghcQ1NtHWp1OZZNVVGpyindMkgbWH9rHK01gFAJe65krfa+Oq3Dqe2Db0IIFfq7/+1eX4q13M+C4xhzOar3gTApwTGQNtFjgx3/7k3gNBjXeCM43gXq9RdCCIsQ4kQAKeVAxZuWAoeklEVSyi7gJeAil2MygY+055847T8H+FBKWS+lbAA+BFYJISYBkVLKLzTz2XPAxV7cw/AQnnjsqr0/0DWOxEwlJNrq+u4fy+VGdLzN5ejn4xjmqKrWaiU0TOb++6adogoNelOddqhObd2XUrRRaTbt9b4JH28ZLo2jep8qgRISObDGcXijCnO2tY/vvjYGgHeC429Ai9PrVm3bYKQAzsuZMm2bM7voNXddAliEEHEDnJuiPR/omgAIIW4WQmwTQmyrqRkmLSE8odeUMZbQBYOefew6GYzVJk7OeLsCdtU4gsJVMtuwaRy1quaSO0xmyLoUDn44eHVaa5n6vH3V8maeqc7LWzs0c5e3hESpz9GbgISBqNqnFiy6j8OTUNArGoNa3BiMa7wRHEJb3QMOE5U3pUrc+R5cf1V3AiuEEDuBFUA5yvnu6VxvrqmP8ykpZY6UMichIcHdIb6jm6rG2opJt+/rzlVXwTEuNA4vVsB2e3+NQwgte3wYTVV61rg7si5TZr996we+jq85HDoBwZB5kfKl1Baobf4wVenXPRaNo6tNOccTM1UdLqQyN7qiV+c1B/W+NhjXeCM4ijQHeaD2+DFQ5MV5ZYDzf04qUOF8gJSyQkp5qZTyBOB/tW3WAc4t0557vKZfiUhUk0ZH44i9pVfogiHJg+AYq02cnAmJVIJtoImsqwWQ/QXgcBY6bKnu7xh3ZvIJEJs+eDLgsURDZa1W97r1afXar4LjGHwcNQcAqX535kC1zZ256uAHKmR8juYS7R5i61uDMYM3guMW4BSUNlCGcljf7MV5W4FZQojpQogg4CqgzzJNCBEvhNDH8AtgjfZ8A3C2ECJGc4qfDWyQUh4FmoUQJ2nRVN8B3vRiLMODI5djjJmr9GzqqCkqZNh1MhgPGgdoE9kAphNPPUWGs5lTa+3AtaWEUCVIDm+BpqOej2sqG3rSXtppqtNf0SfK8WyZNLTrDMaxahx64l/iPKUpAdg6+x+Xt1Z9phnnqNeGxjHu8SYBsFpKeZWUMlFKmSSl/JaUclAPsZTSBtyKEgL7gVeklHuFEPcLIS7UDvsGkC+EKACSgN9p59YDv0EJn63A/do2gO8DTwOHgELgPe9v9xgZq0mAnc292dTuJoOx3MTJmcFWwHp2FBAleQAAIABJREFUvKsADB4mjaO7Xa2MBzJVgdIIkKqHtzs6W5QPZKiagu5LASU09NX8cBOZogIpuoaoAVTvVwuV2OlOGodLZn2HFQo+UPejl/Q3NI5xz6C+CiFECHADMA8I0bdLKa8f7Fwp5buoqCznbfc6PV8HuNX5pZRr6NVAnLdvA7IGe2+/4Cg7MsYiq5yzwt0KDg8T7lgjKhXKtnne71rgUCckUuWAHCuOrPFBqtkmZEDyfLWSPvkH/fc3DYNTO3s1fPnX4S814ow+vp3P98+UNwWo0GB3Jfp1qvaqEHCTudd/4Wqq2v829HSqaLEOzWRqaBzjHm+c3P8GDqBCZO8H/gelQRx/jFmNwynSKCoVDv63736HqWoMJwCCqlfVXu+5p4iuOQW7RIcFR0LngWN/f3d1qjwx7xL46NdqEeEqaIajMOHkRcoE5NrUaThJnKP+vneX+/1n3gvLfup+n60LKnbA3AvUa7NmqnIVHEUbldktZTGUfKm2GRrHuMcbwTFTSnm5EOIiKeWzQogXUOan44+wOECMPcHR4ZRNHTVFJdE5N5zqbIIgi/vchLGEnnTX3uihGZXei8NFAA6Xc7zFB8GRonUgrNrrRnDopdCPQXAIATd80LuS9weTFsDtee4z4d+8FXav9Sw4Cj9W38dczersyTne3aa+VyF6v1ND4xj3eOMc142WjUKILCAKSPPbiMYy5gDV2GesCQ5njSMyBZCqa5zOWC9wqBMSrf56ypEYzDl+rGHSvmgciXpBQjd96K3lw+PUDo7wf7fJ6Kkq/8f1seAqqNnfW1LElby1EBoDM7ROhJ6c47bO3n2BYeqvITjGPd4Ijqe0yKZ7UFFR+4Df+3VUY5mxWK/K1ccBff0cndax798ANRGBZ8HhKTosJFLlDxzrhOSL4IhIUMe5Nl4C9dlbJqmFxngl82KVWOmuqGNXK+S/q47RBZvDx+HiHLd1OAkOXeMwTFXjnQEFhxYq2ySlbJBSbpZSztCiq/4+QuMbe4QnjL2eHM6CwV3pjvGicQwmODqb1GQWFN53+3CVHWmtUZE/QWHeHZ8414PG4afChCNJRILqa573an9NLv89Nflnr+7d5slU1dNlaBwTkAEFh5YlfusIjWV8MNY0DilV1JRD49CicJzDWsd6SXUdbzSOYEtvEycd3b9zrH6OwbLGXUmcp5LgXHuR+6sU+kiTtRqsJVD6dd/teWtVza6pp/RuczjHXU1VHb37DI1jwuCNqepDIcSdQogpQohY/eH3kY1VIhLHluDoalE9t3XBEBgKYfETV+Nwdx/DpXEMljXuSlKmmgQbDvdus9tVOO5EEBxzzlN5Gs5Z8m31cOi/Ki/DuZe6R1OVk8aht/s1NI5xjzeC43rgh8BmYLv2GCDYfoITHq8mqO4xUqjNXaitay5H5xjvxaETbFGmqAE1DjeFGh2l1Y+xJ8dgWeOuuHOQt9Yo84w/ChOONCGRKtt77+uqLDrAvjeVPyn78r7H6r6Ofs5xJx+HyQQBoYbGMQEY1HsnpZw+EgMZN+gr0rZaNUHXH4bt/1KrfoCIJDj5h/3NKf7CXaRRVCrUFfa+dq0oO1YRQmkdnmqB+aJx5L4IM1ao3BBPFG1UPo3UHPW6tab3uTfoeRDV+3vzGZqGIRR3LJF9uRIW63+kFk0F70PcLBXK64ynBMCerl5TFSiNeCgah60LPnukN5k1IBhO/D6Eu+mbMlK0N8IXT7gvs2IOUvNA2MQ0zniTOf4dd9ullM8N/3DGAfqKtKVaTQ4bH4TdLys1XNqVjTdlMUw7eWTG49A4nFbiCbPVP3hrnQrp7OkcH6YqUIJjII3D3YSsa1P6Z9FcBW/cAqf+GM663/N7vf9LVbTyh1+r765tgJLq7ggKh5i0viGruj8gPsP764xlZp4FCXN6y6sIoT5T14WRp5IjzhoHKAf5UATHvjfhk9/1mrtsHeq3sPha3681XGz/F2z+o9KiXLG1q7Gu8JBcOc7xJl5widPzEOBMYAeqidLxh3Ohw+52OPA2LPo2XPi4qlH0x5nKJjxSgsOdxpF5MWz5E+x/E+ZoK2F3Jp6xyECCo9MKIa5NJOm9d/2z0Ht2Vw9S4KCjUfkjKvOUZiLtvvk4QJmrnE1VeetUOZK4dN+uM1YJDIEffjX4cR6d410ugmOIpqo96yAyVSUs2rvht4mjH924Zx2k5MBNH/Xft+ZcFUSw/M6Rsz6MIN4UOfyR0+Mm4ATAz1lJYxg96qa1Rq3qu1q0oneo1f3sczWbcLfnawwnjmxqJ8GRnK1WvHnrxk+BQ53BNA53JrcgCyB6NQ49t8JdjoXr9UD9gztyOHyIqgLlIK8rVD6v+iIo39Y3TPV4wZs8DhiaqcrVIR8QrCLpRjNIpSZfLTg8fdfZl0FtPlTtGdlxjRDeOMddaUP1+j4+0U0ZrdVqYo5IVmWwdbIvVxVHizaNzHjcCQYh1DiOfK71TGB8+DjAs+BwDTt2xmTSmjlp9m9dA2gqU3Zod9h7VCVcgD2v9fY698U5DqqJkexRTZf2aF3usi4b+JyJiDvnuN2utAPzMZqq3DnkwxPU/+BokbdOVQeYd4n7/ZmXqEKR3rQZHocMKjiEEG8JIdZrj7eBfEayB8ZYIyhc/fjrDqkGNVmX9q0B5dz6cyTwVPk26zJAKjssjCONI9r9ZN/dpiZoTwLQuSdH9b7eycqTuUr/3CYtVAJm/1vqta+mqiSnyKrda1Vuw0RxjPuCO+e4/vxYNY68dUqDTs7u3RaeOHp9caRU/99py8CS7P6Y8DhVjmXPq/3zfCYA3mgcDwF/0h4PAMullHf7dVRjnfAEtUrt6eqvqjpaf749MvHqHR6yqePSVYXVgx+q1+NJ4+hsctPXYRCTW8j/b+/so+yoyzz/eXI73el0yAukO8YETMAI6RARyTAg6rI4KCKCb4yw+IKHkdERx3FZd3GOuo6HnaNz9syqK8MuouKwIIYgmp1hZBUB33hJkEDSCWgImDQJ3Z1AQl5IJ508+8dTv77Vt+u+V/Xtvv18zunT99atW13VdW9963mfaW67o0eg/6n80KCkym7Ii8xpl1lwc93t9rxa4Tj2RLtoPrnSXBOT0U0Fya6qMFu8HuHY8zz88TdmbcRjBR1zGzfeYPvvrHanMCW5kOWXWiFu76Ol15uAVCIcW4FHVPVBVf0NsEtEFmW6V+Odjk67A56z2C7OhYTRn7//afb7Emo0kgJwy6OBQzCBLI6oCLCwJmO4pXoZi+Ol5yyjZcn5tqyocEQWxzHzLS51+IAJcPj7lZKbCnNPhmfuM9dE93uqe3+zMCVn/794cLyoxVFFcLznR4COdv81soPD+lUmlCEFuxinXGiZVWPlfRhDKhGOO4G4rXUkWjZ5CXelhXdBgTD6cyz8m6Wqwpe9D4j2byJZHDA6zlFu/G1orR5SY7uWWS+pYgHyuAUT7hw75o6shq6UMOv9pPMaW1fQaHKtI11VweKop45j/Sq7OSvMUpvRZbNbQmHiWHH0iHkblrzd3KqlaDsmSpb58dgly4wRlaTjtqjq8KdBVQ9FM8QnL2FaWjG3RBj9ueZmu3OeFkuFPfgy3P2X1fdVOvuvrAVEIYNFqqkBZs43EXvuVxNfOAYjC6SYSLbNtLhT/yZArDivq9sy3FRHC3x8KNRrzrFzVG1gPNAVCcepk9RNFWhptfTbQHjcMi2/bOr0yi2OnZthxzp4x9+Pfi1kvx3YWTzOkAXP/Rr2vVB5AsTyS+0z+OyDNlGxSajk9mogNiMcEbkEqCgqJSIXiMjTIrJZREbFRUTkBBG5X0QeF5EnReTCaPkVIrIu9nNURN4QvfZAtM3wWo3f9jo49QNwzt9YoV0xlrzd7r62Pz5y+daHrSX14YThOcXYsQ6euCP5tXJ9qM79PJx9zcRp8T0sHAUB8kotjv4eK8pr7bDA9cHdsHfH6PXjFkdLmxW1/clVte1z98Vw2n+ApRfV9v5moZjFEZ8pUo3Fsf5OQCLLuYBGTeNcf6d1G3jdBZWtHzIu+1OYUDmOqORq8gngNhH5VvS8F0isJo8jIjngBuD86D1rRGS1qsZ9B18AVqrqjSLSjc0nX6SqtwG3RdtZDvxEVdfF3ndFNHu8MZz47+ynFPOiseh9G+HEc/PLQ3Hah++u3J9+6/tGzxEPDO4pPZd60Tn2M1EoanGUCY6HGEffxnymU7AE+jaObj0yWFD/Uk8F8rEnwntvrP39zUKudaRLJsQ7Ci2OI4Pm8ik1kVLVCuwWvdks50LiHRzGiqFB2LTaLP9KW++3zSzdf22CUkkB4DOqehbQDSxT1Tep6uYKtn0msFlVt0SurjuASwo3D4QrwSxgO6O5HPhBBX9vfDGj07rU9veMXN630VpSVxOELWxaGGei9KGqlHpiHEcOmbsqCEbXUvtdeA7i25soSQMTgVzryOB4qOnIFVgcUN7q2LHOzmWxzKV4B4exYvPPzfVcLpsqjkiUYj7JhENE/l5EZqvqPlXdKyJzROT6Cra9AIgNhaA3Whbny8CHRKQXszY+nbCdDzJaOL4Xuam+KJJczy8iV4vIWhFZOzDQoOyLed2j6wj6N+WDqZUya6EVOyV15C3W+G+iEuJBiRaHmJsgiWFB0bxgTD/WsqaSajkGX4YpU0feDTv1McpVFSyOggJAKC8c61fZ+em+OPn1EGccyyLA9aug/diRHoRKKNUNYYJSSYzjnao67HBW1ZeACyt4X9IFvXAo9OXALaq6MNrmrdHUQduAyJ8CB1Q1Xrd/haouB94S/Xw46Y+r6k2qukJVV3R2VpmbnxZdy8y3GQqAjhy2XP+uGoQDrK9SHNXmszim5Ew8kiyOtpnFs57i/4PgqgL7XyfNzQ6xoSbsI9QwRgXHk4SjgmFOR49GmUvnF7fM22aaUI1VjGNwn00+XPbefEPHSpmkwpETkeEzLyLtQFuJ9QO9QHwowUJGu6KuAlYCqOpDWBPFeLOgyyiwNlT1+ej3XuB2zCU2PpnXbUHw3c/Z813P2B1Z/MJWCcWEI1RTN5PFAclftHKWVXgt1wbHxlI3u5ZaX6HCtM2JMhVxIlFocQS3VWE6LpS2OLb+FvZuL525JBLVcoyRq+rpe6w+qJYCz0kqHP8HuE9ErhKRq4CfAd+v4H1rgCUisjhK370MWF2wzlas2y4ishQTjoHo+RTgUiw2QrSsRUTmRo+nAhcB47eLWDw4C3lfe60WR2Gco5zff6KS9EUrZ1mF1zpfNzKDbN4yu4C9uGX09ppNcBvNKFdVkXRcKG1xrL8TpnZYDUQpOjrHLji+/k7rznv8WdW/dzIKh6r+A3A9sBQLkP8UeE0F7xvC5pXfC2zCsqd6ROQrsfTea4GPi8gTmGVxpaoGd9ZbgV5VjX/j24B7ReRJYB3wPPDt8ofZIDpjg37Cb8lVP6shZE0VCsdwptEEaZleKdMSgomVWhyFohyeF1aQu8WRPpWm40Jxi2PokDU1POXC0W10Chmr6vH9u+CZX4wel1sp7XOKN9ucoFSa3P8CVj3+58CzwF2VvElV78GC3vFlX4o93ggk5oqq6gPAWQXL9gNnVLjPjadthtUUBEujb6NVwE6tMiDb0maTBfdsG7m8mS2O3VtHLju4xwLdxZgWVfEWCkfnydbFtH8TLIu1Azn4sp0bJz1yrSOnMBZLx4XiwvHsL+2moZJiyhldxVvKpMlT/xJ1562xwLN9jqV/HxmaOPVUZSh6FCLyOsy9dDmwC/ghIKr678do35qDrmUjXVXz31DbdmYuKGFxNKFwFI6PHdxbuuBy1kK46OvWYDLO1HbL+S+MDzVbNtp4oKWtoMlhqXTcIq6q7b8DBBa/pfzf65hrFkdSZ4A02f47uzF51etre3+8/1qTtKQpZXc9hcUf3q2qb1bV/4n1qXKqoWup5aMfeNEa8FUbGA8k1XKUa/w3UQk+4Xg76nKuJRFY8bHkGc9JQdRmy0YbD+SmjpzHMVSDxdEXq/wvR0enucbiVk4WhKLSWsWpWG3SBKaUcLwfc1HdLyLfFpG3kZxi65RiXjToZ1OUF1BtYDww63gTDo1lNDdrEVv7HBvjGgYthbTjWo9zRoEvXNUtjiwoVsdRjcXRv7Hym6vh6vEM4xyq5uas9XsLeTfqZBAOVb1bVT8InAI8AHwWmCciN4rI28do/yY+XdGXIHTKDcVp1TJroX3Z4h++ZrY4IH+sQwdtklzbMbVtr3Ba3KF9gDbf/63RJKXj5lpHBpRLBccPH7SU9Uq/I/Exzlmxe6vdwNT6vYWYq6p5AuSVZFXtV9XbVPUirBZjHTC5BzlVw3En2ZfnuV+bmT5ncW3bSUrJPVimmnqiUigc9SYBFLqqmtVSazRJFkeuoOQruKqGEoRj5+/NOq/07j4+xjkrQkZkrS5mmHSuqlGo6ouq+r9V9bysdqjpyE2N0m/V0nNrSeeDZOEIQ5xq3eZ4pfCLVm/acRi8Nbhv5Pbc4kiXpOB4PBUXTFxkSrLFETKkKnZVjUGH3OHaqxQsjskqHE6NhDuoantUxSlmcTTjxS8LiwPyFxi3OLIhKThe2AtMJJrJkSAcfT0mLMeeNPq1JKZHGUpZVo/3bbT4Yj21UsX6r01gXDjGgiAYXXWYu9Pnmtkfr+Vo1gDvKIujzBCncgy7NKILTHyIk5MewVUVEjhCjKOQlmnJwfH+jTaGt9Jah9xUazoYrx5/ZXfyVMBD+0eKWqX0b6zP2gA7nraE/mtZoDomf8eFYywItRvzT6t9G1OmwKyCWo6Xt1c/I3si0F6QhfJyNIip1mMdDqJGF5iDdQqRk0yuDVArloNkiwOKWxy1dI6OV48f2g/fPB1++Q8j11GF77wD/vXa6rY9dMjiLvVkVAXGqrX6Y7fAP3ZXP2G0Slw4xoITz4Wrfg6veVN924nXcuzeZoVJJzZhPWZLm11cQpuGTf/XZpjMLVEAWIpCV5XHOLIhdI0NAfKkGAdEUwALLI5XXrIizWov0jO68uf16X+zOeTrbh9ZA7RjHfSth22PVLftXZtNBOsJjAfGql/V47fa//blpNFG6eHCMRaIwPF/Un9166zj8xXQPT+y38srnH080Qj9fQ68aAN0au0TBB7jGCtC+/QgHEeKWRwJ42NrzV4K1eOQT3nfsw16H82vE5bveiZ5pk0xQrA+FYtjDITjxS3w/GP2OOMeXi4cE4mZC2x+9pHD1q1zwRk2trQZCV+0TauthqPWPkFgF7S2WflCscGXrdlkSA110iFYHEMxiyMpxpHkquqrMXspuKrCDcYZH4OW9mheOTaidsNdlrKuR8z1VCl9PVFT0iXV7VMSYyEc62MtBF04nGFmLbSK6mcfhBfWVzfCcqIRvmjrV8Fxr629x1cgfmd6MEpj9iFO6RJEYoSrqpjFUeCq6t9k4j6zcEhoGTq6LGa1fpXdYJxxJZx8AfTcbTdYf/yt3Wyd/ano71TRFLF/k4lGSyXjh8qQdYxD1cQyuHNdOJxhQkrur79uufDL3tvY/cmS9tmw6w9WOHnqB+q/yMd94c2ajdZocgWuqqHB5ItuoqtqowXGqz3PIfHhkf8Fxy2xBJTll8KBXbDlQdiwyiycsz9lwpY0DbIY/T3puKkguf9amvRtsOmiZ37crg0uHM4ws6KBis/9Cha9BY55VWP3J0va50Qffq3PTRUYZXF4Km7qFAbHjxQTjukjLQ5Vq5eo5SId4lcvPmOCIQKv/TOrnXjiduj5MZzyLns+9+TKLY7BvdZupJ7aqziF/dfSZv2dMKUFlr3PUvczHnDlwjGRmBUz49O4mI5nQurt/NPS8TF3uMWROYXB8aSWIzDa4nj5eavVqaVeItToQP470dIGSy+22MbB3fnZHvO68yMOyhGC9WlaHJCNuyrMaD/pPGvbPqMr85G6zTFVZLLQ2mEFT4f2wdJ3N3pvsiV80dKK43R0WgD1yJAJx8yF6WzXyRNiHEPlXFUFwfF6+kEFV9WrT7e+cIHll1pqavscu6CCicCTP7SLd/h87d5m+9JZMJUzzYwqiAnHbqi19Gr3Nht0Vcje7ZZJdt4X7Xncus4IF46JxvzX291zMxb+xTluiWXCnJpSunHHXEDN933wZehyiyN1Eus4ilkcMVfVwNP2O4xaroZj5psb6o0fHbl80ZutoejJ78zXkgyPEd6Ur6m66y9MSK55dOT7+zba3PPZZadkV0YaFsfqT8OW+5NfmzbLxu2CXR9eejR5vZTIVDhE5ALgG0AOuFlVv1rw+gnA94HZ0TrXqeo9IrIIm1MefaJ4WFU/Eb3nDOAWoB0bS/uZ2Jzy5ueKVY3eg7HhlHfB556pfsxuMYbbjgz4vPGsGA6OD+Z/F0vHPTpkWU+5qVbU2npMbTdDU9vh2qdHZ29NycFfPZwXM8jHK/p6TDheeg62PQwIHDoArbH07NBqJK0GovUKx94+y6Y8+xo48+qE7c/Ojx1IGlyWMpkJh4jkgBuA84FeYI2IrI7mjAe+AKxU1RtFpBsTgkXRa8+oalIO5o3A1cDD0foXAP+WzVGMQ+JfhGZGJD3RgFgRYH99Q6Gc4gyn4x62gHepdFwwqyM3y9wssxbWnjkXtjdqecHfnrnAkiKCG2pDqHtQGHgKFrwxeqomLqe8q7b9SaJe4ei524Lrb/wIzCljBXXMNXd2oRimSJbB8TOBzaq6RVUPAXcABQOhUSB8g2cBJevkRWQ+MFNVH4qsjH8G3pPubjtNSRCOl/5ohWBucaRPS6yO48hhQIu3HIF8nGNP78jEj6wQGRkgX3+XtbKBfJwFYF+ftS5Jo9VIoN4pgBtWwauWQ2cFbXfi1nVGZCkcC4BYK1d6o2Vxvgx8SER6Mevh07HXFovI4yLyoIiEyfULou2U2iYAInK1iKwVkbUDA9kGipwJQDxtE9ziyILh4Phg3l1VrMkh5OMce3rzNUpZ07XURKKvx+o0zvmM7WM8TTftwDiY9TN1em3C8eKz0Lsmnx1WjmHrOjt3VZbCkWR3FsYiLgduUdWFwIXArSIyBdgBnKCqpwP/EbhdRGZWuE1bqHqTqq5Q1RWdnZ01H4TTJEybZRe2XZFwuMWRPsPB8cOxeeNFguNgFsfhV+DAzjEUjm5L/f3NN62dyKnvt7v4eGFgsEjStDjArI5XahgfG1xqlSaKxN2yGZGlcPQCx8eeL2S0K+oqYCWAqj4ETAPmquqgqu6Klj8GPAO8Ltpm/BOWtE3HGY2IfaFcOLIjHhwPwlEsHRes4WDo4jrr+NHrZUEQgyd/aF2rZ3SamMRdVf0b7bMSUn3TotZ+VetXwQlnw+wK/0djMBkxS+FYAywRkcUi0gpcBqwuWGcr8DYAEVmKCceAiHRGwXVE5ERgCbBFVXcAe0XkLBER4CPATzI8BqeZ6JgLLz1rj91VlT7x4PhQ1IW2WDoumKsqDCYbS1cVMKIjQVc37HvB6nzArI803VSBWoSjrwcGNlVX8DuRhUNVh4BrgHux1NqVqtojIl8RkYuj1a4FPi4iTwA/AK6Mgt5vBZ6Mlq8CPqGq0Vnlk8DNwGbMEpk8GVVOfXR05msM3OJInxHB8ej/XNLieCU/X6ba5oa10j7HAuK5NjjlIlsWT9M9esTqStJ2U0FtjQ7X32kute4qcoCmTrPP977shCPTOg5VvQcLeseXfSn2eCNwTsL77gLuKlwevbYWODXdPXUmBR2x9hRucaRPPDgeLI6SMY4DkXAIzHz1mOwiAK//c0ttDZ+BMNK5f5Ptx9Ar48fi2LbGxidU6zbLuHrcK8edyUP8y+cWR/qMcFWVsjiiTKvDr5irasa8dFqXV8r5fzfy+TGvssB1f0++cWiWwqFaec3K/gHoqqGivqNzwgbHHWd8EXy/iLUzcdJlSs7cKkcGy8Q4Yum4Y5mKWwwRc031bYxScaW2i3U52ufY/yZp3nox9g/EPrdVkHH1uAuHM3kIhVFtM9NrJeGMJNc6MsZRLh13z/ONFw7IZ1b19cCcRdZQNG1C9fjBClNyjxy2QsSahWMCBscdZ9wRXFUe38iOXGsFWVXjzOIAC5Af2mvDn7IIjEP1bUcO7LLftQjHjC57/9Ej1b+3Alw4nMlD+AJ6fCM7Wlqj4HiJOo7cVBs6tKfXAtHjQThCTKPWuSCVUK1whGFMtVocejSfYpwyLhzO5CFkVbnFkR3B4iiVjgtmdez8gz0eF8IRE4ssAuNg6bhQuXAEV1NNwhFZ1xkFyF04nMlD+DK5xZEdudaRwfGkGAdYnGPn7+3xeBCOabPy1evjxVUVgtvxKYeV0pFto0MXDmfykJtqX163OLIjBMdLpeOCCceB6MI4Vu1GytHVbft/7Enl162FIBwhdlGOYC3U0vok40aHXsfhTC7O/TzMfV359ZzaqCQ4DvkAecs0mH7c2OxbOc76BCx+C+Qyuiy2zrCJgs/+Et782fLr7x8wi60WCzmIzb5sXFUuHM7k4k//stF70NyE4HipdFzIp+TWM8ApbU46Lz+fPAtErOfUr79uF/RyLqh9UQ1HLf+f9jmWgOCuKsdxxj3DrqqDVgxY7O49WBxj1aNqvHDqB2yQWM+Py6+7f6D2Dr2hG7QLh+M4455h4SgyNjYwbHGMk/jGWDGv23pjbVhVft39A7UFxgMuHI7jTAhGCEfC2NhA3FU12Vj+ftj2iI0xLkWt7UYCLhyO40wIhus4ylkckatqMgpHmOS3IbEBuKGajnBk1FrdhcNxnPSIV47n3OJIZM4iWHimTfYrxsE9ZrnVIxwzIotDE6dr14ULh+M46THCVVWiVfqwxTHJYhyB5ZdaG/cw37yQUH9Rr8Ux9Aoc2l/7NorgwuE4TnrEu+OWEo7WGYz5AKfxxLL3WNbZjWfDl2fZzwNfzb8eiv9m1CMcoXo8/VoOr+NwHCc94um4xWo4AM64EuafBq3Tx2zXxhUzuuAD34H+p+z5+jth88/h3OvseT19qgLo1Y4TAAALkklEQVSvfgOc+7eZzJ7JVDhE5ALgG0AOuFlVv1rw+gnA94HZ0TrXqeo9InI+8FWgFTgEfE5VfxG95wFgPhCmobxdVbMbdeU4TuUMV44fKh0cnznffiYzy94LoS3WKy/Butvg6FGbFZOGcHQtzazTb2bCISI54AbgfKAXWCMiq6M544EvACtV9UYR6cbmky8CdgLvVtXtInIqcC8QrxS6Ipo97jjOeGI4OH7Qe4JVw7xuOLQP9my14HnIhppeYwFgxmQZ4zgT2KyqW1T1EHAHcEnBOgqET9csYDuAqj6uqtuj5T3ANBEZw6HEjuPURKUFgM5IQiv3/k32e/8AtB+bXd+sOslSOBYA22LPexlpNQB8GfiQiPRi1sanE7bzfuBxVR2MLfueiKwTkS+KJDdyEZGrRWStiKwdGMhuhKLjODFybYDC4f2l03GdkQSXUl+P/d5fQS+rBpKlcCRd0AsTii8HblHVhcCFwK0iMrxPIrIM+BoQ70x3haouB94S/Xw46Y+r6k2qukJVV3R21uEndByncnJT7ffgXrc4qqHtGJh9AvRHnvz9O+uLb2RMlsLRC8STtBcSuaJiXAWsBFDVh4BpwFwAEVkI3A18RFWfCW9Q1eej33uB2zGXmOM444FgZQzuLd1yxBlN17J8XUc9DQ7HgCyFYw2wREQWi0grcBmwumCdrcDbAERkKSYcAyIyG/hX4POq+puwsoi0iEgQlqnARcCGDI/BcZxqCGIxdNAtjmrpWgq7/mAZafsG8nUY45DMhENVh4BrsIyoTVj2VI+IfEVELo5Wuxb4uIg8AfwAuFJVNXrfa4EvRrGMdSLSBbQB94rIk8A64Hng21kdg+M4VRKPa3iMozrmLYOjQ9C3AQb3jGtXVaYhe1W9Bwt6x5d9KfZ4I3BOwvuuB64vstkz0txHx3FSJF70V6py3BlNyKx69pf2u56q8YzxliOO46RHCI6Du6qq5bjXwpSpsOUBez6OLQ4XDsdx0sNdVbXT0gpzl8DWh+25C4fjOJOCuHvKLY7q6eq2jrbgwuE4ziRhhKvKLY6qmdedf+zC4TjOpCDnFkdddEVdD1vaobWjsftSAhcOx3HSw2Mc9RFaj8zohORuSuMCFw7HcdLDs6rqY/YJ0HrMuHZTgQ9ychwnTVq8jqMuRGDxW8d1DQe4cDiOkyYjLA4Xjpq47LZx7aYCd1U5jpMm8eB4qdGxTnHGuWiAC4fjOGkSD4i7xdG0uHA4jpMe7qqaFLhwOI6THnGx8HTcpsWFw3Gc9BjhqvJ03GbFhcNxnPSYkgPJ2WN3VTUtLhyO46RLsDpcOJoWFw7HcdIlCIen4zYtLhyO46RLSysgIzOsnKYiU+EQkQtE5GkR2Swi1yW8foKI3C8ij4vIkyJyYey1z0fve1pE3lHpNh3HaTC5VnNTTYBCNqc2MhMOEckBNwDvBLqBy0Wku2C1LwArVfV04DLgn6L3dkfPlwEXAP8kIrkKt+k4TiMJwuE0LVlaHGcCm1V1i6oeAu4ALilYR4GZ0eNZwPbo8SXAHao6qKrPApuj7VWyTcdxGkmu1eMbTU6WwrEA2BZ73hsti/Nl4EMi0gvcA3y6zHsr2SYAInK1iKwVkbUDAwO1HoPjONWSa/UajiYnS+FIcnBqwfPLgVtUdSFwIXCriEwp8d5KtmkLVW9S1RWquqKzc3y3KHacpqKl1cfGNjlZtlXvBY6PPV9I3hUVuAqLYaCqD4nINGBumfeW26bjOI3ELY6mJ0uLYw2wREQWi0grFuxeXbDOVuBtACKyFJgGDETrXSYibSKyGFgCPFrhNh3HaSS5Vu9T1eRkZnGo6pCIXAPcC+SA76pqj4h8BVirqquBa4Fvi8hnMZfTlaqqQI+IrAQ2AkPAp1T1CEDSNrM6BsdxauDsT8HQwUbvhZMhYtfp5mbFihW6du3aRu+G4zjOhEJEHlPVFYXLvXLccRzHqQoXDsdxHKcqXDgcx3GcqnDhcBzHcarChcNxHMepChcOx3EcpypcOBzHcZyqcOFwHMdxqmJSFACKyADwxxrfPhfYmeLuTBQm43FPxmOGyXncfsyV8RpVHdUldlIIRz2IyNqkyslmZzIe92Q8Zpicx+3HXB/uqnIcx3GqwoXDcRzHqQoXjvLc1OgdaBCT8bgn4zHD5DxuP+Y68BiH4ziOUxVucTiO4zhV4cLhOI7jVIULRwlE5AIReVpENovIdY3enywQkeNF5H4R2SQiPSLymWj5sSLyMxH5Q/R7TqP3NW1EJCcij4vIv0TPF4vII9Ex/zAaT9xUiMhsEVklIk9F5/zsZj/XIvLZ6LO9QUR+ICLTmvFci8h3RaRfRDbEliWeWzG+GV3bnhSRN1bzt1w4iiAiOeAG4J1AN3C5iHQ3dq8yYQi4VlWXAmcBn4qO8zrgPlVdAtwXPW82PgNsij3/GvA/omN+CbiqIXuVLd8AfqqqpwCnYcfftOdaRBYAfw2sUNVTsZHTl9Gc5/oW4IKCZcXO7TuBJdHP1cCN1fwhF47inAlsVtUtqnoIuAO4pMH7lDqqukNVfxc93otdSBZgx/r9aLXvA+9pzB5mg4gsBN4F3Bw9F+A8YFW0SjMe80zgrcB3AFT1kKrupsnPNdACtItICzAd2EETnmtV/SXwYsHiYuf2EuCf1XgYmC0i8yv9Wy4cxVkAbIs9742WNS0isgg4HXgEmKeqO8DEBehq3J5lwteB/wwcjZ4fB+xW1aHoeTOe7xOBAeB7kYvuZhHpoInPtao+D/x3YCsmGHuAx2j+cx0odm7rur65cBRHEpY1be6yiMwA7gL+RlVfbvT+ZImIXAT0q+pj8cUJqzbb+W4B3gjcqKqnA/tpIrdUEpFP/xJgMfBqoANz0xTSbOe6HHV93l04itMLHB97vhDY3qB9yRQRmYqJxm2q+qNocV8wXaPf/Y3avww4B7hYRJ7DXJDnYRbI7MidAc15vnuBXlV9JHq+ChOSZj7XfwY8q6oDqnoY+BHwJpr/XAeKndu6rm8uHMVZAyyJsi9asYDa6gbvU+pEvv3vAJtU9R9jL60GPho9/ijwk7Het6xQ1c+r6kJVXYSd11+o6hXA/cAHotWa6pgBVPUFYJuInBwtehuwkSY+15iL6iwRmR591sMxN/W5jlHs3K4GPhJlV50F7AkurUrwyvESiMiF2J1oDviuqv63Bu9S6ojIm4FfAevJ+/v/FotzrAROwL58l6pqYeBtwiMi5wL/SVUvEpETMQvkWOBx4EOqOtjI/UsbEXkDlhDQCmwBPobdQDbtuRaRvwM+iGUQPg78BebPb6pzLSI/AM7F2qf3Af8V+DEJ5zYS0W9hWVgHgI+p6tqK/5YLh+M4jlMN7qpyHMdxqsKFw3Ecx6kKFw7HcRynKlw4HMdxnKpw4XAcx3GqwoXDcVJARI6IyLrYT2oV2SKyKN7x1HEaTUv5VRzHqYBXVPUNjd4JxxkL3OJwnAwRkedE5Gsi8mj089po+WtE5L5oFsJ9InJCtHyeiNwtIk9EP2+KNpUTkW9HcyX+n4i0N+ygnEmPC4fjpEN7gavqg7HXXlbVM7FK3a9Hy76FtbV+PXAb8M1o+TeBB1X1NKyPVE+0fAlwg6ouA3YD78/4eBynKF457jgpICL7VHVGwvLngPNUdUvUTPIFVT1ORHYC81X1cLR8h6rOFZEBYGG8/UXU7v5n0TAeROS/AFNV9frsj8xxRuMWh+NkjxZ5XGydJOJ9lI7g8UmngbhwOE72fDD2+6Ho8W+xzrwAVwC/jh7fB3wShmeizxyrnXScSvG7FsdJh3YRWRd7/lNVDSm5bSLyCHajdnm07K+B74rI57CpfB+Lln8GuElErsIsi09ik+scZ9zgMQ7HyZAoxrFCVXc2el8cJy3cVeU4juNUhVscjuM4TlW4xeE4juNUhQuH4ziOUxUuHI7jOE5VuHA4juM4VeHC4TiO41TF/wcum60ryT7BLgAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.plot(history.history['accuracy'])\n",
    "plt.plot(history.history['val_accuracy'])\n",
    "plt.title('Model accuracy')\n",
    "plt.ylabel('Accuracy')\n",
    "plt.xlabel('Epoch')\n",
    "plt.legend(['Train', 'Test'], loc='upper left')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.plot(history.history['loss']) \n",
    "plt.plot(history.history['val_loss']) \n",
    "plt.title('Model loss') \n",
    "plt.ylabel('Loss') \n",
    "plt.xlabel('Epoch') \n",
    "plt.legend(['Train', 'Test'], loc='upper left') \n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "methylClinicalNoAttr[methylClinicalNoAttr['status']==1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#autoencoder implementation\n",
    "\n",
    "from tensorflow.python.keras.models import Model\n",
    "\n",
    "# Autoencoder code goes here...\n",
    "encoding_dim = 50    # size of our encoded representations\n",
    "\n",
    "input_dim = X_train.shape[1]\n",
    "# this is our input placeholder\n",
    "input_data = Input(shape=(input_dim, ))\n",
    "\n",
    "# \"encoded\" is the encoded representation of the input\n",
    "encoded = Dense(encoding_dim, activation='relu', name='encoder')(input_data)\n",
    "\n",
    "# \"decoded\" is the lossy reconstruction of the input\n",
    "decoded = Dense(18028, activation='sigmoid', name='decoder')(encoded)\n",
    "# this model maps an input to its reconstruction\n",
    "autoencoder = Model(inputs=input_data, outputs=decoded)\n",
    "\n",
    "autoencoder.compile(optimizer='sgd', loss='mse')\n",
    "\n",
    "# training\n",
    "history=autoencoder.fit(X_train, X_train,\n",
    "            epochs=100,\n",
    "            batch_size=10,\n",
    "            shuffle=True,\n",
    "            validation_data=(X_test, X_test))  # need more tuning\n",
    "\n",
    "# test the autoencoder by encoding and decoding the test dataset\n",
    "reconstructions = autoencoder.predict(X_test)\n",
    "print('Original test data')\n",
    "print(X_test)\n",
    "print('Reconstructed test data')\n",
    "print(reconstructions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.plot(history.history['loss'])\n",
    "plt.plot(history.history['val_loss'])\n",
    "plt.title('model loss')\n",
    "plt.ylabel('loss')\n",
    "plt.xlabel('epoch')\n",
    "plt.legend(['train', 'test'], loc='upper right');"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mse = np.mean(np.power(X_test - reconstructions, 2), axis=1)\n",
    "\n",
    "y_test\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### MLP code goes here...\n",
    "# create model\n",
    "\n",
    "x = autoencoder.get_layer('encoder').output\n",
    "# h = Dense(3, activation='relu', name='hidden')(x)\n",
    "y = Dense(2, activation='softmax', name='predictions')(x)\n",
    "classifier = Model(inputs=autoencoder.inputs, outputs=y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compile model\n",
    "classifier.compile(loss='categorical_crossentropy', optimizer='adam',\n",
    "               metrics=['accuracy'])\n",
    "\n",
    "# Fit the model\n",
    "classifier.fit(X_train, y_train, epochs=100, batch_size=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "y_pred = classifier.predict(X_test)\n",
    "#Converting predictions to label\n",
    "pred = list()\n",
    "for i in range(len(y_pred)):\n",
    "    pred.append(np.argmax(y_pred[i]))\n",
    "#Converting one hot encoded test label to label\n",
    "test = list()\n",
    "for i in range(len(y_test)):\n",
    "    test.append(np.argmax(y_test[i]))\n",
    "    \n",
    "a=accuracy_score(pred,test)\n",
    "\n",
    "print('Accuracy is:', a*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn import feature_selection\n",
    "\n",
    "sel = feature_selection.VarianceThreshold()\n",
    "train_variance = sel.fit_transform(rna1_train)\n",
    "train_variance.shape\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna1_train.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "constant_filter = feature_selection.VarianceThreshold(threshold=0)\n",
    "constant_filter.fit(rna1_train)\n",
    "\n",
    "len(rna1_train.columns[constant_filter.get_support()])\n",
    "\n",
    "constant_columns = [column for column in rna1_train.columns\n",
    "                    if column not in rna1_train.columns[constant_filter.get_support()]]\n",
    "\n",
    "rna1_train.drop(labels=constant_columns, axis=1, inplace=True)\n",
    "rna1_test.drop(labels=constant_columns, axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Removing Quasi-Constant Features Using Variance Threshold\n",
    "from sklearn import feature_selection\n",
    "from sklearn.feature_selection import VarianceThreshold\n",
    "\n",
    "sel = feature_selection.VarianceThreshold()\n",
    "\n",
    "thresholder = VarianceThreshold(threshold=0.01)\n",
    "\n",
    "# Conduct variance thresholding\n",
    "X_high_variance = thresholder.fit_transform(rna1_train)\n",
    "\n",
    "constant_columns = [column for column in rna1_train.columns\n",
    "                    if column not in rna1_train.columns[thresholder.get_support()]]\n",
    "\n",
    "rna1_train.drop(labels=constant_columns, axis=1, inplace=True)\n",
    "rna1_test.drop(labels=constant_columns, axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna1_train.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(rna1_train.columns[thresholder.get_support()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Similarly, you can find the number of constant features with the help of the following script:\n",
    "constant_columns = [column for column in rna1_train.columns\n",
    "                    if column not in rna1_train.columns[thresholder.get_support()]]\n",
    "\n",
    "print(len(constant_columns))\n",
    "\n",
    "for column in constant_columns:\n",
    "    print(column)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna1_train_T = rna1_train.T\n",
    "rna1_train_T.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(rna1_train_T.duplicated().sum())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "correlated_features = set()\n",
    "correlation_matrix = rnaSeqlClinicalNoAttr.corr()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#we will loop through all the columns in the correlation_matrix and will add the columns with a correlation value of 0.8 to the correlated_features set as shown below.\n",
    "\n",
    "for i in range(len(correlation_matrix .columns)):\n",
    "    for j in range(i):\n",
    "        if abs(correlation_matrix.iloc[i, j]) > 0.90:\n",
    "            colname = correlation_matrix.columns[i]\n",
    "            correlated_features.add(colname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(correlated_features)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.feature_selection import SelectKBest\n",
    "from sklearn.feature_selection import f_classif\n",
    "\n",
    "miRna = SelectKBest(f_classif, k=20)\n",
    "fit = miRna.fit(miRna1_train, miRna2_train)\n",
    "\n",
    "\n",
    "dfscores = pd.DataFrame(fit.scores_)\n",
    "dfcolumns = pd.DataFrame(miRna1_train.columns) \n",
    "featureScores = pd.concat([dfcolumns,dfscores],axis=1)\n",
    "featureScores.columns = ['Features','Score']  \n",
    "print(featureScores.nlargest(20,'Score'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "miRna1_train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#trying KNN\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=10)\n",
    "classifierK.fit(miRna1_train, miRna2_train)\n",
    "classifierK.score(miRna1_test, miRna2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scaler = StandardScaler()\n",
    "# Fit on training set only.\n",
    "scaler.fit(rna1_train)\n",
    "\n",
    "# Apply transform to both the training set and the test set.\n",
    "rna1_traint = scaler.transform(rna1_train)\n",
    "rna1_testt = scaler.transform(rna1_test)\n",
    "\n",
    "pca = PCA(n_components = 20)\n",
    "# actually run the fit algorithm\n",
    "eigenbasis = pca.fit(rna1_traint)\n",
    "# transform our data using the learned transform\n",
    "Rna_Trainpca = eigenbasis.transform(rna1_traint)\n",
    "Rna_Testpca = eigenbasis.transform(rna1_testt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#trying KNN\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=10)\n",
    "classifierK.fit(Rna_Trainpca, rna2_train)\n",
    "classifierK.score(Rna_Testpca, rna2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a classifier: a support vector classifier\n",
    "classifierS = svm.SVC(gamma=0.001)\n",
    "classifierS.fit(Rna_Trainpca, rna2_train)\n",
    "classifierS.score(Rna_Testpca, rna2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# random forest model creation\n",
    "classifierR = RandomForestClassifier()\n",
    "classifierR.fit(Rna_Trainpca, rna2_train)\n",
    "classifierR.score(Rna_Testpca, rna2_test)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# the actual components\n",
    "print(\"principal components:\\n\", pca.components_)\n",
    "# let's also look at how much of the total variance we were able to cover with 2 dimensions\n",
    "print('percentage of variance explained by first 2 principal components:', pca.explained_variance_ratio_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.plot(pca.explained_variance_ratio_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Concatenating multiple feature extraction methods\n",
    "from sklearn.pipeline import Pipeline, FeatureUnion\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "from sklearn.svm import SVC\n",
    "from sklearn.model_selection import KFold\n",
    "from sklearn.model_selection import cross_val_score\n",
    "from sklearn.pipeline import Pipeline\n",
    "\n",
    "# create feature union\n",
    "features = []\n",
    "features.append(('pca', PCA(n_components=20)))\n",
    "features.append(('select_best', SelectKBest(k=20)))\n",
    "feature_union = FeatureUnion(features)\n",
    "\n",
    "svm = SVC(kernel=\"linear\")\n",
    "\n",
    "# create pipeline\n",
    "estimators = []\n",
    "estimators.append(('feature_union', feature_union))\n",
    "estimators.append((\"svm\", svm))\n",
    "model = Pipeline(estimators)\n",
    "\n",
    "# evaluate pipeline\n",
    "seed = 7\n",
    "kfold = KFold(n_splits=10, random_state=seed)\n",
    "results = cross_val_score(model, rna1_train, rna1_test, cv=kfold)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use combined features to transform dataset:\n",
    "X_features = combined_features.fit(rna1_traint, rna1_testt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cleanClinicalData[0:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "commonData=methylationData.merge(rnaSeqData, on='attrib_name', how='inner').merge(miRNAData, on='attrib_name', how='inner').merge(cleanClinicalData[['attrib_name','status']], on='attrib_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "commonData[0:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "commonDataNoAttr = commonData.drop(['attrib_name'],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "commonDataNoAttr.drop(commonDataNoAttr.columns[commonDataNoAttr.isna().sum()>0],axis = 1,inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Splitting the Methylation dataset into the Training set and Test set\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "common1_train, common1_test, common2_train, common2_test= train_test_split(commonDataNoAttr.loc[:, commonDataNoAttr.columns != 'status'],commonDataNoAttr['status'],test_size=0.2, random_state=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#trying KNN\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=3)\n",
    "classifierK.fit(common1_train, common2_train)\n",
    "classifierK.score(common1_test, common2_test)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a classifier: a support vector classifier\n",
    "from sklearn import svm, metrics\n",
    "classifierS = svm.SVC(C=100,gamma=0.005)\n",
    "classifierS.fit(common1_train, common2_train)\n",
    "classifierS.score(common1_test, common2_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# random forest model creation\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "classifierR = RandomForestClassifier()\n",
    "classifierR.fit(common1_train, common2_train)\n",
    "classifierR.score(common1_test, common2_test) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "from sklearn.model_selection import GridSearchCV\n",
    "\n",
    "n_estimators = [100,150,200, 300, 500] \n",
    "max_depth = [5,6,7, 8, 15, 25, 30] \n",
    "min_samples_split = [2, 5,6,7, 10, 15, 100] \n",
    "min_samples_leaf = [1, 2,4, 5,6,7, 10,20]\n",
    "\n",
    "hyperF = dict(n_estimators = n_estimators, max_depth = max_depth,\n",
    "min_samples_split = min_samples_split, min_samples_leaf = min_samples_leaf)\n",
    "\n",
    "forest = RandomForestClassifier()\n",
    "\n",
    "gridF = GridSearchCV(forest, hyperF, cv = 3, verbose = 1, n_jobs = -1) \n",
    "bestF = gridF.fit(common1_train, common2_train)\n",
    "\n",
    "bestF.best_params_,bestF.best_score_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "from xgboost import XGBClassifier\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "from sklearn.metrics import roc_auc_score\n",
    "from sklearn.metrics import accuracy_score\n",
    "\n",
    "from sklearn.metrics import roc_auc_score\n",
    "# creating a list of different leaf sizes\n",
    "sample_leaf_options = [1,2,3,4,5,7,10,20,50]\n",
    "n_estimators = [100,150,200,300]\n",
    "max_depth= [4,5,6,7,8,9]\n",
    "\n",
    "# for loop to iterate for each leaf size\n",
    "for estimator in n_estimators :\n",
    "  #  print(\"\\n estimator :\", estimator)\n",
    "    for leaf_size in sample_leaf_options :\n",
    "     #   print(\"\\n Leaf size :\", leaf_size)\n",
    "        for depth_size in max_depth :\n",
    "            model = XGBClassifier(n_estimators = estimator, n_jobs = -1,random_state =0, min_samples_leaf = leaf_size,max_depth=depth_size)\n",
    "            model.fit(common1_train, common2_train)\n",
    "          #  print(\"\\n max depth :\", depth_size)\n",
    "            print (\"\\n\", estimator, leaf_size,depth_size, accuracy_score(model.predict(common1_test),common2_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO: run PCA on the Common data and project the dataset down to 10 dimensions\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scaler = StandardScaler()\n",
    "# Fit on training set only.\n",
    "scaler.fit(common1_train)\n",
    "\n",
    "# Apply transform to both the training set and the test set.\n",
    "common1_traint = scaler.transform(common1_train)\n",
    "common1_testt = scaler.transform(common1_test)\n",
    "\n",
    "pca = PCA(n_components = 20)\n",
    "\n",
    "#pca = PCA(n_components=0.99, svd_solver='full')\n",
    "#my_model.fit_transform(my_matrix)\n",
    "# actually run the fit algorithm\n",
    "eigenbasis = pca.fit(common1_traint)\n",
    "# transform our data using the learned transform\n",
    "common_Trainpca = eigenbasis.transform(common1_traint)\n",
    "common_Testpca = eigenbasis.transform(common1_testt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "classifierK = KNeighborsClassifier(metric='euclidean',n_neighbors=10)\n",
    "classifierK.fit(common_Trainpca, common2_train)\n",
    "print(classifierK.score(common_Testpca, common2_test)*100)\n",
    "\n",
    "# Create a classifier: a support vector classifier\n",
    "from sklearn import svm, metrics\n",
    "classifierS = svm.SVC(gamma=0.001)\n",
    "classifierS.fit(common_Trainpca, common2_train)\n",
    "print(classifierS.score(common_Testpca, common2_test)*100)\n",
    "\n",
    "# random forest model creation\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "classifierR = RandomForestClassifier()\n",
    "classifierR.fit(common_Trainpca, common2_train)\n",
    "print(classifierR.score(common_Testpca, common2_test)*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "plt.ylabel('Explained Variance')\n",
    "plt.xlabel('Principal Components')\n",
    "plt.title('Explained Variance Ratio')\n",
    "plt.plot(pca.explained_variance_ratio_)\n",
    "\n",
    "print(pca.explained_variance_ratio_.cumsum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "commonDataNoAttr['status'].value_counts().plot.bar()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import roc_auc_score\n",
    "from sklearn.metrics import accuracy_score\n",
    "# creating a list of different leaf sizes\n",
    "sample_leaf_options = [1,2,3,4,5,7,10,20,50]\n",
    "n_estimators = [100,150,200,300]\n",
    "max_depth= [4,5,6,7,8,9]\n",
    "\n",
    "# for loop to iterate for each leaf size\n",
    "for estimator in n_estimators :\n",
    "  #  print(\"\\n estimator :\", estimator)\n",
    "    for leaf_size in sample_leaf_options :\n",
    "     #   print(\"\\n Leaf size :\", leaf_size)\n",
    "        for depth_size in max_depth :\n",
    "            model = RandomForestClassifier(n_estimators = estimator, n_jobs = -1,random_state =0, min_samples_leaf = leaf_size,max_depth=depth_size)\n",
    "            model.fit(common_Trainpca, common2_train)\n",
    "          #  print(\"\\n max depth :\", depth_size)\n",
    "            print (\"\\n\", estimator, leaf_size,depth_size, accuracy_score(model.predict(common_Testpca),common2_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = svm.SVC(kernel='linear',C=1,gamma=0.001)\n",
    "model.fit(common_Trainpca, common2_train)\n",
    "test_pred = model.predict(common_Testpca)\n",
    "# prediction for training dataset\n",
    "train_pred = model.predict(common_Trainpca)\n",
    "\n",
    "from sklearn.metrics import accuracy_score\n",
    "print('Accuracy score for Training Dataset = ', accuracy_score(train_pred,common2_train))\n",
    "print('Accuracy score for Testing Dataset = ', accuracy_score(test_pred, common2_test))\n",
    "\n",
    "print('Confusion Matrix - Training Dataset')\n",
    "print(pd.crosstab(common2_train.ravel(), train_pred, rownames = ['True'], colnames = ['Predicted'], margins = True))\n",
    "\n",
    "print('Confusion Matrix - Testing Dataset')\n",
    "print(pd.crosstab(common2_test.ravel(), test_pred.ravel(), rownames = ['True'], colnames = ['Predicted'], margins = True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# standardize the data\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scaler = StandardScaler()\n",
    "X = scaler.fit_transform(commonDataNoAttr.loc[:, commonDataNoAttr.columns != 'status'])\n",
    "#X = (commonDataNoAttr.loc[:, commonDataNoAttr.columns != 'survival'])\n",
    "y=commonDataNoAttr['status']\n",
    "\n",
    "# split into training and testing datasets\n",
    "from sklearn.model_selection import train_test_split\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.33, random_state = 2, shuffle = True)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from imblearn.over_sampling import SMOTE\n",
    "sm = SMOTE(random_state = 33)\n",
    "\n",
    "\n",
    "X_train_new, y_train_new = sm.fit_sample(X_train, y_train.ravel())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# fit the model\n",
    "clf = svm.SVC(kernel='linear',C=1,gamma=0.001)\n",
    "clf.fit(X_train_new, y_train_new)\n",
    "\n",
    "# prediction for Training data\n",
    "train_pred_sm = clf.predict(X_train_new)\n",
    "\n",
    "# prediction for Testing data\n",
    "test_pred_sm = clf.predict(X_test)\n",
    "\n",
    "print('Accuracy score for Training Dataset = ', accuracy_score(train_pred_sm, y_train_new))\n",
    "print('Accuracy score for Testing Dataset = ', accuracy_score(test_pred_sm, y_test))\n",
    "\n",
    "print('Confusion Matrix - Testing Dataset')\n",
    "print(pd.crosstab(y_test.ravel(), test_pred_sm, rownames = ['True'], colnames = ['Predicted'], margins = True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO: run PCA on the Common data and project the dataset down to 10 dimensions\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scaler = StandardScaler()\n",
    "# Fit on training set only.\n",
    "scaler.fit(X_train_new)\n",
    "\n",
    "# Apply transform to both the training set and the test set.\n",
    "common1_traint = scaler.transform(X_train_new)\n",
    "common1_testt = scaler.transform(X_test)\n",
    "\n",
    "pca = PCA(n_components = 20)\n",
    "\n",
    "#pca = PCA(n_components=0.99, svd_solver='full')\n",
    "#my_model.fit_transform(my_matrix)\n",
    "# actually run the fit algorithm\n",
    "eigenbasis = pca.fit(common1_traint)\n",
    "# transform our data using the learned transform\n",
    "common_Trainpca = eigenbasis.transform(common1_traint)\n",
    "common_Testpca = eigenbasis.transform(common1_testt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# fit the model\n",
    "clf = svm.SVC(kernel='linear',C=10,gamma=0.005)\n",
    "clf.fit(common_Trainpca, y_train_new)\n",
    "\n",
    "# prediction for Training data\n",
    "train_pred_sm = clf.predict(common_Trainpca)\n",
    "\n",
    "# prediction for Testing data\n",
    "test_pred_sm = clf.predict(common_Testpca)\n",
    "\n",
    "print('Accuracy score for Training Dataset = ', accuracy_score(train_pred_sm, y_train_new))\n",
    "print('Accuracy score for Testing Dataset = ', accuracy_score(test_pred_sm, y_test))\n",
    "\n",
    "print('Confusion Matrix - Testing Dataset')\n",
    "print(pd.crosstab(y_test.ravel(), test_pred_sm, rownames = ['True'], colnames = ['Predicted'], margins = True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import roc_auc_score\n",
    "from sklearn.metrics import accuracy_score\n",
    "# creating a list of different leaf sizes\n",
    "sample_leaf_options = [1,2,3,4,5,7,10,20,50]\n",
    "n_estimators = [100,150,200,300]\n",
    "max_depth= [4,5,6,7,8,9]\n",
    "\n",
    "# for loop to iterate for each leaf size\n",
    "for estimator in n_estimators :\n",
    "  #  print(\"\\n estimator :\", estimator)\n",
    "    for leaf_size in sample_leaf_options :\n",
    "     #   print(\"\\n Leaf size :\", leaf_size)\n",
    "        for depth_size in max_depth :\n",
    "            model = XGBClassifier(n_estimators = estimator, n_jobs = -1,random_state =0, min_samples_leaf = leaf_size,max_depth=depth_size)\n",
    "            model.fit(common_Trainpca, y_train_new)\n",
    "          #  print(\"\\n max depth :\", depth_size)\n",
    "            print (\"\\n\", estimator, leaf_size,depth_size, accuracy_score(model.predict(common_Testpca),y_test))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "from sklearn.model_selection import validation_curve\n",
    "num_est = [50,100,150,200, 300, 500]\n",
    "max_depth = [4,5,6,7,8,10,15,20,25,30]\n",
    "min_samples_split = [2,3,4, 5,6,7, 10, 15,20, 100]\n",
    "min_samples_leaf = [1, 2, 5,7,8, 10,15,20]\n",
    "learning_rate=[0.1, 0.01, 0.05]\n",
    "train_scores, test_scores = validation_curve(\n",
    "                                XGBClassifier(),\n",
    "                                X = common_Trainpca, y = y_train_new, \n",
    "                                param_name = 'min_samples_split', \n",
    "                                param_range = min_samples_split, cv = 3)\n",
    "\n",
    "train_scores_mean = np.mean(train_scores, axis=1)\n",
    "train_scores_std = np.std(train_scores, axis=1)\n",
    "test_scores_mean = np.mean(test_scores, axis=1)\n",
    "test_scores_std = np.std(test_scores, axis=1)\n",
    "\n",
    "plt.title(\"Validation Curve with XGBoost\")\n",
    "plt.xlabel(\"min_samples_split\")\n",
    "plt.ylabel(\"Score\")\n",
    "#plt.ylim(0.0, 1.1)\n",
    "lw = 2\n",
    "plt.semilogx(min_samples_split, train_scores_mean, label=\"Training score\",\n",
    "             color=\"darkorange\", lw=lw)\n",
    "\n",
    "plt.semilogx(min_samples_split, test_scores_mean, label=\"Cross-validation score\",\n",
    "             color=\"navy\", lw=lw)\n",
    "\n",
    "plt.legend(loc=\"best\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import classification_report\n",
    "print(\"Report : \", \n",
    "    classification_report(y_test, model.predict(common_Testpca))) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = RandomForestClassifier(n_estimators = 150, n_jobs = -1,random_state =0, min_samples_leaf = 1,min_samples_split =4,max_depth=6)\n",
    "model.fit(common_Trainpca, y_train_new)\n",
    "        \n",
    "print (\"Accuracy : \", accuracy_score(model.predict(common_Testpca),y_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import classification_report\n",
    "print(\"Report : \", \n",
    "    classification_report(y_test, model.predict(common_Testpca)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import precision_score\n",
    "\n",
    "print (\"precision : \", precision_score(model.predict(common_Testpca),y_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import f1_score\n",
    "\n",
    "print (\"precision : \", f1_score(model.predict(common_Testpca),y_test))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "from xgboost import XGBClassifier\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "from matplotlib import pyplot\n",
    "import numpy\n",
    "\n",
    "parameters = {\n",
    "    'max_depth': [2,4,6,8],\n",
    "    'n_estimators': [100,150,200,300],\n",
    "    'min_samples_leaf': [1, 2, 5,7,8, 10,15,20],\n",
    "    'learning_rate':[0.1, 0.01, 0.05]\n",
    "\n",
    "}\n",
    "\n",
    "estimator = XGBClassifier(\n",
    "    objective='multi:softmax',\n",
    "    nthread=4,\n",
    "    seed=42\n",
    ")\n",
    "\n",
    "grid_search = GridSearchCV(\n",
    "    estimator=estimator,\n",
    "    param_grid=parameters,\n",
    "    scoring = \"neg_log_loss\",\n",
    "    n_jobs = -1,\n",
    "    cv = 10,\n",
    "    verbose=True\n",
    "    )\n",
    "\n",
    "grid_result = grid_search.fit(common_Trainpca, y_train_new)\n",
    "\n",
    "# summarize results\n",
    "print(\"Best: %f using %s\" % (grid_result.best_score_, grid_result.best_params_))\n",
    "means = grid_result.cv_results_['mean_test_score']\n",
    "stds = grid_result.cv_results_['std_test_score']\n",
    "params = grid_result.cv_results_['params']\n",
    "for mean, stdev, param in zip(means, stds, params):\n",
    "\tprint(\"%f (%f) with: %r\" % (mean, stdev, param))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from xgboost import XGBClassifier\n",
    "from sklearn.metrics import accuracy_score\n",
    "model = XGBClassifier(n_estimators = 100, n_jobs = -1,random_state =0, min_samples_leaf = 1,min_samples_split =4,max_depth=6,learning_rate=0.1)\n",
    "model.fit(common_Trainpca, y_train_new)\n",
    "        \n",
    "print (\"Accuracy : \", accuracy_score(model.predict(common_Testpca),y_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"Report : \", \n",
    "    classification_report(y_test, model.predict(common_Testpca)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.ensemble import VotingClassifier\n",
    "from sklearn import model_selection\n",
    "\n",
    "# Voting Ensemble for Classification\n",
    "import pandas\n",
    "from sklearn import model_selection\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.tree import DecisionTreeClassifier\n",
    "from sklearn.svm import SVC\n",
    "\n",
    "seed = 0\n",
    "kfold = model_selection.KFold(n_splits=10, random_state=seed)\n",
    "# create the sub models\n",
    "estimators = []\n",
    "model1 = SVC(kernel='linear',C=10, gamma=0.001)\n",
    "estimators.append(('svm', model1))\n",
    "\n",
    "model4 = KNeighborsClassifier(metric='euclidean',n_neighbors=5)\n",
    "estimators.append(('knn', model4))\n",
    "\n",
    "model3 = GradientBoostingClassifier(n_estimators = 100,  min_samples_leaf = 1,min_samples_split =4,max_depth=6)\n",
    "estimators.append(('gradboost', model3))\n",
    "\n",
    "\n",
    "# create the ensemble model\n",
    "ensemble = VotingClassifier(estimators)\n",
    "results = model_selection.cross_val_score(ensemble, common_Trainpca, y_train_new, cv=kfold)\n",
    "print(results.mean())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#fit model to training data\n",
    "ensemble.fit(common_Trainpca, y_train_new)\n",
    "#test our model on the test data\n",
    "ensemble.score(common_Testpca, y_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Changing common dataset pandas dataframe to numpy array\n",
    "X = commonDataNoAttr.loc[:, commonDataNoAttr.columns != 'status']\n",
    "y = commonDataNoAttr['status'].values\n",
    "\n",
    "#Normalizing the data\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "sc = StandardScaler()\n",
    "X = sc.fit_transform(X)\n",
    "\n",
    "from sklearn.preprocessing import OneHotEncoder\n",
    "ohe = OneHotEncoder(categories='auto')\n",
    "y = ohe.fit_transform(y.reshape(-1, 1)).toarray()\n",
    "\n",
    "from sklearn.model_selection import train_test_split\n",
    "X_train,X_test,y_train,y_test = train_test_split(X,y,test_size = 0.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "X_train.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#Dependencies\n",
    "from tensorflow.python.keras.models import Sequential\n",
    "from tensorflow.python.keras.layers import Input, Dense,Dropout\n",
    "from tensorflow.python.keras.optimizers import SGD\n",
    "\n",
    "# Neural network\n",
    "model = Sequential()\n",
    "model.add(Dense(500, input_dim=39005, activation='relu'))\n",
    "model.add(Dropout(0.2))\n",
    "model.add(Dense(50, activation='relu'))\n",
    "model.add(Dense(100, activation='relu'))\n",
    "model.add(Dense(2, activation='softmax'))\n",
    "\n",
    "model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "history = model.fit(X_train, y_train,validation_data = (X_test,y_test), epochs=100, batch_size=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "y_pred = model.predict(X_test)\n",
    "#Converting predictions to label\n",
    "pred = list()\n",
    "for i in range(len(y_pred)):\n",
    "    pred.append(np.argmax(y_pred[i]))\n",
    "#Converting one hot encoded test label to label\n",
    "test = list()\n",
    "for i in range(len(y_test)):\n",
    "    test.append(np.argmax(y_test[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import accuracy_score\n",
    "a = accuracy_score(pred,test)\n",
    "print('Accuracy is:', a*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.plot(history.history['accuracy'])\n",
    "plt.plot(history.history['val_accuracy'])\n",
    "plt.title('Model accuracy')\n",
    "plt.ylabel('Accuracy')\n",
    "plt.xlabel('Epoch')\n",
    "plt.legend(['Train', 'Test'], loc='upper left')\n",
    "plt.show()\n",
    "\n",
    "\n",
    "plt.plot(history.history['loss']) \n",
    "plt.plot(history.history['val_loss']) \n",
    "plt.title('Model loss') \n",
    "plt.ylabel('Loss') \n",
    "plt.xlabel('Epoch') \n",
    "plt.legend(['Train', 'Test'], loc='upper left') \n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"Report : \", \n",
    "    classification_report(test,pred))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#autoencoder implementation\n",
    "\n",
    "from tensorflow.python.keras.models import Model\n",
    "\n",
    "# Autoencoder code goes here...\n",
    "encoding_dim = 100    # size of our encoded representations\n",
    "\n",
    "input_dim = X_train.shape[1]\n",
    "# this is our input placeholder\n",
    "input_data = Input(shape=(input_dim, ))\n",
    "\n",
    "# \"encoded\" is the encoded representation of the input\n",
    "encoded = Dense(encoding_dim, activation='relu', name='encoder')(input_data)\n",
    "\n",
    "# \"decoded\" is the lossy reconstruction of the input\n",
    "decoded = Dense(39005, activation='sigmoid', name='decoder')(encoded)\n",
    "# this model maps an input to its reconstruction\n",
    "autoencoder = Model(inputs=input_data, outputs=decoded)\n",
    "\n",
    "autoencoder.compile(optimizer='sgd', loss='mse')\n",
    "\n",
    "# training\n",
    "history=autoencoder.fit(X_train, X_train,\n",
    "            epochs=100,\n",
    "            batch_size=10,\n",
    "            shuffle=True,\n",
    "            validation_data=(X_test, X_test))  # need more tuning\n",
    "\n",
    "# test the autoencoder by encoding and decoding the test dataset\n",
    "reconstructions = autoencoder.predict(X_test)\n",
    "print('Original test data')\n",
    "print(X_test)\n",
    "print('Reconstructed test data')\n",
    "print(reconstructions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#MLP code goes here...\n",
    "#create model\n",
    "x = autoencoder.get_layer('encoder').output\n",
    "\n",
    "h = Dense(600, activation='relu', name='hidden')(x)\n",
    "h2 = Dense(200, activation='relu', name='hidden2')(h)\n",
    "y = Dense(2, activation='softmax', name='predictions')(h2) \n",
    "\n",
    "classifier = Model(inputs=autoencoder.inputs, outputs=y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compile model\n",
    "classifier.compile(loss='categorical_crossentropy', optimizer='adam',\n",
    "               metrics=['accuracy'])\n",
    "\n",
    "# Fit the model\n",
    "classifier.fit(X_train, y_train, epochs=100, batch_size=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "y_pred = classifier.predict(X_test)\n",
    "#Converting predictions to label\n",
    "pred = list()\n",
    "for i in range(len(y_pred)):\n",
    "    pred.append(np.argmax(y_pred[i]))\n",
    "#Converting one hot encoded test label to label\n",
    "test = list()\n",
    "for i in range(len(y_test)):\n",
    "    test.append(np.argmax(y_test[i]))\n",
    "    \n",
    "a=accuracy_score(pred,test)\n",
    "\n",
    "print('Accuracy is:', a*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"Report : \", \n",
    "    classification_report(test,pred))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import itertools\n",
    "\n",
    "\n",
    "def plot_confusion_matrix(cm,\n",
    "                          target_names,\n",
    "                          title='Confusion matrix',\n",
    "                          cmap=None,\n",
    "                          normalize=True):\n",
    "\n",
    "    accuracy = np.trace(cm) / float(np.sum(cm))\n",
    "    misclass = 1 - accuracy\n",
    "\n",
    "    if cmap is None:\n",
    "        cmap = plt.get_cmap('Blues')\n",
    "\n",
    "    plt.figure(figsize=(8, 6))\n",
    "    plt.imshow(cm, interpolation='nearest', cmap=cmap)\n",
    "    plt.title(title)\n",
    "    plt.colorbar()\n",
    "\n",
    "    if target_names is not None:\n",
    "        tick_marks = np.arange(len(target_names))\n",
    "        plt.xticks(tick_marks, target_names, rotation=45)\n",
    "        plt.yticks(tick_marks, target_names)\n",
    "\n",
    "    if normalize:\n",
    "        cm = cm.astype('float') / cm.sum(axis=1)[:, np.newaxis]\n",
    "    thresh = cm.max() / 1.5 if normalize else cm.max() / 2\n",
    "    for i, j in itertools.product(range(cm.shape[0]), range(cm.shape[1])):\n",
    "        if normalize:\n",
    "             plt.text(j, i, \"{:0.4f}\".format(cm[i, j]),\n",
    "                horizontalalignment=\"center\",\n",
    "                color=\"red\" if cm[i, j] > thresh else \"black\")\n",
    "        else:\n",
    "            plt.text(j, i, \"{:,}\".format(cm[i, j]),\n",
    "              horizontalalignment=\"center\",\n",
    "              color=\"red\" if cm[i, j] > thresh else \"black\")\n",
    "\n",
    "\n",
    "    plt.tight_layout()\n",
    "    plt.ylabel('True label')\n",
    "    plt.xlabel('Predicted label\\naccuracy={:0.4f}; misclass={:0.4f}'.format(accuracy, misclass))\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'test' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-80-b2e65d0d061a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0msklearn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmetrics\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mconfusion_matrix\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mcm\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mconfusion_matrix\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'test' is not defined"
     ]
    }
   ],
   "source": [
    "from sklearn.metrics import confusion_matrix\n",
    "cm=confusion_matrix(test, pred)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_confusion_matrix(cm,target_names = ['alive', 'dead'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
